{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4d22d9c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "import timeit\n",
    "import numpy as np\n",
    "import h5py\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ba766e0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# -------- Path STAR\n",
    "path_star = '/cluster/work/grlab/projects/GTEx/rna_gencode32_realign/results'\n",
    "# Star junctions - unique coordinates\n",
    "star_jx = os.path.join(path_star, 'junctions_spladder.all_coords.sorted.uniq.tsv.gz')\n",
    "# Star junctions - projected coordinates and expression\n",
    "#projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "\n",
    "\n",
    "# -------- Intermediate filtering results (threshold and merged)\n",
    "#TODO DO the generation matrix???\n",
    "# Foreground matrix\n",
    "big_matrix = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r_complete.tsv.gz'\n",
    "\n",
    "\n",
    "\n",
    "# -------- GTEX filtering \n",
    "whitelist = '/cluster/work/grlab/projects/projects2020_OHSU/sample_lists/GTEX/GTEx_sample_IDs_10-2021_lib_graph_juliannelist'\n",
    "libsize = '/cluster/work/grlab/projects/TCGA/PanCanAtlas/immunopepper_paper/peptides_ccell_rerun_gtex_151220/ARCHIV_keep_runs/GTEX2019_commit_v3_TEST_merged3_372a147_medium_run_pya.0.17.1_conf2_annot_ref_chrall_cap/expression_counts.libsize.tsv'\n",
    "\n",
    "\n",
    "normalizer = 400000\n",
    "filter_thresholds = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1c3c201",
   "metadata": {},
   "source": [
    "# Load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "63b89a2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def explode_immunopepper_coord(mx):\n",
    "    coord_mx = mx['coord'].str.split(':', expand=True) #7 min\n",
    "\n",
    "    coord_mx[1] = coord_mx[1].astype('int')\n",
    "    coord_mx[2] = coord_mx[2].astype('int')\n",
    "\n",
    "    coord_mx['strand'] = None\n",
    "    coord_mx.loc[coord_mx[1] < coord_mx[2], 'strand'] = '+'\n",
    "    coord_mx.loc[coord_mx[1] > coord_mx[2], 'strand'] = '-'\n",
    "\n",
    "    coord_mx['junction_coordinate1'] = None\n",
    "    coord_mx['junction_coordinate2'] = None\n",
    "\n",
    "\n",
    "    coord_mx = coord_mx.astype(str) # 7 min\n",
    "\n",
    "    coord_mx['+first'] = coord_mx[1] + ':' + coord_mx[2]\n",
    "    coord_mx['+secon'] = coord_mx[3] + ':' + coord_mx[4]\n",
    "    coord_mx['-first'] = coord_mx[3] + ':' + coord_mx[0]\n",
    "    coord_mx['-secon'] = coord_mx[5] + ':' + coord_mx[2]\n",
    "\n",
    "    coord_mx.loc[(coord_mx['strand'] == '+'), 'junction_coordinate1'] = coord_mx['+first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '-'), 'junction_coordinate1'] = coord_mx['-first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '+') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['+secon']\n",
    "    coord_mx.loc[(coord_mx['strand'] == '-') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['-secon']\n",
    "    \n",
    "    return coord_mx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af9d6620",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1218.0</td>\n",
       "      <td>689.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>2025.0</td>\n",
       "      <td>479.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>8641.0</td>\n",
       "      <td>6957.0</td>\n",
       "      <td>4789.0</td>\n",
       "      <td>3211.0</td>\n",
       "      <td>1347.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  RKSTQMPCT                 1218.0                   689.0   \n",
       "1  RKSTQMPCT                 1279.0                   274.0   \n",
       "2  RKSTQMPCT                 7350.0                  4622.0   \n",
       "3  RKSTQMPCT                 2025.0                   479.0   \n",
       "4  RKSTQMPCT                 8641.0                  6957.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   249.0                    89.0                    26.0   \n",
       "1                    70.0                    28.0                     5.0   \n",
       "2                  2653.0                  1745.0                  1126.0   \n",
       "3                    96.0                    35.0                     8.0   \n",
       "4                  4789.0                  3211.0                  1347.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92379857:92379859:92611313:92611338:None:None   \n",
       "1                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "2                    734.0  92379857:92379859:92611313:92611338:None:None   \n",
       "3                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "4                    191.0  92379857:92379859:92611313:92611338:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0               True               False  ...                       372   \n",
       "1               True               False  ...                       372   \n",
       "2               True               False  ...                       372   \n",
       "3               True               False  ...                       372   \n",
       "4               True               False  ...                       372   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                       367                       352   \n",
       "1                       367                       352   \n",
       "2                       367                       352   \n",
       "3                       367                       352   \n",
       "4                       367                       352   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "0                        295                  20.982616   \n",
       "1                        295                  20.982616   \n",
       "2                        295                  20.982616   \n",
       "3                        295                  20.982616   \n",
       "4                        295                  20.982616   \n",
       "\n",
       "   TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "0                   8.422932                  17.304198   \n",
       "1                   8.422932                  17.304198   \n",
       "2                   8.422932                  17.304198   \n",
       "3                   8.422932                  17.304198   \n",
       "4                   8.422932                  17.304198   \n",
       "\n",
       "   TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \n",
       "0                  16.069005                  19.585264          NaN  \n",
       "1                  16.069005                  19.585264          NaN  \n",
       "2                  16.069005                  19.585264          NaN  \n",
       "3                  16.069005                  19.585264          NaN  \n",
       "4                  16.069005                  19.585264          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load cancer generation matrix\n",
    "mx = pd.read_csv(big_matrix, sep = '\\t')\n",
    "display(mx.head())\n",
    "# Add split junction information to generation table\n",
    "coord_mx = explode_immunopepper_coord(mx)\n",
    "display(coord_mx.head())\n",
    "mx = pd.concat([mx, coord_mx[['strand', 'junction_coordinate1', 'junction_coordinate2']]], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75dfdf40",
   "metadata": {},
   "outputs": [],
   "source": [
    "# LOAD STAR junctions\n",
    "star_jx = pd.read_csv(star_jx, sep = '\\t')\n",
    "star_jx.head()\n",
    "# Add STAR junction column 10 min\n",
    "star_jx['junction_coordinate'] = star_jx['junction_start'].astype(str) + ':' + star_jx['junction_end'].astype(str)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc5a3849",
   "metadata": {},
   "source": [
    "# Check junction presence in STAR \n",
    "- 2-exons, 1st junction is present \n",
    "- 3 exons, needs first and second junction present "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "562e6b96",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Foreground Kmers from 1 junction\n",
    "one_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] == 'None')]\n",
    "print(one_jx.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "942231d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Foreground Kmers from 2 junctions\n",
    "two_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] != 'None')]\n",
    "print(two_jx.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15863551",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Foreground 1 junction - NOT IN STAR \n",
    "isstar_one = set(one_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # junction coordinates\n",
    "print(len(isstar_one))\n",
    "is_star_one_jx = set(one_jx.set_index('junction_coordinate1').loc[isstar_one, 'coord']) # corresonding kmer coordinates\n",
    "print(len(is_star_one_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "095f8c21",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Foreground 2 junctions - NOT IN STAR \n",
    "is_star_two1 = set(two_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "is_star_two2 = set(two_jx['junction_coordinate2']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "print(len(is_star_two1))\n",
    "print(len(is_star_two2))\n",
    "\n",
    "is_star_two_jx = set(two_jx.set_index('junction_coordinate1').loc[is_star_two1, 'coord']).intersection(\\\n",
    "set(two_jx.set_index('junction_coordinate2').loc[is_star_two2, 'coord'])) # Corresponding kmer coord\n",
    "print(len(is_star_two_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8770b37e",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(is_star_one_jx))\n",
    "print(len(is_star_two_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "327c77a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Foreground Table - Create FLAG GTEX junctions\n",
    "start_time = timeit.default_timer()\n",
    "mx['STAR_GTEx_jx'] = False\n",
    "mx = mx.set_index('coord')\n",
    "mx.loc[list(is_star_one_jx.union(is_star_two_jx)), 'STAR_GTEx_jx'] = True\n",
    "mx = mx.reset_index()\n",
    "print(timeit.default_timer() - start_time)    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1d2fc09",
   "metadata": {},
   "source": [
    "# Retrieve expression "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7851d93f",
   "metadata": {},
   "source": [
    "### Extract metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37de382f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select junctions to query in STAR file \n",
    "query = mx.loc[mx['STAR_GTEx_jx'] == True, ['strand', 'junction_coordinate1',\n",
    "                                            'junction_coordinate2', 'STAR_GTEx_jx']].drop_duplicates()\\\n",
    "                                                                                    .reset_index()\\\n",
    "                                                                                    .drop('index', axis = 1)\n",
    "\n",
    "assert(query.loc[:, ['junction_coordinate1', 'strand']].drop_duplicates().shape[0] == \\\n",
    "       query.loc[:, ['junction_coordinate1']].drop_duplicates().shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff4d910e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Query-Junction to strand \n",
    "jx_strand = {}\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate1']] = query.loc[i, 'strand']\n",
    "\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate2']] = query.loc[i, 'strand']\n",
    "\n",
    "jx_strand.pop('None')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5b64f47",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Chromosome to query-junction\n",
    "star_jx = star_jx.set_index('junction_coordinate') #Faster\n",
    "\n",
    "intermediate = star_jx.loc[jx_strand.keys(), :]\n",
    "\n",
    "start_time = timeit.default_timer()\n",
    "chr_jx = defaultdict(set)\n",
    "for jx, chrm in zip(intermediate.index, intermediate['chr']):\n",
    "    chr_jx[chrm].add(jx)\n",
    "print(timeit.default_timer() - start_time)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd209d62",
   "metadata": {},
   "source": [
    "### Collect Expression "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2de32fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_libsize_whitelist(libsize, whitelist):\n",
    "    # Read libsize and whitelist\n",
    "    libsize_normal = pd.read_csv(libsize, sep = '\\t')\n",
    "    whitelist_normal = pd.read_csv(whitelist, sep = '\\t', header = None)\n",
    "    whitelist_normal.columns = ['sample']\n",
    "    return libsize_normal, whitelist_normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f0c912e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def preprocess_STAR_projected(chrm, path_star, whitelist_normal, libsize_normal):\n",
    "    # Star junctions - projected coordinates and expression\n",
    "    projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "    star_expr = h5py.File(projected_chr_file, 'r')\n",
    "\n",
    "    # Whitelist samples\n",
    "    index_whitelist_samples = [s for s, sample in enumerate(star_expr['samples']) if sample.decode() + 'all' in whitelist_normal['sample'].values]\n",
    "    samples_decoded = [sample.decode() + 'all' for s, sample in enumerate(star_expr['samples']) if sample.decode() + 'all' in whitelist_normal['sample'].values]\n",
    "\n",
    "    # Libsize sampple     \n",
    "    lib_75_per_sample = libsize_normal.set_index('sample').loc[samples_decoded, 'libsize_75percent'].values\n",
    "\n",
    "    return star_expr, index_whitelist_samples, lib_75_per_sample\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37f3c8ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_junction_counts(junction_start, junction_end, \n",
    "                        chrm, strand, expression_h5, \n",
    "                        index_whitelist_samples, lib_75_per_sample, normalizer):\n",
    "    # Extract junction ID in hdf5 file #TODO SPEEDUP WITH SETS\n",
    "    jx_idx_h5 = np.where((junction_start == expression_h5[f'{chrm}:{strand}:junction_start'][...]) & \n",
    "            (junction_end == expression_h5[f'{chrm}:{strand}:junction_end'][...]))[0]\n",
    "    \n",
    "    if len(jx_idx_h5) > 0:\n",
    "        assert(len(jx_idx_h5) == 1) #TODO check critical\n",
    "        jx_idx_h5 = jx_idx_h5[0]\n",
    "\n",
    "        # Extract Raw count in hdf5 file\n",
    "        start_time = timeit.default_timer()    \n",
    "        raw_counts = expression_h5[f'{chrm}:{strand}:count'][jx_idx_h5, np.array(index_whitelist_samples)]\n",
    "        \n",
    "        print(start_time - timeit.default_timer(), 'raw counts')\n",
    "        \n",
    "        normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n",
    "    else:\n",
    "        normalized_counts = None\n",
    "        print(f'Error: No {chrm}:{strand}:junction_start {chrm}:{strand}:junction_start matching {junction_start}:{junction_end}')\n",
    "    return normalized_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25bb9667",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_recurrence(array_, threshold):\n",
    "    if threshold == 0:\n",
    "        return np.sum(array_ > threshold)\n",
    "    else:\n",
    "        return np.sum(array_ >= threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bee8537c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_multi_thresholds(normalized_counts, filter_thresholds):\n",
    "    recurrence = []\n",
    "    for threshold in filter_thresholds: # Make filter threshold\n",
    "        recurrence.append(filter_recurrence(normalized_counts, threshold))\n",
    "    return recurrence \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d6aaf7",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Thresholding experiments \n",
    "# Remark 1 junction can have multiple chrm --> Which \n",
    "# Remark 1 junction on 1 chrm can have multiple strands --> Take from Immunopepper\n",
    "\n",
    "res = []\n",
    "libsize_normal, whitelist_normal = read_libsize_whitelist(libsize, whitelist)\n",
    "counter = 0\n",
    "for chrm, jxS in chr_jx.items(): # Per chromosome\n",
    "    print(chrm)\n",
    "    # Expression file\n",
    "    expression_h5, index_whitelist_samples, lib_75_per_sample = preprocess_STAR_projected(chrm, \n",
    "                                                                                      path_star, \n",
    "                                                                                      whitelist_normal, \n",
    "                                                                                      libsize_normal)\n",
    "    start_time = timeit.default_timer()    \n",
    "    for jx in jxS: # Per junction\n",
    "        counter +=1\n",
    "        if counter % 500 == 0:\n",
    "            print(f'....{counter}')\n",
    "        junction_start = int(jx.split(':')[0])\n",
    "        junction_end = int(jx.split(':')[1])\n",
    "        normalized_counts = get_junction_counts(junction_start, junction_end, \n",
    "                                                chrm, jx_strand[jx], expression_h5, \n",
    "                                                index_whitelist_samples, \n",
    "                                                lib_75_per_sample, normalizer)\n",
    "        \n",
    "        if normalized_counts is not None:\n",
    "            metadata = [jx, jx_strand[jx], chrm] + filter_multi_thresholds(normalized_counts, filter_thresholds)\n",
    "        \n",
    "        res.append(metadata)\n",
    "    \n",
    "    expression_h5.close() \n",
    "    print(start_time - timeit.default_timer())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "861cb10d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrm chr9 number junctions 8778\n",
      "chrm chr14 number junctions 6377\n",
      "chrm chrX number junctions 5554\n",
      "chrm chr10 number junctions 7228\n",
      "chrm chr1 number junctions 22420\n",
      "chrm chr12 number junctions 11530\n",
      "chrm chr7 number junctions 9538\n",
      "chrm chr3 number junctions 11815\n",
      "chrm chr4 number junctions 5783\n",
      "chrm chr21 number junctions 2454\n",
      "chrm chr13 number junctions 2914\n",
      "chrm chr19 number junctions 18994\n",
      "chrm chr17 number junctions 14150\n",
      "chrm chr16 number junctions 10065\n",
      "chrm chr11 number junctions 13964\n",
      "chrm chr5 number junctions 7976\n",
      "chrm chr15 number junctions 5495\n",
      "chrm chr20 number junctions 6594\n",
      "chrm chr6 number junctions 9060\n",
      "chrm chr18 number junctions 2400\n",
      "chrm chr2 number junctions 13746\n",
      "chrm chr8 number junctions 7737\n",
      "chrm chr22 number junctions 5244\n",
      "chrm chrY number junctions 212\n",
      "chrm chrM number junctions 360\n",
      "Total number junctions 210388\n"
     ]
    }
   ],
   "source": [
    "# OPTIONAL \n",
    "tot = 0 \n",
    "for chrm, jxS in chr_jx.items():\n",
    "    print(f'chrm {chrm } number junctions {len(jxS)}')\n",
    "    tot+= len(jxS)\n",
    "print(f'Total number junctions {tot}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64a76faa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "83a4d816",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Expression to DF\n",
    "df_res = pd.DataFrame(res, columns = ['junction_coordinate', 'strand_STAR', 'chr'] + gtex_cols)\n",
    "print(df_res.shape)\n",
    "display(df_res.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fabc76e4",
   "metadata": {},
   "source": [
    "### Format results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "80c24899",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_cols = ['gtexCohortfilter >0.0', 'gtexCohortfilter >=1.0',\n",
    "       'gtexCohortfilter >=2.0', 'gtexCohortfilter >=3.0',\n",
    "       'gtexCohortfilter >=5.0', 'gtexCohortfilter >=10.0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "0cb41d1c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(44293227, 27)\n",
      "(2431499, 21)\n",
      "Index(['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated',\n",
      "       'isCrossJunction', 'batch', 'cancerCohortfilter >0.0',\n",
      "       'cancerCohortfilter >=1.0', 'cancerCohortfilter >=2.0',\n",
      "       'cancerCohortfilter >=3.0', 'cancerCohortfilter >=5.0',\n",
      "       'cancerCohortfilter >=10.0', 'TCGA25131901A01R156513all',\n",
      "       'TCGA25131301A01R156513all', 'TCGA61200801A02R156813all',\n",
      "       'TCGA24143101A01R156613all', 'TCGA24229801A01R156913all', 'isAnnotated',\n",
      "       'strand', 'junction_coordinate1', 'junction_coordinate2',\n",
      "       'junction_coordinate', 'strand_STAR', 'chr', 'gtexCohortfilter >0.0',\n",
      "       'gtexCohortfilter >=1.0', 'gtexCohortfilter >=2.0',\n",
      "       'gtexCohortfilter >=3.0', 'gtexCohortfilter >=5.0',\n",
      "       'gtexCohortfilter >=10.0'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2431504</th>\n",
       "      <td>ETRSPEKSV</td>\n",
       "      <td>121185317:121185337:121185025:121185032:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>265</td>\n",
       "      <td>258</td>\n",
       "      <td>222</td>\n",
       "      <td>183</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431505</th>\n",
       "      <td>SFKDRHPSY</td>\n",
       "      <td>121183457:121183471:121183340:121183353:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>63</td>\n",
       "      <td>35</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431506</th>\n",
       "      <td>KDSAVDFTG</td>\n",
       "      <td>121183457:121183465:121181329:121181348:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>186</td>\n",
       "      <td>136</td>\n",
       "      <td>68</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431507</th>\n",
       "      <td>WLLLLSLFL</td>\n",
       "      <td>121181270:121181272:121180855:121180880:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>44</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431508</th>\n",
       "      <td>GGETRSPEN</td>\n",
       "      <td>121185317:121185343:121181347:121181348:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>78</td>\n",
       "      <td>36</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              kmer                                              coord  \\\n",
       "2431504  ETRSPEKSV  121185317:121185337:121185025:121185032:None:None   \n",
       "2431505  SFKDRHPSY  121183457:121183471:121183340:121183353:None:None   \n",
       "2431506  KDSAVDFTG  121183457:121183465:121181329:121181348:None:None   \n",
       "2431507  WLLLLSLFL  121181270:121181272:121180855:121180880:None:None   \n",
       "2431508  GGETRSPEN  121185317:121185343:121181347:121181348:None:None   \n",
       "\n",
       "         junctionAnnotated  readFrameAnnotated  isCrossJunction  batch  \\\n",
       "2431504              False                True             True   2858   \n",
       "2431505              False                True             True   2858   \n",
       "2431506               True                True             True   2858   \n",
       "2431507              False               False             True   2858   \n",
       "2431508               True                True             True   2858   \n",
       "\n",
       "         cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "2431504                      265                       258   \n",
       "2431505                       63                        35   \n",
       "2431506                      186                       136   \n",
       "2431507                       44                        28   \n",
       "2431508                       78                        36   \n",
       "\n",
       "         cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "2431504                       222                       183  ...   \n",
       "2431505                        12                         5  ...   \n",
       "2431506                        68                        31  ...   \n",
       "2431507                        10                         4  ...   \n",
       "2431508                        12                         4  ...   \n",
       "\n",
       "         junction_coordinate2  junction_coordinate  strand_STAR  chr  \\\n",
       "2431504                  None                  NaN          NaN  NaN   \n",
       "2431505                  None                  NaN          NaN  NaN   \n",
       "2431506                  None                  NaN          NaN  NaN   \n",
       "2431507                  None                  NaN          NaN  NaN   \n",
       "2431508                  None                  NaN          NaN  NaN   \n",
       "\n",
       "         gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "2431504                    NaN                     NaN   \n",
       "2431505                    NaN                     NaN   \n",
       "2431506                    NaN                     NaN   \n",
       "2431507                    NaN                     NaN   \n",
       "2431508                    NaN                     NaN   \n",
       "\n",
       "         gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  \\\n",
       "2431504                     NaN                     NaN   \n",
       "2431505                     NaN                     NaN   \n",
       "2431506                     NaN                     NaN   \n",
       "2431507                     NaN                     NaN   \n",
       "2431508                     NaN                     NaN   \n",
       "\n",
       "        gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "2431504                    NaN                     NaN  \n",
       "2431505                    NaN                     NaN  \n",
       "2431506                    NaN                     NaN  \n",
       "2431507                    NaN                     NaN  \n",
       "2431508                    NaN                     NaN  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# One junction merge (straightforward merge on junction)\n",
    "print(one_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_one_jx = one_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_one_jx.head()\n",
    "print(base_one_jx.shape)\n",
    "# Add GTEX quantifications from STAR\n",
    "one_jx_quantified = base_one_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left') \n",
    "print(one_jx_quantified.columns)\n",
    "display(one_jx_quantified.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "08c89de9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1839568, 27)\n",
      "(188025, 21)\n",
      "Index(['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated',\n",
      "       'isCrossJunction', 'batch', 'cancerCohortfilter >0.0',\n",
      "       'cancerCohortfilter >=1.0', 'cancerCohortfilter >=2.0',\n",
      "       'cancerCohortfilter >=3.0', 'cancerCohortfilter >=5.0',\n",
      "       'cancerCohortfilter >=10.0', 'TCGA25131901A01R156513all',\n",
      "       'TCGA25131301A01R156513all', 'TCGA61200801A02R156813all',\n",
      "       'TCGA24143101A01R156613all', 'TCGA24229801A01R156913all', 'isAnnotated',\n",
      "       'strand', 'junction_coordinate1', 'junction_coordinate2',\n",
      "       'junction_coordinate', 'strand_STAR', 'chr', 'gtexCohortfilter >0.0',\n",
      "       'gtexCohortfilter >=1.0', 'gtexCohortfilter >=2.0',\n",
      "       'gtexCohortfilter >=3.0', 'gtexCohortfilter >=5.0',\n",
      "       'gtexCohortfilter >=10.0'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>196939</th>\n",
       "      <td>QSPYPCHWW</td>\n",
       "      <td>154464496:154464501:154464774:154464794:154465...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>59971</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>154464794:154465017</td>\n",
       "      <td>154464794:154465017</td>\n",
       "      <td>+</td>\n",
       "      <td>chrX</td>\n",
       "      <td>1398.0</td>\n",
       "      <td>1358.0</td>\n",
       "      <td>757.0</td>\n",
       "      <td>443.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196940</th>\n",
       "      <td>KLLHKFLKE</td>\n",
       "      <td>154468542:154468545:154468708:154468729:154468...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>59971</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>154468729:154468822</td>\n",
       "      <td>154468729:154468822</td>\n",
       "      <td>+</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9447.0</td>\n",
       "      <td>9447.0</td>\n",
       "      <td>9446.0</td>\n",
       "      <td>9438.0</td>\n",
       "      <td>9400.0</td>\n",
       "      <td>9280.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196941</th>\n",
       "      <td>HTKTMQVIS</td>\n",
       "      <td>34609637:34609646:34608029:34608046:34605716:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196942</th>\n",
       "      <td>TKTMQVISF</td>\n",
       "      <td>34609637:34609643:34608029:34608046:34605713:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196943</th>\n",
       "      <td>KTMQVISFH</td>\n",
       "      <td>34609637:34609640:34608029:34608046:34605710:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer                                              coord  \\\n",
       "196939  QSPYPCHWW  154464496:154464501:154464774:154464794:154465...   \n",
       "196940  KLLHKFLKE  154468542:154468545:154468708:154468729:154468...   \n",
       "196941  HTKTMQVIS  34609637:34609646:34608029:34608046:34605716:3...   \n",
       "196942  TKTMQVISF  34609637:34609643:34608029:34608046:34605713:3...   \n",
       "196943  KTMQVISFH  34609637:34609640:34608029:34608046:34605710:3...   \n",
       "\n",
       "       junctionAnnotated readFrameAnnotated isCrossJunction  batch  \\\n",
       "196939             False              False            True  59971   \n",
       "196940              True              False            True  59971   \n",
       "196941              True              False            True  16207   \n",
       "196942              True              False            True  16207   \n",
       "196943              True              False            True  16207   \n",
       "\n",
       "        cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "196939                       14                         5   \n",
       "196940                       31                        12   \n",
       "196941                       13                         9   \n",
       "196942                       13                         9   \n",
       "196943                       13                         9   \n",
       "\n",
       "        cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "196939                         1                         0  ...   \n",
       "196940                         3                         1  ...   \n",
       "196941                         4                         0  ...   \n",
       "196942                         4                         0  ...   \n",
       "196943                         4                         0  ...   \n",
       "\n",
       "        junction_coordinate2  junction_coordinate  strand_STAR    chr  \\\n",
       "196939   154464794:154465017  154464794:154465017            +   chrX   \n",
       "196940   154468729:154468822  154468729:154468822            +   chrX   \n",
       "196941     34605717:34608029    34605717:34608029            -  chr14   \n",
       "196942     34605717:34608029    34605717:34608029            -  chr14   \n",
       "196943     34605717:34608029    34605717:34608029            -  chr14   \n",
       "\n",
       "        gtexCohortfilter >0.0  gtexCohortfilter >=1.0  gtexCohortfilter >=2.0  \\\n",
       "196939                 1398.0                  1358.0                   757.0   \n",
       "196940                 9447.0                  9447.0                  9446.0   \n",
       "196941                 9475.0                  9475.0                  9475.0   \n",
       "196942                 9475.0                  9475.0                  9475.0   \n",
       "196943                 9475.0                  9475.0                  9475.0   \n",
       "\n",
       "        gtexCohortfilter >=3.0 gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "196939                   443.0                  125.0                     7.0  \n",
       "196940                  9438.0                 9400.0                  9280.0  \n",
       "196941                  9475.0                 9475.0                  9473.0  \n",
       "196942                  9475.0                 9475.0                  9473.0  \n",
       "196943                  9475.0                 9475.0                  9473.0  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Two junctions merge (not straight forwrds, consider botrh junctions separately and take the max)\n",
    "print(two_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_two_jx = two_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_two_jx.head()\n",
    "print(base_two_jx.shape)\n",
    "# Add GTEX quantifications from STAR  # first junction\n",
    "two_jx_quantified_left = base_two_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "\n",
    "# Add GTEX quantifications from STAR  # second junction\n",
    "two_jx_quantified_right = base_two_jx.merge(df_res, left_on = 'junction_coordinate2', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "col_merge = [col for col in two_jx_quantified_right if col not in gtex_cols]\n",
    "\n",
    "# Add GTEX quantifications from STAR  # both junctions\n",
    "two_jx_quantified = two_jx_quantified_left.merge(two_jx_quantified_right, on = col_merge, how = 'outer')\n",
    "\n",
    "# Add GTEX quantifications from STAR  # MAX (!!) over 2 junctions\n",
    "for col in gtex_cols:\n",
    "    two_jx_quantified[col] = two_jx_quantified[[col + '_x', col + '_y']].max(skipna = True, axis = 1)\n",
    "    two_jx_quantified = two_jx_quantified.drop([col + '_x', col + '_y'], axis = 1)\n",
    "    \n",
    "print(two_jx_quantified.columns)\n",
    "display(two_jx_quantified.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "36d94add",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2628453, 30)\n"
     ]
    }
   ],
   "source": [
    "# All Junctions quantified on STAR for GTEX\n",
    "print(two_jx_quantified.shape)\n",
    "print(one_jx_quantified.shape)\n",
    "jx_quantified = pd.concat([one_jx_quantified, two_jx_quantified])\n",
    "print(jx_quantified.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "13c77bfb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r_complete.STAR_GTEx.tsv.gz\n"
     ]
    }
   ],
   "source": [
    "### Save \n",
    "save_path = big_matrix.replace('tsv.gz', 'STAR_GTEx.tsv.gz')\n",
    "print(save_path)\n",
    "jx_quantified.to_csv(save_path, compression = 'gzip', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bad6a49",
   "metadata": {},
   "source": [
    "# Quality check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "5adeadd8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/slurm-job.1197223/ipykernel_2447348/1028733944.py:13: RuntimeWarning: invalid value encountered in true_divide\n",
      "  normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "4\n",
      "4\n",
      "4\n",
      "4\n",
      "2\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "chrm_list = ['chr2', 'chr14']\n",
    "strand = '-'\n",
    "jx = '102015975:102016000'\n",
    "junction_start = int(jx.split(':')[0])\n",
    "junction_end = int(jx.split(':')[1])\n",
    "\n",
    "for chrm in chrm_list:\n",
    "    expression_h5, index_whitelist_samples, lib_75_per_sample = preprocess_STAR_projected(chrm, \n",
    "                                                                                      path_star, \n",
    "                                                                                      whitelist_normal, \n",
    "                                                                                      libsize_normal)\n",
    "\n",
    "    normalized_counts = get_junction_counts(junction_start, junction_end, \n",
    "                                            chrm, strand, expression_h5, \n",
    "                                            index_whitelist_samples, \n",
    "                                            lib_75_per_sample, normalizer)\n",
    "    if normalized_counts is not None:\n",
    "        for threshold in filter_thresholds: # Make filter threshold\n",
    "            print(filter_recurrence(normalized_counts, threshold))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "325fdff7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_start</th>\n",
       "      <th>junction_end</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19214000</th>\n",
       "      <td>chr14</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "      <td>102015975:102016000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46460381</th>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "      <td>102015975:102016000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            chr strand  junction_start  junction_end  junction_coordinate\n",
       "19214000  chr14      -       102015975     102016000  102015975:102016000\n",
       "46460381   chr2      -       102015975     102016000  102015975:102016000"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "star_jx.loc[(star_jx['junction_start'] == junction_start) & (star_jx['junction_end'] == junction_end)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20cae9c6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
