{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4d22d9c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "import timeit\n",
    "import numpy as np\n",
    "import h5py\n",
    "from collections import defaultdict\n",
    "from helpers_STAR import * \n",
    "import argparse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9f4c948",
   "metadata": {},
   "outputs": [],
   "source": [
    "##### MAIN #####\n",
    "if __name__ == \"__main__\":\n",
    "    parser = argparse.ArgumentParser(description='run specifications')\n",
    "    parser.add_argument('--path-star', type=str, required=True, help='folder with star junctions')\n",
    "    parser.add_argument('--big-matrix', type=str, required=False, default=None, help='Intermediate filtering matrix')\n",
    "    parser.add_argument('--whitelist', type=str, required=True, help='whitelist of samples to include in recurrence analysis')\n",
    "    parser.add_argument('--libsize', type=str, required=True, help='library size file')\n",
    "    parser.add_argument('--normalizer', type=int, required=True, help='value to rescale after normalisation')\n",
    "    parser.add_argument('--filter-thresholds', nargs='+', type=float, required=False, default=None, \n",
    "                        help='Number of reads to use as expression threshold in recurrence analysis')\n",
    "    parser.add_argument('--jx-target-list', type=str, required=True, help='path to file with whitelist kmers coordinates')\n",
    "    args = parser.parse_args()\n",
    "\n",
    "    print(args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54e9b4c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# -------- Path STAR\n",
    "path_star = arg.path_star\n",
    "# Star junctions - unique coordinates\n",
    "star_jx = os.path.join(path_star, 'junctions_spladder.all_coords.sorted.uniq.tsv.gz')\n",
    "# Star junctions - projected coordinates and expression\n",
    "#projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "\n",
    "\n",
    "# -------- Intermediate filtering results (threshold and merged)\n",
    "#TODO DO the generation matrix???\n",
    "# Foreground matrix\n",
    "big_matrix = arg.big_matrix\n",
    "\n",
    "\n",
    "\n",
    "# -------- GTEX filtering \n",
    "whitelist = arg.whitelist\n",
    "libsize = arg.libsize\n",
    "\n",
    "\n",
    "normalizer = ar.normalizer\n",
    "filter_thresholds = arg.filter_thresholds\n",
    "#Optional if needs to process a junction list\n",
    "jx_target_list = arg.jx_target_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61e69719",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1218.0</td>\n",
       "      <td>689.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>2025.0</td>\n",
       "      <td>479.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>8641.0</td>\n",
       "      <td>6957.0</td>\n",
       "      <td>4789.0</td>\n",
       "      <td>3211.0</td>\n",
       "      <td>1347.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  RKSTQMPCT                 1218.0                   689.0   \n",
       "1  RKSTQMPCT                 1279.0                   274.0   \n",
       "2  RKSTQMPCT                 7350.0                  4622.0   \n",
       "3  RKSTQMPCT                 2025.0                   479.0   \n",
       "4  RKSTQMPCT                 8641.0                  6957.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   249.0                    89.0                    26.0   \n",
       "1                    70.0                    28.0                     5.0   \n",
       "2                  2653.0                  1745.0                  1126.0   \n",
       "3                    96.0                    35.0                     8.0   \n",
       "4                  4789.0                  3211.0                  1347.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92379857:92379859:92611313:92611338:None:None   \n",
       "1                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "2                    734.0  92379857:92379859:92611313:92611338:None:None   \n",
       "3                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "4                    191.0  92379857:92379859:92611313:92611338:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0               True               False  ...                       372   \n",
       "1               True               False  ...                       372   \n",
       "2               True               False  ...                       372   \n",
       "3               True               False  ...                       372   \n",
       "4               True               False  ...                       372   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                       367                       352   \n",
       "1                       367                       352   \n",
       "2                       367                       352   \n",
       "3                       367                       352   \n",
       "4                       367                       352   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "0                        295                  20.982616   \n",
       "1                        295                  20.982616   \n",
       "2                        295                  20.982616   \n",
       "3                        295                  20.982616   \n",
       "4                        295                  20.982616   \n",
       "\n",
       "   TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "0                   8.422932                  17.304198   \n",
       "1                   8.422932                  17.304198   \n",
       "2                   8.422932                  17.304198   \n",
       "3                   8.422932                  17.304198   \n",
       "4                   8.422932                  17.304198   \n",
       "\n",
       "   TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \n",
       "0                  16.069005                  19.585264          NaN  \n",
       "1                  16.069005                  19.585264          NaN  \n",
       "2                  16.069005                  19.585264          NaN  \n",
       "3                  16.069005                  19.585264          NaN  \n",
       "4                  16.069005                  19.585264          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate1</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>+first</th>\n",
       "      <th>+secon</th>\n",
       "      <th>-first</th>\n",
       "      <th>-secon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0         1         2         3     4     5 strand  \\\n",
       "0  92379857  92379859  92611313  92611338  None  None      +   \n",
       "1  92379857  92379859  92611313  92611338  None  None      +   \n",
       "2  92379857  92379859  92611313  92611338  None  None      +   \n",
       "3  92379857  92379859  92611313  92611338  None  None      +   \n",
       "4  92379857  92379859  92611313  92611338  None  None      +   \n",
       "\n",
       "  junction_coordinate1 junction_coordinate2             +first         +secon  \\\n",
       "0    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "1    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "2    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "3    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "4    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "\n",
       "              -first         -secon  \n",
       "0  92611338:92379857  None:92611313  \n",
       "1  92611338:92379857  None:92611313  \n",
       "2  92611338:92379857  None:92611313  \n",
       "3  92611338:92379857  None:92611313  \n",
       "4  92611338:92379857  None:92611313  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "foreground matrix shape (46132795, 27)\n",
      "foreground matrix shape (542295, 27)\n",
      "foreground matrix one junction (406600, 27)\n",
      "foreground matrix two junctions (135695, 27)\n",
      "foreground matrix one junction is in star 19382\n",
      "-> foreground matrix one-junction-kmers are in star 162279\n",
      "foreground matrix two junctions: left junction is in star 9809\n",
      "foreground matrix two junctions: right junction is in star 9876\n",
      "-> foreground matrix two-junctions-kmers are in star 62925\n",
      "time create STAR junction presence flag 1.2987702080281451\n",
      "Number of query junctions: 26460\n",
      "time Query-junction to strand 0.6875324179418385\n",
      "chromosomes involved dict_keys(['chr1', 'chr12', 'chr13', 'chr19', 'chrX', 'chr10', 'chr9', 'chr3', 'chr7', 'chr17', 'chr16', 'chr8', 'chr5', 'chr11', 'chr6', 'chr4', 'chr22', 'chr18', 'chr14', 'chr20', 'chr15', 'chr2', 'chrY', 'chr21', 'chrM'])\n",
      "Chromosome to query-junction 71.80031096201856\n",
      "Start collect expression\n",
      "chr1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cluster/home/prelotla/github/projects2020_ohsu/eth/debug/python_pipeline_filter_draft/helpers_STAR.py:79: RuntimeWarning: invalid value encountered in true_divide\n",
      "  normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "....500\n",
      "....1000\n",
      "....1500\n",
      "....2000\n",
      "Error: No chr1:-:junction_start chr1:-:junction_start matching 155191880:155191939\n",
      "Error: No chr1:-:junction_start chr1:-:junction_start matching 156742873:156742958\n",
      "-8823.614046026953\n",
      "chr12\n",
      "....2500\n",
      "....3000\n",
      "Error: No chr12:+:junction_start chr12:+:junction_start matching 6537374:6537655\n",
      "Error: No chr12:+:junction_start chr12:+:junction_start matching 6537708:6537870\n",
      "Error: No chr12:+:junction_start chr12:+:junction_start matching 6770471:6770688\n",
      "Error: No chr12:+:junction_start chr12:+:junction_start matching 6937330:6937507\n",
      "....3500\n",
      "....4000\n",
      "-4295.845152077964\n",
      "chr13\n",
      "-486.2390420080628\n",
      "chr19\n",
      "....4500\n",
      "....5000\n",
      "Error: No chr19:+:junction_start chr19:+:junction_start matching 54207433:54207582\n",
      "....5500\n",
      "....6000\n",
      "....6500\n",
      "....7000\n",
      "Error: No chr19:-:junction_start chr19:-:junction_start matching 1990046:1990163\n",
      "....7500\n",
      "....8000\n",
      "....8500\n",
      "Error: No chr19:+:junction_start chr19:+:junction_start matching 49173064:49173104\n",
      "Error: No chr19:+:junction_start chr19:+:junction_start matching 10830162:10830207\n",
      "-14435.021114993026\n",
      "chrX\n"
     ]
    }
   ],
   "source": [
    "# -------- Load -------- \n",
    "\n",
    "# Load cancer generation matrix\n",
    "mx = pd.read_csv(big_matrix, sep = '\\t')\n",
    "display(mx.head())\n",
    "# Add split junction information to generation table\n",
    "coord_mx = explode_immunopepper_coord(mx)\n",
    "display(coord_mx.head())\n",
    "mx = pd.concat([mx, coord_mx[['strand', 'junction_coordinate1', 'junction_coordinate2']]], axis = 1)\n",
    "print('foreground matrix shape', mx.shape)\n",
    "\n",
    "if jx_target_list:\n",
    "    target_jx = pd.read_csv(jx_target_list, header = None) \n",
    "    target_jx.columns = ['coord']\n",
    "    mx = mx.merge(target_jx, on = 'coord', how = 'inner')\n",
    "print('foreground matrix shape', mx.shape)   \n",
    "    \n",
    "# LOAD STAR junctions\n",
    "star_jx = pd.read_csv(star_jx, sep = '\\t')\n",
    "star_jx.head()\n",
    "# Add STAR junction column 10 min\n",
    "star_jx['junction_coordinate'] = star_jx['junction_start'].astype(str) + ':' + star_jx['junction_end'].astype(str)\n",
    "\n",
    "\n",
    "# --------  Check junction presence in STAR --------  \n",
    "#- 2-exons, 1st junction is present \n",
    "#- 3 exons, needs first and second junction present \n",
    "\n",
    "# Foreground Kmers from 1 junction\n",
    "one_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] == 'None')]\n",
    "print('foreground matrix one junction', one_jx.shape)\n",
    "\n",
    "# Foreground Kmers from 2 junctions\n",
    "two_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] != 'None')]\n",
    "print('foreground matrix two junctions', two_jx.shape)\n",
    "\n",
    "# Foreground 1 junction - NOT IN STAR \n",
    "isstar_one = set(one_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # junction coordinates\n",
    "print('foreground matrix one junction is in star', len(isstar_one))\n",
    "is_star_one_jx = set(one_jx.set_index('junction_coordinate1').loc[isstar_one, 'coord']) # corresonding kmer coordinates\n",
    "print('-> foreground matrix one-junction-kmers are in star', len(is_star_one_jx))\n",
    "\n",
    "# Foreground 2 junctions - NOT IN STAR \n",
    "is_star_two1 = set(two_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "is_star_two2 = set(two_jx['junction_coordinate2']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "print('foreground matrix two junctions: left junction is in star', len(is_star_two1))\n",
    "print('foreground matrix two junctions: right junction is in star', len(is_star_two2))\n",
    "\n",
    "is_star_two_jx = set(two_jx.set_index('junction_coordinate1').loc[is_star_two1, 'coord']).intersection(\\\n",
    "set(two_jx.set_index('junction_coordinate2').loc[is_star_two2, 'coord'])) # Corresponding kmer coord\n",
    "print('-> foreground matrix two-junctions-kmers are in star', len(is_star_two_jx))\n",
    "\n",
    "\n",
    "# Foreground Table - Create FLAG GTEX junctions\n",
    "start_time = timeit.default_timer()\n",
    "mx['STAR_GTEx_jx'] = False\n",
    "mx = mx.set_index('coord')\n",
    "mx.loc[list(is_star_one_jx.union(is_star_two_jx)), 'STAR_GTEx_jx'] = True\n",
    "mx = mx.reset_index()\n",
    "print('time create STAR junction presence flag', timeit.default_timer() - start_time)    \n",
    "\n",
    "# --------  Retrieve expression -------- \n",
    "\n",
    "# Extract metadata\n",
    "\n",
    "# Select junctions to query in STAR file \n",
    "query = mx.loc[mx['STAR_GTEx_jx'] == True, ['strand', 'junction_coordinate1',\n",
    "                                            'junction_coordinate2', 'STAR_GTEx_jx']].drop_duplicates()\\\n",
    "                                                                                    .reset_index()\\\n",
    "                                                                                    .drop('index', axis = 1)\n",
    "\n",
    "assert(query.loc[:, ['junction_coordinate1', 'strand']].drop_duplicates().shape[0] == \\\n",
    "       query.loc[:, ['junction_coordinate1']].drop_duplicates().shape[0])\n",
    "\n",
    "# Query-Junction to strand \n",
    "start_time = timeit.default_timer()\n",
    "jx_strand = {}\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate1']] = query.loc[i, 'strand']\n",
    "\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate2']] = query.loc[i, 'strand']\n",
    "\n",
    "jx_strand.pop('None')\n",
    "print('Number of query junctions:', len(jx_strand))\n",
    "print('time Query-junction to strand', timeit.default_timer() - start_time)\n",
    "\n",
    "# Chromosome to query-junction\n",
    "start_time = timeit.default_timer()\n",
    "star_jx = star_jx.set_index('junction_coordinate') #Faster\n",
    "intermediate = star_jx.loc[jx_strand.keys(), :]\n",
    "\n",
    "chr_jx = defaultdict(set)\n",
    "for jx, chrm in zip(intermediate.index, intermediate['chr']):\n",
    "    chr_jx[chrm].add(jx)\n",
    "print('chromosomes involved', chr_jx.keys())\n",
    "print('Chromosome to query-junction', timeit.default_timer() - start_time)\n",
    "    \n",
    "\n",
    "### Collect Expression \n",
    "\n",
    "# -------- Thresholding experiments -------- \n",
    "# Remark 1 junction can have multiple chrm --> Which \n",
    "# Remark 1 junction on 1 chrm can have multiple strands --> Take from Immunopepper\n",
    "\n",
    "print('Start collect expression')\n",
    "res = collect_expression_thresholds(libsize, whitelist, normalizer,\n",
    "                                    filter_thresholds, path_star, chr_jx, jx_strand )\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "da2233b6",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'jx_strand' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [1], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28mlen\u001b[39m(\u001b[43mjx_strand\u001b[49m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'jx_strand' is not defined"
     ]
    }
   ],
   "source": [
    "len(jx_strand)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65417179",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_cols = ['gtexCohortfilter >0.0', 'gtexCohortfilter >=1.0',\n",
    "       'gtexCohortfilter >=2.0', 'gtexCohortfilter >=3.0',\n",
    "       'gtexCohortfilter >=5.0', 'gtexCohortfilter >=10.0']\n",
    "\n",
    "# Expression to DF\n",
    "df_res = pd.DataFrame(res, columns = ['junction_coordinate', 'strand_STAR', 'chr'] + gtex_cols)\n",
    "print('Junction - expressin df shape', df_res.shape)\n",
    "display(df_res.head())\n",
    "\n",
    "#-------- Format results -------- \n",
    "\n",
    "\n",
    "\n",
    "# One junction merge (straightforward merge on junction)\n",
    "print('foreground matrix one junction - BIS', one_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_one_jx = one_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_one_jx.head()\n",
    "print('foreground matrix one junction no GTEx', base_one_jx.shape)\n",
    "# Add GTEX quantifications from STAR\n",
    "one_jx_quantified = base_one_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left') \n",
    "print(one_jx_quantified.columns)\n",
    "display(one_jx_quantified.tail())\n",
    "\n",
    "# Two junctions merge (not straight forwrds, consider botrh junctions separately and take the max)\n",
    "print('foreground matrix two junctions - BIS', two_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_two_jx = two_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_two_jx.head()\n",
    "print('foreground matrix two junctions no GTEx', base_two_jx.shape)\n",
    "# Add GTEX quantifications from STAR  # first junction\n",
    "two_jx_quantified_left = base_two_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "\n",
    "# Add GTEX quantifications from STAR  # second junction\n",
    "two_jx_quantified_right = base_two_jx.merge(df_res, left_on = 'junction_coordinate2', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "col_merge = [col for col in two_jx_quantified_right if col not in gtex_cols]\n",
    "\n",
    "# Add GTEX quantifications from STAR  # both junctions\n",
    "two_jx_quantified = two_jx_quantified_left.merge(two_jx_quantified_right, on = col_merge, how = 'outer')\n",
    "\n",
    "# Add GTEX quantifications from STAR  # MAX (!!) over 2 junctions\n",
    "for col in gtex_cols:\n",
    "    two_jx_quantified[col] = two_jx_quantified[[col + '_x', col + '_y']].max(skipna = True, axis = 1)\n",
    "    two_jx_quantified = two_jx_quantified.drop([col + '_x', col + '_y'], axis = 1)\n",
    "    \n",
    "print(two_jx_quantified.columns)\n",
    "display(two_jx_quantified.tail())\n",
    "\n",
    "# All Junctions quantified on STAR for GTEX\n",
    "print('foreground matrix one junction merged with expression', one_jx_quantified.shape)\n",
    "print('foreground matrix two junctions merged with expression', two_jx_quantified.shape)\n",
    "\n",
    "jx_quantified = pd.concat([one_jx_quantified, two_jx_quantified])\n",
    "print('foreground matrix all merged with expression', jx_quantified.shape)\n",
    "\n",
    "### Save \n",
    "save_path = big_matrix.replace('tsv.gz', 'STAR_GTEx.tsv.gz')\n",
    "print('Saving to', save_path)\n",
    "jx_quantified.to_csv(save_path, compression = 'gzip', index = False, sep = '\\t')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "940f9e55",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ed5ab17",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ba766e0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # -------- Path STAR\n",
    "# path_star = '/cluster/work/grlab/projects/GTEx/rna_gencode32_realign/results'\n",
    "# # Star junctions - unique coordinates\n",
    "# star_jx = os.path.join(path_star, 'junctions_spladder.all_coords.sorted.uniq.tsv.gz')\n",
    "# # Star junctions - projected coordinates and expression\n",
    "# #projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "\n",
    "\n",
    "# # -------- Intermediate filtering results (threshold and merged)\n",
    "# #TODO DO the generation matrix???\n",
    "# # Foreground matrix\n",
    "# big_matrix = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r_complete.tsv.gz'\n",
    "\n",
    "\n",
    "\n",
    "# # -------- GTEX filtering \n",
    "# whitelist = '/cluster/work/grlab/projects/projects2020_OHSU/sample_lists/GTEX/GTEx_sample_IDs_10-2021_lib_graph_juliannelist'\n",
    "# libsize = '/cluster/work/grlab/projects/TCGA/PanCanAtlas/immunopepper_paper/peptides_ccell_rerun_gtex_151220/ARCHIV_keep_runs/GTEX2019_commit_v3_TEST_merged3_372a147_medium_run_pya.0.17.1_conf2_annot_ref_chrall_cap/expression_counts.libsize.tsv'\n",
    "\n",
    "\n",
    "# normalizer = 400000\n",
    "# filter_thresholds = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "# #Optional if needs to process a junction list\n",
    "# jx_target_list = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/filters_22March_order_wany_wAnnot/tmp_all_experiments_jx.txt'"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
