{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4d22d9c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "import timeit\n",
    "import numpy as np\n",
    "import h5py\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ba766e0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# -------- Path STAR\n",
    "path_star = '/cluster/work/grlab/projects/GTEx/rna_gencode32_realign/results'\n",
    "# Star junctions - unique coordinates\n",
    "star_jx = os.path.join(path_star, 'junctions_spladder.all_coords.sorted.uniq.tsv.gz')\n",
    "# Star junctions - projected coordinates and expression\n",
    "#projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "\n",
    "\n",
    "# -------- Intermediate filtering results (threshold and merged)\n",
    "#TODO DO the generation matrix???\n",
    "# Foreground matrix\n",
    "big_matrix = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r_complete.tsv.gz'\n",
    "\n",
    "\n",
    "\n",
    "# -------- GTEX filtering \n",
    "whitelist = '/cluster/work/grlab/projects/projects2020_OHSU/sample_lists/GTEX/GTEx_sample_IDs_10-2021_lib_graph_juliannelist'\n",
    "libsize = '/cluster/work/grlab/projects/TCGA/PanCanAtlas/immunopepper_paper/peptides_ccell_rerun_gtex_151220/ARCHIV_keep_runs/GTEX2019_commit_v3_TEST_merged3_372a147_medium_run_pya.0.17.1_conf2_annot_ref_chrall_cap/expression_counts.libsize.tsv'\n",
    "\n",
    "\n",
    "normalizer = 400000\n",
    "filter_thresholds = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1c3c201",
   "metadata": {},
   "source": [
    "# Load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "63b89a2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def explode_immunopepper_coord(mx):\n",
    "    coord_mx = mx['coord'].str.split(':', expand=True) #7 min\n",
    "\n",
    "    coord_mx[1] = coord_mx[1].astype('int')\n",
    "    coord_mx[2] = coord_mx[2].astype('int')\n",
    "\n",
    "    coord_mx['strand'] = None\n",
    "    coord_mx.loc[coord_mx[1] < coord_mx[2], 'strand'] = '+'\n",
    "    coord_mx.loc[coord_mx[1] > coord_mx[2], 'strand'] = '-'\n",
    "\n",
    "    coord_mx['junction_coordinate1'] = None\n",
    "    coord_mx['junction_coordinate2'] = None\n",
    "\n",
    "\n",
    "    coord_mx = coord_mx.astype(str) # 7 min\n",
    "\n",
    "    coord_mx['+first'] = coord_mx[1] + ':' + coord_mx[2]\n",
    "    coord_mx['+secon'] = coord_mx[3] + ':' + coord_mx[4]\n",
    "    coord_mx['-first'] = coord_mx[3] + ':' + coord_mx[0]\n",
    "    coord_mx['-secon'] = coord_mx[5] + ':' + coord_mx[2]\n",
    "\n",
    "    coord_mx.loc[(coord_mx['strand'] == '+'), 'junction_coordinate1'] = coord_mx['+first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '-'), 'junction_coordinate1'] = coord_mx['-first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '+') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['+secon']\n",
    "    coord_mx.loc[(coord_mx['strand'] == '-') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['-secon']\n",
    "    \n",
    "    return coord_mx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "af9d6620",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate1</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>+first</th>\n",
       "      <th>+secon</th>\n",
       "      <th>-first</th>\n",
       "      <th>-secon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92379857</td>\n",
       "      <td>92379859</td>\n",
       "      <td>92611313</td>\n",
       "      <td>92611338</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>+</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>None</td>\n",
       "      <td>92379859:92611313</td>\n",
       "      <td>92611338:None</td>\n",
       "      <td>92611338:92379857</td>\n",
       "      <td>None:92611313</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0         1         2         3     4     5 strand  \\\n",
       "0  92379857  92379859  92611313  92611338  None  None      +   \n",
       "1  92379857  92379859  92611313  92611338  None  None      +   \n",
       "2  92379857  92379859  92611313  92611338  None  None      +   \n",
       "3  92379857  92379859  92611313  92611338  None  None      +   \n",
       "4  92379857  92379859  92611313  92611338  None  None      +   \n",
       "\n",
       "  junction_coordinate1 junction_coordinate2             +first         +secon  \\\n",
       "0    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "1    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "2    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "3    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "4    92379859:92611313                 None  92379859:92611313  92611338:None   \n",
       "\n",
       "              -first         -secon  \n",
       "0  92611338:92379857  None:92611313  \n",
       "1  92611338:92379857  None:92611313  \n",
       "2  92611338:92379857  None:92611313  \n",
       "3  92611338:92379857  None:92611313  \n",
       "4  92611338:92379857  None:92611313  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load cancer generation matrix\n",
    "mx = pd.read_csv(big_matrix, sep = '\\t')\n",
    "display(mx.head())\n",
    "# Add split junction information to generation table\n",
    "coord_mx = explode_immunopepper_coord(mx)\n",
    "display(coord_mx.head())\n",
    "mx = pd.concat([mx, coord_mx[['strand', 'junction_coordinate1', 'junction_coordinate2']]], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "75dfdf40",
   "metadata": {},
   "outputs": [],
   "source": [
    "# LOAD STAR junctions\n",
    "star_jx = pd.read_csv(star_jx, sep = '\\t')\n",
    "star_jx.head()\n",
    "# Add STAR junction column 10 min\n",
    "star_jx['junction_coordinate'] = star_jx['junction_start'].astype(str) + ':' + star_jx['junction_end'].astype(str)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc5a3849",
   "metadata": {},
   "source": [
    "# Check junction presence in STAR \n",
    "- 2-exons, 1st junction is present \n",
    "- 3 exons, needs first and second junction present "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "562e6b96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(44293227, 27)\n"
     ]
    }
   ],
   "source": [
    "# Foreground Kmers from 1 junction\n",
    "one_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] == 'None')]\n",
    "print(one_jx.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "942231d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1839568, 27)\n"
     ]
    }
   ],
   "source": [
    "# Foreground Kmers from 2 junctions\n",
    "two_jx = mx[(mx['junction_coordinate1'] != 'None') & (mx['junction_coordinate2'] != 'None')]\n",
    "print(two_jx.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "15863551",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "208958\n",
      "1992044\n"
     ]
    }
   ],
   "source": [
    "# Foreground 1 junction - NOT IN STAR \n",
    "isstar_one = set(one_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # junction coordinates\n",
    "print(len(isstar_one))\n",
    "is_star_one_jx = set(one_jx.set_index('junction_coordinate1').loc[isstar_one, 'coord']) # corresonding kmer coordinates\n",
    "print(len(is_star_one_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "095f8c21",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13190\n",
      "13720\n",
      "102528\n"
     ]
    }
   ],
   "source": [
    "# Foreground 2 junctions - NOT IN STAR \n",
    "is_star_two1 = set(two_jx['junction_coordinate1']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "is_star_two2 = set(two_jx['junction_coordinate2']).intersection(set(star_jx['junction_coordinate'])) # Junction coordinates\n",
    "print(len(is_star_two1))\n",
    "print(len(is_star_two2))\n",
    "\n",
    "is_star_two_jx = set(two_jx.set_index('junction_coordinate1').loc[is_star_two1, 'coord']).intersection(\\\n",
    "set(two_jx.set_index('junction_coordinate2').loc[is_star_two2, 'coord'])) # Corresponding kmer coord\n",
    "print(len(is_star_two_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8770b37e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1992044\n",
      "102528\n"
     ]
    }
   ],
   "source": [
    "print(len(is_star_one_jx))\n",
    "print(len(is_star_two_jx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "327c77a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247.36660489800852\n"
     ]
    }
   ],
   "source": [
    "# Foreground Table - Create FLAG GTEX junctions\n",
    "start_time = timeit.default_timer()\n",
    "mx['STAR_GTEx_jx'] = False\n",
    "mx = mx.set_index('coord')\n",
    "mx.loc[list(is_star_one_jx.union(is_star_two_jx)), 'STAR_GTEx_jx'] = True\n",
    "mx = mx.reset_index()\n",
    "print(timeit.default_timer() - start_time)    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1d2fc09",
   "metadata": {},
   "source": [
    "# Retrieve expression "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7851d93f",
   "metadata": {},
   "source": [
    "### Extract metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "37de382f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select junctions to query in STAR file \n",
    "query = mx.loc[mx['STAR_GTEx_jx'] == True, ['strand', 'junction_coordinate1',\n",
    "                                            'junction_coordinate2', 'STAR_GTEx_jx']].drop_duplicates()\\\n",
    "                                                                                    .reset_index()\\\n",
    "                                                                                    .drop('index', axis = 1)\n",
    "\n",
    "assert(query.loc[:, ['junction_coordinate1', 'strand']].drop_duplicates().shape[0] == \\\n",
    "       query.loc[:, ['junction_coordinate1']].drop_duplicates().shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ff4d910e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'-'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query-Junction to strand \n",
    "jx_strand = {}\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate1']] = query.loc[i, 'strand']\n",
    "\n",
    "for i in np.arange(query.shape[0]):\n",
    "    jx_strand[query.loc[i, 'junction_coordinate2']] = query.loc[i, 'strand']\n",
    "\n",
    "jx_strand.pop('None')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66a6a66b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b5b64f47",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.11621462600305676\n"
     ]
    }
   ],
   "source": [
    "# Chromosome to query-junction\n",
    "star_jx = star_jx.set_index('junction_coordinate') #Faster\n",
    "\n",
    "intermediate = star_jx.loc[jx_strand.keys(), :]\n",
    "\n",
    "start_time = timeit.default_timer()\n",
    "chr_jx = defaultdict(set)\n",
    "for jx, chrm in zip(intermediate.index, intermediate['chr']):\n",
    "    chr_jx[chrm].add(jx)\n",
    "print(timeit.default_timer() - start_time)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ea3a0fab",
   "metadata": {},
   "outputs": [],
   "source": [
    "######tests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64bfe99f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "ad059948",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "80525491"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#len(set(star_jx['junction_coordinate']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "0f5451bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#foo = star_jx.groupby(['junction_coordinate', 'chr']).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "07712983",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "80550666"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len((foo.index))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "ce89b654",
   "metadata": {},
   "outputs": [],
   "source": [
    "# jx_count = {}\n",
    "# for i in foo.index:\n",
    "#     if i[0] not in jx_count:\n",
    "#         jx_count[i[0]]=1\n",
    "#     else: \n",
    "#         jx_count[i[0]]+=1\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "c23543ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# select = []\n",
    "# for i, j in jx_count.items():\n",
    "#     if j > 1:\n",
    "#         select.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "749e5333",
   "metadata": {},
   "outputs": [],
   "source": [
    "# chr_test = star_jx.set_index('junction_coordinate').loc[select]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "06c92ecd",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_start</th>\n",
       "      <th>junction_end</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78107623</th>\n",
       "      <td>chrX</td>\n",
       "      <td>+</td>\n",
       "      <td>1000293</td>\n",
       "      <td>1189987</td>\n",
       "      <td>1000293:1189987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80346765</th>\n",
       "      <td>chrY</td>\n",
       "      <td>+</td>\n",
       "      <td>1000293</td>\n",
       "      <td>1189987</td>\n",
       "      <td>1000293:1189987</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr strand  junction_start  junction_end junction_coordinate\n",
       "78107623  chrX      +         1000293       1189987     1000293:1189987\n",
       "80346765  chrY      +         1000293       1189987     1000293:1189987"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# star_jx.loc[(star_jx['junction_coordinate'] == select[0]), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "cb12fd20",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_start</th>\n",
       "      <th>junction_end</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>999357:1350993</th>\n",
       "      <td>chrY</td>\n",
       "      <td>+</td>\n",
       "      <td>999357</td>\n",
       "      <td>1350993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999377:1030776</th>\n",
       "      <td>chrX</td>\n",
       "      <td>-</td>\n",
       "      <td>999377</td>\n",
       "      <td>1030776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999377:1030776</th>\n",
       "      <td>chrY</td>\n",
       "      <td>-</td>\n",
       "      <td>999377</td>\n",
       "      <td>1030776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999771:999866</th>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>999771</td>\n",
       "      <td>999866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999771:999866</th>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>999771</td>\n",
       "      <td>999866</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      chr strand  junction_start  junction_end\n",
       "junction_coordinate                                           \n",
       "999357:1350993       chrY      +          999357       1350993\n",
       "999377:1030776       chrX      -          999377       1030776\n",
       "999377:1030776       chrY      -          999377       1030776\n",
       "999771:999866        chr1      -          999771        999866\n",
       "999771:999866        chr4      +          999771        999866"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# chr_test.loc[chr_test]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "a9205dc4",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_start</th>\n",
       "      <th>junction_end</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100153424:100153456</th>\n",
       "      <td>chr7</td>\n",
       "      <td>+</td>\n",
       "      <td>100153424</td>\n",
       "      <td>100153456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100153424:100153456</th>\n",
       "      <td>chr8</td>\n",
       "      <td>+</td>\n",
       "      <td>100153424</td>\n",
       "      <td>100153456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100498207:100499066</th>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "      <td>100498207</td>\n",
       "      <td>100499066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100498207:100499066</th>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>100498207</td>\n",
       "      <td>100499066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005398:1005446</th>\n",
       "      <td>chr4</td>\n",
       "      <td>-</td>\n",
       "      <td>1005398</td>\n",
       "      <td>1005446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005398:1005446</th>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>1005398</td>\n",
       "      <td>1005446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100555227:100555370</th>\n",
       "      <td>chr15</td>\n",
       "      <td>-</td>\n",
       "      <td>100555227</td>\n",
       "      <td>100555370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100555227:100555370</th>\n",
       "      <td>chr3</td>\n",
       "      <td>+</td>\n",
       "      <td>100555227</td>\n",
       "      <td>100555370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100555227:100558404</th>\n",
       "      <td>chr15</td>\n",
       "      <td>-</td>\n",
       "      <td>100555227</td>\n",
       "      <td>100558404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100555227:100558404</th>\n",
       "      <td>chr3</td>\n",
       "      <td>+</td>\n",
       "      <td>100555227</td>\n",
       "      <td>100558404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100558188:100558404</th>\n",
       "      <td>chr15</td>\n",
       "      <td>-</td>\n",
       "      <td>100558188</td>\n",
       "      <td>100558404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100558188:100558404</th>\n",
       "      <td>chr3</td>\n",
       "      <td>+</td>\n",
       "      <td>100558188</td>\n",
       "      <td>100558404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100829557:100829595</th>\n",
       "      <td>chr10</td>\n",
       "      <td>+</td>\n",
       "      <td>100829557</td>\n",
       "      <td>100829595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100829557:100829595</th>\n",
       "      <td>chr14</td>\n",
       "      <td>+</td>\n",
       "      <td>100829557</td>\n",
       "      <td>100829595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100873237:100873350</th>\n",
       "      <td>chr14</td>\n",
       "      <td>-</td>\n",
       "      <td>100873237</td>\n",
       "      <td>100873350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100873237:100873350</th>\n",
       "      <td>chr7</td>\n",
       "      <td>+</td>\n",
       "      <td>100873237</td>\n",
       "      <td>100873350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10106:10128</th>\n",
       "      <td>chr4</td>\n",
       "      <td>-</td>\n",
       "      <td>10106</td>\n",
       "      <td>10128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10106:10128</th>\n",
       "      <td>chrM</td>\n",
       "      <td>-</td>\n",
       "      <td>10106</td>\n",
       "      <td>10128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010979:1011111</th>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "      <td>1010979</td>\n",
       "      <td>1011111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010979:1011111</th>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>1010979</td>\n",
       "      <td>1011111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011296:1011638</th>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>1011296</td>\n",
       "      <td>1011638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011296:1011638</th>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>1011296</td>\n",
       "      <td>1011638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115763:10115784</th>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>10115763</td>\n",
       "      <td>10115784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115763:10115784</th>\n",
       "      <td>chr3</td>\n",
       "      <td>+</td>\n",
       "      <td>10115763</td>\n",
       "      <td>10115784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115781:10115808</th>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>10115781</td>\n",
       "      <td>10115808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115781:10115808</th>\n",
       "      <td>chr3</td>\n",
       "      <td>+</td>\n",
       "      <td>10115781</td>\n",
       "      <td>10115808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115961:10115988</th>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>10115961</td>\n",
       "      <td>10115988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10115961:10115988</th>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>10115961</td>\n",
       "      <td>10115988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011985:1012064</th>\n",
       "      <td>chr17</td>\n",
       "      <td>-</td>\n",
       "      <td>1011985</td>\n",
       "      <td>1012064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011985:1012064</th>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>1011985</td>\n",
       "      <td>1012064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10126883:10126905</th>\n",
       "      <td>chr2</td>\n",
       "      <td>+</td>\n",
       "      <td>10126883</td>\n",
       "      <td>10126905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10126883:10126905</th>\n",
       "      <td>chr3</td>\n",
       "      <td>-</td>\n",
       "      <td>10126883</td>\n",
       "      <td>10126905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10155:10251</th>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>10155</td>\n",
       "      <td>10251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10155:10251</th>\n",
       "      <td>chr12</td>\n",
       "      <td>-</td>\n",
       "      <td>10155</td>\n",
       "      <td>10251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101642437:101642469</th>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>101642437</td>\n",
       "      <td>101642469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101642437:101642469</th>\n",
       "      <td>chr15</td>\n",
       "      <td>-</td>\n",
       "      <td>101642437</td>\n",
       "      <td>101642469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017717:1018085</th>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>1017717</td>\n",
       "      <td>1018085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017717:1018085</th>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>1017717</td>\n",
       "      <td>1018085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101886674:101897862</th>\n",
       "      <td>chr12</td>\n",
       "      <td>+</td>\n",
       "      <td>101886674</td>\n",
       "      <td>101897862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10189:10296</th>\n",
       "      <td>chr9</td>\n",
       "      <td>-</td>\n",
       "      <td>10189</td>\n",
       "      <td>10296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10189:10296</th>\n",
       "      <td>chrM</td>\n",
       "      <td>+</td>\n",
       "      <td>10189</td>\n",
       "      <td>10296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102015975:102016000</th>\n",
       "      <td>chr14</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102015975:102016000</th>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020789:1020810</th>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "      <td>1020789</td>\n",
       "      <td>1020810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020789:1020810</th>\n",
       "      <td>chr20</td>\n",
       "      <td>+</td>\n",
       "      <td>1020789</td>\n",
       "      <td>1020810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102161277:102520411</th>\n",
       "      <td>chr14</td>\n",
       "      <td>+</td>\n",
       "      <td>102161277</td>\n",
       "      <td>102520411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102161277:102520411</th>\n",
       "      <td>chr7</td>\n",
       "      <td>+</td>\n",
       "      <td>102161277</td>\n",
       "      <td>102520411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10221839:10221881</th>\n",
       "      <td>chr12</td>\n",
       "      <td>-</td>\n",
       "      <td>10221839</td>\n",
       "      <td>10221881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10221839:10221881</th>\n",
       "      <td>chr19</td>\n",
       "      <td>+</td>\n",
       "      <td>10221839</td>\n",
       "      <td>10221881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102231404:102231705</th>\n",
       "      <td>chr11</td>\n",
       "      <td>-</td>\n",
       "      <td>102231404</td>\n",
       "      <td>102231705</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       chr strand  junction_start  junction_end\n",
       "junction_coordinate                                            \n",
       "100153424:100153456   chr7      +       100153424     100153456\n",
       "100153424:100153456   chr8      +       100153424     100153456\n",
       "100498207:100499066  chr10      -       100498207     100499066\n",
       "100498207:100499066   chr1      +       100498207     100499066\n",
       "1005398:1005446       chr4      -         1005398       1005446\n",
       "1005398:1005446       chr5      +         1005398       1005446\n",
       "100555227:100555370  chr15      -       100555227     100555370\n",
       "100555227:100555370   chr3      +       100555227     100555370\n",
       "100555227:100558404  chr15      -       100555227     100558404\n",
       "100555227:100558404   chr3      +       100555227     100558404\n",
       "100558188:100558404  chr15      -       100558188     100558404\n",
       "100558188:100558404   chr3      +       100558188     100558404\n",
       "100829557:100829595  chr10      +       100829557     100829595\n",
       "100829557:100829595  chr14      +       100829557     100829595\n",
       "100873237:100873350  chr14      -       100873237     100873350\n",
       "100873237:100873350   chr7      +       100873237     100873350\n",
       "10106:10128           chr4      -           10106         10128\n",
       "10106:10128           chrM      -           10106         10128\n",
       "1010979:1011111      chr10      -         1010979       1011111\n",
       "1010979:1011111       chr4      +         1010979       1011111\n",
       "1011296:1011638      chr11      +         1011296       1011638\n",
       "1011296:1011638       chr4      +         1011296       1011638\n",
       "10115763:10115784    chr19      -        10115763      10115784\n",
       "10115763:10115784     chr3      +        10115763      10115784\n",
       "10115781:10115808    chr19      -        10115781      10115808\n",
       "10115781:10115808     chr3      +        10115781      10115808\n",
       "10115961:10115988    chr12      +        10115961      10115988\n",
       "10115961:10115988    chr19      -        10115961      10115988\n",
       "1011985:1012064      chr17      -         1011985       1012064\n",
       "1011985:1012064      chr19      -         1011985       1012064\n",
       "10126883:10126905     chr2      +        10126883      10126905\n",
       "10126883:10126905     chr3      -        10126883      10126905\n",
       "10155:10251           chr1      -           10155         10251\n",
       "10155:10251          chr12      -           10155         10251\n",
       "101642437:101642469  chr12      +       101642437     101642469\n",
       "101642437:101642469  chr15      -       101642437     101642469\n",
       "1017717:1018085       chr1      +         1017717       1018085\n",
       "1017717:1018085      chr11      +         1017717       1018085\n",
       "101886674:101897862  chr12      +       101886674     101897862\n",
       "10189:10296           chr9      -           10189         10296\n",
       "10189:10296           chrM      +           10189         10296\n",
       "102015975:102016000  chr14      -       102015975     102016000\n",
       "102015975:102016000   chr2      -       102015975     102016000\n",
       "1020789:1020810      chr10      -         1020789       1020810\n",
       "1020789:1020810      chr20      +         1020789       1020810\n",
       "102161277:102520411  chr14      +       102161277     102520411\n",
       "102161277:102520411   chr7      +       102161277     102520411\n",
       "10221839:10221881    chr12      -        10221839      10221881\n",
       "10221839:10221881    chr19      +        10221839      10221881\n",
       "102231404:102231705  chr11      -       102231404     102231705"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# chr_test.loc[(chr_test['chr']!= 'chrX') & (chr_test['chr']!= 'chrY') ].iloc[0:50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "e2ee84dc",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chrY', 'chr8', 'chr14', 'chrM', 'chr4', 'chr19', 'chr12', 'chr11',\n",
       "       'chr2', 'chr7', 'chr5', 'chr20', 'chr9', 'chr6', 'chrX', 'chr13',\n",
       "       'chr3', 'chr16', 'KI270938.1', 'KI270879.1', 'chr17', 'chr1',\n",
       "       'KI270913.1', 'chr22', 'chr21', 'KI270928.1', 'chr18', 'chr15',\n",
       "       'KI270872.1', 'GL949752.1'], dtype=object)"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#chr_test.loc[ chr_test.loc[:, ['strand', 'junction_start', 'junction_end']].duplicated(), 'chr'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "9cc61500",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chrX', 'chrY', 'chr7', 'chr8', 'chr10', 'chr1', 'chr4', 'chr5',\n",
       "       'chr15', 'chr3', 'chr14', 'chrM', 'chr11', 'chr19', 'chr12',\n",
       "       'chr17', 'chr2', 'chr9', 'chr20', 'chr13', 'chr6', 'chr16',\n",
       "       'KV766196.1', 'GL949752.1', 'KI270938.1', 'chr18', 'KI270879.1',\n",
       "       'KI270913.1', 'GL949747.2', 'chr22', 'GL949753.2', 'chr21',\n",
       "       'GL383582.2', 'KI270928.1', 'KI270872.1', 'GL949746.1',\n",
       "       'KV575245.1'], dtype=object)"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#chr_test['chr'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd209d62",
   "metadata": {},
   "source": [
    "### Collect Expression "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d2de32fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_libsize_whitelist(libsize, whitelist):\n",
    "    # Read libsize and whitelist\n",
    "    libsize_normal = pd.read_csv(libsize, sep = '\\t')\n",
    "    whitelist_normal = pd.read_csv(whitelist, sep = '\\t', header = None)\n",
    "    whitelist_normal.columns = ['sample']\n",
    "    return libsize_normal, whitelist_normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5f0c912e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def preprocess_STAR_projected(chrm, path_star, whitelist_normal, libsize_normal):\n",
    "    # Star junctions - projected coordinates and expression\n",
    "    projected_chr_file = os.path.join(path_star, f'junctions_spladder_projected/junctions_spladder.projected.{chrm}.hdf5')\n",
    "    star_expr = h5py.File(projected_chr_file, 'r')\n",
    "\n",
    "    # Whitelist samples\n",
    "    index_whitelist_samples = [s for s, sample in enumerate(star_expr['samples']) if sample.decode() + 'all' in whitelist_normal['sample'].values]\n",
    "    samples_decoded = [sample.decode() + 'all' for s, sample in enumerate(star_expr['samples']) if sample.decode() + 'all' in whitelist_normal['sample'].values]\n",
    "\n",
    "    # Libsize sampple     \n",
    "    lib_75_per_sample = libsize_normal.set_index('sample').loc[samples_decoded, 'libsize_75percent'].values\n",
    "\n",
    "    return star_expr, index_whitelist_samples, lib_75_per_sample\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "37f3c8ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_junction_counts(junction_start, junction_end, \n",
    "                        chrm, strand, expression_h5, \n",
    "                        index_whitelist_samples, lib_75_per_sample, normalizer):\n",
    "    # Extract junction ID in hdf5 file #TODO SPEEDUP WITH SETS\n",
    "    jx_idx_h5 = np.where((junction_start == expression_h5[f'{chrm}:{strand}:junction_start'][...]) & \n",
    "            (junction_end == expression_h5[f'{chrm}:{strand}:junction_end'][...]))[0]\n",
    "    \n",
    "    if len(jx_idx_h5) > 0:\n",
    "        assert(len(jx_idx_h5) == 1) #TODO check critical\n",
    "        jx_idx_h5 = jx_idx_h5[0]\n",
    "\n",
    "        # Extract Raw count in hdf5 file\n",
    "        raw_counts = expression_h5[f'{chrm}:{strand}:count'][jx_idx_h5, np.array(index_whitelist_samples)]\n",
    "        normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n",
    "    else:\n",
    "        normalized_counts = None\n",
    "        print(f'Error: No {chrm}:{strand}:junction_start {chrm}:{strand}:junction_start matching {junction_start}:{junction_end}')\n",
    "    return normalized_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "25bb9667",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_recurrence(array_, threshold):\n",
    "    if threshold == 0:\n",
    "        return np.sum(array_ > threshold)\n",
    "    else:\n",
    "        return np.sum(array_ >= threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "bee8537c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_multi_thresholds(normalized_counts, filter_thresholds):\n",
    "    recurrence = []\n",
    "    for threshold in filter_thresholds: # Make filter threshold\n",
    "        recurrence.append(filter_recurrence(normalized_counts, threshold))\n",
    "    return recurrence \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "86d6aaf7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/slurm-job.1205652/ipykernel_3993045/1095229631.py:14: RuntimeWarning: invalid value encountered in true_divide\n",
      "  normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "....500\n",
      "....1000\n",
      "....1500\n",
      "....2000\n",
      "....2500\n",
      "....3000\n",
      "....3500\n",
      "....4000\n",
      "....4500\n",
      "....5000\n",
      "....5500\n",
      "....6000\n",
      "Error: No chr9:-:junction_start chr9:-:junction_start matching 127507186:127507342\n",
      "....6500\n",
      "....7000\n",
      "....7500\n",
      "....8000\n",
      "....8500\n",
      "-13237.52022917592\n",
      "chr14\n",
      "....9000\n",
      "....9500\n",
      "....10000\n",
      "....10500\n",
      "....11000\n",
      "....11500\n",
      "....12000\n",
      "....12500\n",
      "....13000\n",
      "....13500\n",
      "....14000\n",
      "....14500\n",
      "....15000\n",
      "-8053.89567177603\n",
      "chrX\n",
      "....15500\n",
      "....16000\n",
      "....16500\n",
      "....17000\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [21], line 22\u001b[0m\n\u001b[1;32m     20\u001b[0m junction_start \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(jx\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m0\u001b[39m])\n\u001b[1;32m     21\u001b[0m junction_end \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(jx\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m)[\u001b[38;5;241m1\u001b[39m])\n\u001b[0;32m---> 22\u001b[0m normalized_counts \u001b[38;5;241m=\u001b[39m \u001b[43mget_junction_counts\u001b[49m\u001b[43m(\u001b[49m\u001b[43mjunction_start\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mjunction_end\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     23\u001b[0m \u001b[43m                                        \u001b[49m\u001b[43mchrm\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mjx_strand\u001b[49m\u001b[43m[\u001b[49m\u001b[43mjx\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mexpression_h5\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     24\u001b[0m \u001b[43m                                        \u001b[49m\u001b[43mindex_whitelist_samples\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     25\u001b[0m \u001b[43m                                        \u001b[49m\u001b[43mlib_75_per_sample\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnormalizer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     27\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m normalized_counts \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m     28\u001b[0m     metadata \u001b[38;5;241m=\u001b[39m [jx, jx_strand[jx], chrm] \u001b[38;5;241m+\u001b[39m filter_multi_thresholds(normalized_counts, filter_thresholds)\n",
      "Cell \u001b[0;32mIn [18], line 13\u001b[0m, in \u001b[0;36mget_junction_counts\u001b[0;34m(junction_start, junction_end, chrm, strand, expression_h5, index_whitelist_samples, lib_75_per_sample, normalizer)\u001b[0m\n\u001b[1;32m     10\u001b[0m     jx_idx_h5 \u001b[38;5;241m=\u001b[39m jx_idx_h5[\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m     12\u001b[0m     \u001b[38;5;66;03m# Extract Raw count in hdf5 file\u001b[39;00m\n\u001b[0;32m---> 13\u001b[0m     raw_counts \u001b[38;5;241m=\u001b[39m \u001b[43mexpression_h5\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mchrm\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m:\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mstrand\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m:count\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m[\u001b[49m\u001b[43mjx_idx_h5\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43marray\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindex_whitelist_samples\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m     14\u001b[0m     normalized_counts \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mdivide(raw_counts, lib_75_per_sample) \u001b[38;5;241m*\u001b[39m normalizer\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32mh5py/_objects.pyx:54\u001b[0m, in \u001b[0;36mh5py._objects.with_phil.wrapper\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mh5py/_objects.pyx:55\u001b[0m, in \u001b[0;36mh5py._objects.with_phil.wrapper\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep_other/lib/python3.9/site-packages/h5py-3.7.0-py3.9-linux-x86_64.egg/h5py/_hl/dataset.py:741\u001b[0m, in \u001b[0;36mDataset.__getitem__\u001b[0;34m(self, args, new_dtype)\u001b[0m\n\u001b[1;32m    739\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fast_read_ok \u001b[38;5;129;01mand\u001b[39;00m (new_dtype \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    740\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 741\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_fast_reader\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43margs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    742\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m    743\u001b[0m         \u001b[38;5;28;01mpass\u001b[39;00m  \u001b[38;5;66;03m# Fall back to Python read pathway below\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "### Thresholding experiments \n",
    "# Remark 1 junction can have multiple chrm --> Which \n",
    "# Remark 1 junction on 1 chrm can have multiple strands --> Take from Immunopepper\n",
    "\n",
    "res = []\n",
    "libsize_normal, whitelist_normal = read_libsize_whitelist(libsize, whitelist)\n",
    "counter = 0\n",
    "for chrm, jxS in chr_jx.items(): # Per chromosome\n",
    "    print(chrm)\n",
    "    # Expression file\n",
    "    expression_h5, index_whitelist_samples, lib_75_per_sample = preprocess_STAR_projected(chrm, \n",
    "                                                                                      path_star, \n",
    "                                                                                      whitelist_normal, \n",
    "                                                                                      libsize_normal)\n",
    "    start_time = timeit.default_timer()    \n",
    "    for jx in jxS: # Per junction\n",
    "        counter +=1\n",
    "        if counter % 500 == 0:\n",
    "            print(f'....{counter}')\n",
    "        junction_start = int(jx.split(':')[0])\n",
    "        junction_end = int(jx.split(':')[1])\n",
    "        normalized_counts = get_junction_counts(junction_start, junction_end, \n",
    "                                                chrm, jx_strand[jx], expression_h5, \n",
    "                                                index_whitelist_samples, \n",
    "                                                lib_75_per_sample, normalizer)\n",
    "        \n",
    "        if normalized_counts is not None:\n",
    "            metadata = [jx, jx_strand[jx], chrm] + filter_multi_thresholds(normalized_counts, filter_thresholds)\n",
    "        \n",
    "        res.append(metadata)\n",
    "    \n",
    "    expression_h5.close() \n",
    "    print(start_time - timeit.default_timer())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "5fbc4f87",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrm number junctions 8778\n",
      "chrm number junctions 6377\n",
      "chrm number junctions 5554\n",
      "chrm number junctions 7228\n",
      "chrm number junctions 22420\n",
      "chrm number junctions 11530\n",
      "chrm number junctions 9538\n",
      "chrm number junctions 11815\n",
      "chrm number junctions 5783\n",
      "chrm number junctions 2454\n",
      "chrm number junctions 2914\n",
      "chrm number junctions 18994\n",
      "chrm number junctions 14150\n",
      "chrm number junctions 10065\n",
      "chrm number junctions 13964\n",
      "chrm number junctions 7976\n",
      "chrm number junctions 5495\n",
      "chrm number junctions 6594\n",
      "chrm number junctions 9060\n",
      "chrm number junctions 2400\n",
      "chrm number junctions 13746\n",
      "chrm number junctions 7737\n",
      "chrm number junctions 5244\n",
      "chrm number junctions 212\n",
      "chrm number junctions 360\n",
      "Total number junctions 210388\n"
     ]
    }
   ],
   "source": [
    "tot = 0 \n",
    "for chrm, jxS in chr_jx.items():\n",
    "    print(f'chrm number junctions {len(jxS)}')\n",
    "    tot+= len(jxS)\n",
    "print(f'Total number junctions {tot}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "370f7860",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Expression to DF\n",
    "df_res = pd.DataFrame(res, columns = ['junction_coordinate', 'strand_STAR', 'chr'] + gtex_cols)\n",
    "print(df_res.shape)\n",
    "display(df_res.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64f7b197",
   "metadata": {},
   "source": [
    "### Format results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "baa6f53e",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_cols = ['gtexCohortfilter >0.0', 'gtexCohortfilter >=1.0',\n",
    "       'gtexCohortfilter >=2.0', 'gtexCohortfilter >=3.0',\n",
    "       'gtexCohortfilter >=5.0', 'gtexCohortfilter >=10.0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "ffb11d11",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(44293227, 27)\n",
      "(2431499, 21)\n",
      "Index(['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated',\n",
      "       'isCrossJunction', 'batch', 'cancerCohortfilter >0.0',\n",
      "       'cancerCohortfilter >=1.0', 'cancerCohortfilter >=2.0',\n",
      "       'cancerCohortfilter >=3.0', 'cancerCohortfilter >=5.0',\n",
      "       'cancerCohortfilter >=10.0', 'TCGA25131901A01R156513all',\n",
      "       'TCGA25131301A01R156513all', 'TCGA61200801A02R156813all',\n",
      "       'TCGA24143101A01R156613all', 'TCGA24229801A01R156913all', 'isAnnotated',\n",
      "       'strand', 'junction_coordinate1', 'junction_coordinate2',\n",
      "       'junction_coordinate', 'strand_STAR', 'chr', 'gtexCohortfilter >0.0',\n",
      "       'gtexCohortfilter >=1.0', 'gtexCohortfilter >=2.0',\n",
      "       'gtexCohortfilter >=3.0', 'gtexCohortfilter >=5.0',\n",
      "       'gtexCohortfilter >=10.0'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2431504</th>\n",
       "      <td>ETRSPEKSV</td>\n",
       "      <td>121185317:121185337:121185025:121185032:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>265</td>\n",
       "      <td>258</td>\n",
       "      <td>222</td>\n",
       "      <td>183</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431505</th>\n",
       "      <td>SFKDRHPSY</td>\n",
       "      <td>121183457:121183471:121183340:121183353:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>63</td>\n",
       "      <td>35</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431506</th>\n",
       "      <td>KDSAVDFTG</td>\n",
       "      <td>121183457:121183465:121181329:121181348:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>186</td>\n",
       "      <td>136</td>\n",
       "      <td>68</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431507</th>\n",
       "      <td>WLLLLSLFL</td>\n",
       "      <td>121181270:121181272:121180855:121180880:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>44</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2431508</th>\n",
       "      <td>GGETRSPEN</td>\n",
       "      <td>121185317:121185343:121181347:121181348:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2858</td>\n",
       "      <td>78</td>\n",
       "      <td>36</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              kmer                                              coord  \\\n",
       "2431504  ETRSPEKSV  121185317:121185337:121185025:121185032:None:None   \n",
       "2431505  SFKDRHPSY  121183457:121183471:121183340:121183353:None:None   \n",
       "2431506  KDSAVDFTG  121183457:121183465:121181329:121181348:None:None   \n",
       "2431507  WLLLLSLFL  121181270:121181272:121180855:121180880:None:None   \n",
       "2431508  GGETRSPEN  121185317:121185343:121181347:121181348:None:None   \n",
       "\n",
       "         junctionAnnotated  readFrameAnnotated  isCrossJunction  batch  \\\n",
       "2431504              False                True             True   2858   \n",
       "2431505              False                True             True   2858   \n",
       "2431506               True                True             True   2858   \n",
       "2431507              False               False             True   2858   \n",
       "2431508               True                True             True   2858   \n",
       "\n",
       "         cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "2431504                      265                       258   \n",
       "2431505                       63                        35   \n",
       "2431506                      186                       136   \n",
       "2431507                       44                        28   \n",
       "2431508                       78                        36   \n",
       "\n",
       "         cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "2431504                       222                       183  ...   \n",
       "2431505                        12                         5  ...   \n",
       "2431506                        68                        31  ...   \n",
       "2431507                        10                         4  ...   \n",
       "2431508                        12                         4  ...   \n",
       "\n",
       "         junction_coordinate2  junction_coordinate  strand_STAR  chr  \\\n",
       "2431504                  None                  NaN          NaN  NaN   \n",
       "2431505                  None                  NaN          NaN  NaN   \n",
       "2431506                  None                  NaN          NaN  NaN   \n",
       "2431507                  None                  NaN          NaN  NaN   \n",
       "2431508                  None                  NaN          NaN  NaN   \n",
       "\n",
       "         gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "2431504                    NaN                     NaN   \n",
       "2431505                    NaN                     NaN   \n",
       "2431506                    NaN                     NaN   \n",
       "2431507                    NaN                     NaN   \n",
       "2431508                    NaN                     NaN   \n",
       "\n",
       "         gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  \\\n",
       "2431504                     NaN                     NaN   \n",
       "2431505                     NaN                     NaN   \n",
       "2431506                     NaN                     NaN   \n",
       "2431507                     NaN                     NaN   \n",
       "2431508                     NaN                     NaN   \n",
       "\n",
       "        gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "2431504                    NaN                     NaN  \n",
       "2431505                    NaN                     NaN  \n",
       "2431506                    NaN                     NaN  \n",
       "2431507                    NaN                     NaN  \n",
       "2431508                    NaN                     NaN  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# One junction merge (straightforward merge on junction)\n",
    "print(one_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_one_jx = one_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_one_jx.head()\n",
    "print(base_one_jx.shape)\n",
    "# Add GTEX quantifications from STAR\n",
    "one_jx_quantified = base_one_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left') \n",
    "print(one_jx_quantified.columns)\n",
    "display(one_jx_quantified.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "9df47585",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1839568, 27)\n",
      "(188025, 21)\n",
      "Index(['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated',\n",
      "       'isCrossJunction', 'batch', 'cancerCohortfilter >0.0',\n",
      "       'cancerCohortfilter >=1.0', 'cancerCohortfilter >=2.0',\n",
      "       'cancerCohortfilter >=3.0', 'cancerCohortfilter >=5.0',\n",
      "       'cancerCohortfilter >=10.0', 'TCGA25131901A01R156513all',\n",
      "       'TCGA25131301A01R156513all', 'TCGA61200801A02R156813all',\n",
      "       'TCGA24143101A01R156613all', 'TCGA24229801A01R156913all', 'isAnnotated',\n",
      "       'strand', 'junction_coordinate1', 'junction_coordinate2',\n",
      "       'junction_coordinate', 'strand_STAR', 'chr', 'gtexCohortfilter >0.0',\n",
      "       'gtexCohortfilter >=1.0', 'gtexCohortfilter >=2.0',\n",
      "       'gtexCohortfilter >=3.0', 'gtexCohortfilter >=5.0',\n",
      "       'gtexCohortfilter >=10.0'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>196939</th>\n",
       "      <td>QSPYPCHWW</td>\n",
       "      <td>154464496:154464501:154464774:154464794:154465...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>59971</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>154464794:154465017</td>\n",
       "      <td>154464794:154465017</td>\n",
       "      <td>+</td>\n",
       "      <td>chrX</td>\n",
       "      <td>1398.0</td>\n",
       "      <td>1358.0</td>\n",
       "      <td>757.0</td>\n",
       "      <td>443.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196940</th>\n",
       "      <td>KLLHKFLKE</td>\n",
       "      <td>154468542:154468545:154468708:154468729:154468...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>59971</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>154468729:154468822</td>\n",
       "      <td>154468729:154468822</td>\n",
       "      <td>+</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9447.0</td>\n",
       "      <td>9447.0</td>\n",
       "      <td>9446.0</td>\n",
       "      <td>9438.0</td>\n",
       "      <td>9400.0</td>\n",
       "      <td>9280.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196941</th>\n",
       "      <td>HTKTMQVIS</td>\n",
       "      <td>34609637:34609646:34608029:34608046:34605716:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196942</th>\n",
       "      <td>TKTMQVISF</td>\n",
       "      <td>34609637:34609643:34608029:34608046:34605713:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196943</th>\n",
       "      <td>KTMQVISFH</td>\n",
       "      <td>34609637:34609640:34608029:34608046:34605710:3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>16207</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>34605717:34608029</td>\n",
       "      <td>-</td>\n",
       "      <td>chr14</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>9473.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer                                              coord  \\\n",
       "196939  QSPYPCHWW  154464496:154464501:154464774:154464794:154465...   \n",
       "196940  KLLHKFLKE  154468542:154468545:154468708:154468729:154468...   \n",
       "196941  HTKTMQVIS  34609637:34609646:34608029:34608046:34605716:3...   \n",
       "196942  TKTMQVISF  34609637:34609643:34608029:34608046:34605713:3...   \n",
       "196943  KTMQVISFH  34609637:34609640:34608029:34608046:34605710:3...   \n",
       "\n",
       "       junctionAnnotated readFrameAnnotated isCrossJunction  batch  \\\n",
       "196939             False              False            True  59971   \n",
       "196940              True              False            True  59971   \n",
       "196941              True              False            True  16207   \n",
       "196942              True              False            True  16207   \n",
       "196943              True              False            True  16207   \n",
       "\n",
       "        cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "196939                       14                         5   \n",
       "196940                       31                        12   \n",
       "196941                       13                         9   \n",
       "196942                       13                         9   \n",
       "196943                       13                         9   \n",
       "\n",
       "        cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "196939                         1                         0  ...   \n",
       "196940                         3                         1  ...   \n",
       "196941                         4                         0  ...   \n",
       "196942                         4                         0  ...   \n",
       "196943                         4                         0  ...   \n",
       "\n",
       "        junction_coordinate2  junction_coordinate  strand_STAR    chr  \\\n",
       "196939   154464794:154465017  154464794:154465017            +   chrX   \n",
       "196940   154468729:154468822  154468729:154468822            +   chrX   \n",
       "196941     34605717:34608029    34605717:34608029            -  chr14   \n",
       "196942     34605717:34608029    34605717:34608029            -  chr14   \n",
       "196943     34605717:34608029    34605717:34608029            -  chr14   \n",
       "\n",
       "        gtexCohortfilter >0.0  gtexCohortfilter >=1.0  gtexCohortfilter >=2.0  \\\n",
       "196939                 1398.0                  1358.0                   757.0   \n",
       "196940                 9447.0                  9447.0                  9446.0   \n",
       "196941                 9475.0                  9475.0                  9475.0   \n",
       "196942                 9475.0                  9475.0                  9475.0   \n",
       "196943                 9475.0                  9475.0                  9475.0   \n",
       "\n",
       "        gtexCohortfilter >=3.0 gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "196939                   443.0                  125.0                     7.0  \n",
       "196940                  9438.0                 9400.0                  9280.0  \n",
       "196941                  9475.0                 9475.0                  9473.0  \n",
       "196942                  9475.0                 9475.0                  9473.0  \n",
       "196943                  9475.0                 9475.0                  9473.0  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Two junctions merge (not straight forwrds, consider botrh junctions separately and take the max)\n",
    "print(two_jx.shape)\n",
    "# Remove the GTEX quantifications from immunopepper\n",
    "base_two_jx = two_jx.drop(gtex_cols, axis = 1).drop_duplicates() \n",
    "base_two_jx.head()\n",
    "print(base_two_jx.shape)\n",
    "# Add GTEX quantifications from STAR  # first junction\n",
    "two_jx_quantified_left = base_two_jx.merge(df_res, left_on = 'junction_coordinate1', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "\n",
    "# Add GTEX quantifications from STAR  # second junction\n",
    "two_jx_quantified_right = base_two_jx.merge(df_res, left_on = 'junction_coordinate2', \n",
    "                                      right_on = 'junction_coordinate', how = 'left')\n",
    "\n",
    "col_merge = [col for col in two_jx_quantified_right if col not in gtex_cols]\n",
    "\n",
    "# Add GTEX quantifications from STAR  # both junctions\n",
    "two_jx_quantified = two_jx_quantified_left.merge(two_jx_quantified_right, on = col_merge, how = 'outer')\n",
    "\n",
    "# Add GTEX quantifications from STAR  # MAX (!!) over 2 junctions\n",
    "for col in gtex_cols:\n",
    "    two_jx_quantified[col] = two_jx_quantified[[col + '_x', col + '_y']].max(skipna = True, axis = 1)\n",
    "    two_jx_quantified = two_jx_quantified.drop([col + '_x', col + '_y'], axis = 1)\n",
    "    \n",
    "print(two_jx_quantified.columns)\n",
    "display(two_jx_quantified.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "8d3cdfe9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2628453, 30)\n"
     ]
    }
   ],
   "source": [
    "# All Junctions quantified on STAR for GTEX\n",
    "print(two_jx_quantified.shape)\n",
    "print(one_jx_quantified.shape)\n",
    "jx_quantified = pd.concat([one_jx_quantified, two_jx_quantified])\n",
    "print(jx_quantified.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "c78fe488",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r_complete.STAR_GTEx.tsv.gz\n"
     ]
    }
   ],
   "source": [
    "### Save \n",
    "save_path = big_matrix.replace('tsv.gz', 'STAR_GTEx.tsv.gz')\n",
    "print(save_path)\n",
    "jx_quantified.to_csv(save_path, compression = 'gzip', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bad6a49",
   "metadata": {},
   "source": [
    "# Quality check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "5adeadd8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/slurm-job.1197223/ipykernel_2447348/1028733944.py:13: RuntimeWarning: invalid value encountered in true_divide\n",
      "  normalized_counts = np.divide(raw_counts, lib_75_per_sample) * normalizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "4\n",
      "4\n",
      "4\n",
      "4\n",
      "2\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "chrm_list = ['chr2', 'chr14']\n",
    "strand = '-'\n",
    "jx = '102015975:102016000'\n",
    "junction_start = int(jx.split(':')[0])\n",
    "junction_end = int(jx.split(':')[1])\n",
    "\n",
    "for chrm in chrm_list:\n",
    "    expression_h5, index_whitelist_samples, lib_75_per_sample = preprocess_STAR_projected(chrm, \n",
    "                                                                                      path_star, \n",
    "                                                                                      whitelist_normal, \n",
    "                                                                                      libsize_normal)\n",
    "\n",
    "    normalized_counts = get_junction_counts(junction_start, junction_end, \n",
    "                                            chrm, strand, expression_h5, \n",
    "                                            index_whitelist_samples, \n",
    "                                            lib_75_per_sample, normalizer)\n",
    "    if normalized_counts is not None:\n",
    "        for threshold in filter_thresholds: # Make filter threshold\n",
    "            print(filter_recurrence(normalized_counts, threshold))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "325fdff7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_start</th>\n",
       "      <th>junction_end</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19214000</th>\n",
       "      <td>chr14</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "      <td>102015975:102016000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46460381</th>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>102015975</td>\n",
       "      <td>102016000</td>\n",
       "      <td>102015975:102016000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            chr strand  junction_start  junction_end  junction_coordinate\n",
       "19214000  chr14      -       102015975     102016000  102015975:102016000\n",
       "46460381   chr2      -       102015975     102016000  102015975:102016000"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "star_jx.loc[(star_jx['junction_start'] == junction_start) & (star_jx['junction_end'] == junction_end)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20cae9c6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
