{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cab1d99c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import timeit\n",
    "import glob \n",
    "import pandas as pd\n",
    "import time\n",
    "import multiprocessing as mp \n",
    "import logging\n",
    "import sys \n",
    "import pathlib\n",
    "from pathlib import Path\n",
    "import matplotlib.pyplot as plt "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a8f2e8d",
   "metadata": {},
   "source": [
    "# Order \n",
    "- Uniprot \n",
    "- Sample init\n",
    "- Sample expression \n",
    "- Sample cohort \n",
    "- annotation \n",
    "- GTEX?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "fcf5e3ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_cancer_cohort(df, n_samples, threshold_column ):\n",
    "    df = df.loc[df[threshold_column] >= n_samples]    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d2674c73",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_threshold_colname(threshold, tag):\n",
    "    if (threshold is not None) and (threshold > 0 ):\n",
    "        threshold_column = f'{tag}filter >={threshold}'\n",
    "    else:\n",
    "        threshold_column = f'{tag}filter >{threshold}'\n",
    "    return threshold_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "783a8bfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_single_col(df, threshold, colname):\n",
    "    if threshold:\n",
    "        df = df.loc[df[colname] >= threshold, :]\n",
    "    else:\n",
    "        df = df.loc[df[colname] >threshold, :]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0ab8af74",
   "metadata": {},
   "outputs": [],
   "source": [
    "def max_recurrence_over_kmer(df, threshold_column, new_maxcol):\n",
    "    df = df[['kmer', threshold_column]].groupby('kmer').max()\n",
    "    df = df.reset_index()\n",
    "    df = df.rename({threshold_column: new_maxcol}, axis = 1)\n",
    "    return df "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "898b4316",
   "metadata": {},
   "outputs": [],
   "source": [
    "def output_count(df, report_count, report_step, step_string, perform_count=True):\n",
    "    '''\n",
    "    Performs a count operation on the number of kmers present in spark dataframe after a given filtering step\n",
    "    Note: This operation is expensive but useful if the user is interested in intermediate filtering steps\n",
    "    :param perform_count: bool whether to perform a count operation\n",
    "    :param df: dataframe with kmer expression counts\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param step_string: str name of the counting operation\n",
    "    '''\n",
    "    if perform_count:\n",
    "        mycount = df['kmer'].unique().shape[0]\n",
    "        report_count.append(mycount)\n",
    "        report_step.append(step_string)\n",
    "        print(f'# {step_string} n = {mycount} kmers')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "33b71537",
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_output_count(output_count, report_count, report_steps, prefix, cancer_sample_ori, mutation_mode,\n",
    "                      sample_expr_support_cancer, cohort_expr_support_cancer, n_samples_lim_cancer,\n",
    "                          cohort_expr_support_normal, n_samples_lim_normal, id_normals):\n",
    "    '''\n",
    "    Saves the number of kmers present in spark dataframe after each filtering step in a tabular file\n",
    "    :param output_count: str path for count file of intermediate filtering steps\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param prefix: str information to be added to the result line in an info column\n",
    "    :param cancer_sample_ori: str id of target cancer sample which was filtered\n",
    "    :param mutation_mode: str information about whether mutations where applied or not\n",
    "    :param sample_expr_support_cancer: float normalized expression threshold for the cancer target sample\n",
    "    :param cohort_expr_support_cancer: float normalized expression threshold for the cancer cohort\n",
    "    excluding the target sample\n",
    "    hich should be met in n samples\n",
    "    :param n_samples_lim_cancer: int number of cancer samples in which the cancer cohort expression threshold\n",
    "    should be met\n",
    "    :param cohort_expr_support_normal: float normalized expression threshold for the normal cohort\n",
    "    required in any sample (>=1)\n",
    "    :param n_samples_lim_normal: int number of normal samples in which any number of reads is required (>0)\n",
    "    :param id_normals: str id of the normal cohort (example gtex)\n",
    "    '''\n",
    "    pipeline = 'peptide-based'\n",
    "    if output_count:\n",
    "        header = (f'{\"sample\"}\\t{\"mutation_mode\"}\\t{\"pipeline\"}\\t{\"min_sample_reads\"}\\t{\"#_of_cohort_samples\"}\\t'\n",
    "                  f'{\"reads_per_cohort_sample\"}\\t{\"#_normal_samples_allowed\"}\\t{\"normal_cohort_id\"}'\n",
    "                  f'\\t{\"reads_per_normal_sample\"}')\n",
    "        line =   (f'{cancer_sample_ori}\\t{mutation_mode}\\t{pipeline}\\t{sample_expr_support_cancer}\\t{n_samples_lim_cancer}'\n",
    "                  f'\\t{cohort_expr_support_cancer}\\t{n_samples_lim_normal}\\t{id_normals}'\n",
    "                  f'\\t{cohort_expr_support_normal}')\n",
    "\n",
    "        for idx in np.arange(len(report_count)):\n",
    "            header += f'\\t{report_steps[idx]}'\n",
    "            line += f'\\t{report_count[idx]}'\n",
    "        if prefix:\n",
    "            header += f'\\t{\"info\"}'\n",
    "            line += f'\\t{prefix}'\n",
    "        header += \"\\n\"\n",
    "        line += \"\\n\"\n",
    "        \n",
    "#         print(header, len(header.split('\\t')))\n",
    "#         print(line, len(line.split('\\t')))\n",
    "        if not os.path.exists(output_count):\n",
    "            with open(output_count,\"w\") as f:\n",
    "                f.write(header)\n",
    "        with open(output_count, \"a\") as f:\n",
    "            f.write(line)\n",
    "        logging.info(f'Save intermediate info to {output_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "bd671a7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Julianne sample  mutation_mode   pipeline        min_sample_reads        #_of_cohort_samples     reads_per_cohort_sample #_normal_samples_allowed        normal_cohort_id        reads_per_normal_sample motif_filter    Init_cancer     Filter_Sample   Filter_Sample_Cohort    Filter_Sample_Cohort_CohortBackground   Filter_Motif    Filter_Sample_Cohort_CohortBackground_Uniprot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fcf0675",
   "metadata": {},
   "source": [
    "# MAIN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91281182",
   "metadata": {},
   "source": [
    "Note: Foreground joined with GTEX table \\ 79 genes \n",
    "Genes excluded is a list of 215 genes for now. Should be fine "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "74f4911d",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_type = 'brca'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "33bd2aad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inputs\n",
    "\n",
    "if run_type == 'brca':\n",
    "    target_samples = ['TCGA-C8-A12P-01A-11R-A115-07.all',\n",
    "                      'TCGA-AO-A0JM-01A-21R-A056-07.all',\n",
    "                      'TCGA-BH-A18V-01A-11R-A12D-07.all',\n",
    "                      'TCGA-A2-A0D2-01A-21R-A034-07.all',\n",
    "                      'TCGA-A2-A0SX-01A-12R-A084-07.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz')\n",
    "elif run_type == 'ov':\n",
    "    target_samples = ['TCGA-25-1319-01A-01R-1565-13.all',\n",
    "                      'TCGA-25-1313-01A-01R-1565-13.all',\n",
    "                      'TCGA-61-2008-01A-02R-1568-13.all',\n",
    "                      'TCGA-24-1431-01A-01R-1566-13.all',\n",
    "                      'TCGA-24-2298-01A-01R-1569-13.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz')\n",
    "\n",
    "\n",
    "# Outputs\n",
    "filtering_id = 'chosen_filters_06March_order_wAnnot'\n",
    "\n",
    "output_dir = os.path.join(basedir, 'filtering_samples', filtering_id)\n",
    "pathlib.Path(output_dir).mkdir(exist_ok=True, parents=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "08c6b40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Discussion 02/22 Choices\n",
    "# BACKGROUND cohorts we do (cohort_reads, sample_number)- KEEP pipeline as such\n",
    "# cohort_reads=[0,1,3]\n",
    "# sample_number=[1,2,10]\n",
    "# FOREGROUND  (cohort_reads, sample_number) means\n",
    "# cohort_reads=[0,2]\n",
    "# sample_number(rest of cohort) =[1, 5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e9c1eddc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters\n",
    "Threshold_target = [0.0]\n",
    "Threshold_cancer_cohort = [0.0, 2.0, None] # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_cancer = [1, 5, None] # choices 1 to 1102 for BRCA and 374 for OV   \n",
    "\n",
    "Threshold_normal_cohort = [0.0, 1.0, 3.0]   # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_normal = [1, 2, 10] #choices 1 to max number of samples in Normal whitelist\n",
    "\n",
    "tag_cancer = 'cancerCohort'\n",
    "tag_normal = 'gtexCohort'\n",
    "\n",
    "\n",
    "tag_prefix = 'G_'\n",
    "mutation_mode = 'ref'\n",
    "save_tag = 'GtexCohort'\n",
    "\n",
    "metadata_save = ['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated']\n",
    "\n",
    "filter_annot = False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "c38d0281",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(22348747, 24)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load matrix to be filtered\n",
    "df_load = pd.read_csv(intermediate_output, sep = '\\t')\n",
    "print(f'Loaded {intermediate_output}')\n",
    "df_load = df_load.rename({'batch': f'batch_{run_type}'}, axis = 1)\n",
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "f3d368b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>3290.0</td>\n",
       "      <td>1969.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>572.0</td>\n",
       "      <td>545.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISSQSRVEK</td>\n",
       "      <td>278.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>92379851:92379859:92493866:92493885:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSGDEEKYP</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92600493:92600508:92611313:92611325:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>641</td>\n",
       "      <td>520</td>\n",
       "      <td>348</td>\n",
       "      <td>170</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLKMKMFQI</td>\n",
       "      <td>146.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379850:92379859:92496416:92496434:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  WYITRSGIA                 3290.0                  1969.0   \n",
       "1  WYITRSGIA                  572.0                   545.0   \n",
       "2  ISSQSRVEK                  278.0                   261.0   \n",
       "3  RSGDEEKYP                 7350.0                  4622.0   \n",
       "4  HLKMKMFQI                  146.0                   136.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   861.0                   394.0                    89.0   \n",
       "1                   243.0                   130.0                    27.0   \n",
       "2                    82.0                    29.0                     7.0   \n",
       "3                  2653.0                  1745.0                  1126.0   \n",
       "4                    42.0                    20.0                     3.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92347505:92347506:92349915:92349941:None:None   \n",
       "1                      2.0  92347505:92347506:92349915:92349941:None:None   \n",
       "2                      0.0  92379851:92379859:92493866:92493885:None:None   \n",
       "3                    734.0  92600493:92600508:92611313:92611325:None:None   \n",
       "4                      1.0  92379850:92379859:92496416:92496434:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0              False               False  ...                       138   \n",
       "1              False               False  ...                       138   \n",
       "2              False               False  ...                        10   \n",
       "3               True                True  ...                       641   \n",
       "4              False               False  ...                        16   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                        82                        26   \n",
       "1                        82                        26   \n",
       "2                         2                         0   \n",
       "3                       520                       348   \n",
       "4                         2                         0   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGAC8A12P01A11RA11507all  \\\n",
       "0                          5                   0.000000   \n",
       "1                          5                   0.000000   \n",
       "2                          0                   0.000000   \n",
       "3                        170                   2.922641   \n",
       "4                          0                   2.922641   \n",
       "\n",
       "   TCGAAOA0JM01A21RA05607all  TCGABHA18V01A11RA12D07all  \\\n",
       "0                   0.000000                   0.000000   \n",
       "1                   0.000000                   0.000000   \n",
       "2                   0.000000                   2.474321   \n",
       "3                   2.102386                   1.237161   \n",
       "4                   0.000000                   0.000000   \n",
       "\n",
       "   TCGAA2A0D201A21RA03407all  TCGAA2A0SX01A12RA08407all  isAnnotated  \n",
       "0                   4.102634                        0.0          NaN  \n",
       "1                   4.102634                        0.0          NaN  \n",
       "2                   0.000000                        0.0          NaN  \n",
       "3                   0.000000                        0.0          1.0  \n",
       "4                   0.000000                        0.0          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2331bcea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22348747, 24)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "76817af3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- processing TCGAC8A12P01A11RA11507all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_filtered_df_TCGA-C8-A12P-01A-11R-A115-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 758 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 765 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 991 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 760 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 767 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 993 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 883 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 890 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594740 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1062 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 438 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 439 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 550 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 440 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 441 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 552 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 534 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 535 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 592771 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 620 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 687 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 694 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 873 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 689 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 696 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 875 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 790 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 797 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594185 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 930 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 328 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 329 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 406 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 330 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 331 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 408 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 344 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 345 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 585082 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 415 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 895 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 902 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1166 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 907 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 914 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1168 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1037 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1044 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 594980 kmers\n",
      "# Filter_Sample n = 594980 kmers\n",
      "# Filter_Sample_Cohort n = 594980 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1237 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-C8-A12P-01A-11R-A115-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGAAOA0JM01A21RA05607all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_filtered_df_TCGA-AO-A0JM-01A-21R-A056-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 534 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 555 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 740 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 534 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 555 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 740 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 607 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 607 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 475282 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 770 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 294 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 312 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 355 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 294 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 312 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 355 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 347 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 347 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 473818 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 375 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 422 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 443 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 557 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 422 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 443 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 557 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 487 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 487 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 474548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 586 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 166 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 184 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 208 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 166 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 184 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 208 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 199 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 199 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 469884 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 215 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1121 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1382 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1121 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1382 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1188 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1189 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1412 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGABHA18V01A11RA12D07all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_filtered_df_TCGA-BH-A18V-01A-11R-A12D-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 943 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 968 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1416 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 945 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 970 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1418 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1194 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1195 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1572 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 490 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 499 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 702 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 492 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 501 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 704 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 664 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 665 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 699618 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 826 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 875 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 900 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1278 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 877 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 902 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1280 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1097 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1098 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 701902 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1421 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 319 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 327 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 388 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 321 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 329 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 390 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 356 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 356 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 684842 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 410 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1135 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1175 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1698 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1137 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1177 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1700 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1428 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1430 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 703627 kmers\n",
      "# Filter_Sample n = 703627 kmers\n",
      "# Filter_Sample_Cohort n = 703627 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1858 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGAA2A0D201A21RA03407all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_filtered_df_TCGA-A2-A0D2-01A-21R-A034-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 758 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 774 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1141 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 764 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 774 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1141 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 934 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 942 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633052 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1237 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 423 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 429 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 569 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 429 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 429 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 569 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 561 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 561 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 630667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 631 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 686 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 693 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1035 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 692 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 693 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1035 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 848 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 848 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 632423 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1119 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 251 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 251 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 270 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 251 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 251 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 270 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 276 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 276 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 621176 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 288 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 872 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 890 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1301 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 878 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 890 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1301 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1052 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1060 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 633276 kmers\n",
      "# Filter_Sample n = 633276 kmers\n",
      "# Filter_Sample_Cohort n = 633276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1397 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0D2-01A-21R-A034-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGAA2A0SX01A12RA08407all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_filtered_df_TCGA-A2-A0SX-01A-12R-A084-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 799 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 804 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1080 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 802 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 806 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1082 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 912 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 915 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 707769 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1160 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 476 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 479 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 626 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 479 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 481 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 628 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 532 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 533 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 704701 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 674 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 707 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 712 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 985 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 710 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 714 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 987 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 816 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 819 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 706706 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1064 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 292 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 293 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 378 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 295 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 295 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 380 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 309 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 309 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 691861 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 394 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1018 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1048 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1399 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1036 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1056 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1401 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1169 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1172 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 708211 kmers\n",
      "# Filter_Sample n = 708211 kmers\n",
      "# Filter_Sample_Cohort n = 708211 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1479 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/chosen_filters_06March_order_wAnnot/G_TCGA-A2-A0SX-01A-12R-A084-07_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n"
     ]
    }
   ],
   "source": [
    "for cancer_sample_ori in target_samples: # TODO update\n",
    "    # Sample naming\n",
    "    target_sample = cancer_sample_ori.replace('-', '').replace('.', '')\n",
    "    cancer_sample_ori = cancer_sample_ori.replace('.all', '')\n",
    "    print(f'-------- processing {target_sample} -------- \\n')\n",
    "    \n",
    "    # Summary file for sample\n",
    "    summary_file = f'{tag_prefix}filtered_df_{cancer_sample_ori}_samp_chrt_norm_mot.tsv'\n",
    "    summary_path = os.path.join(output_dir, summary_file)\n",
    "    print(f'Saving to summary file {summary_path}')\n",
    "    \n",
    "\n",
    "    df_expr = []\n",
    "    report_count = [] \n",
    "    report_steps = []\n",
    "    for threshold_target in Threshold_target:\n",
    "        for threshold_cancer_cohort in Threshold_cancer_cohort:\n",
    "            for n_samples_cancer in N_samples_cancer:\n",
    "                for threshold_normal_cohort in Threshold_normal_cohort:\n",
    "                    for n_samples_normal in N_samples_normal:\n",
    "                        if (n_samples_cancer is None) ^ (threshold_cancer_cohort is None):\n",
    "                            continue\n",
    "\n",
    "                        adjusted_threshold_col = 'tmp_cancer_cohort'\n",
    "                        max_threshold_col = 'tmp_normal_Nmax_sup{}'.format(threshold_normal_cohort)\n",
    "                        max_threshold_col_base = 'tmp_normal_Nmax_sup{}'.format(0)\n",
    "\n",
    "\n",
    "                        df = df_load.copy()\n",
    "                        # Make correction for number samples passing theshold in cohort. We want to exclude the target sample in counting\n",
    "                        if (n_samples_cancer is not None) and (threshold_cancer_cohort is not None):\n",
    "                            df[adjusted_threshold_col] = df[get_threshold_colname(threshold_cancer_cohort, tag_cancer)]\n",
    "                            df.loc[df[target_sample] >= threshold_cancer_cohort, adjusted_threshold_col] -=1 \n",
    "\n",
    "                        # Number of kmers expressed in sample \n",
    "                        df = filter_single_col(df, 0, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Init_Sample')\n",
    "\n",
    "                        # Number of kmers >= threshold in sample \n",
    "                        df = filter_single_col(df, threshold_target, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample')\n",
    "\n",
    "                        \n",
    "                        # Filter for cancer cancer cohort \n",
    "                        if (n_samples_cancer is not None) and (threshold_cancer_cohort is not None):\n",
    "                            df = filter_cancer_cohort(df, n_samples_cancer, adjusted_threshold_col)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort')\n",
    "                            \n",
    "\n",
    "\n",
    "                        # Expression in gtex cohort >= threshold \n",
    "                        recurrence_custom =  max_recurrence_over_kmer(df, \n",
    "                                                                      get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                                      max_threshold_col)\n",
    "\n",
    "                        # Expression in gtex cohort > 0  \n",
    "                        recurrence_custom_base = max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(0.0, tag_normal),\n",
    "                                                                          max_threshold_col_base) \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "                        # Perform Background filtering \n",
    "                        df = df.merge(recurrence_custom, on = 'kmer', how = 'left')\n",
    "                        df = df.merge(recurrence_custom_base, on = 'kmer', how = 'left')\n",
    "\n",
    "                        df = df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), :]\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal')\n",
    "                    \n",
    "                        #Perform Annotated junctions filtering \n",
    "                        if filter_annot:\n",
    "                            df = df[df['isAnnotated'].isna()]\n",
    "                            output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot')\n",
    "\n",
    "#                         # DEV: Exclude genes where GTEX is missing\n",
    "#                         df = df.loc[df['exclude'].isna()]\n",
    "#                         output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot_EXPGTEX')\n",
    "\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        # Save outputs \n",
    "                        # outpaths\n",
    "                        base_path_final = os.path.join(output_dir,\n",
    "                                                       (f'{tag_prefix}{cancer_sample_ori}_'\n",
    "                                                        f'SampleLim{threshold_target}'\n",
    "                                                        f'CohortLim{threshold_cancer_cohort}'\n",
    "                                                        f'Across{n_samples_cancer}_'\n",
    "                                                        f'FiltNormals{save_tag}'\n",
    "                                                        f'Cohortlim{threshold_normal_cohort}'\n",
    "                                                        f'Across{n_samples_normal}.tsv.gz'))\n",
    "                        print(f'Saving outputs to: {base_path_final} \\n')\n",
    "                        df.loc[:, metadata_save].to_csv(base_path_final, compression = 'gzip', index = None, sep = '\\t')\n",
    "\n",
    "\n",
    "    save_output_count(summary_path, report_count, report_steps, '', cancer_sample_ori, mutation_mode,\n",
    "                      threshold_target, threshold_cancer_cohort, n_samples_cancer,\n",
    "                          threshold_normal_cohort, n_samples_normal, save_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "279ecb2f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5544c58b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b148871d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98ef99e2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74b560d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'KVDTLGKST'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d85cd18",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f979d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            ['kmer', 'coord', 'exclude', 'batch_brca', 'junctionAnnotated', 'readFrameAnnotated', target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01993640",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer, 'batch_brca']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63aba557",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_batch.loc[(gene_to_batch['batch_brca'] == 54877)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c54267e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = df_load.merge(gene_to_batch[['exclude', f'batch_{run_type}']], on = f'batch_{run_type}', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ada9f080",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo.loc[(foo['kmer'] == kmer)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b35b9cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "final  =df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), \n",
    "       ['kmer', get_threshold_colname(threshold_cancer_cohort, tag_cancer)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97464fa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3d05819",
   "metadata": {},
   "outputs": [],
   "source": [
    "final.loc[final['cancerCohortfilter >0.0']> 403 , ['kmer', 'tmp_cancer_cohort',f'batch_{run_type}'] ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "455fb619",
   "metadata": {},
   "outputs": [],
   "source": [
    "final['cancerCohortfilter >0.0'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3a9ec1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea032687",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            [target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
