{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cab1d99c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Matplotlib created a temporary config/cache directory at /scratch/slurm-job.1322664/matplotlib-mwyzp24e because the default path (/cluster/customapps/biomed/grlab/users/prelotla/.cache/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import timeit\n",
    "import glob \n",
    "import pandas as pd\n",
    "import time\n",
    "import multiprocessing as mp \n",
    "import logging\n",
    "import sys \n",
    "import pathlib\n",
    "from pathlib import Path\n",
    "import matplotlib.pyplot as plt \n",
    "from helpers_ffi import * "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a8f2e8d",
   "metadata": {},
   "source": [
    "# Order \n",
    "- Uniprot \n",
    "- Sample init\n",
    "- Sample expression \n",
    "- Sample cohort \n",
    "- annotation \n",
    "- GTEX?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fcf5e3ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def filter_cancer_cohort(df, n_samples, threshold_column ):\n",
    "#     df = df.loc[df[threshold_column] >= n_samples]    \n",
    "#     return df\n",
    "        \n",
    "\n",
    "# def get_threshold_colname(threshold, tag):\n",
    "#     if (threshold is not None) and (threshold > 0 ):\n",
    "#         threshold_column = f'{tag}filter >={threshold}'\n",
    "#     else:\n",
    "#         threshold_column = f'{tag}filter >{threshold}'\n",
    "#     return threshold_column\n",
    "\n",
    "# def filter_single_col(df, threshold, colname):\n",
    "#     if threshold:\n",
    "#         df = df.loc[df[colname] >= threshold, :]\n",
    "#     else:\n",
    "#         df = df.loc[df[colname] >threshold, :]\n",
    "#     return df\n",
    "\n",
    "# def max_recurrence_over_kmer(df, threshold_column, new_maxcol):\n",
    "#     df = df[['kmer', threshold_column]].groupby('kmer').max()\n",
    "#     df = df.reset_index()\n",
    "#     df = df.rename({threshold_column: new_maxcol}, axis = 1)\n",
    "#     return df \n",
    "\n",
    "# def output_count(df, report_count, report_step, step_string, perform_count=True):\n",
    "#     '''\n",
    "#     Performs a count operation on the number of kmers present in spark dataframe after a given filtering step\n",
    "#     Note: This operation is expensive but useful if the user is interested in intermediate filtering steps\n",
    "#     :param perform_count: bool whether to perform a count operation\n",
    "#     :param df: dataframe with kmer expression counts\n",
    "#     :param report_count: list to store result of successive counting operations\n",
    "#     :param report_step: list to store name of successive counting operations\n",
    "#     :param step_string: str name of the counting operation\n",
    "#     '''\n",
    "#     if perform_count:\n",
    "#         mycount = df['kmer'].unique().shape[0]\n",
    "#         report_count.append(mycount)\n",
    "#         report_step.append(step_string)\n",
    "#         print(f'# {step_string} n = {mycount} kmers')\n",
    "\n",
    "# def save_output_count(output_count, report_count, report_steps, prefix, cancer_sample_ori, mutation_mode,\n",
    "#                       sample_expr_support_cancer, cohort_expr_support_cancer, n_samples_lim_cancer,\n",
    "#                           cohort_expr_support_normal, n_samples_lim_normal, id_normals):\n",
    "#     '''\n",
    "#     Saves the number of kmers present in spark dataframe after each filtering step in a tabular file\n",
    "#     :param output_count: str path for count file of intermediate filtering steps\n",
    "#     :param report_count: list to store result of successive counting operations\n",
    "#     :param report_step: list to store name of successive counting operations\n",
    "#     :param prefix: str information to be added to the result line in an info column\n",
    "#     :param cancer_sample_ori: str id of target cancer sample which was filtered\n",
    "#     :param mutation_mode: str information about whether mutations where applied or not\n",
    "#     :param sample_expr_support_cancer: float normalized expression threshold for the cancer target sample\n",
    "#     :param cohort_expr_support_cancer: float normalized expression threshold for the cancer cohort\n",
    "#     excluding the target sample\n",
    "#     hich should be met in n samples\n",
    "#     :param n_samples_lim_cancer: int number of cancer samples in which the cancer cohort expression threshold\n",
    "#     should be met\n",
    "#     :param cohort_expr_support_normal: float normalized expression threshold for the normal cohort\n",
    "#     required in any sample (>=1)\n",
    "#     :param n_samples_lim_normal: int number of normal samples in which any number of reads is required (>0)\n",
    "#     :param id_normals: str id of the normal cohort (example gtex)\n",
    "#     '''\n",
    "#     pipeline = 'peptide-based'\n",
    "#     if output_count:\n",
    "#         header = (f'{\"sample\"}\\t{\"mutation_mode\"}\\t{\"pipeline\"}\\t{\"min_sample_reads\"}\\t{\"#_of_cohort_samples\"}\\t'\n",
    "#                   f'{\"reads_per_cohort_sample\"}\\t{\"#_normal_samples_allowed\"}\\t{\"normal_cohort_id\"}'\n",
    "#                   f'\\t{\"reads_per_normal_sample\"}')\n",
    "#         line =   (f'{cancer_sample_ori}\\t{mutation_mode}\\t{pipeline}\\t{sample_expr_support_cancer}\\t{n_samples_lim_cancer}'\n",
    "#                   f'\\t{cohort_expr_support_cancer}\\t{n_samples_lim_normal}\\t{id_normals}'\n",
    "#                   f'\\t{cohort_expr_support_normal}')\n",
    "\n",
    "#         for idx in np.arange(len(report_count)):\n",
    "#             header += f'\\t{report_steps[idx]}'\n",
    "#             line += f'\\t{report_count[idx]}'\n",
    "#         if prefix:\n",
    "#             header += f'\\t{\"info\"}'\n",
    "#             line += f'\\t{prefix}'\n",
    "#         header += \"\\n\"\n",
    "#         line += \"\\n\"\n",
    "        \n",
    "# #         print(header, len(header.split('\\t')))\n",
    "# #         print(line, len(line.split('\\t')))\n",
    "#         if not os.path.exists(output_count):\n",
    "#             with open(output_count,\"w\") as f:\n",
    "#                 f.write(header)\n",
    "#         with open(output_count, \"a\") as f:\n",
    "#             f.write(line)\n",
    "#         logging.info(f'Save intermediate info to {output_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bd671a7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Julianne sample  mutation_mode   pipeline        min_sample_reads        #_of_cohort_samples     reads_per_cohort_sample #_normal_samples_allowed        normal_cohort_id        reads_per_normal_sample motif_filter    Init_cancer     Filter_Sample   Filter_Sample_Cohort    Filter_Sample_Cohort_CohortBackground   Filter_Motif    Filter_Sample_Cohort_CohortBackground_Uniprot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fcf0675",
   "metadata": {},
   "source": [
    "# MAIN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91281182",
   "metadata": {},
   "source": [
    "Note: Foreground joined with GTEX table \\ 79 genes \n",
    "Genes excluded is a list of 215 genes for now. Should be fine "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "74f4911d",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_type = 'ov'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "33bd2aad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inputs\n",
    "\n",
    "if run_type == 'brca':\n",
    "    target_samples = ['TCGA-AO-A0JM-01A-21R-A056-07.all',\n",
    "                      'TCGA-C8-A12P-01A-11R-A115-07.all',\n",
    "                      'TCGA-BH-A18V-01A-11R-A12D-07.all',\n",
    "                      'TCGA-A2-A0D2-01A-21R-A034-07.all',\n",
    "                      'TCGA-A2-A0SX-01A-12R-A084-07.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r_complete.tsv.gz')\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r_complete.STAR_GTEx.tsv.gz')\n",
    "elif run_type == 'ov':\n",
    "    target_samples = ['TCGA-25-1319-01A-01R-1565-13.all'] #,\n",
    "#                       'TCGA-25-1313-01A-01R-1565-13.all',\n",
    "#                       'TCGA-61-2008-01A-02R-1568-13.all',\n",
    "#                       'TCGA-24-1431-01A-01R-1566-13.all',\n",
    "#                       'TCGA-24-2298-01A-01R-1569-13.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz')\n",
    "   # intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r_complete.STAR_GTEx.tsv.gz')\n",
    "\n",
    "\n",
    "# Outputs\n",
    "filtering_id = 'filters_16_May_QUICK_INIT'\n",
    "\n",
    "output_dir = os.path.join(basedir, 'filtering_samples', filtering_id)\n",
    "pathlib.Path(output_dir).mkdir(exist_ok=True, parents=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "08c6b40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Discussion 02/22 Choices\n",
    "# BACKGROUND cohorts we do (cohort_reads, sample_number)- KEEP pipeline as such\n",
    "# cohort_reads=[0,1,3, None]\n",
    "# sample_number=[1,2,10, None]\n",
    "# FOREGROUND  (cohort_reads, sample_number) means\n",
    "# cohort_reads=[0,2]\n",
    "# sample_number(rest of cohort) =[1, 5]\n",
    "# Sample reads = [0]\n",
    "# + A case with not foreground"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e9c1eddc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters\n",
    "Threshold_target = [0.0]\n",
    "Threshold_cancer_cohort = [0.0] #, 2.0] # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0] -> Removed None #TODO\n",
    "N_samples_cancer = [None] #, 1, 5] # choices 1 to 1102 for BRCA and 374 for OV   \n",
    "\n",
    "Threshold_normal_cohort = [0.0 ] #, 1.0, 3.0, None] #NONE MEANS ANY  # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_normal = [1] #, 2, 10, None] #choices 1 to max number of samples in Normal whitelist\n",
    "\n",
    "tag_cancer = 'cancerCohort'\n",
    "tag_normal = 'gtexCohort'\n",
    "\n",
    "\n",
    "tag_prefix = 'G_'\n",
    "mutation_mode = 'ref'\n",
    "save_tag = 'GtexCohort'\n",
    "\n",
    "metadata_save = ['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated']\n",
    "\n",
    "filter_annot = False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c38d0281",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(46132623, 24)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load matrix to be filtered\n",
    "df_load = pd.read_csv(intermediate_output, sep = '\\t')\n",
    "print(f'Loaded {intermediate_output}')\n",
    "df_load = df_load.rename({'batch': f'batch_{run_type}'}, axis = 1)\n",
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f3d368b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1218.0</td>\n",
       "      <td>689.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>2025.0</td>\n",
       "      <td>479.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>8641.0</td>\n",
       "      <td>6957.0</td>\n",
       "      <td>4789.0</td>\n",
       "      <td>3211.0</td>\n",
       "      <td>1347.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  RKSTQMPCT                 1218.0                   689.0   \n",
       "1  RKSTQMPCT                 1279.0                   274.0   \n",
       "2  RKSTQMPCT                 7350.0                  4622.0   \n",
       "3  RKSTQMPCT                 2025.0                   479.0   \n",
       "4  RKSTQMPCT                 8641.0                  6957.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   249.0                    89.0                    26.0   \n",
       "1                    70.0                    28.0                     5.0   \n",
       "2                  2653.0                  1745.0                  1126.0   \n",
       "3                    96.0                    35.0                     8.0   \n",
       "4                  4789.0                  3211.0                  1347.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92379857:92379859:92611313:92611338:None:None   \n",
       "1                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "2                    734.0  92379857:92379859:92611313:92611338:None:None   \n",
       "3                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "4                    191.0  92379857:92379859:92611313:92611338:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0               True               False  ...                       372   \n",
       "1               True               False  ...                       372   \n",
       "2               True               False  ...                       372   \n",
       "3               True               False  ...                       372   \n",
       "4               True               False  ...                       372   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                       367                       352   \n",
       "1                       367                       352   \n",
       "2                       367                       352   \n",
       "3                       367                       352   \n",
       "4                       367                       352   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "0                        295                  20.982616   \n",
       "1                        295                  20.982616   \n",
       "2                        295                  20.982616   \n",
       "3                        295                  20.982616   \n",
       "4                        295                  20.982616   \n",
       "\n",
       "   TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "0                   8.422932                  17.304198   \n",
       "1                   8.422932                  17.304198   \n",
       "2                   8.422932                  17.304198   \n",
       "3                   8.422932                  17.304198   \n",
       "4                   8.422932                  17.304198   \n",
       "\n",
       "   TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \n",
       "0                  16.069005                  19.585264          NaN  \n",
       "1                  16.069005                  19.585264          NaN  \n",
       "2                  16.069005                  19.585264          NaN  \n",
       "3                  16.069005                  19.585264          NaN  \n",
       "4                  16.069005                  19.585264          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2331bcea",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(46132623, 24)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b149c73",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "76817af3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- processing TCGA25131901A01R156513all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/filters_16_May_QUICK_INIT/G_filtered_df_TCGA-25-1319-01A-01R-1565-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 113574 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/filters_16_May_QUICK_INIT/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimAnyAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# PREPARE DATA\n",
    "for threshold_normal_cohort in Threshold_normal_cohort:\n",
    "    if threshold_normal_cohort is not None: \n",
    "        reformat_col =  get_threshold_colname(threshold_normal_cohort, tag_normal)\n",
    "        df_load[reformat_col] = df_load[reformat_col].fillna(0.0)\n",
    "\n",
    "for threshold_cancer_cohort in Threshold_cancer_cohort:\n",
    "    if threshold_cancer_cohort is not None: \n",
    "        reformat_col =  get_threshold_colname(threshold_cancer_cohort, tag_cancer)\n",
    "        df_load[reformat_col] = df_load[reformat_col].fillna(0.0)\n",
    "        \n",
    "# FILTER\n",
    "for cancer_sample_ori in target_samples: # TODO update\n",
    "    # Sample naming\n",
    "    target_sample = cancer_sample_ori.replace('-', '').replace('.', '')\n",
    "    cancer_sample_ori = cancer_sample_ori.replace('.all', '')\n",
    "    print(f'-------- processing {target_sample} -------- \\n')\n",
    "    \n",
    "    # Summary file for sample\n",
    "    summary_file = f'{tag_prefix}filtered_df_{cancer_sample_ori}_samp_chrt_norm_mot.tsv'\n",
    "    summary_path = os.path.join(output_dir, summary_file)\n",
    "    print(f'Saving to summary file {summary_path}')\n",
    "    \n",
    "\n",
    "    for threshold_target in Threshold_target:\n",
    "        for threshold_cancer_cohort in Threshold_cancer_cohort:\n",
    "            for n_samples_cancer in N_samples_cancer:\n",
    "                for threshold_normal_cohort in Threshold_normal_cohort:\n",
    "                    for n_samples_normal in N_samples_normal:\n",
    "                        \n",
    "                        df_expr = []\n",
    "                        report_count = [] \n",
    "                        report_steps = []\n",
    "                        if (n_samples_cancer is None) and (threshold_cancer_cohort != 0): #Only None, None as condition \n",
    "                            continue\n",
    "                        if (n_samples_normal is None) and (threshold_normal_cohort is None):\n",
    "                            continue\n",
    "\n",
    "                        adjusted_threshold_col = 'tmp_cancer_cohort'\n",
    "                        max_threshold_col = 'tmp_normal_Nmax_sup{}'.format(threshold_normal_cohort)\n",
    "                        max_threshold_col_base = 'tmp_normal_Nmax_sup{}'.format(0.0) #TODO\n",
    "\n",
    "\n",
    "                        df = df_load.copy()\n",
    "                        \n",
    "                        # Expression in gtex cohort >= threshold \n",
    "                        if threshold_normal_cohort is not None:\n",
    "                            recurrence_custom =  max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                                          max_threshold_col)\n",
    "\n",
    "                        # Expression in gtex cohort > 0  \n",
    "                        recurrence_custom_base = max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(0.0, tag_normal),\n",
    "                                                                          max_threshold_col_base) \n",
    "                        \n",
    "                        # Make correction for number samples passing theshold in cohort. We want to exclude the target sample in counting\n",
    "                        if (n_samples_cancer is not None): \n",
    "                            df[adjusted_threshold_col] = df[get_threshold_colname(threshold_cancer_cohort, tag_cancer)]\n",
    "                            df.loc[df[target_sample] >= threshold_cancer_cohort, adjusted_threshold_col] -=1 \n",
    "\n",
    "                        # Number of kmers expressed in sample \n",
    "                        df = filter_single_col(df, 0, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Init_Sample')\n",
    "\n",
    "                        # Number of kmers >= threshold in sample \n",
    "                        df = filter_single_col(df, threshold_target, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample')\n",
    "\n",
    "                        \n",
    "                        # Filter for cancer cancer cohort \n",
    "                        if (n_samples_cancer is not None): #skip if both \n",
    "                            df = filter_cancer_cohort(df, n_samples_cancer, adjusted_threshold_col)\n",
    "                        if threshold_cancer_cohort: \n",
    "                            threshold_cancer_cohort_save = threshold_cancer_cohort \n",
    "                        else: \n",
    "                            threshold_cancer_cohort_save = 'Any'\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort')\n",
    "                            \n",
    "\n",
    "\n",
    "                        # Perform Background filtering \n",
    "                        if threshold_normal_cohort is not None:\n",
    "                            df = df.merge(recurrence_custom, on = 'kmer', how = 'left')\n",
    "                        df = df.merge(recurrence_custom_base, on = 'kmer', how = 'left')\n",
    "                        \n",
    "                        n_samples_normal_save = n_samples_normal\n",
    "                        threshold_normal_cohort_save = threshold_normal_cohort\n",
    "                        if (threshold_normal_cohort is not None) and (n_samples_normal is not None):\n",
    "                            df = df.loc[ ~ ((df[max_threshold_col] >= 1) | (df[max_threshold_col_base] >= n_samples_normal)), :]\n",
    "                        elif threshold_normal_cohort is None:\n",
    "                            df = df.loc[ ~ (df[max_threshold_col_base] >= n_samples_normal), :]\n",
    "                            threshold_normal_cohort_save = 'Any'\n",
    "                        elif n_samples_normal is None:\n",
    "                            df = df.loc[ ~ (df[max_threshold_col] >= 1), :]\n",
    "                            n_samples_normal_save = 'Any'\n",
    "\n",
    "          \n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal')\n",
    "                    \n",
    "                        #Perform Annotated junctions filtering \n",
    "                        if filter_annot:\n",
    "                            df = df[df['isAnnotated'].isna()]\n",
    "                            output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot')\n",
    "\n",
    "#                         # DEV: Exclude genes where GTEX is missing\n",
    "#                         df = df.loc[df['exclude'].isna()]\n",
    "#                         output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot_EXPGTEX')\n",
    "\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        # Save outputs \n",
    "                        # outpaths \n",
    "                        base_path_final = os.path.join(output_dir,\n",
    "                                                       (f'{tag_prefix}{cancer_sample_ori}_'\n",
    "                                                        f'SampleLim{threshold_target}'\n",
    "                                                        f'CohortLim{threshold_cancer_cohort_save}'\n",
    "                                                        f'Across{n_samples_cancer}_'\n",
    "                                                        f'FiltNormals{save_tag}'\n",
    "                                                        f'Cohortlim{threshold_normal_cohort_save}'\n",
    "                                                        f'Across{n_samples_normal_save}.tsv.gz'))\n",
    "                        print(f'Saving outputs to: {base_path_final} \\n')\n",
    "                        df.loc[:, metadata_save].to_csv(base_path_final, compression = 'gzip', index = None, sep = '\\t')\n",
    "\n",
    "                        \n",
    "                        save_output_count(summary_path, report_count, report_steps, '', cancer_sample_ori, mutation_mode,\n",
    "                                          threshold_target, threshold_cancer_cohort_save, n_samples_cancer,\n",
    "                                              threshold_normal_cohort_save, n_samples_normal_save, save_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d070f529",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(59507, 32)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[ ~ ((df[max_threshold_col] >= 1) | (df[max_threshold_col_base] >= n_samples_normal)), :].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90a349b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[ ~ ((df[max_threshold_col] >= 1) | (df[max_threshold_col_base] >= n_samples_normal)), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "4f8bc1fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21942693"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((df[max_threshold_col] >= 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "3c2d212b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "111130"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((df[max_threshold_col_base] >= n_samples_normal))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8a7cc877",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_samples_normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "f64c5353",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'TAMKAAMAL'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "5a291c2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = df.loc[ ~ ((df[max_threshold_col] >= 1) | (df[max_threshold_col_base] >= n_samples_normal)), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1601ed59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch_ov</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>343168</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343169</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514542</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966376:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>62</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514939</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966358:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>32</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966358</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627581</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966376:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>62</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>+</td>\n",
       "      <td>chr19</td>\n",
       "      <td>1054.0</td>\n",
       "      <td>1018.0</td>\n",
       "      <td>444.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer                                              coord  \\\n",
       "343168  TAMKAAMAL      48965782:48965784:48966315:48966340:None:None   \n",
       "343169  TAMKAAMAL      48965782:48965784:48966315:48966340:None:None   \n",
       "514542  TAMKAAMAL  48965782:48965784:48966315:48966334:48966376:4...   \n",
       "514939  TAMKAAMAL  48965782:48965784:48966315:48966334:48966358:4...   \n",
       "627581  TAMKAAMAL  48965782:48965784:48966315:48966334:48966376:4...   \n",
       "\n",
       "        junctionAnnotated  readFrameAnnotated  isCrossJunction  batch_ov  \\\n",
       "343168              False                True             True     29205   \n",
       "343169              False               False             True     29205   \n",
       "514542              False               False             True     29205   \n",
       "514939              False               False             True     29205   \n",
       "627581              False               False             True     29205   \n",
       "\n",
       "        cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "343168                      148                        89   \n",
       "343169                      148                        89   \n",
       "514542                       62                        24   \n",
       "514939                       32                        17   \n",
       "627581                       62                        24   \n",
       "\n",
       "        cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "343168                        43                        24  ...   \n",
       "343169                        43                        24  ...   \n",
       "514542                        11                         5  ...   \n",
       "514939                         2                         1  ...   \n",
       "627581                        11                         5  ...   \n",
       "\n",
       "        junction_coordinate2  junction_coordinate  strand_STAR    chr  \\\n",
       "343168                  None                  NaN          NaN    NaN   \n",
       "343169                  None                  NaN          NaN    NaN   \n",
       "514542     48966334:48966376                  NaN          NaN    NaN   \n",
       "514939     48966334:48966358                  NaN          NaN    NaN   \n",
       "627581     48966334:48966376    48966334:48966376            +  chr19   \n",
       "\n",
       "        gtexCohortfilter >0.0  gtexCohortfilter >=1.0  gtexCohortfilter >=2.0  \\\n",
       "343168                    0.0                     NaN                     NaN   \n",
       "343169                    0.0                     NaN                     NaN   \n",
       "514542                    0.0                     NaN                     NaN   \n",
       "514939                    0.0                     NaN                     NaN   \n",
       "627581                 1054.0                  1018.0                   444.0   \n",
       "\n",
       "        gtexCohortfilter >=3.0 gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "343168                     NaN                    NaN                     NaN  \n",
       "343169                     NaN                    NaN                     NaN  \n",
       "514542                     NaN                    NaN                     NaN  \n",
       "514939                     NaN                    NaN                     NaN  \n",
       "627581                   235.0                   94.0                    32.0  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "750ce08b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch_ov</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>tmp_normal_Nmax_sup0.0</th>\n",
       "      <th>tmp_normal_Nmax_sup0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92103</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92104</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            kmer                                          coord  \\\n",
       "92103  TAMKAAMAL  48965782:48965784:48966315:48966340:None:None   \n",
       "92104  TAMKAAMAL  48965782:48965784:48966315:48966340:None:None   \n",
       "\n",
       "       junctionAnnotated  readFrameAnnotated  isCrossJunction  batch_ov  \\\n",
       "92103              False                True             True     29205   \n",
       "92104              False               False             True     29205   \n",
       "\n",
       "       cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "92103                      148                        89   \n",
       "92104                      148                        89   \n",
       "\n",
       "       cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  strand_STAR  \\\n",
       "92103                        43                        24  ...          NaN   \n",
       "92104                        43                        24  ...          NaN   \n",
       "\n",
       "       chr  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "92103  NaN                    0.0                     NaN   \n",
       "92104  NaN                    0.0                     NaN   \n",
       "\n",
       "       gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "92103                     NaN                     NaN                     NaN   \n",
       "92104                     NaN                     NaN                     NaN   \n",
       "\n",
       "       gtexCohortfilter >=10.0 tmp_normal_Nmax_sup0.0 tmp_normal_Nmax_sup0  \n",
       "92103                      NaN                    0.0                  0.0  \n",
       "92104                      NaN                    0.0                  0.0  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "foo.loc[foo['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "e2233197",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "      <th>tmp_normal_Nmax_sup0.0</th>\n",
       "      <th>tmp_normal_Nmax_sup0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19096420</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.839305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>1.45076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19098779</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.839305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>1.45076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "19096420  TAMKAAMAL                    0.0                     NaN   \n",
       "19098779  TAMKAAMAL                    0.0                     NaN   \n",
       "\n",
       "          gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  \\\n",
       "19096420                     NaN                     NaN   \n",
       "19098779                     NaN                     NaN   \n",
       "\n",
       "          gtexCohortfilter >=5.0  gtexCohortfilter >=10.0  \\\n",
       "19096420                     NaN                      NaN   \n",
       "19098779                     NaN                      NaN   \n",
       "\n",
       "                                                  coord  junctionAnnotated  \\\n",
       "19096420  48965782:48965784:48966315:48966340:None:None              False   \n",
       "19098779  48965782:48965784:48966315:48966340:None:None              False   \n",
       "\n",
       "          readFrameAnnotated  ...  cancerCohortfilter >=5.0  \\\n",
       "19096420                True  ...                         3   \n",
       "19098779               False  ...                         3   \n",
       "\n",
       "          cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "19096420                          0                   0.839305   \n",
       "19098779                          0                   0.839305   \n",
       "\n",
       "          TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "19096420                        0.0                        0.0   \n",
       "19098779                        0.0                        0.0   \n",
       "\n",
       "          TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \\\n",
       "19096420                   1.147786                    1.45076          NaN   \n",
       "19098779                   1.147786                    1.45076          NaN   \n",
       "\n",
       "          tmp_normal_Nmax_sup0.0  tmp_normal_Nmax_sup0  \n",
       "19096420                     NaN                   NaN  \n",
       "19098779                     NaN                   NaN  \n",
       "\n",
       "[2 rows x 26 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "58e9db57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "      <th>tmp_normal_Nmax_sup0.0</th>\n",
       "      <th>tmp_normal_Nmax_sup0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19096420</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.839305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>1.45076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19098779</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.839305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>1.45076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "19096420  TAMKAAMAL                    0.0                     NaN   \n",
       "19098779  TAMKAAMAL                    0.0                     NaN   \n",
       "\n",
       "          gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  \\\n",
       "19096420                     NaN                     NaN   \n",
       "19098779                     NaN                     NaN   \n",
       "\n",
       "          gtexCohortfilter >=5.0  gtexCohortfilter >=10.0  \\\n",
       "19096420                     NaN                      NaN   \n",
       "19098779                     NaN                      NaN   \n",
       "\n",
       "                                                  coord  junctionAnnotated  \\\n",
       "19096420  48965782:48965784:48966315:48966340:None:None              False   \n",
       "19098779  48965782:48965784:48966315:48966340:None:None              False   \n",
       "\n",
       "          readFrameAnnotated  ...  cancerCohortfilter >=5.0  \\\n",
       "19096420                True  ...                         3   \n",
       "19098779               False  ...                         3   \n",
       "\n",
       "          cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "19096420                          0                   0.839305   \n",
       "19098779                          0                   0.839305   \n",
       "\n",
       "          TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "19096420                        0.0                        0.0   \n",
       "19098779                        0.0                        0.0   \n",
       "\n",
       "          TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \\\n",
       "19096420                   1.147786                    1.45076          NaN   \n",
       "19098779                   1.147786                    1.45076          NaN   \n",
       "\n",
       "          tmp_normal_Nmax_sup0.0  tmp_normal_Nmax_sup0  \n",
       "19096420                     NaN                   NaN  \n",
       "19098779                     NaN                   NaN  \n",
       "\n",
       "[2 rows x 26 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66f4336f",
   "metadata": {},
   "outputs": [],
   "source": [
    "max_recurrence_over_kmer(df,  get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                                          max_threshold_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "2f397707",
   "metadata": {},
   "outputs": [],
   "source": [
    "def max_recurrence_over_kmer(df, threshold_column, new_maxcol):\n",
    "    df = df[['kmer', threshold_column]].groupby('kmer').max()\n",
    "    df = df.reset_index()\n",
    "    df = df.rename({threshold_column: new_maxcol}, axis = 1)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "dbb11c99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19096420    0.0\n",
       "19098779    0.0\n",
       "Name: gtexCohortfilter >0.0, dtype: float64"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['kmer'] == kmer, get_threshold_colname(threshold_normal_cohort, tag_normal)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "ac68b3a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "recurrence_custom =  max_recurrence_over_kmer(df, get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                  max_threshold_col)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27afcb8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "if threshold_normal_cohort is not None:\n",
    "    recurrence_custom =  max_recurrence_over_kmer(df, \n",
    "                                                  get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                  max_threshold_col)\n",
    "\n",
    "# Expression in gtex cohort > 0  \n",
    "recurrence_custom_base = max_recurrence_over_kmer(df, \n",
    "                                                  get_threshold_colname(0.0, tag_normal),\n",
    "                                                  max_threshold_col_base) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2e52d066",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch_ov</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>343168</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343169</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514542</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966376:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>62</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514939</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966358:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>32</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966358</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627581</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966334:48966376:4...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>62</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>48966334:48966376</td>\n",
       "      <td>+</td>\n",
       "      <td>chr19</td>\n",
       "      <td>1054.0</td>\n",
       "      <td>1018.0</td>\n",
       "      <td>444.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>32.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer                                              coord  \\\n",
       "343168  TAMKAAMAL      48965782:48965784:48966315:48966340:None:None   \n",
       "343169  TAMKAAMAL      48965782:48965784:48966315:48966340:None:None   \n",
       "514542  TAMKAAMAL  48965782:48965784:48966315:48966334:48966376:4...   \n",
       "514939  TAMKAAMAL  48965782:48965784:48966315:48966334:48966358:4...   \n",
       "627581  TAMKAAMAL  48965782:48965784:48966315:48966334:48966376:4...   \n",
       "\n",
       "        junctionAnnotated  readFrameAnnotated  isCrossJunction  batch_ov  \\\n",
       "343168              False                True             True     29205   \n",
       "343169              False               False             True     29205   \n",
       "514542              False               False             True     29205   \n",
       "514939              False               False             True     29205   \n",
       "627581              False               False             True     29205   \n",
       "\n",
       "        cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "343168                      148                        89   \n",
       "343169                      148                        89   \n",
       "514542                       62                        24   \n",
       "514939                       32                        17   \n",
       "627581                       62                        24   \n",
       "\n",
       "        cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "343168                        43                        24  ...   \n",
       "343169                        43                        24  ...   \n",
       "514542                        11                         5  ...   \n",
       "514939                         2                         1  ...   \n",
       "627581                        11                         5  ...   \n",
       "\n",
       "        junction_coordinate2  junction_coordinate  strand_STAR    chr  \\\n",
       "343168                  None                  NaN          NaN    NaN   \n",
       "343169                  None                  NaN          NaN    NaN   \n",
       "514542     48966334:48966376                  NaN          NaN    NaN   \n",
       "514939     48966334:48966358                  NaN          NaN    NaN   \n",
       "627581     48966334:48966376    48966334:48966376            +  chr19   \n",
       "\n",
       "        gtexCohortfilter >0.0  gtexCohortfilter >=1.0  gtexCohortfilter >=2.0  \\\n",
       "343168                    0.0                     NaN                     NaN   \n",
       "343169                    0.0                     NaN                     NaN   \n",
       "514542                    0.0                     NaN                     NaN   \n",
       "514939                    0.0                     NaN                     NaN   \n",
       "627581                 1054.0                  1018.0                   444.0   \n",
       "\n",
       "        gtexCohortfilter >=3.0 gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "343168                     NaN                    NaN                     NaN  \n",
       "343169                     NaN                    NaN                     NaN  \n",
       "514542                     NaN                    NaN                     NaN  \n",
       "514939                     NaN                    NaN                     NaN  \n",
       "627581                   235.0                   94.0                    32.0  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e89d34e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>batch_ov</th>\n",
       "      <th>cancerCohortfilter &gt;0.0</th>\n",
       "      <th>cancerCohortfilter &gt;=1.0</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>...</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "      <th>junction_coordinate</th>\n",
       "      <th>strand_STAR</th>\n",
       "      <th>chr</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>343168</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343169</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>48965782:48965784:48966315:48966340:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>29205</td>\n",
       "      <td>148</td>\n",
       "      <td>89</td>\n",
       "      <td>43</td>\n",
       "      <td>24</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer                                          coord  \\\n",
       "343168  TAMKAAMAL  48965782:48965784:48966315:48966340:None:None   \n",
       "343169  TAMKAAMAL  48965782:48965784:48966315:48966340:None:None   \n",
       "\n",
       "        junctionAnnotated  readFrameAnnotated  isCrossJunction  batch_ov  \\\n",
       "343168              False                True             True     29205   \n",
       "343169              False               False             True     29205   \n",
       "\n",
       "        cancerCohortfilter >0.0  cancerCohortfilter >=1.0  \\\n",
       "343168                      148                        89   \n",
       "343169                      148                        89   \n",
       "\n",
       "        cancerCohortfilter >=2.0  cancerCohortfilter >=3.0  ...  \\\n",
       "343168                        43                        24  ...   \n",
       "343169                        43                        24  ...   \n",
       "\n",
       "        junction_coordinate2  junction_coordinate  strand_STAR  chr  \\\n",
       "343168                  None                  NaN          NaN  NaN   \n",
       "343169                  None                  NaN          NaN  NaN   \n",
       "\n",
       "        gtexCohortfilter >0.0  gtexCohortfilter >=1.0  gtexCohortfilter >=2.0  \\\n",
       "343168                    0.0                     NaN                     NaN   \n",
       "343169                    0.0                     NaN                     NaN   \n",
       "\n",
       "        gtexCohortfilter >=3.0 gtexCohortfilter >=5.0 gtexCohortfilter >=10.0  \n",
       "343168                     NaN                    NaN                     NaN  \n",
       "343169                     NaN                    NaN                     NaN  \n",
       "\n",
       "[2 rows x 30 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8311531",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "161ed7f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "recurrence_custom_base.loc[recurrence_custom_base['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a10358a5",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>tmp_normal_Nmax_sup0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>104657</th>\n",
       "      <td>TAMKAAMAL</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             kmer  tmp_normal_Nmax_sup0\n",
       "104657  TAMKAAMAL                   0.0"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recurrence_custom_base.loc[recurrence_custom_base['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3d59130",
   "metadata": {},
   "source": [
    "# TEST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98ef99e2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74b560d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'KVDTLGKST'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d85cd18",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f979d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            ['kmer', 'coord', 'exclude', 'batch_brca', 'junctionAnnotated', 'readFrameAnnotated', target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01993640",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer, 'batch_brca']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63aba557",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_batch.loc[(gene_to_batch['batch_brca'] == 54877)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c54267e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = df_load.merge(gene_to_batch[['exclude', f'batch_{run_type}']], on = f'batch_{run_type}', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ada9f080",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo.loc[(foo['kmer'] == kmer)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b35b9cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "final  =df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), \n",
    "       ['kmer', get_threshold_colname(threshold_cancer_cohort, tag_cancer)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97464fa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3d05819",
   "metadata": {},
   "outputs": [],
   "source": [
    "final.loc[final['cancerCohortfilter >0.0']> 403 , ['kmer', 'tmp_cancer_cohort',f'batch_{run_type}'] ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "455fb619",
   "metadata": {},
   "outputs": [],
   "source": [
    "final['cancerCohortfilter >0.0'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3a9ec1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea032687",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            [target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
