{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cab1d99c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import timeit\n",
    "import glob \n",
    "import pandas as pd\n",
    "import time\n",
    "import multiprocessing as mp \n",
    "import logging\n",
    "import sys \n",
    "import pathlib\n",
    "from pathlib import Path\n",
    "import matplotlib.pyplot as plt \n",
    "from helpers_STAR import * "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a8f2e8d",
   "metadata": {},
   "source": [
    "# Order \n",
    "- Uniprot \n",
    "- Sample init\n",
    "- Sample expression \n",
    "- Sample cohort \n",
    "- annotation \n",
    "- GTEX?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "fcf5e3ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_cancer_cohort(df, n_samples, threshold_column ):\n",
    "    df = df.loc[df[threshold_column] >= n_samples]    \n",
    "    return df\n",
    "        \n",
    "\n",
    "def get_threshold_colname(threshold, tag):\n",
    "    if (threshold is not None) and (threshold > 0 ):\n",
    "        threshold_column = f'{tag}filter >={threshold}'\n",
    "    else:\n",
    "        threshold_column = f'{tag}filter >{threshold}'\n",
    "    return threshold_column\n",
    "\n",
    "def filter_single_col(df, threshold, colname):\n",
    "    if threshold:\n",
    "        df = df.loc[df[colname] >= threshold, :]\n",
    "    else:\n",
    "        df = df.loc[df[colname] >threshold, :]\n",
    "    return df\n",
    "\n",
    "def max_recurrence_over_kmer(df, threshold_column, new_maxcol):\n",
    "    df = df[['kmer', threshold_column]].groupby('kmer').max()\n",
    "    df = df.reset_index()\n",
    "    df = df.rename({threshold_column: new_maxcol}, axis = 1)\n",
    "    return df \n",
    "\n",
    "def output_count(df, report_count, report_step, step_string, perform_count=True):\n",
    "    '''\n",
    "    Performs a count operation on the number of kmers present in spark dataframe after a given filtering step\n",
    "    Note: This operation is expensive but useful if the user is interested in intermediate filtering steps\n",
    "    :param perform_count: bool whether to perform a count operation\n",
    "    :param df: dataframe with kmer expression counts\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param step_string: str name of the counting operation\n",
    "    '''\n",
    "    if perform_count:\n",
    "        mycount = df['kmer'].unique().shape[0]\n",
    "        report_count.append(mycount)\n",
    "        report_step.append(step_string)\n",
    "        print(f'# {step_string} n = {mycount} kmers')\n",
    "\n",
    "def save_output_count(output_count, report_count, report_steps, prefix, cancer_sample_ori, mutation_mode,\n",
    "                      sample_expr_support_cancer, cohort_expr_support_cancer, n_samples_lim_cancer,\n",
    "                          cohort_expr_support_normal, n_samples_lim_normal, id_normals):\n",
    "    '''\n",
    "    Saves the number of kmers present in spark dataframe after each filtering step in a tabular file\n",
    "    :param output_count: str path for count file of intermediate filtering steps\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param prefix: str information to be added to the result line in an info column\n",
    "    :param cancer_sample_ori: str id of target cancer sample which was filtered\n",
    "    :param mutation_mode: str information about whether mutations where applied or not\n",
    "    :param sample_expr_support_cancer: float normalized expression threshold for the cancer target sample\n",
    "    :param cohort_expr_support_cancer: float normalized expression threshold for the cancer cohort\n",
    "    excluding the target sample\n",
    "    hich should be met in n samples\n",
    "    :param n_samples_lim_cancer: int number of cancer samples in which the cancer cohort expression threshold\n",
    "    should be met\n",
    "    :param cohort_expr_support_normal: float normalized expression threshold for the normal cohort\n",
    "    required in any sample (>=1)\n",
    "    :param n_samples_lim_normal: int number of normal samples in which any number of reads is required (>0)\n",
    "    :param id_normals: str id of the normal cohort (example gtex)\n",
    "    '''\n",
    "    pipeline = 'peptide-based'\n",
    "    if output_count:\n",
    "        header = (f'{\"sample\"}\\t{\"mutation_mode\"}\\t{\"pipeline\"}\\t{\"min_sample_reads\"}\\t{\"#_of_cohort_samples\"}\\t'\n",
    "                  f'{\"reads_per_cohort_sample\"}\\t{\"#_normal_samples_allowed\"}\\t{\"normal_cohort_id\"}'\n",
    "                  f'\\t{\"reads_per_normal_sample\"}')\n",
    "        line =   (f'{cancer_sample_ori}\\t{mutation_mode}\\t{pipeline}\\t{sample_expr_support_cancer}\\t{n_samples_lim_cancer}'\n",
    "                  f'\\t{cohort_expr_support_cancer}\\t{n_samples_lim_normal}\\t{id_normals}'\n",
    "                  f'\\t{cohort_expr_support_normal}')\n",
    "\n",
    "        for idx in np.arange(len(report_count)):\n",
    "            header += f'\\t{report_steps[idx]}'\n",
    "            line += f'\\t{report_count[idx]}'\n",
    "        if prefix:\n",
    "            header += f'\\t{\"info\"}'\n",
    "            line += f'\\t{prefix}'\n",
    "        header += \"\\n\"\n",
    "        line += \"\\n\"\n",
    "        \n",
    "#         print(header, len(header.split('\\t')))\n",
    "#         print(line, len(line.split('\\t')))\n",
    "        if not os.path.exists(output_count):\n",
    "            with open(output_count,\"w\") as f:\n",
    "                f.write(header)\n",
    "        with open(output_count, \"a\") as f:\n",
    "            f.write(line)\n",
    "        logging.info(f'Save intermediate info to {output_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "bd671a7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Julianne sample  mutation_mode   pipeline        min_sample_reads        #_of_cohort_samples     reads_per_cohort_sample #_normal_samples_allowed        normal_cohort_id        reads_per_normal_sample motif_filter    Init_cancer     Filter_Sample   Filter_Sample_Cohort    Filter_Sample_Cohort_CohortBackground   Filter_Motif    Filter_Sample_Cohort_CohortBackground_Uniprot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fcf0675",
   "metadata": {},
   "source": [
    "# MAIN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91281182",
   "metadata": {},
   "source": [
    "Note: Foreground joined with GTEX table \\ 79 genes \n",
    "Genes excluded is a list of 215 genes for now. Should be fine "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "74f4911d",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_type = 'brca'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "33bd2aad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inputs\n",
    "\n",
    "if run_type == 'brca':\n",
    "    target_samples = ['TCGA-AO-A0JM-01A-21R-A056-07.all',\n",
    "                      'TCGA-C8-A12P-01A-11R-A115-07.all',\n",
    "                      'TCGA-BH-A18V-01A-11R-A12D-07.all',\n",
    "                      'TCGA-A2-A0D2-01A-21R-A034-07.all',\n",
    "                      'TCGA-A2-A0SX-01A-12R-A084-07.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz')\n",
    "elif run_type == 'ov':\n",
    "    target_samples = ['TCGA-25-1319-01A-01R-1565-13.all',\n",
    "                      'TCGA-25-1313-01A-01R-1565-13.all',\n",
    "                      'TCGA-61-2008-01A-02R-1568-13.all',\n",
    "                      'TCGA-24-1431-01A-01R-1566-13.all',\n",
    "                      'TCGA-24-2298-01A-01R-1569-13.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz')\n",
    "\n",
    "\n",
    "# Outputs\n",
    "filtering_id = 'filters_22March_order_wany_wAnnotTEST'\n",
    "\n",
    "output_dir = os.path.join(basedir, 'filtering_samples', filtering_id)\n",
    "pathlib.Path(output_dir).mkdir(exist_ok=True, parents=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "08c6b40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Discussion 02/22 Choices\n",
    "# BACKGROUND cohorts we do (cohort_reads, sample_number)- KEEP pipeline as such\n",
    "# cohort_reads=[0,1,3, None]\n",
    "# sample_number=[1,2,10, None]\n",
    "# FOREGROUND  (cohort_reads, sample_number) means\n",
    "# cohort_reads=[0,2]\n",
    "# sample_number(rest of cohort) =[1, 5]\n",
    "# Sample reads = [0]\n",
    "# + A case with not foreground"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e9c1eddc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters\n",
    "Threshold_target = [0.0]\n",
    "Threshold_cancer_cohort = [0.0, 2.0] # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0] -> Removed None #TODO\n",
    "N_samples_cancer = [None, 1, 5] # choices 1 to 1102 for BRCA and 374 for OV   \n",
    "\n",
    "Threshold_normal_cohort = [0.0, 1.0, 3.0, None]   # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_normal = [1, 2, 10, None] #choices 1 to max number of samples in Normal whitelist\n",
    "\n",
    "tag_cancer = 'cancerCohort'\n",
    "tag_normal = 'gtexCohort'\n",
    "\n",
    "\n",
    "tag_prefix = 'G_'\n",
    "mutation_mode = 'ref'\n",
    "save_tag = 'GtexCohort'\n",
    "\n",
    "metadata_save = ['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated']\n",
    "\n",
    "filter_annot = False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c38d0281",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_intermediate/complete_cancer_candidates_order_r.tsv.gz\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(22348747, 24)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load matrix to be filtered\n",
    "df_load = pd.read_csv(intermediate_output, sep = '\\t')\n",
    "print(f'Loaded {intermediate_output}')\n",
    "df_load = df_load.rename({'batch': f'batch_{run_type}'}, axis = 1)\n",
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "f3d368b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>3290.0</td>\n",
       "      <td>1969.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>572.0</td>\n",
       "      <td>545.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISSQSRVEK</td>\n",
       "      <td>278.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>92379851:92379859:92493866:92493885:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSGDEEKYP</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92600493:92600508:92611313:92611325:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>641</td>\n",
       "      <td>520</td>\n",
       "      <td>348</td>\n",
       "      <td>170</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLKMKMFQI</td>\n",
       "      <td>146.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379850:92379859:92496416:92496434:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  WYITRSGIA                 3290.0                  1969.0   \n",
       "1  WYITRSGIA                  572.0                   545.0   \n",
       "2  ISSQSRVEK                  278.0                   261.0   \n",
       "3  RSGDEEKYP                 7350.0                  4622.0   \n",
       "4  HLKMKMFQI                  146.0                   136.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   861.0                   394.0                    89.0   \n",
       "1                   243.0                   130.0                    27.0   \n",
       "2                    82.0                    29.0                     7.0   \n",
       "3                  2653.0                  1745.0                  1126.0   \n",
       "4                    42.0                    20.0                     3.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92347505:92347506:92349915:92349941:None:None   \n",
       "1                      2.0  92347505:92347506:92349915:92349941:None:None   \n",
       "2                      0.0  92379851:92379859:92493866:92493885:None:None   \n",
       "3                    734.0  92600493:92600508:92611313:92611325:None:None   \n",
       "4                      1.0  92379850:92379859:92496416:92496434:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0              False               False  ...                       138   \n",
       "1              False               False  ...                       138   \n",
       "2              False               False  ...                        10   \n",
       "3               True                True  ...                       641   \n",
       "4              False               False  ...                        16   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                        82                        26   \n",
       "1                        82                        26   \n",
       "2                         2                         0   \n",
       "3                       520                       348   \n",
       "4                         2                         0   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGAC8A12P01A11RA11507all  \\\n",
       "0                          5                   0.000000   \n",
       "1                          5                   0.000000   \n",
       "2                          0                   0.000000   \n",
       "3                        170                   2.922641   \n",
       "4                          0                   2.922641   \n",
       "\n",
       "   TCGAAOA0JM01A21RA05607all  TCGABHA18V01A11RA12D07all  \\\n",
       "0                   0.000000                   0.000000   \n",
       "1                   0.000000                   0.000000   \n",
       "2                   0.000000                   2.474321   \n",
       "3                   2.102386                   1.237161   \n",
       "4                   0.000000                   0.000000   \n",
       "\n",
       "   TCGAA2A0D201A21RA03407all  TCGAA2A0SX01A12RA08407all  isAnnotated  \n",
       "0                   4.102634                        0.0          NaN  \n",
       "1                   4.102634                        0.0          NaN  \n",
       "2                   0.000000                        0.0          NaN  \n",
       "3                   0.000000                        0.0          1.0  \n",
       "4                   0.000000                        0.0          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "2331bcea",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22348747, 24)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "76817af3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- processing TCGAAOA0JM01A21RA05607all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_22March_order_wany_wAnnotTEST/G_filtered_df_TCGA-AO-A0JM-01A-21R-A056-07_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_22March_order_wany_wAnnotTEST/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimAnyAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_22March_order_wany_wAnnotTEST/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimAnyAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_22March_order_wany_wAnnotTEST/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimAnyAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 1071 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_22March_order_wany_wAnnotTEST/G_TCGA-AO-A0JM-01A-21R-A056-07_SampleLim0.0CohortLimAnyAcrossNone_FiltNormalsGtexCohortCohortlim0.0AcrossAny.tsv.gz \n",
      "\n",
      "# Init_Sample n = 476112 kmers\n",
      "# Filter_Sample n = 476112 kmers\n",
      "# Filter_Sample_Cohort n = 476112 kmers\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [53], line 75\u001b[0m\n\u001b[1;32m     73\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m threshold_normal_cohort \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m     74\u001b[0m     df \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mmerge(recurrence_custom, on \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mkmer\u001b[39m\u001b[38;5;124m'\u001b[39m, how \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mleft\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m---> 75\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmerge\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrecurrence_custom_base\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mon\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mkmer\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhow\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mleft\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     77\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (threshold_normal_cohort \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;129;01mand\u001b[39;00m (n_samples_normal \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m     78\u001b[0m     df \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mloc[ \u001b[38;5;241m~\u001b[39m ((df[max_threshold_col] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m) \u001b[38;5;241m|\u001b[39m (df[max_threshold_col_base] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m n_samples_normal)), :]\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/frame.py:8195\u001b[0m, in \u001b[0;36mDataFrame.merge\u001b[0;34m(self, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m   8176\u001b[0m \u001b[38;5;129m@Substitution\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   8177\u001b[0m \u001b[38;5;129m@Appender\u001b[39m(_merge_doc, indents\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m2\u001b[39m)\n\u001b[1;32m   8178\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmerge\u001b[39m(\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   8191\u001b[0m     validate\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m   8192\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m DataFrame:\n\u001b[1;32m   8193\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcore\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mreshape\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mmerge\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m merge\n\u001b[0;32m-> 8195\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmerge\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   8196\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8197\u001b[0m \u001b[43m        \u001b[49m\u001b[43mright\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8198\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhow\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhow\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8199\u001b[0m \u001b[43m        \u001b[49m\u001b[43mon\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mon\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8200\u001b[0m \u001b[43m        \u001b[49m\u001b[43mleft_on\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mleft_on\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8201\u001b[0m \u001b[43m        \u001b[49m\u001b[43mright_on\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mright_on\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8202\u001b[0m \u001b[43m        \u001b[49m\u001b[43mleft_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mleft_index\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8203\u001b[0m \u001b[43m        \u001b[49m\u001b[43mright_index\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mright_index\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8204\u001b[0m \u001b[43m        \u001b[49m\u001b[43msort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8205\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuffixes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuffixes\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8206\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8207\u001b[0m \u001b[43m        \u001b[49m\u001b[43mindicator\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindicator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8208\u001b[0m \u001b[43m        \u001b[49m\u001b[43mvalidate\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mvalidate\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   8209\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/reshape/merge.py:89\u001b[0m, in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;129m@Substitution\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mleft : DataFrame\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     58\u001b[0m \u001b[38;5;129m@Appender\u001b[39m(_merge_doc, indents\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m)\n\u001b[1;32m     59\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmerge\u001b[39m(\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     72\u001b[0m     validate\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[1;32m     73\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mDataFrame\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m     74\u001b[0m     op \u001b[38;5;241m=\u001b[39m _MergeOperation(\n\u001b[1;32m     75\u001b[0m         left,\n\u001b[1;32m     76\u001b[0m         right,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     87\u001b[0m         validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m     88\u001b[0m     )\n\u001b[0;32m---> 89\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mop\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_result\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/reshape/merge.py:693\u001b[0m, in \u001b[0;36m_MergeOperation.get_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    690\u001b[0m lindexers \u001b[38;5;241m=\u001b[39m {\u001b[38;5;241m1\u001b[39m: left_indexer} \u001b[38;5;28;01mif\u001b[39;00m left_indexer \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m {}\n\u001b[1;32m    691\u001b[0m rindexers \u001b[38;5;241m=\u001b[39m {\u001b[38;5;241m1\u001b[39m: right_indexer} \u001b[38;5;28;01mif\u001b[39;00m right_indexer \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m {}\n\u001b[0;32m--> 693\u001b[0m result_data \u001b[38;5;241m=\u001b[39m \u001b[43mconcatenate_block_managers\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    694\u001b[0m \u001b[43m    \u001b[49m\u001b[43m[\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mleft\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_mgr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlindexers\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mright\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_mgr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrindexers\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    695\u001b[0m \u001b[43m    \u001b[49m\u001b[43maxes\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43mllabels\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mappend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrlabels\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mjoin_index\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    696\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconcat_axis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    697\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    698\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    700\u001b[0m typ \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mleft\u001b[38;5;241m.\u001b[39m_constructor\n\u001b[1;32m    701\u001b[0m result \u001b[38;5;241m=\u001b[39m typ(result_data)\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_merge_type)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/concat.py:83\u001b[0m, in \u001b[0;36mconcatenate_block_managers\u001b[0;34m(mgrs_indexers, axes, concat_axis, copy)\u001b[0m\n\u001b[1;32m     80\u001b[0m         b \u001b[38;5;241m=\u001b[39m make_block(values, placement\u001b[38;5;241m=\u001b[39mplacement, ndim\u001b[38;5;241m=\u001b[39mblk\u001b[38;5;241m.\u001b[39mndim)\n\u001b[1;32m     81\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     82\u001b[0m         b \u001b[38;5;241m=\u001b[39m make_block(\n\u001b[0;32m---> 83\u001b[0m             \u001b[43m_concatenate_join_units\u001b[49m\u001b[43m(\u001b[49m\u001b[43mjoin_units\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconcat_axis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m,\n\u001b[1;32m     84\u001b[0m             placement\u001b[38;5;241m=\u001b[39mplacement,\n\u001b[1;32m     85\u001b[0m             ndim\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mlen\u001b[39m(axes),\n\u001b[1;32m     86\u001b[0m         )\n\u001b[1;32m     87\u001b[0m     blocks\u001b[38;5;241m.\u001b[39mappend(b)\n\u001b[1;32m     89\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m BlockManager(blocks, axes)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/concat.py:313\u001b[0m, in \u001b[0;36m_concatenate_join_units\u001b[0;34m(join_units, concat_axis, copy)\u001b[0m\n\u001b[1;32m    309\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mAssertionError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConcatenating join units along axis0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    311\u001b[0m empty_dtype, upcasted_na \u001b[38;5;241m=\u001b[39m _get_empty_dtype_and_na(join_units)\n\u001b[0;32m--> 313\u001b[0m to_concat \u001b[38;5;241m=\u001b[39m [\n\u001b[1;32m    314\u001b[0m     ju\u001b[38;5;241m.\u001b[39mget_reindexed_values(empty_dtype\u001b[38;5;241m=\u001b[39mempty_dtype, upcasted_na\u001b[38;5;241m=\u001b[39mupcasted_na)\n\u001b[1;32m    315\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m ju \u001b[38;5;129;01min\u001b[39;00m join_units\n\u001b[1;32m    316\u001b[0m ]\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(to_concat) \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;66;03m# Only one block, nothing to concatenate.\u001b[39;00m\n\u001b[1;32m    320\u001b[0m     concat_values \u001b[38;5;241m=\u001b[39m to_concat[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/concat.py:314\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    309\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mAssertionError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConcatenating join units along axis0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    311\u001b[0m empty_dtype, upcasted_na \u001b[38;5;241m=\u001b[39m _get_empty_dtype_and_na(join_units)\n\u001b[1;32m    313\u001b[0m to_concat \u001b[38;5;241m=\u001b[39m [\n\u001b[0;32m--> 314\u001b[0m     \u001b[43mju\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_reindexed_values\u001b[49m\u001b[43m(\u001b[49m\u001b[43mempty_dtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mempty_dtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mupcasted_na\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mupcasted_na\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    315\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m ju \u001b[38;5;129;01min\u001b[39;00m join_units\n\u001b[1;32m    316\u001b[0m ]\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(to_concat) \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;66;03m# Only one block, nothing to concatenate.\u001b[39;00m\n\u001b[1;32m    320\u001b[0m     concat_values \u001b[38;5;241m=\u001b[39m to_concat[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/concat.py:298\u001b[0m, in \u001b[0;36mJoinUnit.get_reindexed_values\u001b[0;34m(self, empty_dtype, upcasted_na)\u001b[0m\n\u001b[1;32m    296\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    297\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m ax, indexer \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindexers\u001b[38;5;241m.\u001b[39mitems():\n\u001b[0;32m--> 298\u001b[0m         values \u001b[38;5;241m=\u001b[39m \u001b[43malgos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtake_nd\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43max\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfill_value\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfill_value\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    300\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m values\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/algorithms.py:1759\u001b[0m, in \u001b[0;36mtake_nd\u001b[0;34m(arr, indexer, axis, out, fill_value, allow_fill)\u001b[0m\n\u001b[1;32m   1754\u001b[0m         out \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mempty(out_shape, dtype\u001b[38;5;241m=\u001b[39mdtype)\n\u001b[1;32m   1756\u001b[0m func \u001b[38;5;241m=\u001b[39m _get_take_nd_function(\n\u001b[1;32m   1757\u001b[0m     arr\u001b[38;5;241m.\u001b[39mndim, arr\u001b[38;5;241m.\u001b[39mdtype, out\u001b[38;5;241m.\u001b[39mdtype, axis\u001b[38;5;241m=\u001b[39maxis, mask_info\u001b[38;5;241m=\u001b[39mmask_info\n\u001b[1;32m   1758\u001b[0m )\n\u001b[0;32m-> 1759\u001b[0m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43marr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfill_value\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1761\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m flip_order:\n\u001b[1;32m   1762\u001b[0m     out \u001b[38;5;241m=\u001b[39m out\u001b[38;5;241m.\u001b[39mT\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for cancer_sample_ori in target_samples: # TODO update\n",
    "    # Sample naming\n",
    "    target_sample = cancer_sample_ori.replace('-', '').replace('.', '')\n",
    "    cancer_sample_ori = cancer_sample_ori.replace('.all', '')\n",
    "    print(f'-------- processing {target_sample} -------- \\n')\n",
    "    \n",
    "    # Summary file for sample\n",
    "    summary_file = f'{tag_prefix}filtered_df_{cancer_sample_ori}_samp_chrt_norm_mot.tsv'\n",
    "    summary_path = os.path.join(output_dir, summary_file)\n",
    "    print(f'Saving to summary file {summary_path}')\n",
    "    \n",
    "\n",
    "    for threshold_target in Threshold_target:\n",
    "        for threshold_cancer_cohort in Threshold_cancer_cohort:\n",
    "            for n_samples_cancer in N_samples_cancer:\n",
    "                for threshold_normal_cohort in Threshold_normal_cohort:\n",
    "                    for n_samples_normal in N_samples_normal:\n",
    "                        \n",
    "                        df_expr = []\n",
    "                        report_count = [] \n",
    "                        report_steps = []\n",
    "                        if (n_samples_cancer is None) and (threshold_cancer_cohort != 0): #Only None, None as condition TODO\n",
    "                            continue\n",
    "                        if (n_samples_normal is None) and (threshold_normal_cohort is None):\n",
    "                            continue\n",
    "\n",
    "                        adjusted_threshold_col = 'tmp_cancer_cohort'\n",
    "                        max_threshold_col = 'tmp_normal_Nmax_sup{}'.format(threshold_normal_cohort)\n",
    "                        max_threshold_col_base = 'tmp_normal_Nmax_sup{}'.format(0)\n",
    "\n",
    "\n",
    "                        df = df_load.copy()\n",
    "                        # Make correction for number samples passing theshold in cohort. We want to exclude the target sample in counting\n",
    "                        if (n_samples_cancer is not None): #TODO \n",
    "                            df[adjusted_threshold_col] = df[get_threshold_colname(threshold_cancer_cohort, tag_cancer)]\n",
    "                            df.loc[df[target_sample] >= threshold_cancer_cohort, adjusted_threshold_col] -=1 \n",
    "\n",
    "                        # Number of kmers expressed in sample \n",
    "                        df = filter_single_col(df, 0, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Init_Sample')\n",
    "\n",
    "                        # Number of kmers >= threshold in sample \n",
    "                        df = filter_single_col(df, threshold_target, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample')\n",
    "\n",
    "                        \n",
    "                        # Filter for cancer cancer cohort #TODO\n",
    "                        if (n_samples_cancer is not None): #skip if both \n",
    "                            df = filter_cancer_cohort(df, n_samples_cancer, adjusted_threshold_col)\n",
    "                        if threshold_cancer_cohort: \n",
    "                            threshold_cancer_cohort_save = threshold_cancer_cohort \n",
    "                        else: \n",
    "                            threshold_cancer_cohort_save = 'Any'\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort')\n",
    "                            \n",
    "\n",
    "\n",
    "                        # Expression in gtex cohort >= threshold \n",
    "                        if threshold_normal_cohort is not None:\n",
    "                            recurrence_custom =  max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                                          max_threshold_col)\n",
    "\n",
    "                        # Expression in gtex cohort > 0  \n",
    "                        recurrence_custom_base = max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(0.0, tag_normal),\n",
    "                                                                          max_threshold_col_base) \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "                        # Perform Background filtering \n",
    "                        if threshold_normal_cohort is not None:\n",
    "                            df = df.merge(recurrence_custom, on = 'kmer', how = 'left')\n",
    "                        df = df.merge(recurrence_custom_base, on = 'kmer', how = 'left')\n",
    "                        \n",
    "                        if (threshold_normal_cohort is not None) and (n_samples_normal is not None):\n",
    "                            df = df.loc[ ~ ((df[max_threshold_col] >= 1) | (df[max_threshold_col_base] >= n_samples_normal)), :]\n",
    "                            threshold_normal_cohort_save = threshold_normal_cohort\n",
    "                            n_samples_normal_save = n_samples_normal\n",
    "                        elif threshold_normal_cohort is None:\n",
    "                            df = df.loc[ ~ (df[max_threshold_col_base] >= n_samples_normal), :]\n",
    "                            threshold_normal_cohort_save = 'Any'\n",
    "                        elif n_samples_normal is None:\n",
    "                            df = df.loc[ ~ (df[max_threshold_col] >= 1), :]\n",
    "                            n_samples_normal_save = 'Any'\n",
    "\n",
    "          \n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal')\n",
    "                    \n",
    "                        #Perform Annotated junctions filtering \n",
    "                        if filter_annot:\n",
    "                            df = df[df['isAnnotated'].isna()]\n",
    "                            output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot')\n",
    "\n",
    "#                         # DEV: Exclude genes where GTEX is missing\n",
    "#                         df = df.loc[df['exclude'].isna()]\n",
    "#                         output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot_EXPGTEX')\n",
    "\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        # Save outputs \n",
    "                        # outpaths #TODO\n",
    "                        base_path_final = os.path.join(output_dir,\n",
    "                                                       (f'{tag_prefix}{cancer_sample_ori}_'\n",
    "                                                        f'SampleLim{threshold_target}'\n",
    "                                                        f'CohortLim{threshold_cancer_cohort_save}'\n",
    "                                                        f'Across{n_samples_cancer}_'\n",
    "                                                        f'FiltNormals{save_tag}'\n",
    "                                                        f'Cohortlim{threshold_normal_cohort_save}'\n",
    "                                                        f'Across{n_samples_normal_save}.tsv.gz'))\n",
    "                        print(f'Saving outputs to: {base_path_final} \\n')\n",
    "                        df.loc[:, metadata_save].to_csv(base_path_final, compression = 'gzip', index = None, sep = '\\t')\n",
    "\n",
    "                        #TODO \n",
    "                        save_output_count(summary_path, report_count, report_steps, '', cancer_sample_ori, mutation_mode,\n",
    "                                          threshold_target, threshold_cancer_cohort_save, n_samples_cancer,\n",
    "                                              threshold_normal_cohort_save, n_samples_normal_save, save_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "279ecb2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "threshold_normal_cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5544c58b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f3d59130",
   "metadata": {},
   "source": [
    "# TEST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98ef99e2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74b560d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'KVDTLGKST'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d85cd18",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f979d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            ['kmer', 'coord', 'exclude', 'batch_brca', 'junctionAnnotated', 'readFrameAnnotated', target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01993640",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer, 'batch_brca']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63aba557",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_batch.loc[(gene_to_batch['batch_brca'] == 54877)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c54267e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = df_load.merge(gene_to_batch[['exclude', f'batch_{run_type}']], on = f'batch_{run_type}', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ada9f080",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo.loc[(foo['kmer'] == kmer)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b35b9cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "final  =df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), \n",
    "       ['kmer', get_threshold_colname(threshold_cancer_cohort, tag_cancer)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97464fa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3d05819",
   "metadata": {},
   "outputs": [],
   "source": [
    "final.loc[final['cancerCohortfilter >0.0']> 403 , ['kmer', 'tmp_cancer_cohort',f'batch_{run_type}'] ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "455fb619",
   "metadata": {},
   "outputs": [],
   "source": [
    "final['cancerCohortfilter >0.0'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3a9ec1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea032687",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            [target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
