{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cab1d99c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Matplotlib created a temporary config/cache directory at /scratch/slurm-job.816371/matplotlib-hj6n8fbw because the default path (/cluster/customapps/biomed/grlab/users/prelotla/.cache/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import timeit\n",
    "import glob \n",
    "import pandas as pd\n",
    "import time\n",
    "import multiprocessing as mp \n",
    "import logging\n",
    "import sys \n",
    "import pathlib\n",
    "from pathlib import Path\n",
    "import matplotlib.pyplot as plt "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a8f2e8d",
   "metadata": {},
   "source": [
    "# Order \n",
    "- Uniprot \n",
    "- Sample init\n",
    "- Sample expression \n",
    "- Sample cohort \n",
    "- annotation \n",
    "- GTEX?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fcf5e3ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_cancer_cohort(df, n_samples, threshold_column ):\n",
    "    df = df.loc[df[threshold_column] >= n_samples]    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d2674c73",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_threshold_colname(threshold, tag):\n",
    "    if (threshold is not None) and (threshold > 0 ):\n",
    "        threshold_column = f'{tag}filter >={threshold}'\n",
    "    else:\n",
    "        threshold_column = f'{tag}filter >{threshold}'\n",
    "    return threshold_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "783a8bfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_single_col(df, threshold, colname):\n",
    "    if threshold:\n",
    "        df = df.loc[df[colname] >= threshold, :]\n",
    "    else:\n",
    "        df = df.loc[df[colname] >threshold, :]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0ab8af74",
   "metadata": {},
   "outputs": [],
   "source": [
    "def max_recurrence_over_kmer(df, threshold_column, new_maxcol):\n",
    "    df = df[['kmer', threshold_column]].groupby('kmer').max()\n",
    "    df = df.reset_index()\n",
    "    df = df.rename({threshold_column: new_maxcol}, axis = 1)\n",
    "    return df "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "898b4316",
   "metadata": {},
   "outputs": [],
   "source": [
    "def output_count(df, report_count, report_step, step_string, perform_count=True):\n",
    "    '''\n",
    "    Performs a count operation on the number of kmers present in spark dataframe after a given filtering step\n",
    "    Note: This operation is expensive but useful if the user is interested in intermediate filtering steps\n",
    "    :param perform_count: bool whether to perform a count operation\n",
    "    :param df: dataframe with kmer expression counts\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param step_string: str name of the counting operation\n",
    "    '''\n",
    "    if perform_count:\n",
    "        mycount = df['kmer'].unique().shape[0]\n",
    "        report_count.append(mycount)\n",
    "        report_step.append(step_string)\n",
    "        print(f'# {step_string} n = {mycount} kmers')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "33b71537",
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_output_count(output_count, report_count, report_steps, prefix, cancer_sample_ori, mutation_mode,\n",
    "                      sample_expr_support_cancer, cohort_expr_support_cancer, n_samples_lim_cancer,\n",
    "                          cohort_expr_support_normal, n_samples_lim_normal, id_normals):\n",
    "    '''\n",
    "    Saves the number of kmers present in spark dataframe after each filtering step in a tabular file\n",
    "    :param output_count: str path for count file of intermediate filtering steps\n",
    "    :param report_count: list to store result of successive counting operations\n",
    "    :param report_step: list to store name of successive counting operations\n",
    "    :param prefix: str information to be added to the result line in an info column\n",
    "    :param cancer_sample_ori: str id of target cancer sample which was filtered\n",
    "    :param mutation_mode: str information about whether mutations where applied or not\n",
    "    :param sample_expr_support_cancer: float normalized expression threshold for the cancer target sample\n",
    "    :param cohort_expr_support_cancer: float normalized expression threshold for the cancer cohort\n",
    "    excluding the target sample\n",
    "    hich should be met in n samples\n",
    "    :param n_samples_lim_cancer: int number of cancer samples in which the cancer cohort expression threshold\n",
    "    should be met\n",
    "    :param cohort_expr_support_normal: float normalized expression threshold for the normal cohort\n",
    "    required in any sample (>=1)\n",
    "    :param n_samples_lim_normal: int number of normal samples in which any number of reads is required (>0)\n",
    "    :param id_normals: str id of the normal cohort (example gtex)\n",
    "    '''\n",
    "    pipeline = 'peptide-based'\n",
    "    if output_count:\n",
    "        header = (f'{\"sample\"}\\t{\"mutation_mode\"}\\t{\"pipeline\"}\\t{\"min_sample_reads\"}\\t{\"#_of_cohort_samples\"}\\t'\n",
    "                  f'{\"reads_per_cohort_sample\"}\\t{\"#_normal_samples_allowed\"}\\t{\"normal_cohort_id\"}'\n",
    "                  f'\\t{\"reads_per_normal_sample\"}')\n",
    "        line =   (f'{cancer_sample_ori}\\t{mutation_mode}\\t{pipeline}\\t{sample_expr_support_cancer}\\t{n_samples_lim_cancer}'\n",
    "                  f'\\t{cohort_expr_support_cancer}\\t{n_samples_lim_normal}\\t{id_normals}'\n",
    "                  f'\\t{cohort_expr_support_normal}')\n",
    "\n",
    "        for idx in np.arange(len(report_count)):\n",
    "            header += f'\\t{report_steps[idx]}'\n",
    "            line += f'\\t{report_count[idx]}'\n",
    "        if prefix:\n",
    "            header += f'\\t{\"info\"}'\n",
    "            line += f'\\t{prefix}'\n",
    "        header += \"\\n\"\n",
    "        line += \"\\n\"\n",
    "        \n",
    "#         print(header, len(header.split('\\t')))\n",
    "#         print(line, len(line.split('\\t')))\n",
    "        if not os.path.exists(output_count):\n",
    "            with open(output_count,\"w\") as f:\n",
    "                f.write(header)\n",
    "        with open(output_count, \"a\") as f:\n",
    "            f.write(line)\n",
    "        logging.info(f'Save intermediate info to {output_count}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bd671a7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Julianne sample  mutation_mode   pipeline        min_sample_reads        #_of_cohort_samples     reads_per_cohort_sample #_normal_samples_allowed        normal_cohort_id        reads_per_normal_sample motif_filter    Init_cancer     Filter_Sample   Filter_Sample_Cohort    Filter_Sample_Cohort_CohortBackground   Filter_Motif    Filter_Sample_Cohort_CohortBackground_Uniprot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fcf0675",
   "metadata": {},
   "source": [
    "# MAIN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91281182",
   "metadata": {},
   "source": [
    "Note: Foreground joined with GTEX table \\ 79 genes \n",
    "Genes excluded is a list of 215 genes for now. Should be fine "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "74f4911d",
   "metadata": {},
   "outputs": [],
   "source": [
    "run_type = 'ov'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "33bd2aad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inputs\n",
    "\n",
    "if run_type == 'brca':\n",
    "    target_samples = ['TCGA-C8-A12P-01A-11R-A115-07.all',\n",
    "                      'TCGA-AO-A0JM-01A-21R-A056-07.all',\n",
    "                      'TCGA-BH-A18V-01A-11R-A12D-07.all',\n",
    "                      'TCGA-A2-A0D2-01A-21R-A034-07.all',\n",
    "                      'TCGA-A2-A0SX-01A-12R-A084-07.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_.tsv.gz')\n",
    "elif run_type == 'ov':\n",
    "    target_samples = ['TCGA-25-1319-01A-01R-1565-13.all',\n",
    "                      'TCGA-25-1313-01A-01R-1565-13.all',\n",
    "                      'TCGA-61-2008-01A-02R-1568-13.all',\n",
    "                      'TCGA-24-1431-01A-01R-1566-13.all',\n",
    "                      'TCGA-24-2298-01A-01R-1569-13.all']\n",
    "    basedir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374'\n",
    "    intermediate_output = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_.tsv.gz')\n",
    "\n",
    "\n",
    "# Outputs\n",
    "filtering_id = 'chosen_filters_06March'\n",
    "\n",
    "output_dir = os.path.join(basedir, 'filtering_samples', filtering_id)\n",
    "pathlib.Path(output_dir).mkdir(exist_ok=True, parents=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "08c6b40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Discussion 02/22 Choices\n",
    "# BACKGROUND cohorts we do (cohort_reads, sample_number)- KEEP pipeline as such\n",
    "# cohort_reads=[0,1,3]\n",
    "# sample_number=[1,2,10]\n",
    "# FOREGROUND  (cohort_reads, sample_number) means\n",
    "# cohort_reads=[0,2]\n",
    "# sample_number(rest of cohort) =[1, 5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e9c1eddc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters\n",
    "Threshold_target = [0.0]\n",
    "Threshold_cancer_cohort = [0.0, 2.0, None] # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_cancer = [1, 5, None] # choices 1 to 1102 for BRCA and 374 for OV   \n",
    "\n",
    "Threshold_normal_cohort = [0.0, 1.0, 3.0]   # choices = [0.0, 1.0, 2.0, 3.0, 5.0, 10.0]\n",
    "N_samples_normal = [1, 2, 10] #choices 1 to max number of samples in Normal whitelist\n",
    "\n",
    "tag_cancer = 'cancerCohort'\n",
    "tag_normal = 'gtexCohort'\n",
    "\n",
    "\n",
    "tag_prefix = 'G_'\n",
    "mutation_mode = 'ref'\n",
    "save_tag = 'GtexCohort'\n",
    "\n",
    "metadata_save = ['kmer', 'coord', 'junctionAnnotated', 'readFrameAnnotated']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c38d0281",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(45760252, 24)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load matrix to be filtered\n",
    "df_load = pd.read_csv(intermediate_output, sep = '\\t')\n",
    "df_load = df_load.rename({'batch': f'batch_{run_type}'}, axis = 1)\n",
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f3d368b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1218.0</td>\n",
       "      <td>689.0</td>\n",
       "      <td>249.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>2025.0</td>\n",
       "      <td>479.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>8641.0</td>\n",
       "      <td>6957.0</td>\n",
       "      <td>4789.0</td>\n",
       "      <td>3211.0</td>\n",
       "      <td>1347.0</td>\n",
       "      <td>191.0</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>372</td>\n",
       "      <td>367</td>\n",
       "      <td>352</td>\n",
       "      <td>295</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  RKSTQMPCT                 1218.0                   689.0   \n",
       "1  RKSTQMPCT                 1279.0                   274.0   \n",
       "2  RKSTQMPCT                 7350.0                  4622.0   \n",
       "3  RKSTQMPCT                 2025.0                   479.0   \n",
       "4  RKSTQMPCT                 8641.0                  6957.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   249.0                    89.0                    26.0   \n",
       "1                    70.0                    28.0                     5.0   \n",
       "2                  2653.0                  1745.0                  1126.0   \n",
       "3                    96.0                    35.0                     8.0   \n",
       "4                  4789.0                  3211.0                  1347.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92379857:92379859:92611313:92611338:None:None   \n",
       "1                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "2                    734.0  92379857:92379859:92611313:92611338:None:None   \n",
       "3                      1.0  92379857:92379859:92611313:92611338:None:None   \n",
       "4                    191.0  92379857:92379859:92611313:92611338:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0               True               False  ...                       372   \n",
       "1               True               False  ...                       372   \n",
       "2               True               False  ...                       372   \n",
       "3               True               False  ...                       372   \n",
       "4               True               False  ...                       372   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                       367                       352   \n",
       "1                       367                       352   \n",
       "2                       367                       352   \n",
       "3                       367                       352   \n",
       "4                       367                       352   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGA25131901A01R156513all  \\\n",
       "0                        295                  20.982616   \n",
       "1                        295                  20.982616   \n",
       "2                        295                  20.982616   \n",
       "3                        295                  20.982616   \n",
       "4                        295                  20.982616   \n",
       "\n",
       "   TCGA25131301A01R156513all  TCGA61200801A02R156813all  \\\n",
       "0                   8.422932                  17.304198   \n",
       "1                   8.422932                  17.304198   \n",
       "2                   8.422932                  17.304198   \n",
       "3                   8.422932                  17.304198   \n",
       "4                   8.422932                  17.304198   \n",
       "\n",
       "   TCGA24143101A01R156613all  TCGA24229801A01R156913all  isAnnotated  \n",
       "0                  16.069005                  19.585264          NaN  \n",
       "1                  16.069005                  19.585264          NaN  \n",
       "2                  16.069005                  19.585264          NaN  \n",
       "3                  16.069005                  19.585264          NaN  \n",
       "4                  16.069005                  19.585264          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2331bcea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(45760252, 24)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76817af3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------- processing TCGA25131901A01R156513all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_filtered_df_TCGA-25-1319-01A-01R-1565-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 132460 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 129003 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 134010 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 130551 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 140779 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 137305 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 132704 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 129247 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 134038 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 130579 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 140779 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 137305 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 136022 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 132550 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 136785 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 133313 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1194632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 141484 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 137998 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 119854 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 116397 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 121329 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 117871 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 127473 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 124000 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 120083 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 116626 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 121350 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 117892 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 127473 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 124000 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 123127 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 119662 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 123862 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 120397 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1173359 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 128080 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 124601 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 115989 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 112562 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 117391 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 113963 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 123372 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 119929 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 116178 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 112751 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 117402 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 113974 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 123372 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 119929 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 119006 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 115571 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 119730 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 116295 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1167145 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 123945 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 120496 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 77214 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 73819 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78080 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 74684 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81966 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78562 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 77307 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 73912 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78080 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 74684 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81966 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78562 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78977 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 75579 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 79457 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 76059 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1079263 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 82281 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78876 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 134975 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 131518 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 136610 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 133151 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 143510 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 140036 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 135226 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 131769 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 136638 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 133179 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 143510 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 140036 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 138643 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 135171 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 139442 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 135970 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1197827 kmers\n",
      "# Filter_Sample n = 1197827 kmers\n",
      "# Filter_Sample_Cohort n = 1197827 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 144221 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 140735 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1319-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGA25131301A01R156513all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_filtered_df_TCGA-25-1313-01A-01R-1565-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78786 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 75474 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 79613 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 76288 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83699 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80365 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78942 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 75630 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 80954 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 77624 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81428 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78098 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1046564 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 84205 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80867 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 70756 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 67444 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 71483 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 68158 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 75118 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 71784 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 70892 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 67580 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 71499 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 68174 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 75118 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 71784 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 72684 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 69354 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 73089 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 69759 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1030934 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 75540 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 72202 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 69401 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 66089 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 70104 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 66779 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 73838 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 70504 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 69519 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 66207 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 70118 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 66793 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 73838 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 70504 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 71300 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 67970 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 71714 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 68384 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1028667 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 74244 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 70906 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 50489 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 47211 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 50952 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 47674 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 53486 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 50206 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 50540 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 47262 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 50952 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 47674 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 53486 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 50206 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 51726 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 48443 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 52013 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 48730 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 976841 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 53768 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 50484 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80199 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 84404 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 81079 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88753 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85419 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83681 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80369 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 84426 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 81101 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88753 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85419 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 85768 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 82438 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 86274 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 82944 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1052807 kmers\n",
      "# Filter_Sample n = 1052807 kmers\n",
      "# Filter_Sample_Cohort n = 1052807 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 89268 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85930 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-25-1313-01A-01R-1565-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGA61200801A02R156813all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_filtered_df_TCGA-61-2008-01A-02R-1568-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 165575 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 162154 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 167703 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 164281 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 176632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 173206 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 166015 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 162594 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 167786 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 164364 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 176632 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 173206 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 170264 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 166839 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 171431 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 168006 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1207927 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 177514 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 174086 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 147173 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 143760 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 149126 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 145712 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 157383 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 153965 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 147580 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 144167 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 149193 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 145779 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 157383 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 153965 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 151538 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 148121 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 152614 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 149197 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1180671 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 158184 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 154764 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 143155 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 139766 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 144986 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 141596 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 152994 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 149600 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 143524 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 140135 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 145061 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 141671 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 152994 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 149600 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 147276 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 143883 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 148316 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 144923 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1174064 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 153801 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 150405 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 92395 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 89068 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 93493 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 90165 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 99262 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 95930 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 92584 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 89257 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 93517 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 90189 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 99262 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 95930 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 95210 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 91880 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 95854 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 92524 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1078546 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 99848 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 96515 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 172056 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 168635 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 174224 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 170802 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 183296 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 179870 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 172498 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 169077 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 174307 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 170885 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 183296 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 179870 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 176786 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 173361 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 177978 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 174553 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1215713 kmers\n",
      "# Filter_Sample n = 1215713 kmers\n",
      "# Filter_Sample_Cohort n = 1215713 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 184178 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 180750 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-61-2008-01A-02R-1568-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGA24143101A01R156613all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_filtered_df_TCGA-24-1431-01A-01R-1566-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 87047 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 84167 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88135 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85255 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 92998 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 90117 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 87297 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 84417 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88151 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85271 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 92998 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 90117 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 89738 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 86856 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 90193 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 87311 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1014511 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 93484 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 90602 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 80335 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 77462 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81337 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78464 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 85783 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 82909 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 80570 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 77697 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81353 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78480 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 85783 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 82909 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 82750 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 79875 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83166 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80291 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1002444 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 86209 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 83334 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim0.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78343 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 75470 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 79293 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 76420 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83592 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80718 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 78548 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 75675 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 79306 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 76433 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 83592 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 80718 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 80705 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 77830 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 81113 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 78238 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 999462 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 84048 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 81173 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 55586 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 52743 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 56198 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 53355 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 59318 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 56474 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 55697 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 52854 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 56205 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 53362 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 59318 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 56474 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 57173 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 54328 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 57425 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 54580 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 946241 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 59590 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 56745 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLim2.0Across5_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88199 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85319 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 89287 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 86407 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 94219 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 91338 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 88449 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 85569 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 89303 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 86423 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 94219 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 91338 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 90937 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 88055 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 91392 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 88510 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1015992 kmers\n",
      "# Filter_Sample n = 1015992 kmers\n",
      "# Filter_Sample_Cohort n = 1015992 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 94709 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 91827 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-1431-01A-01R-1566-13_SampleLim0.0CohortLimNoneAcrossNone_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n",
      "-------- processing TCGA24229801A01R156913all -------- \n",
      "\n",
      "Saving to summary file /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_filtered_df_TCGA-24-2298-01A-01R-1569-13_samp_chrt_norm_mot.tsv\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 178762 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 175340 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 180893 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 177471 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 190428 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 187003 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim0.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 179214 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 175792 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 180937 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 177515 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 190428 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 187003 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim1.0Across10.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 183810 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 180382 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across1.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 184871 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 181443 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across2.tsv.gz \n",
      "\n",
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1299926 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal n = 191296 kmers\n",
      "# Filter_Sample_Cohort_CohortNormal_pepAnnot n = 187866 kmers\n",
      "Saving outputs to: /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Ovarian_374/filtering_samples/chosen_filters_06March/G_TCGA-24-2298-01A-01R-1569-13_SampleLim0.0CohortLim0.0Across1_FiltNormalsGtexCohortCohortlim3.0Across10.tsv.gz \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Init_Sample n = 1310144 kmers\n",
      "# Filter_Sample n = 1310144 kmers\n",
      "# Filter_Sample_Cohort n = 1266018 kmers\n"
     ]
    }
   ],
   "source": [
    "for cancer_sample_ori in target_samples: # TODO update\n",
    "    # Sample naming\n",
    "    target_sample = cancer_sample_ori.replace('-', '').replace('.', '')\n",
    "    cancer_sample_ori = cancer_sample_ori.replace('.all', '')\n",
    "    print(f'-------- processing {target_sample} -------- \\n')\n",
    "    \n",
    "    # Summary file for sample\n",
    "    summary_file = f'{tag_prefix}filtered_df_{cancer_sample_ori}_samp_chrt_norm_mot.tsv'\n",
    "    summary_path = os.path.join(output_dir, summary_file)\n",
    "    print(f'Saving to summary file {summary_path}')\n",
    "    \n",
    "\n",
    "    df_expr = []\n",
    "    report_count = [] \n",
    "    report_steps = []\n",
    "    for threshold_target in Threshold_target:\n",
    "        for threshold_cancer_cohort in Threshold_cancer_cohort:\n",
    "            for n_samples_cancer in N_samples_cancer:\n",
    "                for threshold_normal_cohort in Threshold_normal_cohort:\n",
    "                    for n_samples_normal in N_samples_normal:\n",
    "                        if (n_samples_cancer is None) ^ (threshold_cancer_cohort is None):\n",
    "                            continue\n",
    "\n",
    "                        adjusted_threshold_col = 'tmp_cancer_cohort'\n",
    "                        max_threshold_col = 'tmp_normal_Nmax_sup{}'.format(threshold_normal_cohort)\n",
    "                        max_threshold_col_base = 'tmp_normal_Nmax_sup{}'.format(0)\n",
    "\n",
    "\n",
    "                        df = df_load.copy()\n",
    "                        # Make correction for number samples passing theshold in cohort. We want to exclude the target sample in counting\n",
    "                        if (n_samples_cancer is not None) and (threshold_cancer_cohort is not None):\n",
    "                            df[adjusted_threshold_col] = df[get_threshold_colname(threshold_cancer_cohort, tag_cancer)]\n",
    "                            df.loc[df[target_sample] >= threshold_cancer_cohort, adjusted_threshold_col] -=1 \n",
    "\n",
    "                        # Number of kmers expressed in sample \n",
    "                        df = filter_single_col(df, 0, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Init_Sample')\n",
    "\n",
    "                        # Number of kmers >= threshold in sample \n",
    "                        df = filter_single_col(df, threshold_target, target_sample)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample')\n",
    "\n",
    "                        \n",
    "                        # Filter for cancer cancer cohort \n",
    "                        if (n_samples_cancer is not None) and (threshold_cancer_cohort is not None):\n",
    "                            df = filter_cancer_cohort(df, n_samples_cancer, adjusted_threshold_col)\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort')\n",
    "                            \n",
    "\n",
    "\n",
    "                        # Expression in gtex cohort >= threshold \n",
    "                        recurrence_custom =  max_recurrence_over_kmer(df, \n",
    "                                                                      get_threshold_colname(threshold_normal_cohort, tag_normal), \n",
    "                                                                      max_threshold_col)\n",
    "\n",
    "                        # Expression in gtex cohort > 0  \n",
    "                        recurrence_custom_base = max_recurrence_over_kmer(df, \n",
    "                                                                          get_threshold_colname(0.0, tag_normal),\n",
    "                                                                          max_threshold_col_base) \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "                        # Perform Background filtering \n",
    "                        df = df.merge(recurrence_custom, on = 'kmer', how = 'left')\n",
    "                        df = df.merge(recurrence_custom_base, on = 'kmer', how = 'left')\n",
    "\n",
    "                        df = df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), :]\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal')\n",
    "\n",
    "                        #Perform Annotated junctions filtering \n",
    "                        df = df[df['isAnnotated'].isna()]\n",
    "                        output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot')\n",
    "\n",
    "#                         # DEV: Exclude genes where GTEX is missing\n",
    "#                         df = df.loc[df['exclude'].isna()]\n",
    "#                         output_count(df, report_count, report_steps, 'Filter_Sample_Cohort_CohortNormal_pepAnnot_EXPGTEX')\n",
    "\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        # Save outputs \n",
    "                        # outpaths\n",
    "                        base_path_final = os.path.join(output_dir,\n",
    "                                                       (f'{tag_prefix}{cancer_sample_ori}_'\n",
    "                                                        f'SampleLim{threshold_target}'\n",
    "                                                        f'CohortLim{threshold_cancer_cohort}'\n",
    "                                                        f'Across{n_samples_cancer}_'\n",
    "                                                        f'FiltNormals{save_tag}'\n",
    "                                                        f'Cohortlim{threshold_normal_cohort}'\n",
    "                                                        f'Across{n_samples_normal}.tsv.gz'))\n",
    "                        print(f'Saving outputs to: {base_path_final} \\n')\n",
    "                        df.loc[:, metadata_save].to_csv(base_path_final, compression = 'gzip', index = None, sep = '\\t')\n",
    "\n",
    "\n",
    "    save_output_count(summary_path, report_count, report_steps, '', cancer_sample_ori, mutation_mode,\n",
    "                      threshold_target, threshold_cancer_cohort, n_samples_cancer,\n",
    "                          threshold_normal_cohort, n_samples_normal, save_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "567bfa5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "print('hello')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b148871d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b42885e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58ea428c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2d66abf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83fbc390",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1cba0d91",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98ef99e2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74b560d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'KVDTLGKST'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d85cd18",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f979d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            ['kmer', 'coord', 'exclude', 'batch_brca', 'junctionAnnotated', 'readFrameAnnotated', target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01993640",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer, 'batch_brca']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63aba557",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_batch.loc[(gene_to_batch['batch_brca'] == 54877)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c54267e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = df_load.merge(gene_to_batch[['exclude', f'batch_{run_type}']], on = f'batch_{run_type}', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ada9f080",
   "metadata": {},
   "outputs": [],
   "source": [
    "foo.loc[(foo['kmer'] == kmer)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b35b9cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "final  =df.loc[ ~ ((df[max_threshold_col] >= 1) & (df[max_threshold_col_base] >= n_samples_normal)), \n",
    "       ['kmer', get_threshold_colname(threshold_cancer_cohort, tag_cancer)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97464fa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "final = df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3d05819",
   "metadata": {},
   "outputs": [],
   "source": [
    "final.loc[final['cancerCohortfilter >0.0']> 403 , ['kmer', 'tmp_cancer_cohort',f'batch_{run_type}'] ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "455fb619",
   "metadata": {},
   "outputs": [],
   "source": [
    "final['cancerCohortfilter >0.0'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3a9ec1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea032687",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_load.loc[df_load['kmer'] == kmer,\\\n",
    "            [target_sample,  get_threshold_colname(threshold_cancer_cohort, tag_cancer), \\\n",
    "            get_threshold_colname(threshold_normal_cohort, tag_normal), get_threshold_colname(0.0, tag_normal)]]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
