{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "be763474-6cdf-43cc-8a8a-52113292d86a",
   "metadata": {},
   "source": [
    "## Table of Contents\n",
    "- [Import libraries](#1)\n",
    "- [Download tables](#2)\n",
    "- [Process tables](#3)\n",
    "- [Download plots](#4)\n",
    "- [Download plots with log10](#5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e178b94-edf7-479b-9b4d-79b777ecb97e",
   "metadata": {},
   "source": [
    "<a name='1'></a>\n",
    "## Import libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac9f527b-339f-48f1-8a30-3d9e96a33f2a",
   "metadata": {},
   "source": [
    "The script is focused on setting up an environment for data analysis and visualization. It imports a suite of libraries and modules that are essential for statistical computing, data manipulation, progress tracking, file system operations, and generating visualizations such as plots and Venn diagrams. The specific libraries imported include pandas for data structures, numpy for numerical operations, tqdm for progress bars, glob for file path retrieval, os for operating system interaction, matplotlib and seaborn for plotting and graphical representations, and matplotlib_venn for creating Venn diagrams.\n",
    "\n",
    "Additionally, the script modifies the system path to include a custom directory, which suggests that the script will use additional custom modules and configuration settings located in this directory. These custom modules, imported with wildcard imports (from config import * and from functions import *)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "88e8baca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# %load /cluster/home/myurchikova/github/projects2020_ohsu/eth/learning_Master_thesis/TASKS/func/base_imports.py\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import tqdm \n",
    "import glob\n",
    "import os\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import tarfile\n",
    "import re\n",
    "from matplotlib_venn import venn2, venn2_circles, venn2_unweighted\n",
    "from matplotlib_venn import venn3, venn3_circles\n",
    "import sys\n",
    "sys.path.append(r\"/cluster/home/prelotla/github/projects2020_ohsu/eth/MY_Master_thesis_rerun_LP/TASKS/func/\")\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "from config import *\n",
    "from functions import *\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c5eb18b-bada-4b0f-97a3-abc6c7307d63",
   "metadata": {},
   "source": [
    "<a name='2'></a>\n",
    "## Download tables"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b618898-3472-4a95-9b8b-b8eb76d2e48a",
   "metadata": {},
   "source": [
    "Definition constants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d8f388f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "samples = ['TCGA-25-1319-01A-01R-1565-13',\n",
    "               'TCGA-25-1313-01A-01R-1565-13',\n",
    "               'TCGA-61-2008-01A-02R-1568-13',\n",
    "               'TCGA-24-1431-01A-01R-1566-13',\n",
    "               'TCGA-24-2298-01A-01R-1569-13']\n",
    "MAIN = [\n",
    "        'kmer',\n",
    "        'batch',\n",
    "]\n",
    "                                 \n",
    "COLS_ETH = [\n",
    "            'coord',\n",
    "            'kmer',\n",
    "            'batch',\n",
    "            'TCGA25131901A01R156513all',\n",
    "            'TCGA25131301A01R156513all',\n",
    "            'TCGA61200801A02R156813all',\n",
    "            'TCGA24143101A01R156613all',\n",
    "            'TCGA24229801A01R156913all',\n",
    "           ]\n",
    "ETH_COLMNS= {\n",
    "             'TCGA25131901A01R156513all':'TCGA25131901A01R156513',\n",
    "             'TCGA25131301A01R156513all':'TCGA25131301A01R156513',\n",
    "             'TCGA61200801A02R156813all':'TCGA61200801A02R156813',\n",
    "             'TCGA24143101A01R156613all':'TCGA24143101A01R156613',\n",
    "             'TCGA24229801A01R156913all':'TCGA24229801A01R156913',\n",
    "            }\n",
    "EXPRESSION=['TCGA25131901A01R156513',\n",
    "            'TCGA25131301A01R156513',\n",
    "            'TCGA61200801A02R156813',\n",
    "            'TCGA24143101A01R156613',\n",
    "            'TCGA24229801A01R156913',]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b48048f-0e05-4b80-b2b7-ae8a8bff71b8",
   "metadata": {},
   "source": [
    "eth_df=pd.read_csv(OVARIAN_ETH, usecols = COLS_ETH, sep = '\\t', low_memory=False):\n",
    "This line reads a tab-separated values (TSV) file (likely due to sep='\\t'), which is specified by the OVARIAN_ETH path. It only reads certain columns (COLS_ETH) to conserve memory, and it turns off optimizations for low memory usage.\n",
    "batch_to_gene = pd.read_csv(BATCH_PATH, names = ['gene_id', 'batch'], low_memory=False):\n",
    "Another CSV file is read from BATCH_PATH where the columns are explicitly named 'gene_id' and 'batch'. The low_memory=False parameter is also used here, which is helpful when processing large files.\n",
    "eth_df.head():\n",
    "Displays the first few rows of the DataFrame eth_df to get a preview of the data structure. The DataFrame seems to contain columns like 'kmer', 'coord', 'batch', and several others that look like identifiers, possibly for gene expression data.\n",
    "ohsu_df=pd.read_csv(OVARIAN_OHSU, sep = '\\t', low_memory=False):\n",
    "Reads another TSV file specified by OVARIAN_OHSU. Similar to the first read operation, it disables memory optimization.\n",
    "ohsu_df.head():\n",
    "Similar to the eth_df.head(), this previews the first few rows of the ohsu_df DataFrame.\n",
    "The data shown in the head of the eth_df DataFrame contains:\n",
    "\n",
    "'kmer' column: This typically refers to substrings of length 'k' found in DNA or RNA sequences.\n",
    "'coord' column: Likely refers to genomic coordinates.\n",
    "'batch' column: Could be related to the batch processing of samples.\n",
    "Other columns with TCGA (The Cancer Genome Atlas) in their headers: These columns likely contain gene expression values or other related data for different samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "aaefaea2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>batch</th>\n",
       "      <th>TCGA25131901A01R156513all</th>\n",
       "      <th>TCGA25131301A01R156513all</th>\n",
       "      <th>TCGA61200801A02R156813all</th>\n",
       "      <th>TCGA24143101A01R156613all</th>\n",
       "      <th>TCGA24229801A01R156913all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RKSTQMPCT</td>\n",
       "      <td>92379857:92379859:92611313:92611338:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>20.982616</td>\n",
       "      <td>8.422932</td>\n",
       "      <td>17.304198</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>19.585264</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer                                          coord  batch  \\\n",
       "0  RKSTQMPCT  92379857:92379859:92611313:92611338:None:None  56543   \n",
       "1  RKSTQMPCT  92379857:92379859:92611313:92611338:None:None  56543   \n",
       "2  RKSTQMPCT  92379857:92379859:92611313:92611338:None:None  56543   \n",
       "3  RKSTQMPCT  92379857:92379859:92611313:92611338:None:None  56543   \n",
       "4  RKSTQMPCT  92379857:92379859:92611313:92611338:None:None  56543   \n",
       "\n",
       "   TCGA25131901A01R156513all  TCGA25131301A01R156513all  \\\n",
       "0                  20.982616                   8.422932   \n",
       "1                  20.982616                   8.422932   \n",
       "2                  20.982616                   8.422932   \n",
       "3                  20.982616                   8.422932   \n",
       "4                  20.982616                   8.422932   \n",
       "\n",
       "   TCGA61200801A02R156813all  TCGA24143101A01R156613all  \\\n",
       "0                  17.304198                  16.069005   \n",
       "1                  17.304198                  16.069005   \n",
       "2                  17.304198                  16.069005   \n",
       "3                  17.304198                  16.069005   \n",
       "4                  17.304198                  16.069005   \n",
       "\n",
       "   TCGA24229801A01R156913all  \n",
       "0                  19.585264  \n",
       "1                  19.585264  \n",
       "2                  19.585264  \n",
       "3                  19.585264  \n",
       "4                  19.585264  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eth_df=pd.read_csv(OVARIAN_ETH, usecols = COLS_ETH, sep = '\\t', low_memory=False)\n",
    "batch_to_gene = pd.read_csv(BATCH_PATH, names = ['gene_id', 'batch'], low_memory=False)\n",
    "eth_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6b8483b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;STCTTTRPAVAGQCGA</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;STCTTTRPAVAGQCGA</td>\n",
       "      <td>NVYYNEASG;VYYNEASGG;YYNEASGGR;YNEASGGRY;NEASGG...</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA</td>\n",
       "      <td>4.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10;277578;281199;-</td>\n",
       "      <td>2.295572</td>\n",
       "      <td>30.465966</td>\n",
       "      <td>16.003571</td>\n",
       "      <td>10.071656</td>\n",
       "      <td>26.917642</td>\n",
       "      <td>51.966700</td>\n",
       "      <td>53.589040</td>\n",
       "      <td>31.535793</td>\n",
       "      <td>50.723584</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.277578.281199.M...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RAHKSVTECAVFTWTNL</td>\n",
       "      <td>RAHKSVTECAVFTWTNL</td>\n",
       "      <td>RAHKSVTEC;AHKSVTECA;HKSVTECAV;KSVTECAVF;SVTECA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10;280261;281199;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.307159</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.280261.281199.M...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10;281324;283271;-</td>\n",
       "      <td>3.443358</td>\n",
       "      <td>23.212165</td>\n",
       "      <td>18.530451</td>\n",
       "      <td>15.107483</td>\n",
       "      <td>20.188231</td>\n",
       "      <td>32.821074</td>\n",
       "      <td>33.065578</td>\n",
       "      <td>44.150110</td>\n",
       "      <td>40.826300</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.281324.283271.M...</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "      <td>TELTDANGERHDALYVV</td>\n",
       "      <td>TELTDANGERHDALYVV</td>\n",
       "      <td>TELTDANGE;ELTDANGER;LTDANGERH;TDANGERHD;DANGER...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10;283447;286272;-</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>41.346668</td>\n",
       "      <td>19.372744</td>\n",
       "      <td>6.714437</td>\n",
       "      <td>28.840330</td>\n",
       "      <td>61.539513</td>\n",
       "      <td>62.710579</td>\n",
       "      <td>37.842952</td>\n",
       "      <td>92.787044</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.283447.286272.M...</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "      <td>PLGDSHLGEIWVHSAH</td>\n",
       "      <td>PLGDSHLGEIWVHSAH</td>\n",
       "      <td>PLGDSHLGE;LGDSHLGEI;GDSHLGEIW;DSHLGEIWV;SHLGEI...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      jx  TCGA-24-1431-01A-01R-1566-13  \\\n",
       "0    chr10;48726;48803;-                      0.000000   \n",
       "1  chr10;277578;281199;-                      2.295572   \n",
       "2  chr10;280261;281199;-                      0.000000   \n",
       "3  chr10;281324;283271;-                      3.443358   \n",
       "4  chr10;283447;286272;-                      1.147786   \n",
       "\n",
       "   TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \\\n",
       "0                      2.176140                      0.842293   \n",
       "1                     30.465966                     16.003571   \n",
       "2                      0.000000                      0.000000   \n",
       "3                     23.212165                     18.530451   \n",
       "4                     41.346668                     19.372744   \n",
       "\n",
       "   TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \\\n",
       "0                      2.517914                      0.961344   \n",
       "1                     10.071656                     26.917642   \n",
       "2                      0.000000                      0.961344   \n",
       "3                     15.107483                     20.188231   \n",
       "4                      6.714437                     28.840330   \n",
       "\n",
       "   TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \\\n",
       "0                      0.000000                      5.700962   \n",
       "1                     51.966700                     53.589040   \n",
       "2                      0.000000                      0.000000   \n",
       "3                     32.821074                     33.065578   \n",
       "4                     61.539513                     62.710579   \n",
       "\n",
       "   TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \\\n",
       "0                      2.102386                      1.237161  ...   \n",
       "1                     31.535793                     50.723584  ...   \n",
       "2                      6.307159                      2.474321  ...   \n",
       "3                     44.150110                     40.826300  ...   \n",
       "4                     37.842952                     92.787044  ...   \n",
       "\n",
       "                               modified_upstream_txs  \\\n",
       "0  ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "1  ['ENST00000280886.12.MOD.CHR10.277578.281199.M...   \n",
       "2  ['ENST00000280886.12.MOD.CHR10.280261.281199.M...   \n",
       "3  ['ENST00000280886.12.MOD.CHR10.281324.283271.M...   \n",
       "4  ['ENST00000280886.12.MOD.CHR10.283447.286272.M...   \n",
       "\n",
       "                     in-frame_all-transcript_biexons  \\\n",
       "0  LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...   \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...   \n",
       "\n",
       "                      in-frame_nonhanging-tx_biexons  \\\n",
       "0  LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...   \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...   \n",
       "\n",
       "            in-frame_peptide_sequence  hanging_txs_included_inframe_pepseqs  \\\n",
       "0  NVYYNEASGGRYVPRAV;STCTTTRPAVAGQCGA    NVYYNEASGGRYVPRAV;STCTTTRPAVAGQCGA   \n",
       "1                   RAHKSVTECAVFTWTNL                     RAHKSVTECAVFTWTNL   \n",
       "2                           RAHKSVTEC                             RAHKSVTEC   \n",
       "3                   TELTDANGERHDALYVV                     TELTDANGERHDALYVV   \n",
       "4                    PLGDSHLGEIWVHSAH                      PLGDSHLGEIWVHSAH   \n",
       "\n",
       "                       prefiltered_in-frame_epitopes  \\\n",
       "0  NVYYNEASG;VYYNEASGG;YYNEASGGR;YNEASGGRY;NEASGG...   \n",
       "1  RAHKSVTEC;AHKSVTECA;HKSVTECAV;KSVTECAVF;SVTECA...   \n",
       "2                                          RAHKSVTEC   \n",
       "3  TELTDANGE;ELTDANGER;LTDANGERH;TDANGERHD;DANGER...   \n",
       "4  PLGDSHLGE;LGDSHLGEI;GDSHLGEIW;DSHLGEIWV;SHLGEI...   \n",
       "\n",
       "   prefiltered_inframe_epitope_count                     in-frame_neoepitopes  \\\n",
       "0                               17.0  TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA   \n",
       "1                                9.0                                      NaN   \n",
       "2                                1.0                                      NaN   \n",
       "3                                9.0                                      NaN   \n",
       "4                                8.0                                      NaN   \n",
       "\n",
       "   in-frame_neoepitope_count  \\\n",
       "0                        4.0   \n",
       "1                        0.0   \n",
       "2                        0.0   \n",
       "3                        0.0   \n",
       "4                        0.0   \n",
       "\n",
       "               frame-agnostic_all-transcript_biexons  \n",
       "0  LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...  \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...  \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...  \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...  \n",
       "\n",
       "[5 rows x 50 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_df=pd.read_csv(OVARIAN_OHSU, sep = '\\t', low_memory=False)\n",
    "ohsu_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2229da36-c372-4d8b-8955-e3efb09842b2",
   "metadata": {},
   "source": [
    "<a name='3'></a>\n",
    "## Process tables"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5249eb72-f2ad-4c19-b954-a693b007b1db",
   "metadata": {},
   "source": [
    "eth_df=table_processing.get_junction_coordinates(eth_df, 'coord'):\n",
    "This line suggests that the eth_df DataFrame is being passed to a function to get junction coordinates based on the 'coord' column. The function is likely part of a custom module named table_processing.\n",
    "ETH_df = pd.merge(batch_to_gene, eth_df, on=['batch']):\n",
    "The pandas merge function is used to combine two DataFrames, batch_to_gene and eth_df, on the 'batch' column. This is commonly done to join related data that share a common identifier.\n",
    "ETH_df.drop(columns=['batch'], inplace=True):\n",
    "The drop method is used to remove the 'batch' column from ETH_df after the merge operation, indicating that the column is no longer needed for further analysis.\n",
    "ohsu_df.rename(columns = ETH_COLUMNS, inplace = True):\n",
    "The rename method is used to rename the columns of ohsu_df DataFrame according to a dictionary or list of column names provided by ETH_COLUMNS. The inplace=True parameter indicates that the changes are to be applied directly to the DataFrame without creating a copy.\n",
    "ETH_df.rename(columns = ETH_COLUMNS, inplace = True):\n",
    "Similarly, the rename method is used to rename the columns of ETH_df DataFrame according to ETH_COLUMNS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5ed1c4a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>TCGA25131901A01R156513</th>\n",
       "      <th>TCGA25131301A01R156513</th>\n",
       "      <th>TCGA61200801A02R156813</th>\n",
       "      <th>TCGA24143101A01R156613</th>\n",
       "      <th>TCGA24229801A01R156913</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>NQTTSPAPF</td>\n",
       "      <td>1333118:1333124:1333505:1333526:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>0.72538</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>NQTTSPAPF</td>\n",
       "      <td>1333118:1333124:1333505:1333526:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>0.72538</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>DNQTTSPAP</td>\n",
       "      <td>1333115:1333124:1333505:1333523:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>0.72538</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>DNQTTSPAP</td>\n",
       "      <td>1333115:1333124:1333505:1333523:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>0.72538</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>QTTSPAPFV</td>\n",
       "      <td>1333121:1333124:1333505:1333529:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>0.72538</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46118846</th>\n",
       "      <td>ENSG00000171533.11</td>\n",
       "      <td>KSCTSVEFP</td>\n",
       "      <td>75608108:75608111:75607238:75607262:None:None</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>-</td>\n",
       "      <td>75607262:75608108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46118847</th>\n",
       "      <td>ENSG00000171533.11</td>\n",
       "      <td>KSCTSVEFP</td>\n",
       "      <td>75608108:75608111:75607238:75607262:None:None</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>-</td>\n",
       "      <td>75607262:75608108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46118848</th>\n",
       "      <td>ENSG00000171533.11</td>\n",
       "      <td>KSCTSVEFP</td>\n",
       "      <td>75608108:75608111:75607238:75607262:None:None</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>-</td>\n",
       "      <td>75607262:75608108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46118849</th>\n",
       "      <td>ENSG00000171533.11</td>\n",
       "      <td>KSCTSVEFP</td>\n",
       "      <td>75608108:75608111:75607238:75607262:None:None</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>-</td>\n",
       "      <td>75607262:75608108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46118850</th>\n",
       "      <td>ENSG00000171533.11</td>\n",
       "      <td>PFKEPPKSC</td>\n",
       "      <td>75608108:75608129:75607256:75607262:None:None</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>-</td>\n",
       "      <td>75607262:75608108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>46118851 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     gene_id       kmer  \\\n",
       "0          ENSG00000169962.5  NQTTSPAPF   \n",
       "1          ENSG00000169962.5  NQTTSPAPF   \n",
       "2          ENSG00000169962.5  DNQTTSPAP   \n",
       "3          ENSG00000169962.5  DNQTTSPAP   \n",
       "4          ENSG00000169962.5  QTTSPAPFV   \n",
       "...                      ...        ...   \n",
       "46118846  ENSG00000171533.11  KSCTSVEFP   \n",
       "46118847  ENSG00000171533.11  KSCTSVEFP   \n",
       "46118848  ENSG00000171533.11  KSCTSVEFP   \n",
       "46118849  ENSG00000171533.11  KSCTSVEFP   \n",
       "46118850  ENSG00000171533.11  PFKEPPKSC   \n",
       "\n",
       "                                                  coord  \\\n",
       "0             1333118:1333124:1333505:1333526:None:None   \n",
       "1             1333118:1333124:1333505:1333526:None:None   \n",
       "2             1333115:1333124:1333505:1333523:None:None   \n",
       "3             1333115:1333124:1333505:1333523:None:None   \n",
       "4             1333121:1333124:1333505:1333529:None:None   \n",
       "...                                                 ...   \n",
       "46118846  75608108:75608111:75607238:75607262:None:None   \n",
       "46118847  75608108:75608111:75607238:75607262:None:None   \n",
       "46118848  75608108:75608111:75607238:75607262:None:None   \n",
       "46118849  75608108:75608111:75607238:75607262:None:None   \n",
       "46118850  75608108:75608129:75607256:75607262:None:None   \n",
       "\n",
       "          TCGA25131901A01R156513  TCGA25131301A01R156513  \\\n",
       "0                       0.000000                     0.0   \n",
       "1                       0.000000                     0.0   \n",
       "2                       0.000000                     0.0   \n",
       "3                       0.000000                     0.0   \n",
       "4                       0.000000                     0.0   \n",
       "...                          ...                     ...   \n",
       "46118846                4.196523                     0.0   \n",
       "46118847                4.196523                     0.0   \n",
       "46118848                4.196523                     0.0   \n",
       "46118849                4.196523                     0.0   \n",
       "46118850                4.196523                     0.0   \n",
       "\n",
       "          TCGA61200801A02R156813  TCGA24143101A01R156613  \\\n",
       "0                            0.0                4.591144   \n",
       "1                            0.0                4.591144   \n",
       "2                            0.0                4.591144   \n",
       "3                            0.0                4.591144   \n",
       "4                            0.0                4.591144   \n",
       "...                          ...                     ...   \n",
       "46118846                     0.0                0.000000   \n",
       "46118847                     0.0                0.000000   \n",
       "46118848                     0.0                0.000000   \n",
       "46118849                     0.0                0.000000   \n",
       "46118850                     0.0                0.000000   \n",
       "\n",
       "          TCGA24229801A01R156913 strand junction_coordinate  \n",
       "0                        0.72538      +     1333124:1333505  \n",
       "1                        0.72538      +     1333124:1333505  \n",
       "2                        0.72538      +     1333124:1333505  \n",
       "3                        0.72538      +     1333124:1333505  \n",
       "4                        0.72538      +     1333124:1333505  \n",
       "...                          ...    ...                 ...  \n",
       "46118846                 0.00000      -   75607262:75608108  \n",
       "46118847                 0.00000      -   75607262:75608108  \n",
       "46118848                 0.00000      -   75607262:75608108  \n",
       "46118849                 0.00000      -   75607262:75608108  \n",
       "46118850                 0.00000      -   75607262:75608108  \n",
       "\n",
       "[46118851 rows x 10 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LP: Changed ETH table processing table_processing.get_junction_coordinates_updated\n",
    "eth_df=table_processing.get_junction_coordinates_updated(eth_df,'coord')\n",
    "ETH_df = pd.merge(batch_to_gene,eth_df, on=['batch'])\n",
    "ETH_df.drop(columns=['batch'],inplace=True)\n",
    "ohsu_df.rename(columns = ETH_COLMNS, inplace = True)\n",
    "ETH_df.rename(columns = ETH_COLMNS, inplace = True)\n",
    "ETH_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d345b48-05eb-4350-8a06-30e64c64cf4e",
   "metadata": {},
   "source": [
    "ohsu_df.rename(columns = {'in-frame_neopeptides': 'kmer'}, inplace = True):\n",
    "This line renames the column 'in-frame_neopeptides' to 'kmer' in the DataFrame ohsu_df. The inplace=True argument makes the change in the existing DataFrame without the need to assign the result to a new variable.\n",
    "ohsu_df = table_processing.ohsu_to_eth_coord(ohsu_df):\n",
    "The ohsu_df DataFrame is passed to a function ohsu_to_eth_coord from a custom module table_processing. This suggests that the function is transforming or mapping OHSU coordinates to ETH coordinates.\n",
    "ohsu_df = table_processing.change_column_names(ohsu_df):\n",
    "Another function from table_processing is used to change the column names in ohsu_df. The specific changes are not detailed, but this is likely part of data standardization or cleanup.\n",
    "ohsu_df = table_processing.preprocess_ohsu(ohsu_df):\n",
    "The DataFrame ohsu_df is further processed by a preprocess_ohsu function from the table_processing module, which likely performs a series of preprocessing steps on the data.\n",
    "ohsu_df['junction_coordinate'] = ohsu_df['x_shifted'].apply(lambda x: '::'.join(x.split('::')[1:3])):\n",
    "This line creates or modifies the 'junction_coordinate' column in ohsu_df by applying a lambda function to each value in the 'x_shifted' column. The lambda function splits each string on '::', slices to get the second and third elements (if available), and then joins these elements with '::'. This could be formatting a genomic coordinate or similar structured data.\n",
    "print(ohsu_df['junction_coordinate']):\n",
    "Finally, the 'junction_coordinate' column is printed to the output, likely for verification or inspection purposes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b8428cb5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0                  48725:48803\n",
      "0                  48725:48803\n",
      "0                  48725:48803\n",
      "0                  48725:48803\n",
      "29               390095:390274\n",
      "                  ...         \n",
      "1344358    109916656:109919488\n",
      "1344358    109916656:109919488\n",
      "1344358    109916656:109919488\n",
      "1344358    109916656:109919488\n",
      "1344358    109916656:109919488\n",
      "Name: junction_coordinate, Length: 3182186, dtype: object\n"
     ]
    }
   ],
   "source": [
    "# LP: Changed table_processing.ohsu_to_eth_coord (See function)\n",
    "ohsu_df.rename(columns = {'in-frame_neoepitopes': 'kmer'}, inplace = True)\n",
    "ohsu_df = table_processing.ohsu_to_eth_coord(ohsu_df)\n",
    "ohsu_df = table_processing.change_column_names(ohsu_df)\n",
    "ohsu_df = table_processing.preprocess_ohsu(ohsu_df)\n",
    "ohsu_df['junction_coordinate'] = ohsu_df['jx_shifted'].apply(lambda x: ':'.join(x.split(';')[1:3]))\n",
    "print(ohsu_df['junction_coordinate'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3001a77b",
   "metadata": {},
   "outputs": [],
   "source": [
    "oshu_df=ohsu_df[['gene_id','kmer','junction_coordinate']+EXPRESSION]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f357ad34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'kmer', 'TCGA25131901A01R156513', 'TCGA25131301A01R156513',\n",
       "       'TCGA61200801A02R156813', 'TCGA24143101A01R156613',\n",
       "       'TCGA24229801A01R156913', 'junction_coordinate'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ETH_df.drop(columns=['strand'],inplace=True)\n",
    "ETH_df.drop(columns=['coord'],inplace=True)\n",
    "ETH_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a046646b-613b-4431-9edd-d29deca2cda6",
   "metadata": {},
   "source": [
    "<a name='4'></a>\n",
    "## Plotting data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76db8438-9b53-4b6c-9b93-da027a8aa03e",
   "metadata": {},
   "source": [
    "A dictionary named out_table is initialized with various keys, each associated with an empty list. These keys are suggestive of data attributes such as sample identifiers, k-mer counts, sizes, and coordinates which are often used in genomic data analysis.\n",
    "A for loop begins, iterating over columns of ETH_df DataFrame using the tqdm library, which provides a progress bar for loops.\n",
    "Within the loop, multiple conditional and data manipulation steps are performed:\n",
    "Checking if the current column's k-mers are zero and performing set operations.\n",
    "Appending data to the lists in out_table, such as 'junction_coordinate' and various size and coordinate related data.\n",
    "Calculating differences between sets of k-mers and appending the results to the appropriate lists in out_table.\n",
    "The snippet contains several operations that transform and aggregate data, possibly to prepare for statistical analysis or visualization. For example, it deals with intersection and union of genomic coordinates, suggesting a comparison between different datasets or conditions within the study.\n",
    "The code ends with appending array data converted to lists, likely for uniformity in data structure within the out_table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0f3be676-7ffe-49dd-82e9-082868ff2a69",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_eth_df=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_OVARIAN,NAME_TABLES,NAME_ETH_OVARIAN])\n",
    "path_ohsu_df=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_OVARIAN,NAME_TABLES,NAME_OHSU_OVARIAN])\n",
    "ETH_df.to_csv(path_eth_df, header = True, sep='\\t')\n",
    "ohsu_df.to_csv(path_ohsu_df, header = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fc1207bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [07:02<00:00, 84.59s/it]\n"
     ]
    }
   ],
   "source": [
    "# LP Updated separated the coordinates\n",
    "sample=[]\n",
    "inter = []\n",
    "kmer_inter=[]\n",
    "eth_without_ohsu =  []\n",
    "ohsu_withou_eth =  []\n",
    "fs_ohsu =  []\n",
    "kmer_ohsu=[]\n",
    "fs_eth =  []\n",
    "kmer_eth=[]\n",
    "batch =  []\n",
    "gene_id =  []\n",
    "LANG='ENG'\n",
    "OHSU_COLOR ='red' #'red'\n",
    "ETH_COLOR = 'green'#'green'\n",
    "OHSU_ETH_COLOR ='yellow' #'yellow'\n",
    "\n",
    "if LANG == 'ENG':\n",
    "    title_venn = 'Comparison of unique kmers ({exp_col})'\n",
    "    title_venn_perc = 'Comparison of unique kmers in percentages ({exp_col})'\n",
    "    title_barplot = 'Comparison of unique kmers ({exp_col})'\n",
    "else:\n",
    "    title_venn = 'Порівняння унікальних kmers ({exp_col})'\n",
    "    title_venn_perc = 'Порівняння унікальних kmers у відсотках ({exp_col})'\n",
    "    title_barplot = 'Порівняння унікальних kmers ({exp_col})'\n",
    "\n",
    "out_table = {\n",
    "    'sample':sample,\n",
    "    'eth_without_ohsu':eth_without_ohsu,\n",
    "    'kmer_eth':kmer_eth,\n",
    "    'ohsu_without_eth':ohsu_withou_eth,\n",
    "    'kmer_ohsu':kmer_ohsu,\n",
    "    'inter':inter,\n",
    "    'kmer_inter':kmer_inter,\n",
    "    'full_size_ohsu': fs_ohsu,\n",
    "    'full_size_eth': fs_eth,\n",
    "\n",
    "    'coord_OHSU':[],\n",
    "    'coord_ETH':[],\n",
    "    'size_ohsu_coor' : [], \n",
    "    'size_eth_coor' : [], \n",
    "    'size_intersection_coor' : [], \n",
    "    'size_ohsu\\eth_coor' : [], \n",
    "    'size_eth\\ohsu_coor' : [],\n",
    "    'eth_coor\\inter_coor':[],\n",
    "    'ohsu_coor\\inter_coor':[],\n",
    "    'eth_coor\\ohsu_coor':[],\n",
    "    'ohsu_coor\\eth_coor':[],\n",
    "    'inter_coor':[],\n",
    "}\n",
    "for exp_col in tqdm.tqdm(EXPRESSION):\n",
    "    df1=ETH_df[ETH_df[exp_col] > 0]\n",
    "    df2=oshu_df[oshu_df[exp_col] > 0]\n",
    "    eth_kmers=set(df1['kmer'])\n",
    "    ohsu_kmers=set(df2['kmer'])\n",
    "    \n",
    "    # df1, df2 = filter_df_common_kmers(eth_df[['gene_id', 'kmer', 'junction_coordinate']], ohsu_df[['gene_id', 'kmer', 'junction_coordinate']])\n",
    "    out_table['coord_ETH'].append(df1['junction_coordinate'] if not df1.empty else 'None' )\n",
    "    out_table['coord_OHSU'].append(df2['junction_coordinate'] if not df2.empty else 'None')\n",
    "    ser1 = set(table_processing.separate_ETH_3exons(df1['junction_coordinate'])) # LP\n",
    "    ser2 = set(df2['junction_coordinate'])\n",
    "    out_table['size_ohsu_coor'].append(len(ser2))\n",
    "    out_table['size_eth_coor'].append(len(ser1))\n",
    "    out_table['size_ohsu\\eth_coor'].append(len(df_ohsu_filter_coor:=ser2.difference(ser1)))\n",
    "    out_table['size_eth\\ohsu_coor'].append(len(df_eth_filter_coor:=ser1.difference(ser2)))\n",
    "    out_table['size_intersection_coor'].append(len(df_inter_filter_coor:=ser2 & ser1))\n",
    "    out_table['eth_coor\\inter_coor'].append(df_eth_witout_inter_coor:=list(df_eth_filter_coor.difference(df_inter_filter_coor)))\n",
    "    out_table['ohsu_coor\\inter_coor'].append(df_ohsu_witout_inter_coor:=list(df_ohsu_filter_coor.difference(df_inter_filter_coor)))\n",
    "    out_table['eth_coor\\ohsu_coor'].append(df_eth_filter_coor)\n",
    "    out_table['ohsu_coor\\eth_coor'].append(df_ohsu_filter_coor)\n",
    "    out_table['inter_coor'].append(list(df_inter_filter_coor))\n",
    "    \n",
    "    sample.append(exp_col)\n",
    "    fs_ohsu.append(n_ohsu := len(ohsu_kmers))\n",
    "    kmer_ohsu.append(list(ohsu_kmers))\n",
    "    fs_eth.append(n_eth:=len(eth_kmers))\n",
    "    kmer_eth.append(list(eth_kmers))\n",
    "    eth_without_ohsu.append(len(eth_without_ohsu_kmers:=eth_kmers.difference(ohsu_kmers)))\n",
    "    ohsu_withou_eth.append(len(oshu_without_ohsu_kmers:=ohsu_kmers.difference(eth_kmers)))\n",
    "    inter.append(intersection:=len(intersection_kmers:= (eth_kmers & ohsu_kmers)))\n",
    "    kmer_inter.append(list(intersection_kmers))\n",
    "out_df_original=pd.DataFrame(data=out_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d8d0013c",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_non_filtering=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_OVARIAN,NAME_TABLES,NAME_NON_FILTERING_OVARIAN])\n",
    "out_df_original.to_csv(path_non_filtering, header = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "98a0fe20",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/202301_myurchikova_MT_rerun_LP/DATA/OVARIAN/TABLES/out_df_non_filtering_OVARIAN.csv\n"
     ]
    }
   ],
   "source": [
    "print(path_non_filtering)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "807be0d0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
