{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2ee2ef11",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import timeit\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b578e635",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "11ed020d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def get_junction_coordinates(df, coordinates_col, sep=':', shift_to_1based=True):\n",
    "#     df['strand'] = None\n",
    "#     df['junction_coordinate'] = None\n",
    "\n",
    "#     for idx, row in df.iterrows():\n",
    "#         kmer_coordinates = [int(x) for x in row[coordinates_col].split(sep) if x !='None']\n",
    "\n",
    "#         if kmer_coordinates[1] < kmer_coordinates[2]: # order strand +\n",
    "\n",
    "#             df.loc[idx, 'strand'] = '+'\n",
    "#             if len(kmer_coordinates) == 4:  # 2 exons\n",
    "#                 df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in kmer_coordinates[1:3]])\n",
    "#             elif len(kmer_coordinates) == 6:\n",
    "#                 df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in kmer_coordinates[1:5]])\n",
    "#         else: # order strand +\n",
    "#             df.loc[idx, 'strand'] = '-'\n",
    "#             if len(kmer_coordinates) == 4:  # 2 exons\n",
    "#                 df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in [kmer_coordinates[3],\n",
    "#                                                                                 kmer_coordinates[0]]])\n",
    "#             elif len(kmer_coordinates) == 6:\n",
    "#                 df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in [kmer_coordinates[3],\n",
    "#                                                                                 kmer_coordinates[0],\n",
    "#                                                                                 kmer_coordinates[2],\n",
    "#                                                                                 kmer_coordinates[5]\n",
    "#                                                                                ]])\n",
    "#     return df\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4504f873",
   "metadata": {},
   "outputs": [],
   "source": [
    "def explode_immunopepper_coord2(mx, coord_col = 'coord', sep=':'):\n",
    "    \n",
    "    coord_mx = mx[coord_col].str.split(sep, expand=True) #7 min\n",
    "    \n",
    "    dummy_fill = -9999\n",
    "    coord_mx = coord_mx.astype(float)\n",
    "    coord_mx.fillna(dummy_fill, inplace=True)\n",
    "    coord_mx = coord_mx.astype(int)\n",
    "\n",
    "    coord_mx['strand'] = None\n",
    "    coord_mx.loc[coord_mx[1] < coord_mx[2], 'strand'] = '+'\n",
    "    coord_mx.loc[coord_mx[1] > coord_mx[2], 'strand'] = '-'\n",
    "\n",
    "    coord_mx['junction_coordinate1'] = None\n",
    "    coord_mx['junction_coordinate2'] = None\n",
    "\n",
    "    if 4 not in coord_mx.columns:\n",
    "        coord_mx[4] = dummy_fill\n",
    "        coord_mx[5] = dummy_fill\n",
    "    \n",
    "    coord_mx = coord_mx.astype(str) # 7 min\n",
    "    \n",
    "    coord_mx['+first'] = coord_mx[1] + ':' + coord_mx[2]\n",
    "    coord_mx['+secon'] = coord_mx[3] + ':' + coord_mx[4]\n",
    "    coord_mx['-first'] = coord_mx[3] + ':' + coord_mx[0]\n",
    "    coord_mx['-secon'] = coord_mx[5] + ':' + coord_mx[2]\n",
    "    \n",
    "        \n",
    "    coord_mx[4] = ['None' if str(dummy_fill) in jx2 else jx2 for jx2 in coord_mx[4]]\n",
    "    coord_mx[5] = ['None' if str(dummy_fill) in jx2 else jx2 for jx2 in coord_mx[5]]\n",
    "\n",
    "    coord_mx.loc[(coord_mx['strand'] == '+'), 'junction_coordinate1'] = coord_mx['+first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '-'), 'junction_coordinate1'] = coord_mx['-first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '+') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['+secon']\n",
    "    coord_mx.loc[(coord_mx['strand'] == '-') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['-secon']\n",
    "    \n",
    "    return coord_mx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5dd79411",
   "metadata": {},
   "outputs": [],
   "source": [
    "def explode_immunopepper_coord(mx, coord_col = 'coord', sep=':'):\n",
    "    coord_mx = mx[coord_col].str.split(sep, expand=True) #7 min\n",
    "\n",
    "    coord_mx[1] = coord_mx[1].astype('int')\n",
    "    coord_mx[2] = coord_mx[2].astype('int')\n",
    "\n",
    "    coord_mx['strand'] = None\n",
    "    coord_mx.loc[coord_mx[1] < coord_mx[2], 'strand'] = '+'\n",
    "    coord_mx.loc[coord_mx[1] > coord_mx[2], 'strand'] = '-'\n",
    "\n",
    "    coord_mx['junction_coordinate1'] = None\n",
    "    coord_mx['junction_coordinate2'] = None\n",
    "\n",
    "    \n",
    "    coord_mx = coord_mx.astype(str) # 7 min\n",
    "    \n",
    "    coord_mx['+first'] = coord_mx[1] + ':' + coord_mx[2]\n",
    "    coord_mx['+secon'] = coord_mx[3] + ':' + coord_mx[4]\n",
    "    coord_mx['-first'] = coord_mx[3] + ':' + coord_mx[0]\n",
    "    coord_mx['-secon'] = coord_mx[5] + ':' + coord_mx[2]\n",
    "    \n",
    "\n",
    "\n",
    "    coord_mx.loc[(coord_mx['strand'] == '+'), 'junction_coordinate1'] = coord_mx['+first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '-'), 'junction_coordinate1'] = coord_mx['-first'] \n",
    "    coord_mx.loc[(coord_mx['strand'] == '+') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['+secon']\n",
    "    coord_mx.loc[(coord_mx['strand'] == '-') & (coord_mx[4] != 'None') , 'junction_coordinate2'] = coord_mx['-secon']\n",
    "    \n",
    "    return coord_mx"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ffee8a80",
   "metadata": {},
   "source": [
    "# Main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "638e8de6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102\n"
     ]
    }
   ],
   "source": [
    "# Paths INPUTS\n",
    "\n",
    "sample_type = 'brca'\n",
    "\n",
    "if sample_type == 'brca':\n",
    "    sample_dir = 'TCGA_Breast_1102'\n",
    "elif sample_type == 'ov':\n",
    "    sample_dir = 'TCGA_Ovarian_374'\n",
    "\n",
    "basedir = f'/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/{sample_dir}'\n",
    "print(basedir)\n",
    "intermediate_file = os.path.join(basedir, 'filtering_intermediate/complete_cancer_candidates_order_r_complete.tsv.gz')\n",
    "\n",
    "longlist = os.path.join(basedir, 'filtering_samples/filters_19May_order_5ge_wAnnot_GPstar/All_samples_longlist.tsv.gz')\n",
    "\n",
    "save_ = True\n",
    "dump_ = True\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "62cd0f0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Paths OUTPUTS\n",
    "csv_res = os.path.join(basedir, 'filtering_samples/filters_19May_order_5ge_wAnnot_GPstar/peptides_25012024.tsv.gz') \n",
    "pickle_res = os.path.join(basedir, 'filtering_samples/filters_19May_order_5ge_wAnnot_GPstar/peptides25012024.pickle')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "3bc215e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load\n",
    "inter = pd.read_csv(intermediate_file, sep = '\\t')\n",
    "long_list = pd.read_csv(longlist, sep = '\\t', header = None)\n",
    "long_list.columns = ['kmer', 'coord']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "df1a714b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>3290.0</td>\n",
       "      <td>1969.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>572.0</td>\n",
       "      <td>545.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISSQSRVEK</td>\n",
       "      <td>278.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>92379851:92379859:92493866:92493885:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSGDEEKYP</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92600493:92600508:92611313:92611325:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>641</td>\n",
       "      <td>520</td>\n",
       "      <td>348</td>\n",
       "      <td>170</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLKMKMFQI</td>\n",
       "      <td>146.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379850:92379859:92496416:92496434:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  WYITRSGIA                 3290.0                  1969.0   \n",
       "1  WYITRSGIA                  572.0                   545.0   \n",
       "2  ISSQSRVEK                  278.0                   261.0   \n",
       "3  RSGDEEKYP                 7350.0                  4622.0   \n",
       "4  HLKMKMFQI                  146.0                   136.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   861.0                   394.0                    89.0   \n",
       "1                   243.0                   130.0                    27.0   \n",
       "2                    82.0                    29.0                     7.0   \n",
       "3                  2653.0                  1745.0                  1126.0   \n",
       "4                    42.0                    20.0                     3.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92347505:92347506:92349915:92349941:None:None   \n",
       "1                      2.0  92347505:92347506:92349915:92349941:None:None   \n",
       "2                      0.0  92379851:92379859:92493866:92493885:None:None   \n",
       "3                    734.0  92600493:92600508:92611313:92611325:None:None   \n",
       "4                      1.0  92379850:92379859:92496416:92496434:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0              False               False  ...                       138   \n",
       "1              False               False  ...                       138   \n",
       "2              False               False  ...                        10   \n",
       "3               True                True  ...                       641   \n",
       "4              False               False  ...                        16   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                        82                        26   \n",
       "1                        82                        26   \n",
       "2                         2                         0   \n",
       "3                       520                       348   \n",
       "4                         2                         0   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGAC8A12P01A11RA11507all  \\\n",
       "0                          5                   0.000000   \n",
       "1                          5                   0.000000   \n",
       "2                          0                   0.000000   \n",
       "3                        170                   2.922641   \n",
       "4                          0                   2.922641   \n",
       "\n",
       "   TCGAAOA0JM01A21RA05607all  TCGABHA18V01A11RA12D07all  \\\n",
       "0                   0.000000                   0.000000   \n",
       "1                   0.000000                   0.000000   \n",
       "2                   0.000000                   2.474321   \n",
       "3                   2.102386                   1.237161   \n",
       "4                   0.000000                   0.000000   \n",
       "\n",
       "   TCGAA2A0D201A21RA03407all  TCGAA2A0SX01A12RA08407all  isAnnotated  \n",
       "0                   4.102634                        0.0          NaN  \n",
       "1                   4.102634                        0.0          NaN  \n",
       "2                   0.000000                        0.0          NaN  \n",
       "3                   0.000000                        0.0          1.0  \n",
       "4                   0.000000                        0.0          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAASFSMDG</td>\n",
       "      <td>62492703:62492724:62477828:62477834:None:None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AACFLVHPS</td>\n",
       "      <td>62270267:62270271:62270854:62270877:None:None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AADDISAQL</td>\n",
       "      <td>224929828:224929835:225085709:225085729:None:None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAETLQDDF</td>\n",
       "      <td>129536369:129536377:129540389:129540408:None:None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAGDSPIPG</td>\n",
       "      <td>177436312:177436319:177434461:177434481:None:None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer                                              coord\n",
       "0  AAASFSMDG      62492703:62492724:62477828:62477834:None:None\n",
       "1  AACFLVHPS      62270267:62270271:62270854:62270877:None:None\n",
       "2  AADDISAQL  224929828:224929835:225085709:225085729:None:None\n",
       "3  AAETLQDDF  129536369:129536377:129540389:129540408:None:None\n",
       "4  AAGDSPIPG  177436312:177436319:177434461:177434481:None:None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(5579, 2)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "display(inter.head()) \n",
    "inter.shape\n",
    "\n",
    "display(long_list.head()) \n",
    "long_list.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e56cd4ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6080, 24)\n",
      "662\n"
     ]
    }
   ],
   "source": [
    "# Merge on metadata \n",
    "batch_interest = inter.merge(long_list, on = ['kmer', 'coord'], how = 'right')\n",
    "\n",
    "print(batch_interest.shape)\n",
    "\n",
    "print(len(batch_interest['batch'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "be01b431",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add Coordinate\n",
    "coord_mx = explode_immunopepper_coord(batch_interest)\n",
    "\n",
    "batch_interest = pd.concat([batch_interest, \n",
    "                            coord_mx[['strand', 'junction_coordinate1', 'junction_coordinate2']]], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e23bbee8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6080, 27)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "batch_interest.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "5fba42ee",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "      <th>isAnnotated</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate1</th>\n",
       "      <th>junction_coordinate2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAASFSMDG</td>\n",
       "      <td>9.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>62492703:62492724:62477828:62477834:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1.461321</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>62477834:62492703</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AACFLVHPS</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>62270267:62270271:62270854:62270877:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>10.229245</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>62270271:62270854</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AADDISAQL</td>\n",
       "      <td>12.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>224929828:224929835:225085709:225085729:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.367545</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>224929835:225085709</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAETLQDDF</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>129536369:129536377:129540389:129540408:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>129536377:129540389</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAETLQDDF</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>129536369:129536377:129540389:129540408:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>129536377:129540389</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  AAASFSMDG                    9.0                     6.0   \n",
       "1  AACFLVHPS                    4.0                     4.0   \n",
       "2  AADDISAQL                   12.0                    11.0   \n",
       "3  AAETLQDDF                    9.0                     9.0   \n",
       "4  AAETLQDDF                    9.0                     9.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                     4.0                     3.0                     1.0   \n",
       "1                     2.0                     2.0                     0.0   \n",
       "2                     3.0                     0.0                     0.0   \n",
       "3                     5.0                     4.0                     0.0   \n",
       "4                     5.0                     4.0                     0.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                              coord  \\\n",
       "0                      0.0      62492703:62492724:62477828:62477834:None:None   \n",
       "1                      0.0      62270267:62270271:62270854:62270877:None:None   \n",
       "2                      0.0  224929828:224929835:225085709:225085729:None:None   \n",
       "3                      0.0  129536369:129536377:129540389:129540408:None:None   \n",
       "4                      0.0  129536369:129536377:129540389:129540408:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=10.0  \\\n",
       "0              False               False  ...                          0   \n",
       "1              False                True  ...                          6   \n",
       "2              False               False  ...                          0   \n",
       "3              False                True  ...                          0   \n",
       "4              False               False  ...                          0   \n",
       "\n",
       "   TCGAC8A12P01A11RA11507all  TCGAAOA0JM01A21RA05607all  \\\n",
       "0                   1.461321                        0.0   \n",
       "1                  10.229245                        0.0   \n",
       "2                   0.000000                        0.0   \n",
       "3                   2.922641                        0.0   \n",
       "4                   2.922641                        0.0   \n",
       "\n",
       "   TCGABHA18V01A11RA12D07all  TCGAA2A0D201A21RA03407all  \\\n",
       "0                        0.0                   0.000000   \n",
       "1                        0.0                   0.000000   \n",
       "2                        0.0                   1.367545   \n",
       "3                        0.0                   0.000000   \n",
       "4                        0.0                   0.000000   \n",
       "\n",
       "   TCGAA2A0SX01A12RA08407all  isAnnotated  strand  junction_coordinate1  \\\n",
       "0                        0.0          NaN       -     62477834:62492703   \n",
       "1                        0.0          NaN       +     62270271:62270854   \n",
       "2                        0.0          NaN       +   224929835:225085709   \n",
       "3                        0.0          NaN       +   129536377:129540389   \n",
       "4                        0.0          NaN       +   129536377:129540389   \n",
       "\n",
       "   junction_coordinate2  \n",
       "0                  None  \n",
       "1                  None  \n",
       "2                  None  \n",
       "3                  None  \n",
       "4                  None  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "batch_interest.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bf9e9f45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "....0 gene number 1841 total peptides 22 add 22\n",
      "....100 gene number 34142 total peptides 5198 add 13\n",
      "....200 gene number 53915 total peptides 7191 add 24\n",
      "....300 gene number 28987 total peptides 8508 add 4\n",
      "....400 gene number 22200 total peptides 10849 add 21\n",
      "....500 gene number 55165 total peptides 12379 add 20\n",
      "....600 gene number 1167 total peptides 13969 add 5\n",
      "69.7245497489348\n",
      "14529\n"
     ]
    }
   ],
   "source": [
    "# Merge peptides \n",
    "debut_time = timeit.default_timer()\n",
    "tot_shape = 0\n",
    "extracted_metadata = []\n",
    "for idx, gene in enumerate(batch_interest['batch'].unique()):\n",
    "    # Load metadata\n",
    "    base_meta = os.path.join(basedir, 'cohort_mutNone', f'tmp_out_ref_batch_{gene}', 'ref_sample_peptides_meta.gz')\n",
    "    meta = pd.read_csv(base_meta, sep = '\\t') # Note they are peptides 'NA' confused with Nan \n",
    "    meta = meta[~meta['peptide'].isna()]\n",
    "    meta = meta.set_index(np.arange(len(meta)))\n",
    "    meta_coord = explode_immunopepper_coord2(meta, coord_col='modifiedExonsCoord', sep=';')\n",
    "    meta = pd.concat([meta, meta_coord[['strand', 'junction_coordinate1', 'junction_coordinate2']]], axis = 1)\n",
    "    \n",
    "    # Merge with coordinates interest\n",
    "    batch_col = ['coord', 'kmer', 'junction_coordinate1', 'junction_coordinate2']\n",
    "#     selected_pep = batch_interest.loc[:, batch_col].merge(meta, \n",
    "#                                         on = ['junction_coordinate1', 'junction_coordinate2'], \n",
    "#                                         how = 'inner')\n",
    "\n",
    "    selected_pep0 = batch_interest.loc[:, batch_col].merge(meta, \n",
    "                                                         on = ['junction_coordinate1', \n",
    "                                                               'junction_coordinate2'], \n",
    "                                                         how = 'inner')\n",
    "    selected_pep1 = batch_interest.loc[:, batch_col].merge(meta, #correct issue with second coord only   \n",
    "                                                          right_on = [ 'junction_coordinate2'], \n",
    "                                                          left_on = ['junction_coordinate1'], \n",
    "                                                         how = 'inner')\n",
    "    selected_pep2 = batch_interest.loc[:, batch_col].merge(meta.loc[meta['kmerType'] == '3-exons'], #correct issue with first coordinate from 3-exons\n",
    "                                                          on = ['junction_coordinate1'], \n",
    "                                                         how = 'inner')\n",
    "    selected_pep = pd.concat([selected_pep2, selected_pep1, selected_pep0]).\\\n",
    "                drop_duplicates().reset_index()\n",
    "    # Restrict merge on kmer presence\n",
    "    selected_pep['keep'] = [selected_pep.loc[ix, 'kmer'] in pep \n",
    "                            for ix, pep in enumerate(selected_pep['peptide'])]\n",
    "    selected_pep = selected_pep.loc[selected_pep['keep'] == True]\n",
    "    \n",
    "    # Prepare save\n",
    "    extracted_metadata.append(selected_pep)\n",
    "    tot_shape+=selected_pep.shape[0]\n",
    "    \n",
    "    if idx % 100 == 0:\n",
    "        print(f'....{idx} gene number {gene} total peptides {tot_shape} add {selected_pep.shape[0]}')\n",
    "        print\n",
    "print(timeit.default_timer() - debut_time)\n",
    "print(tot_shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1217ad6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "662"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(extracted_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "f7f83f5d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saving to /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_19May_order_5ge_wAnnot_GPstar/peptides25012024.pickle\n"
     ]
    }
   ],
   "source": [
    "if dump_: \n",
    "    print(f'Saving to {pickle_res}')\n",
    "    file_ = open(pickle_res, 'wb')\n",
    "\n",
    "    # dump information to that file\n",
    "    pickle.dump(extracted_metadata, file_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a6fb3ba5",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = pd.concat(extracted_metadata, axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d7e2e144",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saving to /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_19May_order_5ge_wAnnot_GPstar/peptides_25012024.tsv.gz\n"
     ]
    }
   ],
   "source": [
    "if save_:\n",
    "    print(f'Saving to {csv_res}')\n",
    "    res.to_csv(csv_res, \n",
    "              sep = '\\t', \n",
    "              index= None, \n",
    "              compression = 'gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "0588ff65",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18378    True\n",
       "Name: peptide, dtype: bool"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.loc[meta['id'] == 'ENSG00000167996.16:1775_1290:0:61965419:2-exons', 'peptide'].isna()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23f85983",
   "metadata": {},
   "source": [
    "# TESTS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d7624e7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
