{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:01.739750Z",
     "start_time": "2022-06-24T08:20:59.397721Z"
    }
   },
   "outputs": [],
   "source": [
    "import os \n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "import numpy as np \n",
    "import glob\n",
    "import gzip\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Previous steps\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 1: "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Get a list of kmers passing the filtering.\n",
    "Can also be a kmer \"longlist\" which is a pool of kmers from multiple filter experiments. The peptide to experiment correspondance will then be stored in a table.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given a list of kmers, grep the corresponding metadata information from the ImmunoPepper metadata file (very big, no kmer information) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-25T15:16:53.158318Z",
     "start_time": "2022-02-25T15:16:53.155867Z"
    }
   },
   "source": [
    "See /GitHub/projects2020_ohsu/eth/peptide_search_format/p20220115_format_peptides_longlist_pr\n",
    "ior.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3. \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-10T16:33:59.242716Z",
     "start_time": "2022-06-10T16:33:59.161360Z"
    }
   },
   "source": [
    "This notebook performs the following tasks * Given a kmer_metadata file, generate a fasta file with metadata relevant for the tryptic digestion 2. A table containing the peptide IDs per sample \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Define Samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:01.749461Z",
     "start_time": "2022-06-24T08:21:01.744414Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_19May_order_5ge_wAnnot\n"
     ]
    }
   ],
   "source": [
    "samples = ['TCGA-BH-A18V-01A-11R-A12D-07', \n",
    "        'TCGA-C8-A12P-01A-11R-A115-07', \n",
    "        'TCGA-AO-A0JM-01A-21R-A056-07', \n",
    "        'TCGA-A2-A0SX-01A-12R-A084-07',\n",
    "        'TCGA-A2-A0D2-01A-21R-A034-07']\n",
    "\n",
    "basefolder = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102'\n",
    "filter_folder = 'filtering_samples/filters_19May_order_5ge_wAnnot'\n",
    "\n",
    "\n",
    "experiment_folder = os.path.join(basefolder, filter_folder)\n",
    "\n",
    "print(experiment_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preprocess the annotation \n",
    "### Input: Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:01.805779Z",
     "start_time": "2022-06-24T08:21:01.800912Z"
    }
   },
   "outputs": [],
   "source": [
    "ann_path = '/cluster/work/grlab/projects/projects2020_OHSU/annotation/gencode.v32.annotation.gtf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:10.602236Z",
     "start_time": "2022-06-24T08:21:01.806792Z"
    }
   },
   "outputs": [],
   "source": [
    "from immunopepper.preprocess import attribute_item_to_dict\n",
    "from immunopepper.preprocess import leq_strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:37.413482Z",
     "start_time": "2022-06-24T08:21:10.603570Z"
    }
   },
   "outputs": [],
   "source": [
    "# Partial copy From Immunopepper preprocess.py \n",
    "transcript_to_gene_dict = {}    # transcript -> gene id\n",
    "\n",
    "\n",
    "gene_to_transcript_dict = {}    # gene_id -> list of transcripts\n",
    "gene_cds_begin_dict = {}        # gene -> list of first CDS exons\n",
    "\n",
    "transcript_to_cds_dict = {}     # transcript -> list of CDS exons\n",
    "transcript_cds_begin_dict = {}  # transcript -> first exon of the CDS\n",
    "transcript_to_strand = {}\n",
    "\n",
    "file_type = ann_path.split('.')[-1]\n",
    "chromesome_set = set()\n",
    "# collect information from annotation file\n",
    "for line in open(ann_path, 'r'):\n",
    "    if line[0] == '#':\n",
    "        continue\n",
    "    item = line.strip().split('\\t')\n",
    "    chromesome_set.add(item[0])\n",
    "    feature_type = item[2]\n",
    "    attribute_item = item[-1]\n",
    "    attribute_dict = attribute_item_to_dict(attribute_item, file_type, feature_type)\n",
    "    # store relationship between gene ID and its transcript IDs\n",
    "    if feature_type in ['transcript', 'mRNA']:\n",
    "        gene_id = attribute_dict['gene_id']\n",
    "        transcript_id = attribute_dict['transcript_id']\n",
    "        if attribute_dict['gene_type'] != 'protein_coding' or attribute_dict['transcript_type']  != 'protein_coding':\n",
    "            continue\n",
    "        assert (transcript_id not in transcript_to_gene_dict)\n",
    "        transcript_to_gene_dict[transcript_id] = gene_id\n",
    "        if gene_id in gene_to_transcript_dict and transcript_id not in gene_to_transcript_dict[gene_id]:\n",
    "            gene_to_transcript_dict[gene_id].append(transcript_id)\n",
    "        else:\n",
    "            gene_to_transcript_dict[gene_id] = [transcript_id]\n",
    "        # Todo python is 0-based while gene annotation file(.gtf, .vcf, .maf) is one based\n",
    "    elif feature_type == \"CDS\":\n",
    "        parent_ts = attribute_dict['transcript_id']\n",
    "        strand_mode = item[6]\n",
    "        cds_left = int(item[3])-1\n",
    "        cds_right = int(item[4])\n",
    "        frameshift = int(item[7])\n",
    "        transcript_to_strand[parent_ts] = strand_mode\n",
    "        if parent_ts in transcript_to_cds_dict:\n",
    "            transcript_to_cds_dict[parent_ts].append((cds_left, cds_right, frameshift))\n",
    "        else:\n",
    "            transcript_to_cds_dict[parent_ts] = [(cds_left, cds_right, frameshift)]\n",
    "        if strand_mode == \"+\" :\n",
    "            cds_start, cds_stop = cds_left, cds_right\n",
    "        else:\n",
    "            cds_start, cds_stop = cds_right, cds_left\n",
    "\n",
    "        # we only consider the start of the whole CoDing Segment\n",
    "        if parent_ts not in transcript_cds_begin_dict or \\\n",
    "           leq_strand(cds_start, transcript_cds_begin_dict[parent_ts][0], strand_mode):\n",
    "            transcript_cds_begin_dict[parent_ts] = (cds_start, cds_stop, item)\n",
    "\n",
    "# collect first CDS exons for all transcripts of a gene\n",
    "for ts_key in transcript_to_gene_dict:\n",
    "    target_gene = transcript_to_gene_dict[ts_key]\n",
    "    if target_gene not in gene_cds_begin_dict:\n",
    "        gene_cds_begin_dict[target_gene] = []\n",
    "    if ts_key in transcript_cds_begin_dict:\n",
    "        gene_cds_begin_dict[target_gene].append(transcript_cds_begin_dict[ts_key])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:37.653285Z",
     "start_time": "2022-06-24T08:21:37.414890Z"
    }
   },
   "outputs": [],
   "source": [
    "# Custom collection of CDS \n",
    "transcript_cds_begin_dict_bis = {}\n",
    "transcript_cds_end_dict_bis = {}\n",
    "\n",
    "gene_cds_begin_dict_bis = defaultdict(list)\n",
    "gene_cds_end_dict_bis = defaultdict(list)\n",
    "\n",
    "# will be in reading order \n",
    "for ts_key in transcript_to_cds_dict:\n",
    "    if transcript_to_strand[ts_key] == '+': # '+'\n",
    "        transcript_cds_begin_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][0][0],\n",
    "                                                 transcript_to_cds_dict[ts_key][0][1], '+')\n",
    "        transcript_cds_end_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][-1][0],\n",
    "                                                 transcript_to_cds_dict[ts_key][-1][1], '+')\n",
    "        \n",
    "\n",
    "    else: \n",
    "        transcript_cds_begin_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][0][1],\n",
    "                                                 transcript_to_cds_dict[ts_key][0][0], '-')\n",
    "        transcript_cds_end_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][-1][1],\n",
    "                                                 transcript_to_cds_dict[ts_key][-1][0], '-')\n",
    "    \n",
    "    assert(transcript_cds_begin_dict_bis[ts_key][0] == transcript_cds_begin_dict[ts_key][0])\n",
    "    assert(transcript_cds_begin_dict_bis[ts_key][1] == transcript_cds_begin_dict[ts_key][1])\n",
    "\n",
    "# collect first, last CDS exons for all transcripts of a gene\n",
    "for ts_key in transcript_to_gene_dict:\n",
    "    target_gene = transcript_to_gene_dict[ts_key]\n",
    "    gene_cds_begin_dict_bis[target_gene].append(transcript_cds_begin_dict_bis[ts_key])\n",
    "    gene_cds_end_dict_bis[target_gene].append(transcript_cds_end_dict_bis[ts_key])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 4. Format the peptide raw file  \n",
    "### Input:  file meta with kmers and bi-exons matching the longlist of the samples "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Restrict the fasta to a whitelist\n",
    "whitelist_kmer = False  \n",
    "if whitelist_kmer:\n",
    "    with open(os.path.join(experiment_folder, 'G_TCGA_Allsamples_intersect.pickle'), 'rb') as handle:\n",
    "        whitelist_kmer = pickle.load(handle)\n",
    "    print(len(whitelist_kmer))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample = samples[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:50.188524Z",
     "start_time": "2022-06-24T08:21:50.185913Z"
    }
   },
   "outputs": [],
   "source": [
    "# Meta Input File\n",
    "file_meta = os.path.join(experiment_folder, f'G_{sample}_grep_metadata_raw.tsv.gz')\n",
    "\n",
    "# Experiment Input File \n",
    "file_save_experiement= None\n",
    "\n",
    "# Fasta output File\n",
    "if not whitelist_kmer:\n",
    "    file_save = os.path.join(experiment_folder, f'G_{sample}_pool_kmer.fa')\n",
    "else:\n",
    "    file_save = os.path.join(experiment_folder, f'G_{sample}_pool_kmer.fa')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_peptide_fields(pep):\n",
    "    cols_correct = ['kmer', 'kmer_coord', 'peptide','id','readFrame','readFrameAnnotated','geneName','geneChr','geneStrand',\n",
    "    'mutationMode','hasStopCodon','isInJunctionList',\n",
    "    'isIsolated','variantComb','variantSegExpr','modifiedExonsCoord',\n",
    "    'originalExonsCoord',\n",
    "    'vertexIdx','kmerType', 'dummy1', 'dummy2', 'dummy3']\n",
    "    res = {}\n",
    "    for idx, p in enumerate(pep):\n",
    "        res[cols_correct[idx]] = p\n",
    "    return res \n",
    "\n",
    "\n",
    "def extract_end_starts(pep_orig_coord, strand):\n",
    "    ''' Get peptide end and start coordinates'''\n",
    "    if strand == '+': # Do - strand \n",
    "        pep_start = int(pep_orig_coord[0])\n",
    "        pep_end = int(pep_orig_coord[-1])\n",
    "    else: \n",
    "        pep_start = int(pep_orig_coord[1])\n",
    "        pep_end = int(pep_orig_coord[-2])\n",
    "    return pep_start, pep_end\n",
    "\n",
    "\n",
    "def get_include_flag(start_cds, end_cds, pep_start, pep_end, has_stop_codon ):\n",
    "    '''Use end and start coordinates for 3' 5' include flag'''\n",
    "    if pep_start in start_cds: # We will always miss things that are new in the graph \n",
    "        pep_5include = 1\n",
    "    else: \n",
    "        pep_5include = 0 \n",
    "    if (pep_end in end_cds) or (has_stop_codon) == '1':\n",
    "        pep_3include = 1\n",
    "    else: \n",
    "        pep_3include = 0 \n",
    "    return pep_5include, pep_3include\n",
    "\n",
    "\n",
    "def get_nt_len_with_aa_shift(pep_modi_coord):\n",
    "    '''Get nt length of each exon involved -> jx_list, shift_list'''\n",
    "    tot_len = 0 \n",
    "    shift = 0 \n",
    "    jx_list = []\n",
    "    jx_list_ori = []\n",
    "    shift_list = []\n",
    "    for pair in np.arange(0, len(pep_modi_coord), 2):\n",
    "        cds = int(pep_modi_coord[pair + 1]) - int(pep_modi_coord[pair])  # 0 based, open right \n",
    "        jx_list_ori.append(cds)\n",
    "        cds += shift \n",
    "        shift = cds % 3\n",
    "        jx_list.append(cds - shift)\n",
    "        shift_list.append(shift)\n",
    "        \n",
    "    return jx_list, shift_list, jx_list_ori\n",
    "\n",
    "\n",
    "def get_aaPos_betweenFlag(shift_list, jx_list):\n",
    "    '''Get aa position of the junction\n",
    "    the junction coordinate jx_pos is the 0-based position in the peptide \n",
    "    of the amino acid that either overlaps the junction (if the junction is \n",
    "    in the middle of a codon), or is immediately before it if the junction \n",
    "    occurs between codons'''\n",
    "    if shift_list[0]: # junction is inside an amino acid\n",
    "        aa_junction_pos0 = int((jx_list[0] / 3)) # because 0 based\n",
    "        between_codons0 = 0 \n",
    "    else: # junction is between amino acids \n",
    "        aa_junction_pos0 = int((jx_list[0] / 3) - 1)  # because 0 based\n",
    "        between_codons0 = 1\n",
    "        \n",
    "    if len(shift_list) > 2: #third exon \n",
    "        if shift_list[1]: # junction is inside an amino acid\n",
    "            aa_junction_pos1 = int((jx_list[1] / 3)) # because 0 based\n",
    "            between_codons1 = 0 \n",
    "        else: # junction is between amino acids \n",
    "            aa_junction_pos1 = int((jx_list[1] / 3) - 1)  # because 0 based\n",
    "            between_codons1 = 1 \n",
    "        aa_junction_pos1_from_start = aa_junction_pos1 + aa_junction_pos0 + 1 \n",
    "    else:\n",
    "        aa_junction_pos1 = None\n",
    "        between_codons1 = None\n",
    "        aa_junction_pos1_from_start = None\n",
    "    \n",
    "    return aa_junction_pos0, between_codons0, aa_junction_pos1, between_codons1, \\\n",
    "           aa_junction_pos1_from_start\n",
    "\n",
    "\n",
    "def get_genomic_coordinates(pep_modi_coord, strand):\n",
    "    '''We have in + case: exon1_start, exon 1_stop, exon2_start, exon2_stop, exon3_start, exon3_stop\n",
    "     In the - case: exon1_stop, exon 1_start, exon 2_stop, exon2_start, exon3_stop, exon3_start'''\n",
    "    genome_junction_pos1 = None\n",
    "    if strand == '+':\n",
    "        genome_junction_pos0 = '{}_{}'.format(pep_modi_coord[1], pep_modi_coord[2])\n",
    "        if len(pep_modi_coord) > 4:\n",
    "            genome_junction_pos1 = '{}_{}'.format(pep_modi_coord[3], pep_modi_coord[4])\n",
    "    else:\n",
    "        genome_junction_pos0 = '{}_{}'.format(pep_modi_coord[0], pep_modi_coord[3])\n",
    "        if len(pep_modi_coord) > 4:\n",
    "            genome_junction_pos1 = '{}_{}'.format(pep_modi_coord[2], pep_modi_coord[5])\n",
    "    return genome_junction_pos0, genome_junction_pos1\n",
    "\n",
    "\n",
    "def split_coord(pep_coord):\n",
    "    pep_coord = pep_coord.split(';')\n",
    "    pep_coord = [coord for coord in pep_coord if (coord != 'None') and (coord != 'nan')]\n",
    "    return pep_coord\n",
    "\n",
    "\n",
    "def write_peptide_to_experiment(filepointer, pep_idx=None, pep_seq=None,\\\n",
    "                                idx=None, header=False):\n",
    "    if header:\n",
    "        header_exp = 'peptide_id\\tpeptide_sequence\\texperiment_ids\\n'\n",
    "        filepointer.write(header_exp)\n",
    "    elif pep_idx is not None:\n",
    "        exp_line = '{}\\t{}\\t{}\\n'.format(pep_idx, \n",
    "                                         pep_seq,\n",
    "                                         idx)\n",
    "        filepointer.write(exp_line)\n",
    "\n",
    "            \n",
    "def write_fasta(write_, filepointer, pep_seq, pep_idx, aa_junction_pos, \n",
    "                aa_junction_pos1_from_start, between_codons, between_codons1,\n",
    "                pep_5include, pep_3include, pep_gene, \n",
    "                genome_junction_pos, genome_junction_pos1, \n",
    "                kmer, jx_pep1, jx_pep2, readFrameAnnotated, \\\n",
    "                kmer_coord, kmer_type, strand, do_write=True):\n",
    "        \n",
    "    if write_:\n",
    "        pep_handle1 = (f'>pepID-{pep_idx};jx_pos-{aa_junction_pos};between_codons-{between_codons}'\n",
    "                       f';includes_5\\'-{pep_5include};includes_3\\'-{pep_3include};gene-{pep_gene};'\n",
    "                       f'jx_coord-{genome_junction_pos};kmer-{kmer};readFrameAnnotated-{readFrameAnnotated};'\n",
    "                       f'kmer_coord-{kmer_coord};origin-{kmer_type};strand{strand}')\n",
    "\n",
    "        pep_handle2 = (f'>pepID-{pep_idx};jx_pos-{aa_junction_pos1_from_start};between_codons-{between_codons1}'\n",
    "                       f';includes_5\\'-{pep_5include};includes_3\\'-{pep_3include};gene-{pep_gene};'\n",
    "                       f'jx_coord-{genome_junction_pos1};kmer-{kmer};readFrameAnnotated-{readFrameAnnotated};'\n",
    "                       f'kmer_coord-{kmer_coord};origin-{kmer_type};strand{strand}')\n",
    "\n",
    "        if kmer in jx_pep1:\n",
    "            pep_idx+=1\n",
    "            sp.write(pep_handle1 + '\\n')\n",
    "            sp.write(pep_seq + '\\n')\n",
    "        elif kmer in jx_pep2:\n",
    "            pep_idx+=1\n",
    "            sp.write(pep_handle2 + '\\n')\n",
    "            sp.write(pep_seq + '\\n')\n",
    "        return pep_idx\n",
    "\n",
    "\n",
    "\n",
    "def cut_peptides(pep_seq, jx_list, between_codons, between_codons1, aa_junction_pos, \n",
    "                 aa_junction_pos1, aa_junction_pos1_from_start, \n",
    "                 print_ = False):\n",
    "    peptide_cut = []\n",
    "    aa_junction_pos_shift = aa_junction_pos + 1 \n",
    "    exon1, aa_jx1, exon2, aa_jx2, exon3 = '', '', '', '', ''\n",
    "\n",
    "    if len(jx_list) == 2:\n",
    "        if between_codons:\n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:]\n",
    "        else:\n",
    "            exon1 = pep_seq[:aa_junction_pos]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:]\n",
    "    elif len(jx_list) == 3:\n",
    "        aa_junction2_pos_shift =  aa_junction_pos1 + aa_junction_pos + 1\n",
    "        aa_junction2_pos_sshift = aa_junction_pos1 + aa_junction_pos + 2\n",
    "        assert(aa_junction_pos1_from_start == aa_junction2_pos_shift)\n",
    "        if between_codons and between_codons1: \n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:\n",
    "                          aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (not between_codons) and between_codons1: \n",
    "            exon1 = pep_seq[:aa_junction_pos ]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:\n",
    "                          aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (between_codons) and (not between_codons1): \n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift :\n",
    "                          aa_junction2_pos_shift]\n",
    "            aa_jx2 = pep_seq[aa_junction2_pos_shift:\n",
    "                         aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (not between_codons) and (not between_codons1): \n",
    "            exon1 = pep_seq[:aa_junction_pos ]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift :\n",
    "                          aa_junction2_pos_shift]\n",
    "            aa_jx2 = pep_seq[aa_junction2_pos_shift:\n",
    "                         aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "    if print_:\n",
    "        print(f'exon1:{exon1}, aa_containing_jx1:{aa_jx1}, exon2:{exon2}, aa_containing_jx2:{aa_jx2}, exon3:{exon3}')\n",
    "        print(f'junction positions jx1: {aa_junction_pos}, jx2:{aa_junction_pos1_from_start}')\n",
    "        print(f'is junction between a codon jx1: {between_codons}, jx2: {between_codons1}')\n",
    "        print('\\n')\n",
    "    return exon1 + aa_jx1 + exon2, exon2 + aa_jx2 + exon3 \n",
    "\n",
    "def print_stats(print_, kmer, pep_seq, strand, pep_orig_coord, pep_modi_coord, jx_list,\n",
    "               jx_list_ori, genome_junction_pos, genome_junction_pos1,\n",
    "               aa_junction_pos, aa_junction_pos1, between_codons, between_codons1):\n",
    "    \n",
    "    if print_:\n",
    "        p_ori_coord = ';'.join(pep_orig_coord)\n",
    "        p_modif_coord = ';'.join(pep_modi_coord)\n",
    "        print(f'INSTANCE: \\n kmer {kmer}/ sequence {pep_seq}/ strand {strand} / \\n original coordinates {p_ori_coord} / \\n modif coordinates {p_modif_coord} /  \\n junction list origin {jx_list_ori}/ junction list {jx_list} / \\n junction coordinates 1 {genome_junction_pos} / junction coordinates 2 {genome_junction_pos1}')\n",
    "        #print(aa_junction_pos, between_codons, aa_junction_pos1, between_codons1)\n",
    "        print('peptide length', len(pep_seq))\n",
    "\n",
    "\n",
    "def readlines_custom(file_):\n",
    "    with gzip.open(file_, 'rb') as fp: \n",
    "        lines = fp.readlines()\n",
    "        for j, line in enumerate(lines):\n",
    "            line = line.decode().replace('\\n', '').split('\\t')\n",
    "            if '\\\\t' in line[0]:\n",
    "                splitted_p = line[0].split('\\\\t') #correction temp\n",
    "                kmer, coord, peptide = splitted_p\n",
    "                line = [kmer, coord, peptide] + line[1:]\n",
    "            else:\n",
    "                line = [kmer, coord] + line #Use previous kmer instance\n",
    "            lines[j] = line\n",
    "    return lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T08:21:51.529894Z",
     "start_time": "2022-06-24T08:21:50.333790Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "written to:\n",
      "None\n",
      "/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_samples/filters_19May_order_5ge_wAnnot/G_TCGA-BH-A18V-01A-11R-A12D-07_pool_kmer.fa\n"
     ]
    }
   ],
   "source": [
    "subset_run_test = False\n",
    "write_ = True\n",
    "print_= False\n",
    "test = 0 \n",
    "\n",
    "### Main \n",
    "pep_idx = 0 \n",
    "peptide_junctions = set()\n",
    "\n",
    "if file_save_experiement is not None: \n",
    "    ep = open(file_save_experiement, 'w') # Experiment file to save\n",
    "    write_peptide_to_experiment(ep, header=True)\n",
    "    \n",
    "with open(file_save, 'w') as sp: # Fasta file to save\n",
    "    \n",
    "    ### Iterate over peptides \n",
    "    lines = readlines_custom(file_meta)\n",
    "    for line in lines:\n",
    "        res = extract_peptide_fields(line)\n",
    "        \n",
    "        if whitelist_kmer and res['kmer'] not in whitelist_kmer:\n",
    "            continue\n",
    "            \n",
    "\n",
    "        # some peptides do not contain a junction\n",
    "        if int(res['isIsolated']):\n",
    "            continue \n",
    "            \n",
    "        # Genomic coordinates : get include 3' or 5' flag \n",
    "        res['originalExonsCoord'] = split_coord(res['originalExonsCoord'])\n",
    "        start_cds = [ first_exon[0] for first_exon in gene_cds_begin_dict_bis[res['geneName']] ] \n",
    "        end_cds = [ last_exon[1] for last_exon in gene_cds_end_dict_bis[res['geneName']] ] \n",
    "        pep_start, pep_end = extract_end_starts(res['originalExonsCoord'], res['geneStrand'])\n",
    "        pep_5include, pep_3include = get_include_flag(start_cds, end_cds, \n",
    "                                                      pep_start, pep_end, \n",
    "                                                      res['hasStopCodon'] )\n",
    "\n",
    "        # Modified genomic coordinates: \n",
    "        res['modifiedExonsCoord'] = split_coord(res['modifiedExonsCoord'])\n",
    "        if len(res['modifiedExonsCoord']) <=2:\n",
    "            continue\n",
    "            \n",
    "            \n",
    "        jx_list, shift_list, jx_list_ori = get_nt_len_with_aa_shift(res['modifiedExonsCoord'])\n",
    "        # get \"junction position\" flag\n",
    "        aa_junction_pos, between_codons, \\\n",
    "        aa_junction_pos1, between_codons1, \\\n",
    "        aa_junction_pos1_from_start = get_aaPos_betweenFlag(shift_list, jx_list)\n",
    "        #get \"genomic coordinates\" flag \n",
    "        genome_junction_pos, \\\n",
    "        genome_junction_pos1 = get_genomic_coordinates(res['modifiedExonsCoord'], res['geneStrand'])\n",
    "\n",
    "\n",
    "#             print_stats(print_, res['kmer'], res['peptide'], res['geneStrand'], res['originalExonsCoord'], \n",
    "#                         res['modifiedExonsCoord'], jx_list, jx_list_ori, genome_junction_pos,\n",
    "#                         genome_junction_pos1, aa_junction_pos, aa_junction_pos1, \n",
    "#                         between_codons, between_codons1)\n",
    "        \n",
    "        # Get the 2 peptides sequences \n",
    "        jx_pep1, jx_pep2 = cut_peptides(res['peptide'], jx_list, between_codons,\n",
    "                                            between_codons1, aa_junction_pos, \n",
    "                                            aa_junction_pos1, \n",
    "                                            aa_junction_pos1_from_start, \n",
    "                                            print_=print_)\n",
    "\n",
    "        # skip duplicates (Issue 3-exons?)\n",
    "        if (genome_junction_pos in peptide_junctions) and \\\n",
    "           (genome_junction_pos1 in peptide_junctions):\n",
    "            continue\n",
    "        else:\n",
    "            peptide_junctions.add(genome_junction_pos)\n",
    "            peptide_junctions.add(genome_junction_pos1)\n",
    "            pep_idx +=1\n",
    "\n",
    "\n",
    "        # write fasta file \n",
    "        write_fasta(write_, sp, res['peptide'], pep_idx, aa_junction_pos, \n",
    "                    aa_junction_pos1_from_start, between_codons, between_codons1,\n",
    "                    pep_5include, pep_3include, res['geneName'], \n",
    "                    genome_junction_pos, genome_junction_pos1, \n",
    "                    res['kmer'], jx_pep1, jx_pep2, res['readFrameAnnotated'], \\\n",
    "                    res['kmer_coord'], res['kmerType'], res['geneStrand'], do_write=True)\n",
    "    \n",
    "#TODO \n",
    "#             if file_save_experiement is not None: \n",
    "#                 write_peptide_to_experiment(ep, pep_idx, \n",
    "#                                             res['peptide'],';'.join(kmer_file_idx[kmer]), \n",
    "#                                             peptide_junctions) \n",
    "\n",
    "\n",
    "if write_:\n",
    "    print('written to:')\n",
    "    print(file_save_experiement)\n",
    "    print(file_save)\n",
    "    if file_save_experiement is not None:\n",
    "        ep.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'100646614_100631213',\n",
       " '100704925_100704390',\n",
       " '100705006_100704946',\n",
       " '100705588_100705029',\n",
       " '100709484_100709223',\n",
       " '101338274_101338812',\n",
       " '102028141_102029466',\n",
       " '102448235_102410801',\n",
       " '102527756_102525064',\n",
       " '10262618_10263151',\n",
       " '102970218_102962760',\n",
       " '103174735_103123089',\n",
       " '103415516_103441573',\n",
       " '10497716_10484246',\n",
       " '10498710_10497772',\n",
       " '10498727_10484633',\n",
       " '10634794_10634973',\n",
       " '106974577_106972736',\n",
       " '107186681_107189315',\n",
       " '108765521_108765543',\n",
       " '108765521_108765554',\n",
       " '108765521_108765565',\n",
       " '108765521_108765569',\n",
       " '108766047_108768349',\n",
       " '108766294_108768349',\n",
       " '109255510_109217145',\n",
       " '109280769_109280489',\n",
       " '109322452_109296569',\n",
       " '109333892_109322463',\n",
       " '109554887_109556901',\n",
       " '109659961_109661164',\n",
       " '110499534_110496203',\n",
       " '110499534_110497002',\n",
       " '110499534_110498858',\n",
       " '110499534_110498859',\n",
       " '110502008_110499546',\n",
       " '110502048_110499546',\n",
       " '113219488_113233176',\n",
       " '113777259_113780570',\n",
       " '115415047_115412467',\n",
       " '115667873_115667137',\n",
       " '116110707_116068136',\n",
       " '116502209_116495514',\n",
       " '116770366_116777762',\n",
       " '119025190_119024607',\n",
       " '119240319_119243416',\n",
       " '120198856_120198631',\n",
       " '120200729_120198631',\n",
       " '121799195_121798029',\n",
       " '122857215_122859425',\n",
       " '124362285_124362246',\n",
       " '124363678_124362297',\n",
       " '124524841_124523002',\n",
       " '124539024_124526824',\n",
       " '124585086_124580562',\n",
       " '124912504_124912400',\n",
       " '124912831_124912778',\n",
       " '124912962_124912506',\n",
       " '124913175_124912857',\n",
       " '124913175_124913085',\n",
       " '124913190_124912050',\n",
       " '124913190_124912278',\n",
       " '124913190_124912506',\n",
       " '124913212_124912778',\n",
       " '124913267_124912583',\n",
       " '124913267_124913039',\n",
       " '124913287_124913195',\n",
       " '124913363_124913274',\n",
       " '124913395_124913094',\n",
       " '124913418_124913190',\n",
       " '124913522_124913213',\n",
       " '127807215_127710016',\n",
       " '128504904_128508875',\n",
       " '128508891_128509068',\n",
       " '12915721_12911650',\n",
       " '12920429_12915733',\n",
       " '13198807_13204218',\n",
       " '13201504_13204218',\n",
       " '133205542_133205411',\n",
       " '133208468_133207945',\n",
       " '13338704_13319356',\n",
       " '134141583_134141964',\n",
       " '134798461_134802887',\n",
       " '135914468_135911487',\n",
       " '136226773_136221941',\n",
       " '138059737_138069757',\n",
       " '138069763_138074000',\n",
       " '139386230_139391882',\n",
       " '141176757_141176408',\n",
       " '142181460_142165770',\n",
       " '143739948_143741751',\n",
       " '143741768_143741789',\n",
       " '144424945_144424819',\n",
       " '144425727_144424971',\n",
       " '145110444_145123020',\n",
       " '1461602_1460948',\n",
       " '149371686_149369862',\n",
       " '150546830_150558387',\n",
       " '150719472_150717469',\n",
       " '150896393_150831976',\n",
       " '151102957_151103058',\n",
       " '151199081_151206307',\n",
       " '151206329_151224244',\n",
       " '151206329_151224247',\n",
       " '151344416_151344257',\n",
       " '151669664_151667599',\n",
       " '152672458_152659356',\n",
       " '153628867_153629168',\n",
       " '154380517_154380597',\n",
       " '154464886_154465017',\n",
       " '155965701_155965671',\n",
       " '155965701_155965674',\n",
       " '156548347_156548168',\n",
       " '156551704_156544472',\n",
       " '156551973_156551883',\n",
       " '156700921_156700613',\n",
       " '15764775_15767608',\n",
       " '16008948_16000619',\n",
       " '160417956_160423545',\n",
       " '161722703_161722988',\n",
       " '161723009_161723334',\n",
       " '161723009_161723423',\n",
       " '16526091_16497678',\n",
       " '165759524_165754262',\n",
       " '16781704_16780741',\n",
       " '168302299_168302058',\n",
       " '169842539_169847689',\n",
       " '170805853_170811482',\n",
       " '170808752_170809982',\n",
       " '170911134_171240927',\n",
       " '171737531_171676820',\n",
       " '173834519_173838199',\n",
       " '176116623_176106988',\n",
       " '177038312_177038171',\n",
       " '179705852_179709872',\n",
       " '179705852_179716214',\n",
       " '179765789_179766648',\n",
       " '182860035_182866960',\n",
       " '183001952_183001263',\n",
       " '183094558_183079283',\n",
       " '183717748_183718572',\n",
       " '184245197_184243625',\n",
       " '184685684_184694521',\n",
       " '184711444_184709731',\n",
       " '185308799_185301026',\n",
       " '18555504_18547024',\n",
       " '185582817_185590113',\n",
       " '189063016_189060772',\n",
       " '189707146_189744741',\n",
       " '195312946_195308837',\n",
       " '195320700_195312967',\n",
       " '197003900_196997110',\n",
       " '197003900_196997113',\n",
       " '200554467_200552102',\n",
       " '201629728_201626147',\n",
       " '20176600_20171122',\n",
       " '20176600_20174509',\n",
       " '202530954_202552715',\n",
       " '203417081_203437095',\n",
       " '20402357_20402609',\n",
       " '20402625_20479845',\n",
       " '206593618_206586870',\n",
       " '21020116_20787308',\n",
       " '21474840_21474526',\n",
       " '21537119_21537121',\n",
       " '2164001_2165915',\n",
       " '2165936_2168076',\n",
       " '2175150_2175301',\n",
       " '218265830_218264366',\n",
       " '224916694_224907133',\n",
       " '235908906_235909401',\n",
       " '244855595_244855527',\n",
       " '244855903_244855608',\n",
       " '24533254_24535011',\n",
       " '24533254_24540373',\n",
       " '24704469_24706231',\n",
       " '26439126_26428088',\n",
       " '26494152_26443650',\n",
       " '26777064_26771089',\n",
       " '27066486_27064057',\n",
       " '27202427_27192242',\n",
       " '28407463_28436289',\n",
       " '28452654_28476180',\n",
       " '28929710_28935521',\n",
       " '291284_288075',\n",
       " '29148566_29148172',\n",
       " '29550857_29553224',\n",
       " '29828316_29829820',\n",
       " '30888814_30889201',\n",
       " '30967919_30961043',\n",
       " '31184390_31188324',\n",
       " '31798552_31800978',\n",
       " '32544779_32548846',\n",
       " '32548860_32558928',\n",
       " '32764347_32759269',\n",
       " '32774843_32786233',\n",
       " '32932902_32955295',\n",
       " '33403955_33406195',\n",
       " '354075_334422',\n",
       " '36137932_36146307',\n",
       " '36423943_36424133',\n",
       " '36479230_36481561',\n",
       " '36771424_36758617',\n",
       " '37109223_37107395',\n",
       " '3739804_3739604',\n",
       " '37872929_37868626',\n",
       " '38347935_38347775',\n",
       " '38909153_38830328',\n",
       " '39098421_39098372',\n",
       " '39265421_39279336',\n",
       " '39279358_39279448',\n",
       " '39366349_39357437',\n",
       " '39629230_39629539',\n",
       " '40260985_40281342',\n",
       " '40853627_40854638',\n",
       " '41527956_41525273',\n",
       " '41528204_41527967',\n",
       " '41621730_41620825',\n",
       " '41621730_41621649',\n",
       " '41622354_41621751',\n",
       " '41622354_41621754',\n",
       " '41636577_41633271',\n",
       " '42056034_42063185',\n",
       " '42382064_42380578',\n",
       " '42689627_42695520',\n",
       " '43687072_43692201',\n",
       " '43742456_43730702',\n",
       " '43992190_43993798',\n",
       " '4408180_4407684',\n",
       " '4416479_4408197',\n",
       " '44715701_44717606',\n",
       " '44716356_44717606',\n",
       " '44717617_44717678',\n",
       " '44717704_44718757',\n",
       " '44736710_44759639',\n",
       " '44759662_44759747',\n",
       " '44759662_44767230',\n",
       " '44894066_44894637',\n",
       " '45151831_45175642',\n",
       " '47280945_47280646',\n",
       " '47590881_47619736',\n",
       " '47689386_47636136',\n",
       " '49857935_49858056',\n",
       " '50094572_50098713',\n",
       " '50096972_50098713',\n",
       " '50098728_50099486',\n",
       " '50191820_50187119',\n",
       " '50192639_50191849',\n",
       " '50225677_50226878',\n",
       " '50546820_50547567',\n",
       " '51014231_51009142',\n",
       " '51029319_51029174',\n",
       " '52043624_52035721',\n",
       " '52237569_52238679',\n",
       " '52237717_52238679',\n",
       " '52238706_52243011',\n",
       " '52372669_52372796',\n",
       " '52374536_52375568',\n",
       " '52527696_52527375',\n",
       " '52527828_52527718',\n",
       " '52528511_52527651',\n",
       " '52528511_52528266',\n",
       " '52528636_52528539',\n",
       " '5332152_5335155',\n",
       " '53546787_53538454',\n",
       " '53547672_53546808',\n",
       " '54912815_54913759',\n",
       " '55588878_55585109',\n",
       " '56470082_56470486',\n",
       " '56470509_56470544',\n",
       " '56834442_56841704',\n",
       " '5781219_5765717',\n",
       " '58411469_58433159',\n",
       " '58458488_58458463',\n",
       " '58855333_58895611',\n",
       " '58891831_58895611',\n",
       " '58891865_58895611',\n",
       " '58895626_58909161',\n",
       " '59774188_59764233',\n",
       " '59810787_59666846',\n",
       " '60226872_60222761',\n",
       " '61813242_61811075',\n",
       " '620261_633768',\n",
       " '62029540_62020061',\n",
       " '62521390_62520180',\n",
       " '62521909_62521524',\n",
       " '62529781_62524861',\n",
       " '62531728_62529829',\n",
       " '62532112_62529829',\n",
       " '62957580_62973705',\n",
       " '63574403_63576855',\n",
       " '63865384_63869295',\n",
       " '63869311_63869350',\n",
       " '65528918_65521528',\n",
       " '65562888_65570090',\n",
       " '6609825_6607275',\n",
       " '66835846_66840002',\n",
       " '67431515_67432492',\n",
       " '6769660_6769945',\n",
       " '67932692_67925187',\n",
       " '68306760_68292039',\n",
       " '68829797_68833265',\n",
       " '69340332_69341510',\n",
       " '69350272_69352252',\n",
       " '69426457_69435482',\n",
       " '70234214_70235084',\n",
       " '70528452_70531916',\n",
       " '71011274_71005587',\n",
       " '71297461_71297835',\n",
       " '71297852_71297876',\n",
       " '71769622_71767885',\n",
       " '73148094_73173565',\n",
       " '73795391_73804622',\n",
       " '74084993_74101154',\n",
       " '74986117_75025545',\n",
       " '75012805_75018393',\n",
       " '75175037_75165701',\n",
       " '75693482_75694355',\n",
       " '7669670_7667425',\n",
       " '78037303_78037430',\n",
       " '78037441_78040203',\n",
       " '78040225_78040510',\n",
       " '78040225_78040577',\n",
       " '78040225_78040605',\n",
       " '78040225_78040614',\n",
       " '78040225_78040615',\n",
       " '78040225_78040620',\n",
       " '78040225_78040628',\n",
       " '78040225_78040636',\n",
       " '78040225_78040642',\n",
       " '78580962_78588313',\n",
       " '80207085_80208127',\n",
       " '80207085_80208573',\n",
       " '81205636_81204075',\n",
       " '82025507_82026904',\n",
       " '836073_836768',\n",
       " '85577649_85579008',\n",
       " '85579024_85581810',\n",
       " '86040738_86039443',\n",
       " '87060687_87060193',\n",
       " '87589921_87518205',\n",
       " '87589921_87569582',\n",
       " '8877029_8868512',\n",
       " '89110163_89109867',\n",
       " '89722954_89721495',\n",
       " '89805279_89799232',\n",
       " '89980632_89984516',\n",
       " '91359641_91342463',\n",
       " '94029517_94025048',\n",
       " '94395101_94397747',\n",
       " '94397758_94399048',\n",
       " '94397758_94400195',\n",
       " '94397758_94414221',\n",
       " '94397758_94426422',\n",
       " '94397758_94427626',\n",
       " '94397759_94398383',\n",
       " '94623300_94625523',\n",
       " '9468575_9469106',\n",
       " '9770262_9776567',\n",
       " '97805463_97815327',\n",
       " '98898178_98897637',\n",
       " '9943515_9945304',\n",
       " '99603166_99517183',\n",
       " '9990747_9992007',\n",
       " '99951290_99960104',\n",
       " None}"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptide_junctions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pep3",
   "language": "python",
   "name": "pep3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
