{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.013781Z",
     "start_time": "2022-06-27T09:30:39.429314Z"
    }
   },
   "outputs": [],
   "source": [
    "import os \n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "import numpy as np \n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-10T16:33:59.242716Z",
     "start_time": "2022-06-10T16:33:59.161360Z"
    }
   },
   "source": [
    "This notebook performs the following task: * Given a kmer_metadata file, generate a fasta file with metadata relevant for the tryptic digestion\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.021964Z",
     "start_time": "2022-06-27T09:30:45.017685Z"
    }
   },
   "outputs": [],
   "source": [
    "experiment_folder = '/cluster/work/grlab/projects/projects2022-PCa_Immuno/peptides_generation/v3_84dc237_conf2_annotFrame_cap0_runs'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.029263Z",
     "start_time": "2022-06-27T09:30:45.024214Z"
    }
   },
   "outputs": [],
   "source": [
    "sample_file = '/cluster/work/grlab/projects/projects2022-PCa_Immuno/samples_list/PCa_samples_20220616'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.199334Z",
     "start_time": "2022-06-27T09:30:45.030706Z"
    }
   },
   "outputs": [],
   "source": [
    "samples = pd.read_csv(sample_file, names = ['sample'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.252845Z",
     "start_time": "2022-06-27T09:30:45.201977Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C_scr_96_n1.all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C_scr_96_n2.all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C_scr_96_n3.all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C_scr_96_n4.all</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C_siU6atac_96_n1.all</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample\n",
       "0       C_scr_96_n1.all\n",
       "1       C_scr_96_n2.all\n",
       "2       C_scr_96_n3.all\n",
       "3       C_scr_96_n4.all\n",
       "4  C_siU6atac_96_n1.all"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.256211Z",
     "start_time": "2022-06-27T09:30:45.254048Z"
    }
   },
   "outputs": [],
   "source": [
    "sample = samples.iloc[0][0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preprocess the annotation \n",
    "### Input: Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:45.261363Z",
     "start_time": "2022-06-27T09:30:45.257827Z"
    }
   },
   "outputs": [],
   "source": [
    "ann_path = '/cluster/work/grlab/projects/projects2022-PCa_Immuno/annotation/gencode.v32.annotation.gtf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:30:56.106032Z",
     "start_time": "2022-06-27T09:30:45.263948Z"
    }
   },
   "outputs": [],
   "source": [
    "from immunopepper.preprocess import attribute_item_to_dict\n",
    "from immunopepper.preprocess import leq_strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:31:25.716182Z",
     "start_time": "2022-06-27T09:30:56.109116Z"
    }
   },
   "outputs": [],
   "source": [
    "# Partial copy From Immunopepper preprocess.py \n",
    "transcript_to_gene_dict = {}    # transcript -> gene id\n",
    "\n",
    "\n",
    "gene_to_transcript_dict = {}    # gene_id -> list of transcripts\n",
    "gene_cds_begin_dict = {}        # gene -> list of first CDS exons\n",
    "\n",
    "transcript_to_cds_dict = {}     # transcript -> list of CDS exons\n",
    "transcript_cds_begin_dict = {}  # transcript -> first exon of the CDS\n",
    "transcript_to_strand = {}\n",
    "\n",
    "file_type = ann_path.split('.')[-1]\n",
    "chromesome_set = set()\n",
    "# collect information from annotation file\n",
    "for line in open(ann_path, 'r'):\n",
    "    if line[0] == '#':\n",
    "        continue\n",
    "    item = line.strip().split('\\t')\n",
    "    chromesome_set.add(item[0])\n",
    "    feature_type = item[2]\n",
    "    attribute_item = item[-1]\n",
    "    attribute_dict = attribute_item_to_dict(attribute_item, file_type, feature_type)\n",
    "    # store relationship between gene ID and its transcript IDs\n",
    "    if feature_type in ['transcript', 'mRNA']:\n",
    "        gene_id = attribute_dict['gene_id']\n",
    "        transcript_id = attribute_dict['transcript_id']\n",
    "        if attribute_dict['gene_type'] != 'protein_coding' or attribute_dict['transcript_type']  != 'protein_coding':\n",
    "            continue\n",
    "        assert (transcript_id not in transcript_to_gene_dict)\n",
    "        transcript_to_gene_dict[transcript_id] = gene_id\n",
    "        if gene_id in gene_to_transcript_dict and transcript_id not in gene_to_transcript_dict[gene_id]:\n",
    "            gene_to_transcript_dict[gene_id].append(transcript_id)\n",
    "        else:\n",
    "            gene_to_transcript_dict[gene_id] = [transcript_id]\n",
    "        # Todo python is 0-based while gene annotation file(.gtf, .vcf, .maf) is one based\n",
    "    elif feature_type == \"CDS\":\n",
    "        parent_ts = attribute_dict['transcript_id']\n",
    "        strand_mode = item[6]\n",
    "        cds_left = int(item[3])-1\n",
    "        cds_right = int(item[4])\n",
    "        frameshift = int(item[7])\n",
    "        transcript_to_strand[parent_ts] = strand_mode\n",
    "        if parent_ts in transcript_to_cds_dict:\n",
    "            transcript_to_cds_dict[parent_ts].append((cds_left, cds_right, frameshift))\n",
    "        else:\n",
    "            transcript_to_cds_dict[parent_ts] = [(cds_left, cds_right, frameshift)]\n",
    "        if strand_mode == \"+\" :\n",
    "            cds_start, cds_stop = cds_left, cds_right\n",
    "        else:\n",
    "            cds_start, cds_stop = cds_right, cds_left\n",
    "\n",
    "        # we only consider the start of the whole CoDing Segment\n",
    "        if parent_ts not in transcript_cds_begin_dict or \\\n",
    "           leq_strand(cds_start, transcript_cds_begin_dict[parent_ts][0], strand_mode):\n",
    "            transcript_cds_begin_dict[parent_ts] = (cds_start, cds_stop, item)\n",
    "\n",
    "# collect first CDS exons for all transcripts of a gene\n",
    "for ts_key in transcript_to_gene_dict:\n",
    "    target_gene = transcript_to_gene_dict[ts_key]\n",
    "    if target_gene not in gene_cds_begin_dict:\n",
    "        gene_cds_begin_dict[target_gene] = []\n",
    "    if ts_key in transcript_cds_begin_dict:\n",
    "        gene_cds_begin_dict[target_gene].append(transcript_cds_begin_dict[ts_key])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:31:26.064988Z",
     "start_time": "2022-06-27T09:31:25.764929Z"
    }
   },
   "outputs": [],
   "source": [
    "# Custom collection of CDS \n",
    "transcript_cds_begin_dict_bis = {}\n",
    "transcript_cds_end_dict_bis = {}\n",
    "\n",
    "gene_cds_begin_dict_bis = defaultdict(list)\n",
    "gene_cds_end_dict_bis = defaultdict(list)\n",
    "\n",
    "# will be in reading order \n",
    "for ts_key in transcript_to_cds_dict:\n",
    "    if transcript_to_strand[ts_key] == '+': # '+'\n",
    "        transcript_cds_begin_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][0][0],\n",
    "                                                 transcript_to_cds_dict[ts_key][0][1], '+')\n",
    "        transcript_cds_end_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][-1][0],\n",
    "                                                 transcript_to_cds_dict[ts_key][-1][1], '+')\n",
    "        \n",
    "\n",
    "    else: \n",
    "        transcript_cds_begin_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][0][1],\n",
    "                                                 transcript_to_cds_dict[ts_key][0][0], '-')\n",
    "        transcript_cds_end_dict_bis[ts_key] = (transcript_to_cds_dict[ts_key][-1][1],\n",
    "                                                 transcript_to_cds_dict[ts_key][-1][0], '-')\n",
    "    \n",
    "    assert(transcript_cds_begin_dict_bis[ts_key][0] == transcript_cds_begin_dict[ts_key][0])\n",
    "    assert(transcript_cds_begin_dict_bis[ts_key][1] == transcript_cds_begin_dict[ts_key][1])\n",
    "\n",
    "# collect first, last CDS exons for all transcripts of a gene\n",
    "for ts_key in transcript_to_gene_dict:\n",
    "    target_gene = transcript_to_gene_dict[ts_key]\n",
    "    gene_cds_begin_dict_bis[target_gene].append(transcript_cds_begin_dict_bis[ts_key])\n",
    "    gene_cds_end_dict_bis[target_gene].append(transcript_cds_end_dict_bis[ts_key])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Format the peptide raw file to fasta \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-27T09:31:26.068998Z",
     "start_time": "2022-06-27T09:31:26.066375Z"
    }
   },
   "outputs": [],
   "source": [
    "fields_meta_peptide_dict = ['peptide', 'id', 'readFrame', 'geneName', 'geneChr', 'geneStrand',\n",
    "                                'mutationMode',\n",
    "                                'junctionAnnotated', 'hasStopCodon', 'isInJunctionList',\n",
    "                                'isIsolated', 'variantComb', 'variantSegExpr', 'modifiedExonsCoord',\n",
    "                                'originalExonsCoord', 'vertexIdx',\n",
    "                                'kmerType']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.692670Z",
     "start_time": "2022-06-24T13:58:04.687128Z"
    }
   },
   "outputs": [],
   "source": [
    "def extract_peptide_fields_pq(pep):\n",
    "    '''0 peptide\n",
    "    1 id\n",
    "    2 readFrame\n",
    "    3 readFrameAnnotated\n",
    "    4 geneName\n",
    "    5 geneChr\n",
    "    6 geneStrand\n",
    "    7 mutationMode\n",
    "    8 junctionAnnotated\n",
    "    9 hasStopCodon\n",
    "    10 isInJunctionList\n",
    "    11 isIsolated\n",
    "    12 variantComb\n",
    "    13 variantSegExpr\n",
    "    14 modifiedExonsCoord\n",
    "    15 originalExonsCoord\n",
    "    16 vertexIdx\n",
    "    ?junctionExpr # sample mode only\n",
    "    ?segmentExpr #sample mode only\n",
    "    17 kmerType '''\n",
    "    \n",
    "    pep_seq = pep['peptide']\n",
    "    pep_gene = pep['geneName']\n",
    "    pep_orig_coordS = pep['originalExonsCoord'].split('/')\n",
    "    pep_modif_coordS = pep['modifiedExonsCoord'].split('/')\n",
    "    strand = pep['geneStrand']\n",
    "    has_stop_codon = pep['hasStopCodon']\n",
    "    readFrameAnnotated = pep['readFrameAnnotated']\n",
    "    junctionAnnotated = pep['junctionAnnotated']\n",
    "    kmer_type = pep['kmerType'].replace('-','').replace('\\n','')\n",
    "    is_isolated = pep['isIsolated']\n",
    "    junctionExpr = pep['junctionExpr']\n",
    "\n",
    "    return pep, pep_seq, pep_gene, pep_orig_coordS, pep_modif_coordS, \\\n",
    "           strand, has_stop_codon, readFrameAnnotated, junctionAnnotated, \\\n",
    "           kmer_type, is_isolated, junctionExpr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.729705Z",
     "start_time": "2022-06-24T13:58:04.695530Z"
    }
   },
   "outputs": [],
   "source": [
    "def extract_end_starts(pep_orig_coord):\n",
    "    ''' Get peptide end and start coordinates'''\n",
    "    if strand == '+': # Do - strand \n",
    "        pep_start = np.int(pep_orig_coord[0])\n",
    "        pep_end = np.int(pep_orig_coord[-1])\n",
    "    else: \n",
    "        pep_start = np.int(pep_orig_coord[1])\n",
    "        pep_end = np.int(pep_orig_coord[-2])\n",
    "    return pep_start, pep_end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.740379Z",
     "start_time": "2022-06-24T13:58:04.732055Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_include_flag(start_cds, end_cds, pep_start, pep_end, has_stop_codon ):\n",
    "    '''Use end and start coordinates for 3' 5' include flag'''\n",
    "    if pep_start in start_cds: # We will always miss things that are new in the graph \n",
    "        pep_5include = 1\n",
    "    else: \n",
    "        pep_5include = 0 \n",
    "    if (pep_end in end_cds) or (has_stop_codon) == '1':\n",
    "        pep_3include = 1\n",
    "    else: \n",
    "        pep_3include = 0 \n",
    "    return pep_5include, pep_3include"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.750137Z",
     "start_time": "2022-06-24T13:58:04.745021Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_nt_len_with_aa_shift(pep_modi_coord):\n",
    "    '''Get nt length of each exon involved -> jx_list, shift_list'''\n",
    "    tot_len = 0 \n",
    "    shift = 0 \n",
    "    jx_list = []\n",
    "    jx_list_ori = []\n",
    "    shift_list = []\n",
    "    for pair in np.arange(0, len(pep_modi_coord), 2):\n",
    "        cds = int(pep_modi_coord[pair + 1]) - int(pep_modi_coord[pair])  # 0 based, open right \n",
    "        jx_list_ori.append(cds)\n",
    "        cds += shift \n",
    "        shift = cds % 3\n",
    "        jx_list.append(cds - shift)\n",
    "        shift_list.append(shift)\n",
    "        \n",
    "    return jx_list, shift_list, jx_list_ori"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.756266Z",
     "start_time": "2022-06-24T13:58:04.752054Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_aaPos_betweenFlag(shift_list, jx_list):\n",
    "    '''Get aa position of the junction\n",
    "    the junction coordinate jx_pos is the 0-based position in the peptide \n",
    "    of the amino acid that either overlaps the junction (if the junction is \n",
    "    in the middle of a codon), or is immediately before it if the junction \n",
    "    occurs between codons'''\n",
    "    if shift_list[0]: # junction is inside an amino acid\n",
    "        aa_junction_pos0 = int((jx_list[0] / 3)) # because 0 based\n",
    "        between_codons0 = 0 \n",
    "    else: # junction is between amino acids \n",
    "        aa_junction_pos0 = int((jx_list[0] / 3) - 1)  # because 0 based\n",
    "        between_codons0 = 1\n",
    "        \n",
    "    if len(shift_list) > 2: #third exon \n",
    "        if shift_list[1]: # junction is inside an amino acid\n",
    "            aa_junction_pos1 = int((jx_list[1] / 3)) # because 0 based\n",
    "            between_codons1 = 0 \n",
    "        else: # junction is between amino acids \n",
    "            aa_junction_pos1 = int((jx_list[1] / 3) - 1)  # because 0 based\n",
    "            between_codons1 = 1 \n",
    "        aa_junction_pos1_from_start = aa_junction_pos1 + aa_junction_pos0 + 1 \n",
    "    else:\n",
    "        aa_junction_pos1 = None\n",
    "        between_codons1 = None\n",
    "        aa_junction_pos1_from_start = None\n",
    "    \n",
    "    return aa_junction_pos0, between_codons0, aa_junction_pos1, between_codons1, \\\n",
    "           aa_junction_pos1_from_start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.765085Z",
     "start_time": "2022-06-24T13:58:04.757815Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_genomic_coordinates(pep_modi_coord):\n",
    "    '''We have in + case: exon1_start, exon 1_stop, exon2_start, exon2_stop, exon3_start, exon3_stop\n",
    "     In the - case: exon1_stop, exon 1_start, exon 2_stop, exon2_start, exon3_stop, exon3_start'''\n",
    "    genome_junction_pos1 = None\n",
    "    if strand == '+':\n",
    "        genome_junction_pos0 = '{}_{}'.format(pep_modi_coord[1], pep_modi_coord[2])\n",
    "        if len(pep_modi_coord) > 4:\n",
    "            genome_junction_pos1 = '{}_{}'.format(pep_modi_coord[3], pep_modi_coord[4])\n",
    "    else:\n",
    "        genome_junction_pos0 = '{}_{}'.format(pep_modi_coord[0], pep_modi_coord[3])\n",
    "        if len(pep_modi_coord) > 4:\n",
    "            genome_junction_pos1 = '{}_{}'.format(pep_modi_coord[2], pep_modi_coord[5])\n",
    "    return genome_junction_pos0, genome_junction_pos1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.774587Z",
     "start_time": "2022-06-24T13:58:04.767016Z"
    }
   },
   "outputs": [],
   "source": [
    "def split_coord(pep_coord):\n",
    "    pep_coord = pep_coord.split(';')\n",
    "    pep_coord = [coord for coord in pep_coord if (coord != 'None') and (coord != 'nan')]\n",
    "    return pep_coord"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.783777Z",
     "start_time": "2022-06-24T13:58:04.778075Z"
    }
   },
   "outputs": [],
   "source": [
    "def preprocess_line(line):\n",
    "    assert(False)\n",
    "    line = line.replace('3-exons_9-mer ', '3-exons_9-mer@').replace('2-exons ', '2-exons@')\n",
    "    kmer = line.split(',')[0]\n",
    "    peptides = ','.join(line.split(',')[1:])\n",
    "    peptides = peptides.split('@')\n",
    "    return line, kmer, peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.793180Z",
     "start_time": "2022-06-24T13:58:04.785775Z"
    }
   },
   "outputs": [],
   "source": [
    "def write_peptide_to_experiment(filepointer, pep_idx=None, pep_seq=None,\\\n",
    "                                idx=None, header=False):\n",
    "    if header:\n",
    "        header_exp = 'peptide_id\\tpeptide_sequence\\texperiment_ids\\n'\n",
    "        filepointer.write(header_exp)\n",
    "    elif pep_idx is not None:\n",
    "        exp_line = '{}\\t{}\\t{}\\n'.format(pep_idx, \n",
    "                                         pep_seq,\n",
    "                                         idx)\n",
    "        filepointer.write(exp_line)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.803714Z",
     "start_time": "2022-06-24T13:58:04.794755Z"
    }
   },
   "outputs": [],
   "source": [
    "def write_fasta_option(filepointer, pep_idx, aa_junction_pos, \n",
    "                aa_junction_pos1_from_start, between_codons, between_codons1,\n",
    "                pep_5include, pep_3include, pep_gene, \n",
    "                genome_junction_pos, genome_junction_pos1, \n",
    "                kmer, jx_pep1, jx_pep2, readFrameAnnotated, \\\n",
    "                junctionAnnotated, kmer_type, do_write=True):\n",
    "    \n",
    "        \n",
    "    if (kmer is not None):\n",
    "        assert(kmer in jx_pep1)\n",
    "        \n",
    "    pep_handle1 = '>pepID-{};jx_pos-{};between_codons-{};includes_5\\'-{};includes_3\\'-{};gene-{};jx_coord-{};kmer-{};readFrameAnnotated-{};junctionAnnotated-{};origin-{}'.format(\n",
    "    pep_idx, aa_junction_pos, between_codons, pep_5include, \n",
    "    pep_3include, pep_gene, genome_junction_pos, kmer, readFrameAnnotated, \n",
    "    junctionAnnotated, kmer_type)\n",
    "\n",
    "    pep_handle2 = '>pepID-{};jx_pos-{};between_codons-{};includes_5\\'-{};includes_3\\'-{};gene-{};jx_coord-{};kmer-{};readFrameAnnotated-{};junctionAnnotated-{};origin-{}'.format(\n",
    "    pep_idx, aa_junction_pos1_from_start, between_codons1, pep_5include, \n",
    "    pep_3include, pep_gene, genome_junction_pos1, kmer, readFrameAnnotated, \n",
    "    junctionAnnotated, kmer_type)\n",
    "    \n",
    "        \n",
    "    pep_idx+=1\n",
    "    sp.write(pep_handle1 + '\\n')\n",
    "    sp.write(pep_seq + '\\n')\n",
    "    if aa_junction_pos1_from_start is not None:\n",
    "        if (kmer is None) or (kmer in jx_pep2):\n",
    "            pep_idx+=1\n",
    "            sp.write(pep_handle2 + '\\n')\n",
    "            sp.write(pep_seq + '\\n')\n",
    "        \n",
    "    return pep_idx\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.814167Z",
     "start_time": "2022-06-24T13:58:04.805741Z"
    }
   },
   "outputs": [],
   "source": [
    "def cut_peptides(jx_list, between_codons, between_codons1, aa_junction_pos, \n",
    "                 aa_junction_pos1, aa_junction_pos1_from_start, \n",
    "                 print_ = False):\n",
    "    peptide_cut = []\n",
    "    aa_junction_pos_shift = aa_junction_pos + 1 \n",
    "    exon1, aa_jx1, exon2, aa_jx2, exon3 = '', '', '', '', ''\n",
    "\n",
    "    if len(jx_list) == 2:\n",
    "        if between_codons:\n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:]\n",
    "        else:\n",
    "            exon1 = pep_seq[:aa_junction_pos]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:]\n",
    "    elif len(jx_list) == 3:\n",
    "        aa_junction2_pos_shift =  aa_junction_pos1 + aa_junction_pos + 1\n",
    "        aa_junction2_pos_sshift = aa_junction_pos1 + aa_junction_pos + 2\n",
    "        assert(aa_junction_pos1_from_start == aa_junction2_pos_shift)\n",
    "        if between_codons and between_codons1: \n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:\n",
    "                          aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (not between_codons) and between_codons1: \n",
    "            exon1 = pep_seq[:aa_junction_pos ]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift:\n",
    "                          aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (between_codons) and (not between_codons1): \n",
    "            exon1 = pep_seq[:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift :\n",
    "                          aa_junction2_pos_shift]\n",
    "            aa_jx2 = pep_seq[aa_junction2_pos_shift:\n",
    "                         aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "        if (not between_codons) and (not between_codons1): \n",
    "            exon1 = pep_seq[:aa_junction_pos ]\n",
    "            aa_jx1 = pep_seq[aa_junction_pos:aa_junction_pos_shift]\n",
    "            exon2 = pep_seq[aa_junction_pos_shift :\n",
    "                          aa_junction2_pos_shift]\n",
    "            aa_jx2 = pep_seq[aa_junction2_pos_shift:\n",
    "                         aa_junction2_pos_sshift]\n",
    "            exon3 = pep_seq[aa_junction2_pos_sshift:]\n",
    "    if print_:\n",
    "        print(f'exon1:{exon1}, aa_containing_jx1:{aa_jx1}, exon2:{exon2}, aa_containing_jx2:{aa_jx2}, exon3:{exon3}')\n",
    "        print(f'junction positions jx1: {aa_junction_pos}, jx2:{aa_junction_pos1_from_start}')\n",
    "        print(f'is junction between a codon jx1: {between_codons}, jx2: {between_codons1}')\n",
    "        print('\\n')\n",
    "    return exon1 + aa_jx1 + exon2, exon2 + aa_jx2 + exon3 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.818881Z",
     "start_time": "2022-06-24T13:58:04.815630Z"
    }
   },
   "outputs": [],
   "source": [
    "def print_stats(print_, kmer, pep_seq, strand, pep_orig_coord, pep_modi_coord, jx_list,\n",
    "               jx_list_ori, genome_junction_pos, genome_junction_pos1,\n",
    "               aa_junction_pos, aa_junction_pos1, between_codons, between_codons1):\n",
    "    \n",
    "    if print_:\n",
    "        p_ori_coord = ';'.join(pep_orig_coord)\n",
    "        p_modif_coord = ';'.join(pep_modi_coord)\n",
    "        print(f'INSTANCE: \\n kmer {kmer}/ sequence {pep_seq}/ strand {strand} / \\n original coordinates {p_ori_coord} / \\n modif coordinates {p_modif_coord} /  \\n junction list origin {jx_list_ori}/ junction list {jx_list} / \\n junction coordinates 1 {genome_junction_pos} / junction coordinates 2 {genome_junction_pos1}')\n",
    "        #print(aa_junction_pos, between_codons, aa_junction_pos1, between_codons1)\n",
    "        print('peptide length', len(pep_seq))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:58:04.827098Z",
     "start_time": "2022-06-24T13:58:04.820340Z"
    }
   },
   "outputs": [],
   "source": [
    "def update_expression_dict(genome_junction_pos, expr_val, n_samples, id_sample, pep_seq):\n",
    "         # Check that each junction position in one sample has one expression value\n",
    "        if genome_junction_pos in jx_to_expression:\n",
    "            assert(jx_to_expression[genome_junction_pos][id_sample] == float(expr_val))\n",
    "        # Update the expression \n",
    "        jx_to_expression.setdefault(genome_junction_pos, \n",
    "                                    np.zeros(n_samples))[id_sample] = expr_val\n",
    "        # Update the peptides\n",
    "        jx_to_peptides[genome_junction_pos].append(pep_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:25:54.444968Z",
     "start_time": "2022-06-24T14:25:54.253870Z"
    }
   },
   "outputs": [],
   "source": [
    "def expression_dict(pep_seq, id_sample, n_samples, junctionExpr, \n",
    "                    genome_junction_pos, genome_junction_pos1, \n",
    "                   id_modi_coord, pep_orig_coord):\n",
    "    ''' \n",
    "       junctionExpr: string: example (1,1)/(2,0)' \n",
    "    - each tuple represents an expression value instance.\n",
    "    - First entry of the tuple is first junction expression value and\n",
    "    - the second entry is the second junction expressiion value  '''\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "        \n",
    "    \n",
    "    expression = [pair.split(',') for pair in junctionExpr.replace(',)', '').\\\n",
    "                                                                   replace('(', '').\\\n",
    "                                                                   replace(')', '').\\\n",
    "                                                                   replace(' ', '').\\\n",
    "                                                                  split('/')]   \n",
    "    \n",
    "    # Edge case: some junctions have the same expression values \n",
    "    # because they have the same position inside the same peptide!\n",
    "    id_modi_coord = min(len(expression) - 1 , id_modi_coord) \n",
    "    expr_val = expression[id_modi_coord]\n",
    "    # Update 2-exon case\n",
    "    update_expression_dict(genome_junction_pos, expr_val[0], \n",
    "                           n_samples, id_sample, pep_seq)\n",
    "    # Update 3-exon case\n",
    "    if genome_junction_pos1: \n",
    "        update_expression_dict(genome_junction_pos1, expr_val[1], \n",
    "                       n_samples, id_sample, pep_seq)\n",
    "\n",
    "    #TEST prints\n",
    "    #if len(expression)> 1:\n",
    "    print(junctionExpr)\n",
    "    print(pep['peptide'])\n",
    "    print('strand', pep['geneStrand'])\n",
    "    print('modif', pep['modifiedExonsCoord'], 'orig', pep_orig_coord) \n",
    "    print(expression )\n",
    "    print(id_modi_coord, 'id of coordinate')\n",
    "    print(expr_val, 'expression value')\n",
    "    print( '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:25:55.061342Z",
     "start_time": "2022-06-24T14:25:55.058499Z"
    }
   },
   "outputs": [],
   "source": [
    "## Format and create flags \n",
    "pep_idx = 0 \n",
    "kmer_len = 9 \n",
    "file_save_experiement = None \n",
    "\n",
    "print_ = False\n",
    "test_query = 'MRGQRSLLLGPARLCLRLLLLLGYRRRCPPLLRGLVQRWRYGKVCLRSLLYNSFGGSDTAVDAAFEPVYWLVDNVIRWFGVVSDVQGAGKGVLWGAEGHVSHRPNPGFGGRPPPPGKLRVSFGHSSCAMSP'\n",
    "subset_run_test = False\n",
    "write_= False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:25:55.879999Z",
     "start_time": "2022-06-24T14:25:55.422234Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C_scr_96_n1.all\n",
      "open /cluster/work/grlab/projects/projects2022-PCa_Immuno/peptides_generation/v3_37b9a8f_conf2_annotFrame_cap0_runs/C_scr_96_n1.all_mutNone/ref_sample_peptides_meta.mig_filtered.pq\n",
      "new\n",
      "aa position 34 None\n",
      "genome position 60536859_60014152 None\n",
      "(17.0,)\n",
      "MSFRFGQHLIKPSVVFLKTELSFALVNRKPVVPGHVLVCPLRPVERFHDLRPDEVADLFQTTQRVGTVVEKHFHGTSLTFSMQ\n",
      "strand -\n",
      "modif 60536859;60536962;60014006;60014152;None;None orig ['60536859', '60536979', '60014006', '60014152']\n",
      "[['17.0']]\n",
      "0 id of coordinate\n",
      "['17.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 34 None\n",
      "genome position 60536859_60014152 None\n",
      "(17.0,)\n",
      "MSFRFGQHLIKPSVVFLKTELSFALVNRKPVVPGHVLVCPLRPVERFHDLRPDEVADLFQTTQRVGTVVEKHFHGTSL\n",
      "strand -\n",
      "modif 60536859;60536962;60014021;60014152;None;None orig ['60536859', '60536979', '60014019', '60014152']\n",
      "[['17.0']]\n",
      "0 id of coordinate\n",
      "['17.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 30 None\n",
      "genome position 19608272_19608973 None\n",
      "(0.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRLRGGIHPEM\n",
      "strand +\n",
      "modif 19608179;19608272;19608973;19609021;None;None orig ['19608177', '19608272', '19608973', '19609022']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 19608274_19608973 None\n",
      "(437.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRFYVEASILKCNKFLSK\n",
      "strand +\n",
      "modif 19608179;19608274;19608973;19609019;None;None orig ['19608177', '19608274', '19608973', '19609021']\n",
      "[['437.0']]\n",
      "0 id of coordinate\n",
      "['437.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 19608274_19608973 None\n",
      "(437.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRFYVEASILKCNKFLSKT\n",
      "strand +\n",
      "modif 19608179;19608274;19608973;19609022;None;None/19608179;19608274;19608973;19610414;None;None/19608179;19608274;19608973;19609034;None;None/19608179;19608274;19608973;19610543;None;None orig ['19608177', '19608274', '19608973', '19610543']\n",
      "[['437.0']]\n",
      "0 id of coordinate\n",
      "['437.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 19608274_19609337 None\n",
      "(0.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRFLRTSGSLSWEHDPKKRSEGVLWVTHATMRVTPGTRWAHPPCWLLLMMMMMKMTSEAEAFALSGLGP\n",
      "strand +\n",
      "modif 19608179;19608274;19609337;19609536;None;None orig ['19608177', '19608274', '19609337', '19609537']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 19608274_19610321 None\n",
      "(0.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRLGISECLPRLTCMIRGIGDPLVSVYARAYLCR\n",
      "strand +\n",
      "modif 19608179;19608274;19610321;19610415;None;None orig ['19608177', '19608274', '19610321', '19610415']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 19608274_19610321 None\n",
      "(0.0,)\n",
      "HFSPENANDTAKETCLNWFFKIASIRELIPRLGISECLPRLTCMIRGIGDPLVSVYARAYLCRVGHASHCP\n",
      "strand +\n",
      "modif 19608179;19608274;19610321;19610541;None;None/19608179;19608274;19610321;19610442;None;None orig ['19608177', '19608274', '19610321', '19610444']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42774398 None\n",
      "(0.0,)\n",
      "MTVMEAQESPLFNNVKLQRSSSLAQAGMQWHNHIKLLLIGRVSLYLPPLNIQEMPIHIMLGNCTHMCVGEEGAGVSQHGTGSENFYPTQLMCMPFP\n",
      "strand +\n",
      "modif 42773823;42773878;42774398;42774697;None;None orig ['42773732', '42773878', '42774398', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42774584 None\n",
      "(0.0,)\n",
      "MTVMEAQESPLFNNVKLQRSENFYPTQLMCMPFP\n",
      "strand +\n",
      "modif 42773823;42773878;42774584;42774697;None;None orig ['42773479', '42773878', '42774584', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42774585 None\n",
      "(0.0,)\n",
      "MTVMEAQESPLFNNVKLQPLKIFILPNSCVCRSLNLKPGKLPVESIQIVLEELRKK\n",
      "strand +\n",
      "modif 42773823;42773878;42774585;42774698;None;None orig ['42773732', '42773878', '42774585', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42774645 None\n",
      "(491.0,)\n",
      "MTVMEAQESPLFNNVKLQRKLPVESIQIVLEELRKK\n",
      "strand +\n",
      "modif 42773823;42773878;42774645;42774698;None;None orig ['42773479', '42773878', '42774645', '42774699']\n",
      "[['491.0']]\n",
      "0 id of coordinate\n",
      "['491.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42774645 None\n",
      "(491.0,)\n",
      "MTVMEAQESPLFNNVKLQRKLPVESIQIVLEELRKKGGFSSPDSLFFFLVGFS\n",
      "strand +\n",
      "modif 42773823;42773878;42774645;42775346;None;None orig ['42773732', '42773878', '42774645', '42775348']\n",
      "[['491.0']]\n",
      "0 id of coordinate\n",
      "['491.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 18 None\n",
      "genome position 42773878_42775257 None\n",
      "(0.0,)\n",
      "MTVMEAQESPLFNNVKLQRSHHVAQASFELLGSSSPPALASQSVSHCA\n",
      "strand +\n",
      "modif 42773823;42773878;42775257;42775346;None;None orig ['42773479', '42773878', '42775257', '42775348']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42774398 None\n",
      "(0.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRSSSLAQAGMQWHNHIKLLLIGRVSLYLPPLNIQEMPIHIMLGNCTHMCVGEEGAGVSQHGTGSENFYPTQLMCMPFP\n",
      "strand +\n",
      "modif 42773733;42773878;42774398;42774697;None;None orig ['42773732', '42773878', '42774398', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42774584 None\n",
      "(0.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRSENFYPTQLMCMPFP\n",
      "strand +\n",
      "modif 42773733;42773878;42774584;42774697;None;None orig ['42773732', '42773878', '42774584', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42774585 None\n",
      "(0.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQPLKIFILPNSCVCRSLNLKPGKLPVESIQIVLEELRKK\n",
      "strand +\n",
      "modif 42773733;42773878;42774585;42774698;None;None orig ['42773732', '42773878', '42774585', '42774699']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42774645 None\n",
      "(491.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRKLPVESIQIVLEELRKK\n",
      "strand +\n",
      "modif 42773733;42773878;42774645;42774698;None;None orig ['42773732', '42773878', '42774645', '42774699']\n",
      "[['491.0']]\n",
      "0 id of coordinate\n",
      "['491.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42774645 None\n",
      "(491.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRKLPVESIQIVLEELRKKGGFSSPDSLFFFLVGFS\n",
      "strand +\n",
      "modif 42773733;42773878;42774645;42775346;None;None orig ['42773732', '42773878', '42774645', '42775348']\n",
      "[['491.0']]\n",
      "0 id of coordinate\n",
      "['491.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 42773878_42775257 None\n",
      "(0.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRSHHVAQASFELLGSSSPPALASQSVSHCA\n",
      "strand +\n",
      "modif 42773733;42773878;42775257;42775346;None;None orig ['42773732', '42773878', '42775257', '42775348']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 80 None\n",
      "genome position 42773976_42774645 None\n",
      "(0.0,)\n",
      "LQPNVDTRQKQLAAWCSLVLSFCRLHKQSSMTVMEAQESPLFNNVKLQRILPQALPQPIHMHFCAFTHAQTPPRASPLLPMESFLWSRSRLY\n",
      "strand +\n",
      "modif 42773733;42773976;42774645;42774699;None;None/42773733;42773976;42774645;42775347;None;None orig ['42773732', '42773976', '42774645', '42775348']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 50 None\n",
      "genome position 42773976_42774645 None\n",
      "(0.0,)\n",
      "MTVMEAQESPLFNNVKLQRILPQALPQPIHMHFCAFTHAQTPPRASPLLPMESFLWSRSRLY\n",
      "strand +\n",
      "modif 42773823;42773976;42774645;42774699;None;None/42773823;42773976;42774645;42775347;None;None orig ['42773732', '42773976', '42774645', '42775348']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212965721 None\n",
      "(7.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLQYNHTGTQFFEIRKMRPLS\n",
      "strand +\n",
      "modif 212961165;212961254;212965721;212965776;None;None orig ['212961165', '212961254', '212965721', '212965778']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212965721 None\n",
      "(7.0,)\n",
      "MTIPDWLQAIQNYMKTLQYNHTGTQFFEIRKMRPLS\n",
      "strand +\n",
      "modif 212961201;212961254;212965721;212965776;None;None orig ['212961165', '212961254', '212965721', '212965778']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212965721 None\n",
      "(7.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLQYNHTGTQFFEIRKMRPLSG\n",
      "strand +\n",
      "modif 212961165;212961254;212965721;212966343;None;None/212961165;212961254;212965721;212966340;None;None/212961165;212961254;212965721;212966154;None;None orig ['212961165', '212961254', '212965721', '212966341']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212965721 None\n",
      "(7.0,)\n",
      "MTIPDWLQAIQNYMKTLQYNHTGTQFFEIRKMRPLSG\n",
      "strand +\n",
      "modif 212961201;212961254;212965721;212966340;None;None/212961201;212961254;212965721;212966154;None;None/212961201;212961254;212965721;212966343;None;None orig ['212961165', '212961254', '212965721', '212966341']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212972529 None\n",
      "(0.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLHFKFSRPPFSSLTQIV\n",
      "strand +\n",
      "modif 212961165;212961254;212972529;212972728;None;None orig ['212961165', '212961254', '212972529', '212972729']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212972529 None\n",
      "(0.0,)\n",
      "MTIPDWLQAIQNYMKTLHFKFSRPPFSSLTQIV\n",
      "strand +\n",
      "modif 212961201;212961254;212972529;212972728;None;None orig ['212961165', '212961254', '212972529', '212972729']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRAD\n",
      "strand +\n",
      "modif 212961165;212961254;212972579;212972916;None;None orig ['212961165', '212961254', '212972579', '212972918']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "MTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRAD\n",
      "strand +\n",
      "modif 212961201;212961254;212972579;212972916;None;None orig ['212961165', '212961254', '212972579', '212972918']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMK\n",
      "strand +\n",
      "modif 212961165;212961254;212972579;212972961;None;None orig ['212961165', '212961254', '212972579', '212972961']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "MTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMK\n",
      "strand +\n",
      "modif 212961201;212961254;212972579;212972961;None;None orig ['212961165', '212961254', '212972579', '212972961']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "PSIPQVPNYRLSMTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMKVGAGKTRKPCRRALFPFLSLCLLLQASFF\n",
      "strand +\n",
      "modif 212961165;212961254;212972579;212974068;None;None orig ['212961165', '212961254', '212972579', '212974070']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 17 None\n",
      "genome position 212961254_212972579 None\n",
      "(2.0,)\n",
      "MTIPDWLQAIQNYMKTLHYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMKVGAGKTRKPCRRALFPFLSLCLLLQASFF\n",
      "strand +\n",
      "modif 212961201;212961254;212972579;212974068;None;None orig ['212961165', '212961254', '212972579', '212974070']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 23 None\n",
      "genome position 212966341_212972579 None\n",
      "(0.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILAT\n",
      "strand +\n",
      "modif 212966271;212966341;212972579;212972959;None;None orig ['212966270', '212966341', '212972579', '212972961']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 24 None\n",
      "genome position 212966345_212966509 None\n",
      "(0.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILGIIEPNQCLASWTNISLTCELQPPSGPALRPVTLQTH\n",
      "strand +\n",
      "modif 212966271;212966345;212966509;212966708;None;None orig ['212966270', '212966345', '212966509', '212966709']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 24 None\n",
      "genome position 212966345_212967113 None\n",
      "(0.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILGITHS\n",
      "strand +\n",
      "modif 212966271;212966345;212967113;212969247;None;None orig ['212966270', '212966345', '212967113', '212969249']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 24 None\n",
      "genome position 212966345_212972579 None\n",
      "(39.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILGIYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRAD\n",
      "strand +\n",
      "modif 212966271;212966345;212972579;212972916;None;None orig ['212966270', '212966345', '212972579', '212972918']\n",
      "[['39.0']]\n",
      "0 id of coordinate\n",
      "['39.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 24 None\n",
      "genome position 212966345_212972579 None\n",
      "(39.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILGIYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMK\n",
      "strand +\n",
      "modif 212966271;212966345;212972579;212972961;None;None orig ['212966270', '212966345', '212972579', '212972961']\n",
      "[['39.0']]\n",
      "0 id of coordinate\n",
      "['39.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 24 None\n",
      "genome position 212966345_212972579 None\n",
      "(39.0,)\n",
      "LMETAKEMTRESLPIKCLEAVILGIYLTNGQPSIERFPISFKTYFSGNYFHHVVLGIYCNGRYGSLGMSRRAELMDKPLTFRTLSDLIFDFEDSYKKYLHTVKKVKIGLYVPHEPHSFQPIEWKQLVLNVSKMLRADIRKELEKYARDMRMKVGAGKTRKPCRRALFPFLSLCLLLQASFF\n",
      "strand +\n",
      "modif 212966271;212966345;212972579;212974068;None;None orig ['212966270', '212966345', '212972579', '212974070']\n",
      "[['39.0']]\n",
      "0 id of coordinate\n",
      "['39.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 10 None\n",
      "genome position 97438669_97437275 None\n",
      "(2.0,)\n",
      "KEDVRATEKDKISLGDAQSNYLLTTAENELGVVVAHSES\n",
      "strand -\n",
      "modif 97438669;97438702;97437191;97437275;None;None orig ['97438669', '97438703', '97437190', '97437275']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 35 None\n",
      "genome position 97437641_97437275 None\n",
      "(0.0,)\n",
      "LKFIRVSAQVTLSWPKWYPLFPADFWSFVKRREGGRISLGDAQSNYLLTTAENELGVVVAHSES\n",
      "strand -\n",
      "modif 97437641;97437749;97437191;97437275;None;None orig ['97437641', '97437750', '97437190', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 36 None\n",
      "genome position 97437641_97437275 None\n",
      "(0.0,)\n",
      "VEIYKSFRPGDIVLAKVVSFIPSRLLVLCQKKRRRPDLLR\n",
      "strand -\n",
      "modif 97437641;97437750;97437261;97437275;None;None/97437641;97437750;97437192;97437275;None;None orig ['97437641', '97437750', '97437190', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 35 None\n",
      "genome position 97437641_97437275 None\n",
      "(0.0,)\n",
      "LKFIRVSAQVTLSWPKWYPLFPADFWSFVKRREGGRISLGD\n",
      "strand -\n",
      "modif 97437641;97437749;97437260;97437275;None;None orig ['97437641', '97437750', '97437259', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 8 None\n",
      "genome position 97437722_97437275 None\n",
      "(0.0,)\n",
      "LKFIRVSAQISLGDAQSNYLLTTAENELGVVVAHSES\n",
      "strand -\n",
      "modif 97437722;97437749;97437191;97437275;None;None orig ['97437722', '97437750', '97437190', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97437722_97437275 None\n",
      "(0.0,)\n",
      "VEIYKSFRPDLLR\n",
      "strand -\n",
      "modif 97437722;97437750;97437192;97437275;None;None/97437722;97437750;97437261;97437275;None;None orig ['97437722', '97437750', '97437259', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 8 None\n",
      "genome position 97437722_97437275 None\n",
      "(0.0,)\n",
      "LKFIRVSAQISLGD\n",
      "strand -\n",
      "modif 97437722;97437749;97437260;97437275;None;None orig ['97437722', '97437750', '97437259', '97437275']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 16 None\n",
      "genome position 97437699_97437275 None\n",
      "(243.0,)\n",
      "VEIYKSFRPGDIVLAKVISLGDAQSNYLLTTAENELGVVVAHSES\n",
      "strand -\n",
      "modif 97437699;97437750;97437191;97437275;None;None orig ['97437699', '97437750', '97437190', '97437275']\n",
      "[['243.0']]\n",
      "0 id of coordinate\n",
      "['243.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 16 None\n",
      "genome position 97437699_97437282 None\n",
      "(1.0,)\n",
      "LKFIRVSAQVTLSWPKCD\n",
      "strand -\n",
      "modif 97437699;97437749;97437191;97437282;None;None orig ['97437699', '97437750', '97437190', '97437282']\n",
      "[['1.0']]\n",
      "0 id of coordinate\n",
      "['1.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 16 None\n",
      "genome position 97437699_97437275 None\n",
      "(243.0,)\n",
      "VEIYKSFRPGDIVLAKVISLGD\n",
      "strand -\n",
      "modif 97437699;97437750;97437260;97437275;None;None orig ['97437699', '97437750', '97437259', '97437275']\n",
      "[['243.0']]\n",
      "0 id of coordinate\n",
      "['243.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 29 39\n",
      "genome position 97441170_97437750 97437722_97437275\n",
      "(0.0, 0.0)\n",
      "VSSINSRFAKVHILYVGSMPLKNSFRGTIRLKFIRVSAQISLGDAQSNYLLTTAENELGVVVAHSES\n",
      "strand -\n",
      "modif 97441170;97441259;97437722;97437750;97437191;97437275 orig ['97441170', '97441259', '97437722', '97437750', '97437190', '97437275']\n",
      "[['0.0', '0.0']]\n",
      "0 id of coordinate\n",
      "['0.0', '0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 29 39\n",
      "genome position 97441170_97437750 97437722_97437275\n",
      "(0.0, 0.0)\n",
      "VSSINSRFAKVHILYVGSMPLKNSFRGTIRLKFIRVSAQISLGD\n",
      "strand -\n",
      "modif 97441170;97441259;97437722;97437750;97437260;97437275 orig ['97441170', '97441259', '97437722', '97437750', '97437259', '97437275']\n",
      "[['0.0', '0.0']]\n",
      "0 id of coordinate\n",
      "['0.0', '0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 10 20\n",
      "genome position 97438669_97437750 97437722_97437275\n",
      "(249.0, 0.0)\n",
      "KEDVRATEKDKVEIYKSFRPDLLR\n",
      "strand -\n",
      "modif 97438669;97438702;97437722;97437750;97437261;97437275/97438669;97438702;97437722;97437750;97437192;97437275 orig ['97438669', '97438703', '97437722', '97437750', '97437259', '97437275']\n",
      "[['249.0', '0.0']]\n",
      "0 id of coordinate\n",
      "['249.0', '0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "new\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97453529 None\n",
      "(220.0,)\n",
      "CVLKMDHHCPWLNNCVGHYNH\n",
      "strand +\n",
      "modif 97452895;97452923;97453529;97453564;None;None orig ['97452894', '97452923', '97453529', '97453566']\n",
      "[['220.0']]\n",
      "0 id of coordinate\n",
      "['220.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97453529 None\n",
      "(220.0,)\n",
      "CVLKMDHHCPWLNNCVGHYNHRYFFSFCFFMTLGCVY\n",
      "strand +\n",
      "modif 97452895;97452923;97453529;97453612;None;None orig ['97452894', '97452923', '97453529', '97453614']\n",
      "[['220.0']]\n",
      "0 id of coordinate\n",
      "['220.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97453529 None\n",
      "(220.0,)\n",
      "CVLKMDHHCPWLNNCVGHYNHRYFFSFCFFMTLGCVYCSYGSWDLFREAYAAIE\n",
      "strand +\n",
      "modif 97452895;97452923;97453529;97453663;None;None orig ['97452894', '97452923', '97453529', '97453663']\n",
      "[['220.0']]\n",
      "0 id of coordinate\n",
      "['220.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97453529 None\n",
      "(220.0,)\n",
      "CVLKMDHHCPWLNNCVGHYNHRYFFSFCFFMTLGCVYCSYGSWDLFREAYAAIEVSSSGTGQLSSAELRDVEPVPKEWGW\n",
      "strand +\n",
      "modif 97452895;97452923;97453529;97457359;None;None/97452895;97452923;97453529;97453846;None;None/97452895;97452923;97453529;97457311;None;None/97452895;97452923;97453529;97455781;None;None/97452895;97452923;97453529;97454797;None;None/97452895;97452923;97453529;97457083;None;None/97452895;97452923;97453529;97453759;None;None/97452895;97452923;97453529;97457248;None;None/97452895;97452923;97453529;97455757;None;None/97452895;97452923;97453529;97456042;None;None/97452895;97452923;97453529;97456924;None;None/97452895;97452923;97453529;97457365;None;None/97452895;97452923;97453529;97456984;None;None/97452895;97452923;97453529;97457368;None;None/97452895;97452923;97453529;97457293;None;None/97452895;97452923;97453529;97453981;None;None orig ['97452894', '97452923', '97453529', '97455757']\n",
      "[['220.0']]\n",
      "0 id of coordinate\n",
      "['220.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97454599 None\n",
      "(0.0,)\n",
      "CVLKMDHHCPLPCFLCLDLDHFLPIGAGGSCWRPTEQMSQLCCFLFLDLSPDPTTHLLLSRKDDSQESCLPLVPVQ\n",
      "strand +\n",
      "modif 97452895;97452923;97454599;97454799;None;None orig ['97452894', '97452923', '97454599', '97454799']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455590 None\n",
      "(7.0,)\n",
      "CVLKMDHHCPQFQTYFLQPSLNYPPVFAPLFLVLWHLPWVP\n",
      "strand +\n",
      "modif 97452895;97452923;97455590;97455781;None;None orig ['97452894', '97452923', '97455590', '97455783']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455601 None\n",
      "(58.0,)\n",
      "CVLKMDHHCHILPPALSKLPTSLRPSFLSSVALALGALTVWHAVLISRGETSIERHINKKERRRLQAKGR\n",
      "strand +\n",
      "modif 97452895;97452923;97455601;97455783;None;None orig ['97452894', '97452923', '97455601', '97455783']\n",
      "[['58.0']]\n",
      "0 id of coordinate\n",
      "['58.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455601 None\n",
      "(58.0,)\n",
      "CVLKMDHHCHILPPALSKLPTSLRPSFLSSVALALGALTVWHAVLISRGETSIERHINKKERRRLQAKGRVSRVEGSGWVGG\n",
      "strand +\n",
      "modif 97452895;97452923;97455601;97456017;None;None/97452895;97452923;97455601;97457370;None;None/97452895;97452923;97455601;97457361;None;None/97452895;97452923;97455601;97457247;None;None/97452895;97452923;97455601;97456923;None;None/97452895;97452923;97455601;97456044;None;None/97452895;97452923;97455601;97457310;None;None/97452895;97452923;97455601;97457364;None;None/97452895;97452923;97455601;97457292;None;None/97452895;97452923;97455601;97456983;None;None/97452895;97452923;97455601;97457085;None;None orig ['97452894', '97452923', '97455601', '97456924']\n",
      "[['58.0']]\n",
      "0 id of coordinate\n",
      "['58.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455616 None\n",
      "(0.0,)\n",
      "CVLKMDHHCPLSKLPTSLRPSFLSSVALALGALTVWHAVLISRGETSIERHINKKERRRLQAKGR\n",
      "strand +\n",
      "modif 97452895;97452923;97455616;97455783;None;None orig ['97452894', '97452923', '97455616', '97455783']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455913 None\n",
      "(0.0,)\n",
      "CVLKMDHHCP\n",
      "strand +\n",
      "modif 97452895;97452923;97455913;97456017;None;None orig ['97452894', '97452923', '97455913', '97456018']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455918 None\n",
      "(0.0,)\n",
      "CVLKMDHHCPNLV\n",
      "strand +\n",
      "modif 97452895;97452923;97455918;97456016;None;None orig ['97452894', '97452923', '97455918', '97456018']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455981 None\n",
      "(1.0,)\n",
      "CVLKMDHHCRILTTTAAWTTG\n",
      "strand +\n",
      "modif 97452895;97452923;97455981;97456016;None;None orig ['97452894', '97452923', '97455981', '97456018']\n",
      "[['1.0']]\n",
      "0 id of coordinate\n",
      "['1.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455981 None\n",
      "(1.0,)\n",
      "CVLKMDHHCRILTTTAAWTTGRYSWVWIQE\n",
      "strand +\n",
      "modif 97452895;97452923;97455981;97456043;None;None orig ['97452894', '97452923', '97455981', '97456044']\n",
      "[['1.0']]\n",
      "0 id of coordinate\n",
      "['1.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455982 None\n",
      "(2.0,)\n",
      "CVLKMDHHCQSLQLRLLGQLE\n",
      "strand +\n",
      "modif 97452895;97452923;97455982;97456017;None;None orig ['97452894', '97452923', '97455982', '97456018']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 9 None\n",
      "genome position 97452923_97455982 None\n",
      "(2.0,)\n",
      "CVLKMDHHCQSLQLRLLGQLEGIPGCGYRK\n",
      "strand +\n",
      "modif 97452895;97452923;97455982;97456044;None;None orig ['97452894', '97452923', '97455982', '97456044']\n",
      "[['2.0']]\n",
      "0 id of coordinate\n",
      "['2.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 55 None\n",
      "genome position 61724954_61723784 None\n",
      "(0.0,)/(168.0,)\n",
      "MAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQYICKDPKINDATQEPVNCTNYTAH\n",
      "strand -\n",
      "modif 61724954;61725120;61723713;61723784;None;None/61724956;61725120;61723713;61723786;None;None orig ['61724956', '61725120', '61723712', '61723786']\n",
      "[['0.0'], ['168.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "aa position 54 None\n",
      "genome position 61724956_61723786 None\n",
      "(0.0,)/(168.0,)\n",
      "MAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQYICKDPKINDATQEPVNCTNYTAH\n",
      "strand -\n",
      "modif 61724954;61725120;61723713;61723784;None;None/61724956;61725120;61723713;61723786;None;None orig ['61724956', '61725120', '61723712', '61723786']\n",
      "[['0.0'], ['168.0']]\n",
      "1 id of coordinate\n",
      "['168.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 117 None\n",
      "genome position 61724954_61723784 None\n",
      "(0.0,)/(168.0,)\n",
      "MHILKGSPNVIPRAHGQKNTRRDGTGLYPMRGPFKNLALLPFSLPLLGGGGSGSGEKVSVSKMAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQYICKDPKINDATQEPVNCTNYTAH\n",
      "strand -\n",
      "modif 61724954;61725306;61723713;61723784;None;None/61724956;61725306;61723713;61723786;None;None orig ['61724954', '61725423', '61723712', '61723784']\n",
      "[['0.0'], ['168.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "aa position 116 None\n",
      "genome position 61724956_61723786 None\n",
      "(0.0,)/(168.0,)\n",
      "MHILKGSPNVIPRAHGQKNTRRDGTGLYPMRGPFKNLALLPFSLPLLGGGGSGSGEKVSVSKMAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQYICKDPKINDATQEPVNCTNYTAH\n",
      "strand -\n",
      "modif 61724954;61725306;61723713;61723784;None;None/61724956;61725306;61723713;61723786;None;None orig ['61724954', '61725423', '61723712', '61723784']\n",
      "[['0.0'], ['168.0']]\n",
      "1 id of coordinate\n",
      "['168.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 54 None\n",
      "genome position 61724956_61724031 None\n",
      "(0.0,)\n",
      "MAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQDRLLVMLPRLFFPNSWAPVILLPQVLLSNTKSIFVHMLIDLDKLAASIYM\n",
      "strand -\n",
      "modif 61724956;61725120;61723712;61724031;None;None orig ['61724956', '61725120', '61723712', '61724031']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 116 None\n",
      "genome position 61724956_61724031 None\n",
      "(0.0,)\n",
      "MHILKGSPNVIPRAHGQKNTRRDGTGLYPMRGPFKNLALLPFSLPLLGGGGSGSGEKVSVSKMAAAWPSGPSAPEAVTARLVGVLWFVSVTTGPWGAVATSAGGEESLKCEDLKVGQDRLLVMLPRLFFPNSWAPVILLPQVLLSNTKSIFVHMLIDLDKLAASIYM\n",
      "strand -\n",
      "modif 61724956;61725306;61723712;61724031;None;None orig ['61724956', '61725423', '61723712', '61724031']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 None\n",
      "genome position 87665833_87667010 None\n",
      "(6.0,)\n",
      "AFATVVTAAEKMDANSATSEEMNEIIHLQCSVY\n",
      "strand +\n",
      "modif 87665753;87665833;87667010;87667449;None;None orig ['87665753', '87665833', '87667010', '87667451']\n",
      "[['6.0']]\n",
      "0 id of coordinate\n",
      "['6.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 None\n",
      "genome position 87665833_87667017 None\n",
      "(32.0,)\n",
      "AFATVVTAAEKMDANSATSEEMNEIIHARFIRAVSELELLGFIKPTKQKTDHVARLTWGGC\n",
      "strand +\n",
      "modif 87665753;87665833;87667017;87667450;None;None/87665753;87665833;87667017;87667453;None;None orig ['87665753', '87665833', '87667017', '87667451']\n",
      "[['32.0']]\n",
      "0 id of coordinate\n",
      "['32.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 33 None\n",
      "genome position 33516644_33519951 None\n",
      "(0.0,)\n",
      "DNGELDNSYLPRIVLLMHRWYLSSTELAEKLLCMYRNATGESCNEFRLKICYF\n",
      "strand +\n",
      "modif 33516543;33516644;33519951;33520009;None;None orig ['33516541', '33516644', '33519951', '33520009']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 33 None\n",
      "genome position 33516644_33519951 None\n",
      "(0.0,)\n",
      "DNGELDNSYLPRIVLLMHRWYLSSTELAEKLLCMYRNATGESCNEFRLKICYFM\n",
      "strand +\n",
      "modif 33516543;33516644;33519951;33520012;None;None orig ['33516541', '33516644', '33519951', '33520014']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 48 None\n",
      "genome position 120451317_120453846 None\n",
      "(0.0,)\n",
      "FCSSDWQEIHFDTEKDKIEDVLQTNIEECLGAVECFEEEDSNSRESLSLSL\n",
      "strand +\n",
      "modif 120451170;120451317;120453846;120453903;None;None orig ['120451170', '120451317', '120453846', '120453905']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 50 None\n",
      "genome position 120451322_120451607 None\n",
      "(0.0,)\n",
      "FCSSDWQEIHFDTEKDKIEDVLQTNIEECLGAVECFEEEDSNSRESLSLAEVLLCHPGCSAVVQSRLATA\n",
      "strand +\n",
      "modif 120451170;120451322;120451607;120452829;None;None/120451170;120451322;120451607;120451722;None;None orig ['120451170', '120451322', '120451607', '120451722']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 50 None\n",
      "genome position 120451322_120453846 None\n",
      "(33.0,)\n",
      "FCSSDWQEIHFDTEKDKIEDVLQTNIEECLGAVECFEEEDSNSRESLSLADLYEEAAENLHQLSDKLPAP\n",
      "strand +\n",
      "modif 120451170;120451322;120453846;120453904;None;None orig ['120451170', '120451322', '120453846', '120453905']\n",
      "[['33.0']]\n",
      "0 id of coordinate\n",
      "['33.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 50 None\n",
      "genome position 120451322_120459214 None\n",
      "(0.0,)\n",
      "FCSSDWQEIHFDTEKDKIEDVLQTNIEECLGAVECFEEEDSNSRESLSLADLDLKLVFLNFV\n",
      "strand +\n",
      "modif 120451170;120451322;120459214;120459347;None;None orig ['120451170', '120451322', '120459214', '120459349']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 38 None\n",
      "genome position 120451722_120453846 None\n",
      "(0.0,)\n",
      "GLAVSPRLQCSGAVTACYSLDLGSSDPPTSASRVAGTTVSMKKLQKICISCQTSFLLL\n",
      "strand +\n",
      "modif 120451607;120451722;120453846;120453905;None;None orig ['120451607', '120451722', '120453846', '120453905']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 120461255_120463686 None\n",
      "(7.0,)\n",
      "VFHYYGPALEFVQMIKLSDLPSCYMSDIEFEFTED\n",
      "strand +\n",
      "modif 120461160;120461255;120463686;120463759;None;None orig ['120461160', '120461255', '120463686', '120463761']\n",
      "[['7.0']]\n",
      "0 id of coordinate\n",
      "['7.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 120461255_120463691 None\n",
      "(40.0,)\n",
      "VFHYYGPALEFVQMIKLSDLPSCYMSDIEFELGLTNSTKQNSVLLLEQISSLCSK\n",
      "strand +\n",
      "modif 120461160;120461255;120463691;120463761;None;None orig ['120461160', '120461255', '120463691', '120463761']\n",
      "[['40.0']]\n",
      "0 id of coordinate\n",
      "['40.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 31 None\n",
      "genome position 120461255_120463693 None\n",
      "(0.0,)\n",
      "VFHYYGPALEFVQMIKLSDLPSCYMSDIEFELIDKQYQTEFCVAVGADFFSV\n",
      "strand +\n",
      "modif 120461160;120461255;120463693;120463760;None;None orig ['120461160', '120461255', '120463693', '120463761']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 26 None\n",
      "genome position 17274185_17275776 None\n",
      "(4.0,)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFTFSLAPPPDLEGLFSL\n",
      "strand +\n",
      "modif 17274106;17274185;17275776;17275823;None;None orig ['17274106', '17274185', '17275776', '17275825']\n",
      "[['4.0']]\n",
      "0 id of coordinate\n",
      "['4.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 None\n",
      "genome position 17274185_17275788 None\n",
      "(4.0,)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFTPPPDLEGLFSL\n",
      "strand +\n",
      "modif 17274106;17274185;17275788;17275823;None;None orig ['17274106', '17274185', '17275788', '17275825']\n",
      "[['4.0']]\n",
      "0 id of coordinate\n",
      "['4.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 None\n",
      "genome position 17274185_17275800 None\n",
      "(541.0,)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSL\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275823;None;None orig ['17274106', '17274185', '17275800', '17275825']\n",
      "[['541.0']]\n",
      "0 id of coordinate\n",
      "['541.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 35\n",
      "genome position 17274185_17275800 17275825_17276494\n",
      "(541.0, 658.0)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSLIQQKTELPVTE\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275825;17276494;17276525 orig ['17274106', '17274185', '17275800', '17275825', '17276494', '17276525']\n",
      "[['541.0', '658.0']]\n",
      "0 id of coordinate\n",
      "['541.0', '658.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 35\n",
      "genome position 17274185_17275800 17275825_17276494\n",
      "(541.0, 658.0)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSLIQQKTELPVTENVQTIPPP\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275825;17276494;17276549 orig ['17274106', '17274185', '17275800', '17275825', '17276494', '17276550']\n",
      "[['541.0', '658.0']]\n",
      "0 id of coordinate\n",
      "['541.0', '658.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 35\n",
      "genome position 17274185_17275800 17275825_17276494\n",
      "(541.0, 658.0)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSLIQQKTELPVTENVQTIPPPYVVRTILVYSRPPCQPQ\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275825;17276494;17276600 orig ['17274106', '17274185', '17275800', '17275825', '17276494', '17276602']\n",
      "[['541.0', '658.0']]\n",
      "0 id of coordinate\n",
      "['541.0', '658.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 35\n",
      "genome position 17274185_17275800 17275825_17276494\n",
      "(541.0, 658.0)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSLIQQKTELPVTENVQTIPPPYVVRTILVYSRPPCQPQFSLTEPMK\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275825;17276494;17276624 orig ['17274106', '17274185', '17275800', '17275825', '17276494', '17276624']\n",
      "[['541.0', '658.0']]\n",
      "0 id of coordinate\n",
      "['541.0', '658.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 26 35\n",
      "genome position 17274185_17275800 17275825_17276494\n",
      "(541.0, 658.0)\n",
      "LSGLTSDPRELCSCLYDLETASCSTFNLEGLFSLIQQKTELPVTENVQTIPPPYVVRTILVYSRPPCQPQFSLTEPMKVGEAWEREGCLQP\n",
      "strand +\n",
      "modif 17274106;17274185;17275800;17275825;17276494;17276909 orig ['17274106', '17274185', '17275800', '17275825', '17276494', '17276909']\n",
      "[['541.0', '658.0']]\n",
      "0 id of coordinate\n",
      "['541.0', '658.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 45 None\n",
      "genome position 82228190_82227470 None\n",
      "(0.0,)\n",
      "DKDKIDKIVTSLGLKIGAREARHSDPKVQINAICSQWLPISHAVLETQALKA\n",
      "strand -\n",
      "modif 82228190;82228326;82227450;82227470;None;None orig ['82228190', '82228327', '82227449', '82227470']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 45 None\n",
      "genome position 82228190_82227530 None\n",
      "(0.0,)\n",
      "DKDKIDKIVTSLGLKIGAREARHSDPKVQINAICSQWLPISHAVLAERVERLMCTGSQTFDSFPPETQALKA\n",
      "strand -\n",
      "modif 82228190;82228326;82227450;82227530;None;None orig ['82228190', '82228327', '82227449', '82227530']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 45 None\n",
      "genome position 82228190_82227572 None\n",
      "(64.0,)\n",
      "DKDKIDKIVTSLGLKIGAREARHSDPKVQINAICSQWLPISHAVLAMVCQKLPSPLDITAERVERLMCTGSQTFDSFPPETQALKA\n",
      "strand -\n",
      "modif 82228190;82228326;82227450;82227572;None;None orig ['82228190', '82228327', '82227449', '82227572']\n",
      "[['64.0']]\n",
      "0 id of coordinate\n",
      "['64.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 45 None\n",
      "genome position 82228190_82227579 None\n",
      "(0.0,)\n",
      "DKDKIDKIVTSLGLKIGAREARHSDPKVQINAICSQWLPISHAVLDFSYGVSETS\n",
      "strand -\n",
      "modif 82228190;82228326;82227451;82227579;None;None orig ['82228190', '82228327', '82227449', '82227579']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 45 None\n",
      "genome position 82228190_82227572 None\n",
      "(64.0,)\n",
      "DKDKIDKIVTSLGLKIGAREARHSDPKVQINAICSQWLPISHAVLAMVCQKLPSPLDITAERVERLMCTGSQTFDSFPPET\n",
      "strand -\n",
      "modif 82228190;82228326;82227465;82227572;None;None orig ['82228190', '82228327', '82227465', '82227572']\n",
      "[['64.0']]\n",
      "0 id of coordinate\n",
      "['64.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "new\n",
      "aa position 88 None\n",
      "genome position 154618952_154618273 None\n",
      "(0.0,)\n",
      "MQAEGRGTGGSTGDADGPGGPGIPDGPGGNAGGPGEAGATGGRGPRGAGAARASGPGGGAPRGPHGGAASGLNGCCRCGARGPESRLLDYLAMPFATPMEAELARRSLAQDAPPLPVPGVLLKEFTVSGNILT\n",
      "strand +/-\n",
      "modif 154618952;154619218;154618140;154618273;None;None/154585206;154585472;154586151;154586284;None;None orig ['154618952', '154619271', '154618138', '154618273']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "aa position 88 None\n",
      "genome position 154585206_154586284 None\n",
      "(0.0,)\n",
      "MQAEGRGTGGSTGDADGPGGPGIPDGPGGNAGGPGEAGATGGRGPRGAGAARASGPGGGAPRGPHGGAASGLNGCCRCGARGPESRLLDYLAMPFATPMEAELARRSLAQDAPPLPVPGVLLKEFTVSGNILT\n",
      "strand +/-\n",
      "modif 154618952;154619218;154618140;154618273;None;None/154585206;154585472;154586151;154586284;None;None orig ['154618952', '154619271', '154618138', '154618273']\n",
      "[['0.0']]\n",
      "0 id of coordinate\n",
      "['0.0'] expression value\n",
      "\n",
      "\n",
      "new\n",
      "aa position 89 None\n",
      "genome position 154585206_154586284 None\n",
      "(1.0,)/(15.0,)\n",
      "MQAEGRGTGGSTGDADGPGGPGIPDGPGGNAGGPGEAGATGGRGPRGAGAARASGPGGGAPRGPHGGAASGLNGCCRCGARGPESRLLEFYLAMPFATPMEAELARRSLAQDAPPLPVPGVLLKEFTVSGNILT\n",
      "strand +/-\n",
      "modif 154585206;154585475;154586151;154586284;None;None/154618949;154619218;154618140;154618273;None;None orig ['154618949', '154619271', '154618138', '154618273']\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-47-107bd70efd2b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m    114\u001b[0m                         kmer_file_idx[kmer]), peptide_junctions) \n\u001b[1;32m    115\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 116\u001b[0;31m                 expression_dict(pep_seq, id_sample, n_samples, \n\u001b[0m\u001b[1;32m    117\u001b[0m                                 \u001b[0mjunctionExpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome_junction_pos\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    118\u001b[0m                                 \u001b[0mgenome_junction_pos1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mid_modi_coord\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-45-71d282fa4e62>\u001b[0m in \u001b[0;36mexpression_dict\u001b[0;34m(pep_seq, id_sample, n_samples, junctionExpr, genome_junction_pos, genome_junction_pos1, id_modi_coord, pep_orig_coord)\u001b[0m\n\u001b[1;32m     30\u001b[0m     \u001b[0mexpr_val\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mexpression\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mid_modi_coord\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m     \u001b[0;31m# Update 2-exon case\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 32\u001b[0;31m     update_expression_dict(genome_junction_pos, expr_val[0], \n\u001b[0m\u001b[1;32m     33\u001b[0m                            n_samples, id_sample, pep_seq)\n\u001b[1;32m     34\u001b[0m     \u001b[0;31m# Update 3-exon case\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-24-bbb426420050>\u001b[0m in \u001b[0;36mupdate_expression_dict\u001b[0;34m(genome_junction_pos, expr_val, n_samples, id_sample, pep_seq)\u001b[0m\n\u001b[1;32m      2\u001b[0m          \u001b[0;31m# Check that each junction position in one sample has one expression value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mgenome_junction_pos\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mjx_to_expression\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m             \u001b[0;32massert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjx_to_expression\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgenome_junction_pos\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mid_sample\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexpr_val\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m         \u001b[0;31m# Update the expression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m         jx_to_expression.setdefault(genome_junction_pos, \n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "jx_to_expression = dict()\n",
    "jx_to_peptides = defaultdict(list)\n",
    "n_samples = len(samples['sample'])\n",
    "\n",
    "for id_sample, sample in enumerate(samples['sample']):\n",
    "    print(sample)\n",
    "\n",
    "    file_meta = os.path.join(experiment_folder,\n",
    "                             f'{sample}_mutNone/ref_sample_peptides_meta.mig_filtered.pq')\n",
    "\n",
    "    file_save = os.path.join(experiment_folder,\n",
    "                             f'{sample}_mutNone/ref_sample_peptides_meta.mig_filtered.fasta')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    if file_save_experiement is not None: \n",
    "        ep = open(file_save_experiement, 'w') # Experiment file to save\n",
    "        write_peptide_to_experiment(ep, header=True)\n",
    "    with open(file_save, 'w') as sp: # Fasta file to save\n",
    "        fp = pd.read_parquet(file_meta) # metadata from immunopepper as parquet\n",
    "        print(\"open {}\".format(file_meta))\n",
    "        ### Iterate over peptides \n",
    "        for j, line in fp.iterrows():\n",
    "            print('new')\n",
    "            peptide_junctions = set() #each line is one peptide / peptides matching a kmer\n",
    "                                      #each peptide contains at most one kmer\n",
    "\n",
    "            # --- TEST ---\n",
    "            if subset_run_test and (test_query in line['peptide']):\n",
    "                print(f'\\n METADATA (ImmunoPepper derived): \\n {line}')\n",
    "                break\n",
    "            # --- END TEST ---\n",
    "\n",
    "\n",
    "            pep, pep_seq, pep_gene, pep_orig_coordS, pep_modif_coordS, \\\n",
    "            strand, has_stop_codon, readFrameAnnotated, \\\n",
    "            junctionAnnotated, kmer_type, is_isolated, \\\n",
    "            junctionExpr = extract_peptide_fields_pq(line)\n",
    "        \n",
    "            # some peptides do not contain a junction\n",
    "            if int(is_isolated):\n",
    "                continue \n",
    "\n",
    "            kmer = None\n",
    "\n",
    "            # Iterate over original genomic coordinates : get include 3' or 5' flag\n",
    "            for pep_orig_coord in pep_orig_coordS:\n",
    "                pep_orig_coord = split_coord(pep_orig_coord)\n",
    "                start_cds = [ first_exon[0] for first_exon in \n",
    "                             gene_cds_begin_dict_bis[pep_gene] ] \n",
    "                end_cds = [ last_exon[1] for last_exon in \n",
    "                           gene_cds_end_dict_bis[pep_gene] ] \n",
    "                pep_start, pep_end = extract_end_starts(pep_orig_coord)\n",
    "                pep_5include, pep_3include = get_include_flag(start_cds, end_cds, \n",
    "                                                              pep_start, pep_end, \n",
    "                                                              has_stop_codon )\n",
    "\n",
    "            # Iterate over modified genomic coordinates: get \n",
    "            # \"junction position\" flag and \"between codon\" flag\n",
    "            #\"and genomic coordinates\"\n",
    "            for id_modi_coord, pep_modi_coord in enumerate(pep_modif_coordS):\n",
    "                pep_modi_coord = split_coord(pep_modi_coord)\n",
    "                if len(pep_modi_coord) <=2:\n",
    "                    continue\n",
    "                jx_list, shift_list, \\\n",
    "                jx_list_ori = get_nt_len_with_aa_shift(pep_modi_coord)\n",
    "\n",
    "                aa_junction_pos, between_codons, \\\n",
    "                aa_junction_pos1, between_codons1, \\\n",
    "                aa_junction_pos1_from_start = get_aaPos_betweenFlag(shift_list,\n",
    "                                                                          jx_list)\n",
    "\n",
    "                genome_junction_pos, \\\n",
    "                genome_junction_pos1 = get_genomic_coordinates(pep_modi_coord)\n",
    "\n",
    "                # --- TEST ---\n",
    "                if subset_run_test and (test_query not in pep_seq): \n",
    "                    continue\n",
    "                # --- END TEST ---\n",
    "\n",
    "                print_stats(print_, kmer, pep_seq, strand, pep_orig_coord, \n",
    "                            pep_modi_coord, jx_list, jx_list_ori, genome_junction_pos,\n",
    "                            genome_junction_pos1, aa_junction_pos, aa_junction_pos1, \n",
    "                            between_codons, between_codons1)\n",
    "\n",
    "                jx_pep1, jx_pep2 = cut_peptides(jx_list, between_codons,\n",
    "                                                between_codons1, aa_junction_pos, \n",
    "                                                aa_junction_pos1, \n",
    "                                                aa_junction_pos1_from_start, \n",
    "                                                print_=print_)\n",
    "\n",
    "                # skip duplicates\n",
    "                if (genome_junction_pos in peptide_junctions) and \\\n",
    "                   (genome_junction_pos1 in peptide_junctions):\n",
    "                    continue\n",
    "                else:\n",
    "                    peptide_junctions.add(genome_junction_pos)\n",
    "                    peptide_junctions.add(genome_junction_pos1)\n",
    "                print('aa position', aa_junction_pos, aa_junction_pos1_from_start)\n",
    "                print('genome position', genome_junction_pos, genome_junction_pos1 )\n",
    "                # write fasta file \n",
    "                pep_idx = write_fasta_option(sp, pep_idx, aa_junction_pos,\n",
    "                            aa_junction_pos1_from_start, between_codons, \n",
    "                            between_codons1, pep_5include, pep_3include, pep_gene, \n",
    "                            genome_junction_pos, genome_junction_pos1, \n",
    "                            kmer, jx_pep1, jx_pep2, \n",
    "                            readFrameAnnotated, \\\n",
    "                            junctionAnnotated, \\\n",
    "                            kmer_type, \\\n",
    "                            do_write=write_)\n",
    "                if file_save_experiement is not None: \n",
    "                    write_peptide_to_experiment(ep, pep_idx, pep_seq,';'.join(\n",
    "                        kmer_file_idx[kmer]), peptide_junctions) \n",
    "\n",
    "                expression_dict(pep_seq, id_sample, n_samples, \n",
    "                                junctionExpr, genome_junction_pos, \n",
    "                                genome_junction_pos1, id_modi_coord, \n",
    "                                pep_orig_coord)\n",
    "    if write_:\n",
    "        print('written to:')\n",
    "        print(file_save_experiement)\n",
    "        print(file_save)\n",
    "        if file_save_experiement is not None: \n",
    "            ep.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:26:45.259644Z",
     "start_time": "2022-06-24T14:26:45.254845Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33743"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "154618949 - 154585206"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:19:37.362307Z",
     "start_time": "2022-06-24T14:19:37.325943Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'MQAEGRGTGGSTGDADGPGGPGIPDGPGGNAGGPGEAGATGGRGPRGAGAARASGPGGGAPRGPHGGAASGLNGCCRCGARGPESRLLDYLAMPFATPMEAELARRSLAQDAPPLPVPGVLLK'\\\n",
    "== 'MQAEGRGTGGSTGDADGPGGPGIPDGPGGNAGGPGEAGATGGRGPRGAGAARASGPGGGAPRGPHGGAASGLNGCCRCGARGPESRLLEFYLAMPFATPMEAELARRSLAQDAPPLPVPGVLLKEFTVSGNILT'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T14:14:57.141629Z",
     "start_time": "2022-06-24T14:14:56.987Z"
    }
   },
   "outputs": [],
   "source": [
    "defaultdict([1,1,1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:16:01.697471Z",
     "start_time": "2022-06-24T13:16:01.694354Z"
    }
   },
   "outputs": [],
   "source": [
    "foo = dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:16:18.791687Z",
     "start_time": "2022-06-24T13:16:18.788812Z"
    }
   },
   "outputs": [],
   "source": [
    "foo.setdefault(123, np.zeros(32))[5] = 22"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:16:08.951111Z",
     "start_time": "2022-06-24T13:16:08.948983Z"
    }
   },
   "outputs": [],
   "source": [
    "foo.setdefault(123, np.zeros(32))[10] = 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:16:20.881225Z",
     "start_time": "2022-06-24T13:16:20.876952Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{123: array([ 0.,  0.,  0.,  0.,  0., 22.,  0.,  0.,  0.,  0.,  8.,  0.,  0.,\n",
       "         0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,\n",
       "         0.,  0.,  0.,  0.,  0.,  0.])}"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "foo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T13:07:39.884657Z",
     "start_time": "2022-06-24T13:07:39.881422Z"
    }
   },
   "outputs": [],
   "source": [
    "if genome_junction_pos1: \n",
    "    print('t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T12:47:18.969383Z",
     "start_time": "2022-06-24T12:47:18.820924Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-269"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "154618949 - 154619218"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T12:47:35.100240Z",
     "start_time": "2022-06-24T12:47:35.093275Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-169"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "154618104 - 154618273"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T12:47:51.024279Z",
     "start_time": "2022-06-24T12:47:51.017784Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-269"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "154585206 - 154585475"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T12:48:02.009723Z",
     "start_time": "2022-06-24T12:48:02.004769Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-169"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "154586151 - 154586320"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T09:50:29.147316Z",
     "start_time": "2022-06-24T09:50:29.145044Z"
    }
   },
   "outputs": [],
   "source": [
    "test = '(1,1)/(2,0)'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-24T09:50:29.157885Z",
     "start_time": "2022-06-24T09:50:29.148908Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1., 1.],\n",
       "       [2., 0.]])"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#'(1,1)/(2,0)' each tuple represents an expression value instance.\n",
    "#- First entry of the tuple is first junction expression value and\n",
    "#- the second entry is the second junction expressiion value \n",
    "np.array([pair.split(',') for pair in \n",
    " test.replace(',)', '').replace('(', '').replace(')', '').replace(' ', '').split('/')],\n",
    "         dtype = float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myimmuno3_light",
   "language": "python",
   "name": "myimmuno3_light"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
