{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "73245bef",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "001d1db7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_junction_coordinates(df, coordinates_col, sep=':', shift_to_1based=True):\n",
    "    df['strand'] = None\n",
    "    df['junction_coordinate'] = None\n",
    "\n",
    "    for idx, row in df.iterrows():\n",
    "        kmer_coordinates = [int(x) for x in row[coordinates_col].split(sep) if x !='None']\n",
    "\n",
    "        if kmer_coordinates[1] < kmer_coordinates[2]: # order strand +\n",
    "\n",
    "            df.loc[idx, 'strand'] = '+'\n",
    "            if len(kmer_coordinates) == 4:  # 2 exons\n",
    "                df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in kmer_coordinates[1:3]])\n",
    "            elif len(kmer_coordinates) == 6:\n",
    "                df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in kmer_coordinates[1:5]])\n",
    "        else: # order strand +\n",
    "            df.loc[idx, 'strand'] = '-'\n",
    "            if len(kmer_coordinates) == 4:  # 2 exons\n",
    "                df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in [kmer_coordinates[3],\n",
    "                                                                                kmer_coordinates[0]]])\n",
    "            elif len(kmer_coordinates) == 6:\n",
    "                df.loc[idx, 'junction_coordinate'] = ':'.join([str(x) for x in [kmer_coordinates[3],\n",
    "                                                                                kmer_coordinates[0],\n",
    "                                                                                kmer_coordinates[2],\n",
    "                                                                                kmer_coordinates[5]\n",
    "                                                                               ]])\n",
    "    return df\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37f13094",
   "metadata": {},
   "source": [
    "### Load ETH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "039b7250",
   "metadata": {},
   "outputs": [],
   "source": [
    "batch = 'tmp_out_ref_batch_601'\n",
    "kmer_files = glob.glob(f'/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/cohort_mutNone/{batch}/ref_graph_kmer_JuncExpr/part*')\n",
    "kmer_df = pd.concat([ pd.read_csv(kmer_file, sep = '\\t') for kmer_file in kmer_files], axis = 0) \n",
    "protein_meta = pd.read_csv(f'/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/cohort_mutNone/{batch}/ref_sample_peptides_meta.gz', sep = '\\t')\n",
    "\n",
    "kmer_df = get_junction_coordinates(kmer_df, 'coord')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "7e55f514",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>TCGA3CAAAU01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALI01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALJ01A31RA41B07all</th>\n",
       "      <th>TCGA3CAALK01A11RA41B07all</th>\n",
       "      <th>TCGA4HAAAK01A12RA41B07all</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGAV7A7HQ01A11RA33J07all</th>\n",
       "      <th>TCGAW8A86G01A21RA36F07all</th>\n",
       "      <th>TCGAWTAB4101A11RA41B07all</th>\n",
       "      <th>TCGAWTAB4401A11RA41B07all</th>\n",
       "      <th>TCGAXXA89901A11RA36F07all</th>\n",
       "      <th>TCGAXXA89A01A11RA36F07all</th>\n",
       "      <th>TCGAZ7A8R501A42RA41B07all</th>\n",
       "      <th>TCGAZ7A8R601A11RA41B07all</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VFHICDRVS</td>\n",
       "      <td>17267946:17267962:17269071:17269082:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>17267962:17269071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PSCSRGLLM</td>\n",
       "      <td>17276852:17276876:17280349:17280352:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>17276876:17280349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSRMRWSWA</td>\n",
       "      <td>17271170:17271178:17273339:17273358:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>17271178:17273339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DFPYKRILV</td>\n",
       "      <td>17273423:17273447:17276518:17276521:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>17273447:17276518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SWSSGRRVT</td>\n",
       "      <td>17273442:17273447:17276518:17276540:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>17273447:17276518</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 1109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer                                          coord  isCrossJunction  \\\n",
       "0  VFHICDRVS  17267946:17267962:17269071:17269082:None:None             True   \n",
       "1  PSCSRGLLM  17276852:17276876:17280349:17280352:None:None             True   \n",
       "2  GSRMRWSWA  17271170:17271178:17273339:17273358:None:None             True   \n",
       "3  DFPYKRILV  17273423:17273447:17276518:17276521:None:None             True   \n",
       "4  SWSSGRRVT  17273442:17273447:17276518:17276540:None:None             True   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  TCGA3CAAAU01A11RA41B07all  \\\n",
       "0              False               False                        0.0   \n",
       "1               True               False                        0.0   \n",
       "2               True               False                        0.0   \n",
       "3              False               False                        0.0   \n",
       "4              False               False                        0.0   \n",
       "\n",
       "   TCGA3CAALI01A11RA41B07all  TCGA3CAALJ01A31RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        6.0                        0.0   \n",
       "2                        8.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGA3CAALK01A11RA41B07all  TCGA4HAAAK01A12RA41B07all  ...  \\\n",
       "0                        0.0                        0.0  ...   \n",
       "1                        0.0                        0.0  ...   \n",
       "2                        0.0                        0.0  ...   \n",
       "3                        0.0                        0.0  ...   \n",
       "4                        0.0                        0.0  ...   \n",
       "\n",
       "   TCGAV7A7HQ01A11RA33J07all  TCGAW8A86G01A21RA36F07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAWTAB4101A11RA41B07all  TCGAWTAB4401A11RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAXXA89901A11RA36F07all  TCGAXXA89A01A11RA36F07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        6.0                        1.0   \n",
       "2                        1.0                        4.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAZ7A8R501A42RA41B07all  TCGAZ7A8R601A11RA41B07all  strand  \\\n",
       "0                        0.0                        0.0       +   \n",
       "1                        0.0                        0.0       +   \n",
       "2                        0.0                        1.0       +   \n",
       "3                        0.0                        0.0       +   \n",
       "4                        0.0                        0.0       +   \n",
       "\n",
       "   junction_coordinate  \n",
       "0    17267962:17269071  \n",
       "1    17276876:17280349  \n",
       "2    17271178:17273339  \n",
       "3    17273447:17276518  \n",
       "4    17273447:17276518  \n",
       "\n",
       "[5 rows x 1109 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmer_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "cfe7dc6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peptide</th>\n",
       "      <th>id</th>\n",
       "      <th>readFrame</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>geneName</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>mutationMode</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>hasStopCodon</th>\n",
       "      <th>isInJunctionList</th>\n",
       "      <th>isIsolated</th>\n",
       "      <th>variantComb</th>\n",
       "      <th>variantSegExpr</th>\n",
       "      <th>modifiedExonsCoord</th>\n",
       "      <th>originalExonsCoord</th>\n",
       "      <th>vertexIdx</th>\n",
       "      <th>junctionExpr</th>\n",
       "      <th>segmentExpr</th>\n",
       "      <th>kmerType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIASLQQQVDFQETQLRKINTENETLQKELRERRQQLQAMTDKFSN...</td>\n",
       "      <td>ENSG00000162592.10:15_16:0:3763707:2-exons</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000162592.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>ref</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3763707;3763836;3766534;3766612</td>\n",
       "      <td>3763705;3763836;3766534;3766612</td>\n",
       "      <td>15;16</td>\n",
       "      <td>2-exons</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>QVSELERKLTKRDCVISELDTKVSQLQEQVELDQNHLQRWKQLQED...</td>\n",
       "      <td>ENSG00000162592.10:18_19:0:3767232:2-exons</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000162592.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>ref</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3767232;3767445;3769782;3769887</td>\n",
       "      <td>3767232;3767445;3769782;3769887</td>\n",
       "      <td>18;19</td>\n",
       "      <td>2-exons</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LERLRNKIIQATFSISGTKSLANEISDNDILEALQRIISERSDYYN...</td>\n",
       "      <td>ENSG00000162592.10:19_20:0:3769782:2-exons</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000162592.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>ref</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3769782;3769887;3771400;3771643</td>\n",
       "      <td>3769782;3769887;3771400;3771645</td>\n",
       "      <td>19;20</td>\n",
       "      <td>2-exons</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             peptide  \\\n",
       "0  VIASLQQQVDFQETQLRKINTENETLQKELRERRQQLQAMTDKFSN...   \n",
       "1  QVSELERKLTKRDCVISELDTKVSQLQEQVELDQNHLQRWKQLQED...   \n",
       "2  LERLRNKIIQATFSISGTKSLANEISDNDILEALQRIISERSDYYN...   \n",
       "\n",
       "                                           id  readFrame  readFrameAnnotated  \\\n",
       "0  ENSG00000162592.10:15_16:0:3763707:2-exons          0               False   \n",
       "1  ENSG00000162592.10:18_19:0:3767232:2-exons          0               False   \n",
       "2  ENSG00000162592.10:19_20:0:3769782:2-exons          0               False   \n",
       "\n",
       "             geneName geneChr geneStrand mutationMode  junctionAnnotated  \\\n",
       "0  ENSG00000162592.10    chr1          +          ref                  0   \n",
       "1  ENSG00000162592.10    chr1          +          ref                  0   \n",
       "2  ENSG00000162592.10    chr1          +          ref                  1   \n",
       "\n",
       "   hasStopCodon  isInJunctionList  isIsolated  variantComb  \\\n",
       "0           NaN                 0         NaN          NaN   \n",
       "1           NaN                 0         NaN          NaN   \n",
       "2           NaN                 0         NaN          NaN   \n",
       "\n",
       "                    variantSegExpr               modifiedExonsCoord  \\\n",
       "0  3763707;3763836;3766534;3766612  3763705;3763836;3766534;3766612   \n",
       "1  3767232;3767445;3769782;3769887  3767232;3767445;3769782;3769887   \n",
       "2  3769782;3769887;3771400;3771643  3769782;3769887;3771400;3771645   \n",
       "\n",
       "  originalExonsCoord vertexIdx  junctionExpr  segmentExpr  kmerType  \n",
       "0              15;16   2-exons           NaN          NaN       NaN  \n",
       "1              18;19   2-exons           NaN          NaN       NaN  \n",
       "2              19;20   2-exons           NaN          NaN       NaN  "
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_meta.head(3)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "673f93c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_meta = get_junction_coordinates(protein_meta, 'modifiedExonsCoord', sep=';')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d9ca411",
   "metadata": {},
   "source": [
    "### Load OHSU"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "d1ba0f2b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3139: DtypeWarning: Columns (57,58,60,62,63,64,65,66,67,68,70,72) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "ohsu_df = pd.read_csv('/cluster/work/grlab/projects/projects2020_OHSU/share_OHUS_PNLL/OHSU_Feb2023_complete-annotated-shortlist.tsv.gz', sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "dc86715f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['jx', 'TCGA-24-1431-01A-01R-1566-13', 'TCGA-24-2298-01A-01R-1569-13',\n",
       "       'TCGA-25-1313-01A-01R-1565-13', 'TCGA-25-1319-01A-01R-1565-13',\n",
       "       'TCGA-61-2008-01A-02R-1568-13', 'TCGA-A2-A0D2-01A-21R-A034-07',\n",
       "       'TCGA-A2-A0SX-01A-12R-A084-07', 'TCGA-AO-A0JM-01A-21R-A056-07',\n",
       "       'TCGA-BH-A18V-01A-11R-A12D-07', 'TCGA-C8-A12P-01A-11R-A115-07',\n",
       "       'GTEx_brain_testis_over0', 'GTEx_brain_testis_over1',\n",
       "       'GTEx_brain_testis_over10', 'GTEx_brain_testis_over2',\n",
       "       'GTEx_brain_testis_over3', 'GTEx_brain_testis_over5',\n",
       "       'GTEx_breast_over0', 'GTEx_breast_over1', 'GTEx_breast_over10',\n",
       "       'GTEx_breast_over2', 'GTEx_breast_over3', 'GTEx_breast_over5',\n",
       "       'GTEx_core_over0', 'GTEx_core_over1', 'GTEx_core_over10',\n",
       "       'GTEx_core_over2', 'GTEx_core_over3', 'GTEx_core_over5',\n",
       "       'GTEx_ovary_over0', 'GTEx_ovary_over1', 'GTEx_ovary_over10',\n",
       "       'GTEx_ovary_over2', 'GTEx_ovary_over3', 'GTEx_ovary_over5',\n",
       "       'all_TCGA_over0', 'all_TCGA_over1', 'all_TCGA_over10', 'all_TCGA_over2',\n",
       "       'all_TCGA_over3', 'all_TCGA_over5', 'BRCA_over_5', 'OV_over_2',\n",
       "       'OV_over_0', 'BRCA_over_3', 'BRCA_over_0', 'OV_over_10', 'BRCA_over_2',\n",
       "       'OV_over_3', 'BRCA_over_10', 'OV_over_1', 'OV_over_5', 'BRCA_over_1',\n",
       "       'motif', 'annotation', '5';3'_coding_regions', 'has_cds', 'gene',\n",
       "       'gene_id', 'upstream_exon', 'upstream_txs', 'downstream_exon',\n",
       "       'downstream_txs', 'modified_upstream_txs',\n",
       "       'in-frame_all-transcript_biexons', 'in-frame_nonhanging-tx_biexons',\n",
       "       'in-frame_peptide_sequence', 'hanging_txs_included_inframe_pepseqs',\n",
       "       'prefiltered_in-frame_epitopes', 'prefiltered_inframe_epitope_count',\n",
       "       'in-frame_neoepitopes', 'in-frame_neoepitope_count',\n",
       "       'frame-agnostic_all-transcript_biexons'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b3e14cd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ohsu_df['in-frame_neoepitopes'] = ohsu_df['in-frame_neoepitopes'].str.split(';')\n",
    "ohsu_df = ohsu_df.explode(\"in-frame_neoepitopes\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3fd16bfc",
   "metadata": {},
   "source": [
    "### Compare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "821286b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'INPRQATNP' #'NMASWGRAT' in batch 101 -   #HTSDNQCQE, NQTTSPAPF , QTTSPAPFV in batch 100 +\n",
    "search = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5993eb7e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "451536c6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "c4f43ad8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if search:\n",
    "    for kmer in kmer_df['kmer']:\n",
    "        foo = ohsu_df.loc[ohsu_df['in-frame_neoepitopes'] == kmer]\n",
    "        if foo.shape[0] and kmer_df.loc[kmer_df['kmer'] == kmer, 'junctionAnnotated'].values[0]:\n",
    "            display(foo)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "68530ada",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [jx, TCGA-24-1431-01A-01R-1566-13, TCGA-24-2298-01A-01R-1569-13, TCGA-25-1313-01A-01R-1565-13, TCGA-25-1319-01A-01R-1565-13, TCGA-61-2008-01A-02R-1568-13, TCGA-A2-A0D2-01A-21R-A034-07, TCGA-A2-A0SX-01A-12R-A084-07, TCGA-AO-A0JM-01A-21R-A056-07, TCGA-BH-A18V-01A-11R-A12D-07, TCGA-C8-A12P-01A-11R-A115-07, GTEx_brain_testis_over0, GTEx_brain_testis_over1, GTEx_brain_testis_over10, GTEx_brain_testis_over2, GTEx_brain_testis_over3, GTEx_brain_testis_over5, GTEx_breast_over0, GTEx_breast_over1, GTEx_breast_over10, GTEx_breast_over2, GTEx_breast_over3, GTEx_breast_over5, GTEx_core_over0, GTEx_core_over1, GTEx_core_over10, GTEx_core_over2, GTEx_core_over3, GTEx_core_over5, GTEx_ovary_over0, GTEx_ovary_over1, GTEx_ovary_over10, GTEx_ovary_over2, GTEx_ovary_over3, GTEx_ovary_over5, all_TCGA_over0, all_TCGA_over1, all_TCGA_over10, all_TCGA_over2, all_TCGA_over3, all_TCGA_over5, BRCA_over_5, OV_over_2, OV_over_0, BRCA_over_3, BRCA_over_0, OV_over_10, BRCA_over_2, OV_over_3, BRCA_over_10, OV_over_1, OV_over_5, BRCA_over_1, motif, annotation, 5';3'_coding_regions, has_cds, gene, gene_id, upstream_exon, upstream_txs, downstream_exon, downstream_txs, modified_upstream_txs, in-frame_all-transcript_biexons, in-frame_nonhanging-tx_biexons, in-frame_peptide_sequence, hanging_txs_included_inframe_pepseqs, prefiltered_in-frame_epitopes, prefiltered_inframe_epitope_count, in-frame_neoepitopes, in-frame_neoepitope_count, frame-agnostic_all-transcript_biexons]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 73 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_df.loc[ohsu_df['in-frame_neoepitopes'] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "e77e25de",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>TCGA3CAAAU01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALI01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALJ01A31RA41B07all</th>\n",
       "      <th>TCGA3CAALK01A11RA41B07all</th>\n",
       "      <th>TCGA4HAAAK01A12RA41B07all</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGAV7A7HQ01A11RA33J07all</th>\n",
       "      <th>TCGAW8A86G01A21RA36F07all</th>\n",
       "      <th>TCGAWTAB4101A11RA41B07all</th>\n",
       "      <th>TCGAWTAB4401A11RA41B07all</th>\n",
       "      <th>TCGAXXA89901A11RA36F07all</th>\n",
       "      <th>TCGAXXA89A01A11RA36F07all</th>\n",
       "      <th>TCGAZ7A8R501A42RA41B07all</th>\n",
       "      <th>TCGAZ7A8R601A11RA41B07all</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>GVSRVKGAK</td>\n",
       "      <td>1337976:1337994:1336506:1336515:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>...</td>\n",
       "      <td>73.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1336515:1337976</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 1109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          kmer                                      coord  isCrossJunction  \\\n",
       "219  GVSRVKGAK  1337976:1337994:1336506:1336515:None:None             True   \n",
       "\n",
       "     junctionAnnotated  readFrameAnnotated  TCGA3CAAAU01A11RA41B07all  \\\n",
       "219               True               False                      200.0   \n",
       "\n",
       "     TCGA3CAALI01A11RA41B07all  TCGA3CAALJ01A31RA41B07all  \\\n",
       "219                       44.0                       73.0   \n",
       "\n",
       "     TCGA3CAALK01A11RA41B07all  TCGA4HAAAK01A12RA41B07all  ...  \\\n",
       "219                      158.0                      124.0  ...   \n",
       "\n",
       "     TCGAV7A7HQ01A11RA33J07all  TCGAW8A86G01A21RA36F07all  \\\n",
       "219                       73.0                      117.0   \n",
       "\n",
       "     TCGAWTAB4101A11RA41B07all  TCGAWTAB4401A11RA41B07all  \\\n",
       "219                       17.0                       31.0   \n",
       "\n",
       "     TCGAXXA89901A11RA36F07all  TCGAXXA89A01A11RA36F07all  \\\n",
       "219                      104.0                       76.0   \n",
       "\n",
       "     TCGAZ7A8R501A42RA41B07all  TCGAZ7A8R601A11RA41B07all  strand  \\\n",
       "219                       58.0                      131.0       -   \n",
       "\n",
       "     junction_coordinate  \n",
       "219      1336515:1337976  \n",
       "\n",
       "[1 rows x 1109 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmer_df.loc[kmer_df['kmer'] == kmer]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60b58263",
   "metadata": {},
   "source": [
    "### Use ETH FULL TABLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "f4421a68",
   "metadata": {},
   "outputs": [],
   "source": [
    "eth_df = pd.read_csv('/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/CANCER_eth/commit_c4dd02c_conf2_Frame_cap0_runs/TCGA_Breast_1102/filtering_intermediate/complete_cancer_candidates_.tsv.gz', sep = '\\t')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "2da58090",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>gtexCohortfilter &gt;0.0</th>\n",
       "      <th>gtexCohortfilter &gt;=1.0</th>\n",
       "      <th>gtexCohortfilter &gt;=2.0</th>\n",
       "      <th>gtexCohortfilter &gt;=3.0</th>\n",
       "      <th>gtexCohortfilter &gt;=5.0</th>\n",
       "      <th>gtexCohortfilter &gt;=10.0</th>\n",
       "      <th>coord</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "      <th>...</th>\n",
       "      <th>cancerCohortfilter &gt;=2.0</th>\n",
       "      <th>cancerCohortfilter &gt;=3.0</th>\n",
       "      <th>cancerCohortfilter &gt;=5.0</th>\n",
       "      <th>cancerCohortfilter &gt;=10.0</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "      <th>isAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>3290.0</td>\n",
       "      <td>1969.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>394.0</td>\n",
       "      <td>89.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>572.0</td>\n",
       "      <td>545.0</td>\n",
       "      <td>243.0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>138</td>\n",
       "      <td>82</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISSQSRVEK</td>\n",
       "      <td>278.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>92379851:92379859:92493866:92493885:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSGDEEKYP</td>\n",
       "      <td>7350.0</td>\n",
       "      <td>4622.0</td>\n",
       "      <td>2653.0</td>\n",
       "      <td>1745.0</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>734.0</td>\n",
       "      <td>92600493:92600508:92611313:92611325:None:None</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>641</td>\n",
       "      <td>520</td>\n",
       "      <td>348</td>\n",
       "      <td>170</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLKMKMFQI</td>\n",
       "      <td>146.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>92379850:92379859:92496416:92496434:None:None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  gtexCohortfilter >0.0  gtexCohortfilter >=1.0  \\\n",
       "0  WYITRSGIA                 3290.0                  1969.0   \n",
       "1  WYITRSGIA                  572.0                   545.0   \n",
       "2  ISSQSRVEK                  278.0                   261.0   \n",
       "3  RSGDEEKYP                 7350.0                  4622.0   \n",
       "4  HLKMKMFQI                  146.0                   136.0   \n",
       "\n",
       "   gtexCohortfilter >=2.0  gtexCohortfilter >=3.0  gtexCohortfilter >=5.0  \\\n",
       "0                   861.0                   394.0                    89.0   \n",
       "1                   243.0                   130.0                    27.0   \n",
       "2                    82.0                    29.0                     7.0   \n",
       "3                  2653.0                  1745.0                  1126.0   \n",
       "4                    42.0                    20.0                     3.0   \n",
       "\n",
       "   gtexCohortfilter >=10.0                                          coord  \\\n",
       "0                      4.0  92347505:92347506:92349915:92349941:None:None   \n",
       "1                      2.0  92347505:92347506:92349915:92349941:None:None   \n",
       "2                      0.0  92379851:92379859:92493866:92493885:None:None   \n",
       "3                    734.0  92600493:92600508:92611313:92611325:None:None   \n",
       "4                      1.0  92379850:92379859:92496416:92496434:None:None   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  ...  cancerCohortfilter >=2.0  \\\n",
       "0              False               False  ...                       138   \n",
       "1              False               False  ...                       138   \n",
       "2              False               False  ...                        10   \n",
       "3               True                True  ...                       641   \n",
       "4              False               False  ...                        16   \n",
       "\n",
       "   cancerCohortfilter >=3.0  cancerCohortfilter >=5.0  \\\n",
       "0                        82                        26   \n",
       "1                        82                        26   \n",
       "2                         2                         0   \n",
       "3                       520                       348   \n",
       "4                         2                         0   \n",
       "\n",
       "   cancerCohortfilter >=10.0  TCGAC8A12P01A11RA11507all  \\\n",
       "0                          5                   0.000000   \n",
       "1                          5                   0.000000   \n",
       "2                          0                   0.000000   \n",
       "3                        170                   2.922641   \n",
       "4                          0                   2.922641   \n",
       "\n",
       "   TCGAAOA0JM01A21RA05607all  TCGABHA18V01A11RA12D07all  \\\n",
       "0                   0.000000                   0.000000   \n",
       "1                   0.000000                   0.000000   \n",
       "2                   0.000000                   2.474321   \n",
       "3                   2.102386                   1.237161   \n",
       "4                   0.000000                   0.000000   \n",
       "\n",
       "   TCGAA2A0D201A21RA03407all  TCGAA2A0SX01A12RA08407all  isAnnotated  \n",
       "0                   4.102634                        0.0          NaN  \n",
       "1                   4.102634                        0.0          NaN  \n",
       "2                   0.000000                        0.0          NaN  \n",
       "3                   0.000000                        0.0          1.0  \n",
       "4                   0.000000                        0.0          NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eth_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "cff89dd4",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [160], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m eth_df \u001b[38;5;241m=\u001b[39m \u001b[43mget_junction_coordinates\u001b[49m\u001b[43m(\u001b[49m\u001b[43meth_df\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mcoord\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn [26], line 10\u001b[0m, in \u001b[0;36mget_junction_coordinates\u001b[0;34m(df, coordinates_col, sep)\u001b[0m\n\u001b[1;32m      6\u001b[0m kmer_coordinates \u001b[38;5;241m=\u001b[39m [\u001b[38;5;28mint\u001b[39m(x) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m row[coordinates_col]\u001b[38;5;241m.\u001b[39msplit(sep) \u001b[38;5;28;01mif\u001b[39;00m x \u001b[38;5;241m!=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNone\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m kmer_coordinates[\u001b[38;5;241m1\u001b[39m] \u001b[38;5;241m<\u001b[39m kmer_coordinates[\u001b[38;5;241m2\u001b[39m]: \u001b[38;5;66;03m# order strand +\u001b[39;00m\n\u001b[0;32m---> 10\u001b[0m     \u001b[43mdf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mloc\u001b[49m\u001b[43m[\u001b[49m\u001b[43midx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mstrand\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m+\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m     11\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(kmer_coordinates) \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m4\u001b[39m:  \u001b[38;5;66;03m# 2 exons\u001b[39;00m\n\u001b[1;32m     12\u001b[0m         df\u001b[38;5;241m.\u001b[39mloc[idx, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mjunction_coordinate\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin([\u001b[38;5;28mstr\u001b[39m(x) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m kmer_coordinates[\u001b[38;5;241m1\u001b[39m:\u001b[38;5;241m3\u001b[39m]])\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:692\u001b[0m, in \u001b[0;36m_LocationIndexer.__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    689\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_has_valid_setitem_indexer(key)\n\u001b[1;32m    691\u001b[0m iloc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mname \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miloc\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39miloc\n\u001b[0;32m--> 692\u001b[0m \u001b[43miloc\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_with_indexer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1635\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_with_indexer\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1632\u001b[0m \u001b[38;5;66;03m# align and set the values\u001b[39;00m\n\u001b[1;32m   1633\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m take_split_path:\n\u001b[1;32m   1634\u001b[0m     \u001b[38;5;66;03m# We have to operate column-wise\u001b[39;00m\n\u001b[0;32m-> 1635\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_with_indexer_split_path\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1636\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1637\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_setitem_single_block(indexer, value, name)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1720\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_with_indexer_split_path\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1716\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1717\u001b[0m \n\u001b[1;32m   1718\u001b[0m     \u001b[38;5;66;03m# scalar value\u001b[39;00m\n\u001b[1;32m   1719\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m loc \u001b[38;5;129;01min\u001b[39;00m ilocs:\n\u001b[0;32m-> 1720\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_single_column\u001b[49m\u001b[43m(\u001b[49m\u001b[43mloc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpi\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1812\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_single_column\u001b[0;34m(self, loc, value, plane_indexer)\u001b[0m\n\u001b[1;32m   1809\u001b[0m     ser \u001b[38;5;241m=\u001b[39m value\n\u001b[1;32m   1810\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1811\u001b[0m     \u001b[38;5;66;03m# set the item, possibly having a dtype change\u001b[39;00m\n\u001b[0;32m-> 1812\u001b[0m     ser \u001b[38;5;241m=\u001b[39m \u001b[43mser\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1813\u001b[0m     ser\u001b[38;5;241m.\u001b[39m_mgr \u001b[38;5;241m=\u001b[39m ser\u001b[38;5;241m.\u001b[39m_mgr\u001b[38;5;241m.\u001b[39msetitem(indexer\u001b[38;5;241m=\u001b[39m(pi,), value\u001b[38;5;241m=\u001b[39mvalue)\n\u001b[1;32m   1814\u001b[0m     ser\u001b[38;5;241m.\u001b[39m_maybe_update_cacher(clear\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/generic.py:5995\u001b[0m, in \u001b[0;36mNDFrame.copy\u001b[0;34m(self, deep)\u001b[0m\n\u001b[1;32m   5889\u001b[0m \u001b[38;5;129m@final\u001b[39m\n\u001b[1;32m   5890\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcopy\u001b[39m(\u001b[38;5;28mself\u001b[39m: FrameOrSeries, deep: bool_t \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m FrameOrSeries:\n\u001b[1;32m   5891\u001b[0m     \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   5892\u001b[0m \u001b[38;5;124;03m    Make a copy of this object's indices and data.\u001b[39;00m\n\u001b[1;32m   5893\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   5993\u001b[0m \u001b[38;5;124;03m    dtype: object\u001b[39;00m\n\u001b[1;32m   5994\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 5995\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_mgr\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdeep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdeep\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   5996\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_clear_item_cache()\n\u001b[1;32m   5997\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_constructor(data)\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcopy\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/managers.py:821\u001b[0m, in \u001b[0;36mBlockManager.copy\u001b[0;34m(self, deep)\u001b[0m\n\u001b[1;32m    818\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    819\u001b[0m     new_axes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maxes)\n\u001b[0;32m--> 821\u001b[0m res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcopy\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdeep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdeep\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    822\u001b[0m res\u001b[38;5;241m.\u001b[39maxes \u001b[38;5;241m=\u001b[39m new_axes\n\u001b[1;32m    823\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/managers.py:427\u001b[0m, in \u001b[0;36mBlockManager.apply\u001b[0;34m(self, f, align_keys, ignore_failures, **kwargs)\u001b[0m\n\u001b[1;32m    425\u001b[0m         applied \u001b[38;5;241m=\u001b[39m b\u001b[38;5;241m.\u001b[39mapply(f, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[1;32m    426\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 427\u001b[0m         applied \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mgetattr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mf\u001b[49m\u001b[43m)\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    428\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mTypeError\u001b[39;00m, \u001b[38;5;167;01mNotImplementedError\u001b[39;00m):\n\u001b[1;32m    429\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m ignore_failures:\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/internals/blocks.py:756\u001b[0m, in \u001b[0;36mBlock.copy\u001b[0;34m(self, deep)\u001b[0m\n\u001b[1;32m    754\u001b[0m values \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvalues\n\u001b[1;32m    755\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m deep:\n\u001b[0;32m--> 756\u001b[0m     values \u001b[38;5;241m=\u001b[39m \u001b[43mvalues\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    757\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmake_block_same_class(values, ndim\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mndim)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "eth_df = get_junction_coordinates(eth_df, 'coord')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "a57d1d04",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/slurm-job.816371/ipykernel_1760357/1607268209.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['strand'] = None\n",
      "/scratch/slurm-job.816371/ipykernel_1760357/1607268209.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['junction_coordinate'] = None\n",
      "/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1720: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_column(loc, value, pi)\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [168], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m foo \u001b[38;5;241m=\u001b[39m eth_df\u001b[38;5;241m.\u001b[39mloc[(eth_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mjunctionAnnotated\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m) ]\n\u001b[0;32m----> 2\u001b[0m foo \u001b[38;5;241m=\u001b[39m \u001b[43mget_junction_coordinates\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfoo\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mcoord\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn [26], line 10\u001b[0m, in \u001b[0;36mget_junction_coordinates\u001b[0;34m(df, coordinates_col, sep)\u001b[0m\n\u001b[1;32m      6\u001b[0m kmer_coordinates \u001b[38;5;241m=\u001b[39m [\u001b[38;5;28mint\u001b[39m(x) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m row[coordinates_col]\u001b[38;5;241m.\u001b[39msplit(sep) \u001b[38;5;28;01mif\u001b[39;00m x \u001b[38;5;241m!=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNone\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m      8\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m kmer_coordinates[\u001b[38;5;241m1\u001b[39m] \u001b[38;5;241m<\u001b[39m kmer_coordinates[\u001b[38;5;241m2\u001b[39m]: \u001b[38;5;66;03m# order strand +\u001b[39;00m\n\u001b[0;32m---> 10\u001b[0m     \u001b[43mdf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mloc\u001b[49m\u001b[43m[\u001b[49m\u001b[43midx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mstrand\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m+\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m     11\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(kmer_coordinates) \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m4\u001b[39m:  \u001b[38;5;66;03m# 2 exons\u001b[39;00m\n\u001b[1;32m     12\u001b[0m         df\u001b[38;5;241m.\u001b[39mloc[idx, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mjunction_coordinate\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin([\u001b[38;5;28mstr\u001b[39m(x) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m kmer_coordinates[\u001b[38;5;241m1\u001b[39m:\u001b[38;5;241m3\u001b[39m]])\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:692\u001b[0m, in \u001b[0;36m_LocationIndexer.__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    689\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_has_valid_setitem_indexer(key)\n\u001b[1;32m    691\u001b[0m iloc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mname \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miloc\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39miloc\n\u001b[0;32m--> 692\u001b[0m \u001b[43miloc\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_with_indexer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1635\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_with_indexer\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1632\u001b[0m \u001b[38;5;66;03m# align and set the values\u001b[39;00m\n\u001b[1;32m   1633\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m take_split_path:\n\u001b[1;32m   1634\u001b[0m     \u001b[38;5;66;03m# We have to operate column-wise\u001b[39;00m\n\u001b[0;32m-> 1635\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_with_indexer_split_path\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindexer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1636\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1637\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_setitem_single_block(indexer, value, name)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1720\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_with_indexer_split_path\u001b[0;34m(self, indexer, value, name)\u001b[0m\n\u001b[1;32m   1716\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1717\u001b[0m \n\u001b[1;32m   1718\u001b[0m     \u001b[38;5;66;03m# scalar value\u001b[39;00m\n\u001b[1;32m   1719\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m loc \u001b[38;5;129;01min\u001b[39;00m ilocs:\n\u001b[0;32m-> 1720\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_setitem_single_column\u001b[49m\u001b[43m(\u001b[49m\u001b[43mloc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpi\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1817\u001b[0m, in \u001b[0;36m_iLocIndexer._setitem_single_column\u001b[0;34m(self, loc, value, plane_indexer)\u001b[0m\n\u001b[1;32m   1814\u001b[0m     ser\u001b[38;5;241m.\u001b[39m_maybe_update_cacher(clear\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m   1816\u001b[0m \u001b[38;5;66;03m# reset the sliced object if unique\u001b[39;00m\n\u001b[0;32m-> 1817\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_iset_item\u001b[49m\u001b[43m(\u001b[49m\u001b[43mloc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mser\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/frame.py:3222\u001b[0m, in \u001b[0;36mDataFrame._iset_item\u001b[0;34m(self, loc, value)\u001b[0m\n\u001b[1;32m   3218\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_ensure_valid_index(value)\n\u001b[1;32m   3220\u001b[0m \u001b[38;5;66;03m# technically _sanitize_column expects a label, not a position,\u001b[39;00m\n\u001b[1;32m   3221\u001b[0m \u001b[38;5;66;03m#  but the behavior is the same as long as we pass broadcast=False\u001b[39;00m\n\u001b[0;32m-> 3222\u001b[0m value \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sanitize_column\u001b[49m\u001b[43m(\u001b[49m\u001b[43mloc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbroadcast\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m   3223\u001b[0m NDFrame\u001b[38;5;241m.\u001b[39m_iset_item(\u001b[38;5;28mself\u001b[39m, loc, value)\n\u001b[1;32m   3225\u001b[0m \u001b[38;5;66;03m# check if we are modifying a copy\u001b[39;00m\n\u001b[1;32m   3226\u001b[0m \u001b[38;5;66;03m# try to set first as we want an invalid\u001b[39;00m\n\u001b[1;32m   3227\u001b[0m \u001b[38;5;66;03m# value exception to occur first\u001b[39;00m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/frame.py:3876\u001b[0m, in \u001b[0;36mDataFrame._sanitize_column\u001b[0;34m(self, key, value, broadcast)\u001b[0m\n\u001b[1;32m   3873\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m value\n\u001b[1;32m   3875\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(value, Series):\n\u001b[0;32m-> 3876\u001b[0m     value \u001b[38;5;241m=\u001b[39m \u001b[43mreindexer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalue\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3878\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(value, DataFrame):\n\u001b[1;32m   3879\u001b[0m     \u001b[38;5;66;03m# align right-hand-side columns if self.columns\u001b[39;00m\n\u001b[1;32m   3880\u001b[0m     \u001b[38;5;66;03m# is multi-index and self[key] is a sub-frame\u001b[39;00m\n\u001b[1;32m   3881\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns, MultiIndex) \u001b[38;5;129;01mand\u001b[39;00m key \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns:\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/frame.py:3857\u001b[0m, in \u001b[0;36mDataFrame._sanitize_column.<locals>.reindexer\u001b[0;34m(value)\u001b[0m\n\u001b[1;32m   3853\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mreindexer\u001b[39m(value):\n\u001b[1;32m   3854\u001b[0m     \u001b[38;5;66;03m# reindex if necessary\u001b[39;00m\n\u001b[1;32m   3856\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m value\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39mequals(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex) \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex):\n\u001b[0;32m-> 3857\u001b[0m         value \u001b[38;5;241m=\u001b[39m \u001b[43mvalue\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_values\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3858\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   3859\u001b[0m \n\u001b[1;32m   3860\u001b[0m         \u001b[38;5;66;03m# GH 4107\u001b[39;00m\n\u001b[1;32m   3861\u001b[0m         \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "foo = eth_df.loc[(eth_df['junctionAnnotated'] == True) ]\n",
    "foo = get_junction_coordinates(foo, 'coord')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "57eed671",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmers_eth = set(foo.loc[foo['strand'] == '-', 'kmer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "150d13af",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmers_ohsu = set(ohsu_df['in-frame_neoepitopes'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "4dee1acd",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = kmers_ohsu.intersection(kmers_eth)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "355e2873",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ed1844c",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "5fa484de",
   "metadata": {},
   "source": [
    "### Use old data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "2fc43621",
   "metadata": {},
   "outputs": [],
   "source": [
    "eth_old = pd.read_parquet('/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/ARCHIV_keep_runs/commit_v3_TEST_merged3_372a147_medium_run_conf2_annotFrame_cap0_runs/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_101/ref_graph_kmer_JuncExpr.pq')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "c1f8925b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>TCGA3CAAAU01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALI01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALJ01A31RA41B07all</th>\n",
       "      <th>TCGA3CAALK01A11RA41B07all</th>\n",
       "      <th>TCGA4HAAAK01A12RA41B07all</th>\n",
       "      <th>TCGA5LAAT001A12RA41B07all</th>\n",
       "      <th>TCGA5LAAT101A12RA41B07all</th>\n",
       "      <th>TCGA5TA9QA01A11RA41B07all</th>\n",
       "      <th>TCGAA1A0SB01A11RA14407all</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGAW8A86G01A21RA36F07all</th>\n",
       "      <th>TCGAWTAB4101A11RA41B07all</th>\n",
       "      <th>TCGAWTAB4401A11RA41B07all</th>\n",
       "      <th>TCGAXXA89901A11RA36F07all</th>\n",
       "      <th>TCGAXXA89A01A11RA36F07all</th>\n",
       "      <th>TCGAZ7A8R501A42RA41B07all</th>\n",
       "      <th>TCGAZ7A8R601A11RA41B07all</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KSMDQDFGW</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMDQDFGWS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MDQDFGWSW</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DQDFGWSWL</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>QDFGWSWLR</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 1106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        kmer  TCGA3CAAAU01A11RA41B07all  TCGA3CAALI01A11RA41B07all  \\\n",
       "0  KSMDQDFGW                        0.0                        0.0   \n",
       "1  SMDQDFGWS                        0.0                        0.0   \n",
       "2  MDQDFGWSW                        0.0                        0.0   \n",
       "3  DQDFGWSWL                        0.0                        0.0   \n",
       "4  QDFGWSWLR                        0.0                        0.0   \n",
       "\n",
       "   TCGA3CAALJ01A31RA41B07all  TCGA3CAALK01A11RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGA4HAAAK01A12RA41B07all  TCGA5LAAT001A12RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGA5LAAT101A12RA41B07all  TCGA5TA9QA01A11RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAA1A0SB01A11RA14407all  ...  TCGAW8A86G01A21RA36F07all  \\\n",
       "0                        0.0  ...                        0.0   \n",
       "1                        0.0  ...                        0.0   \n",
       "2                        0.0  ...                        0.0   \n",
       "3                        0.0  ...                        0.0   \n",
       "4                        0.0  ...                        0.0   \n",
       "\n",
       "   TCGAWTAB4101A11RA41B07all  TCGAWTAB4401A11RA41B07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAXXA89901A11RA36F07all  TCGAXXA89A01A11RA36F07all  \\\n",
       "0                        0.0                        0.0   \n",
       "1                        0.0                        0.0   \n",
       "2                        0.0                        0.0   \n",
       "3                        0.0                        0.0   \n",
       "4                        0.0                        0.0   \n",
       "\n",
       "   TCGAZ7A8R501A42RA41B07all  TCGAZ7A8R601A11RA41B07all  isCrossJunction  \\\n",
       "0                        0.0                        0.0             True   \n",
       "1                        0.0                        0.0             True   \n",
       "2                        0.0                        0.0             True   \n",
       "3                        0.0                        0.0             True   \n",
       "4                        0.0                        0.0             True   \n",
       "\n",
       "   junctionAnnotated  readFrameAnnotated  \n",
       "0                0.0                 1.0  \n",
       "1                0.0                 1.0  \n",
       "2                0.0                 1.0  \n",
       "3                0.0                 1.0  \n",
       "4                0.0                 1.0  \n",
       "\n",
       "[5 rows x 1106 columns]"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eth_old.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "ff20d566",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>TCGA3CAAAU01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALI01A11RA41B07all</th>\n",
       "      <th>TCGA3CAALJ01A31RA41B07all</th>\n",
       "      <th>TCGA3CAALK01A11RA41B07all</th>\n",
       "      <th>TCGA4HAAAK01A12RA41B07all</th>\n",
       "      <th>TCGA5LAAT001A12RA41B07all</th>\n",
       "      <th>TCGA5LAAT101A12RA41B07all</th>\n",
       "      <th>TCGA5TA9QA01A11RA41B07all</th>\n",
       "      <th>TCGAA1A0SB01A11RA14407all</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGAW8A86G01A21RA36F07all</th>\n",
       "      <th>TCGAWTAB4101A11RA41B07all</th>\n",
       "      <th>TCGAWTAB4401A11RA41B07all</th>\n",
       "      <th>TCGAXXA89901A11RA36F07all</th>\n",
       "      <th>TCGAXXA89A01A11RA36F07all</th>\n",
       "      <th>TCGAZ7A8R501A42RA41B07all</th>\n",
       "      <th>TCGAZ7A8R601A11RA41B07all</th>\n",
       "      <th>isCrossJunction</th>\n",
       "      <th>junctionAnnotated</th>\n",
       "      <th>readFrameAnnotated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GWSWLRVLT</td>\n",
       "      <td>198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>...</td>\n",
       "      <td>95.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>PGWSWLRVL</td>\n",
       "      <td>198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>...</td>\n",
       "      <td>95.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>ASTAAWSPG</td>\n",
       "      <td>198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>...</td>\n",
       "      <td>95.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>STAAWSPGW</td>\n",
       "      <td>198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>...</td>\n",
       "      <td>95.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>TAAWSPGWS</td>\n",
       "      <td>198.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>...</td>\n",
       "      <td>95.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>SRVKGAKAV</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>...</td>\n",
       "      <td>117.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1301</th>\n",
       "      <td>RVKGAKAVG</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>...</td>\n",
       "      <td>117.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1302</th>\n",
       "      <td>VKGAKAVGP</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>...</td>\n",
       "      <td>117.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1303</th>\n",
       "      <td>KGAKAVGPP</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>...</td>\n",
       "      <td>117.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420</th>\n",
       "      <td>APGVSRVKP</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>103 rows Ã— 1106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           kmer  TCGA3CAAAU01A11RA41B07all  TCGA3CAALI01A11RA41B07all  \\\n",
       "7     GWSWLRVLT                      198.0                       47.0   \n",
       "205   PGWSWLRVL                      198.0                       47.0   \n",
       "206   ASTAAWSPG                      198.0                       47.0   \n",
       "207   STAAWSPGW                      198.0                       47.0   \n",
       "208   TAAWSPGWS                      198.0                       47.0   \n",
       "...         ...                        ...                        ...   \n",
       "1300  SRVKGAKAV                      200.0                       44.0   \n",
       "1301  RVKGAKAVG                      200.0                       44.0   \n",
       "1302  VKGAKAVGP                      200.0                       44.0   \n",
       "1303  KGAKAVGPP                      200.0                       44.0   \n",
       "1420  APGVSRVKP                       14.0                        1.0   \n",
       "\n",
       "      TCGA3CAALJ01A31RA41B07all  TCGA3CAALK01A11RA41B07all  \\\n",
       "7                          55.0                      143.0   \n",
       "205                        55.0                      143.0   \n",
       "206                        55.0                      143.0   \n",
       "207                        55.0                      143.0   \n",
       "208                        55.0                      143.0   \n",
       "...                         ...                        ...   \n",
       "1300                       73.0                      158.0   \n",
       "1301                       73.0                      158.0   \n",
       "1302                       73.0                      158.0   \n",
       "1303                       73.0                      158.0   \n",
       "1420                        1.0                        7.0   \n",
       "\n",
       "      TCGA4HAAAK01A12RA41B07all  TCGA5LAAT001A12RA41B07all  \\\n",
       "7                         124.0                       77.0   \n",
       "205                       124.0                       77.0   \n",
       "206                       124.0                       77.0   \n",
       "207                       124.0                       77.0   \n",
       "208                       124.0                       77.0   \n",
       "...                         ...                        ...   \n",
       "1300                      124.0                       59.0   \n",
       "1301                      124.0                       59.0   \n",
       "1302                      124.0                       59.0   \n",
       "1303                      124.0                       59.0   \n",
       "1420                        5.0                       25.0   \n",
       "\n",
       "      TCGA5LAAT101A12RA41B07all  TCGA5TA9QA01A11RA41B07all  \\\n",
       "7                          40.0                       32.0   \n",
       "205                        40.0                       32.0   \n",
       "206                        40.0                       32.0   \n",
       "207                        40.0                       32.0   \n",
       "208                        40.0                       32.0   \n",
       "...                         ...                        ...   \n",
       "1300                       41.0                       37.0   \n",
       "1301                       41.0                       37.0   \n",
       "1302                       41.0                       37.0   \n",
       "1303                       41.0                       37.0   \n",
       "1420                        3.0                        0.0   \n",
       "\n",
       "      TCGAA1A0SB01A11RA14407all  ...  TCGAW8A86G01A21RA36F07all  \\\n",
       "7                          96.0  ...                       95.0   \n",
       "205                        96.0  ...                       95.0   \n",
       "206                        96.0  ...                       95.0   \n",
       "207                        96.0  ...                       95.0   \n",
       "208                        96.0  ...                       95.0   \n",
       "...                         ...  ...                        ...   \n",
       "1300                       65.0  ...                      117.0   \n",
       "1301                       65.0  ...                      117.0   \n",
       "1302                       65.0  ...                      117.0   \n",
       "1303                       65.0  ...                      117.0   \n",
       "1420                        0.0  ...                       19.0   \n",
       "\n",
       "      TCGAWTAB4101A11RA41B07all  TCGAWTAB4401A11RA41B07all  \\\n",
       "7                           5.0                        8.0   \n",
       "205                         5.0                        8.0   \n",
       "206                         5.0                        8.0   \n",
       "207                         5.0                        8.0   \n",
       "208                         5.0                        8.0   \n",
       "...                         ...                        ...   \n",
       "1300                       17.0                       31.0   \n",
       "1301                       17.0                       31.0   \n",
       "1302                       17.0                       31.0   \n",
       "1303                       17.0                       31.0   \n",
       "1420                        1.0                        0.0   \n",
       "\n",
       "      TCGAXXA89901A11RA36F07all  TCGAXXA89A01A11RA36F07all  \\\n",
       "7                         125.0                       56.0   \n",
       "205                       125.0                       56.0   \n",
       "206                       125.0                       56.0   \n",
       "207                       125.0                       56.0   \n",
       "208                       125.0                       56.0   \n",
       "...                         ...                        ...   \n",
       "1300                      104.0                       76.0   \n",
       "1301                      104.0                       76.0   \n",
       "1302                      104.0                       76.0   \n",
       "1303                      104.0                       76.0   \n",
       "1420                        3.0                        5.0   \n",
       "\n",
       "      TCGAZ7A8R501A42RA41B07all  TCGAZ7A8R601A11RA41B07all  isCrossJunction  \\\n",
       "7                          17.0                      126.0             True   \n",
       "205                        17.0                      126.0             True   \n",
       "206                        17.0                      126.0             True   \n",
       "207                        17.0                      126.0             True   \n",
       "208                        17.0                      126.0             True   \n",
       "...                         ...                        ...              ...   \n",
       "1300                       58.0                      131.0             True   \n",
       "1301                       58.0                      131.0             True   \n",
       "1302                       58.0                      131.0             True   \n",
       "1303                       58.0                      131.0             True   \n",
       "1420                        4.0                       29.0             True   \n",
       "\n",
       "      junctionAnnotated  readFrameAnnotated  \n",
       "7                   1.0                 1.0  \n",
       "205                 1.0                 0.0  \n",
       "206                 1.0                 0.0  \n",
       "207                 1.0                 0.0  \n",
       "208                 1.0                 0.0  \n",
       "...                 ...                 ...  \n",
       "1300                1.0                 0.0  \n",
       "1301                1.0                 0.0  \n",
       "1302                1.0                 0.0  \n",
       "1303                1.0                 0.0  \n",
       "1420                1.0                 0.0  \n",
       "\n",
       "[103 rows x 1106 columns]"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eth_old.loc[eth_old['junctionAnnotated'] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "27f62b73",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3139: DtypeWarning: Columns (57,58,60,62,63,64,65,66,67,68,70,72) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "ohsu_old = pd.read_csv('/cluster/work/grlab/projects/projects2020_OHSU/share_OHUS_PNLL/OHSU_Feb2023_complete-annotated-shortlist.tsv.gz', sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "722e213c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ep_col = 'prefiltered_in-frame_epitopes'\n",
    "ohsu_old[ep_col] = ohsu_old[ep_col].str.split(';')\n",
    "ohsu_old = ohsu_old.explode(ep_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "8cf7523b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASG</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>VYYNEASGG</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>YYNEASGGR</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>YNEASGGRY</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48726;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NEASGGRYV</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346995</th>\n",
       "      <td>chr2;109934895;109947363;+</td>\n",
       "      <td>14.921219</td>\n",
       "      <td>3.626901</td>\n",
       "      <td>175.196991</td>\n",
       "      <td>87.287682</td>\n",
       "      <td>1.922689</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.140192</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346996</th>\n",
       "      <td>chr2;109987191;109987560;+</td>\n",
       "      <td>34.433582</td>\n",
       "      <td>53.678131</td>\n",
       "      <td>138.136089</td>\n",
       "      <td>42.804536</td>\n",
       "      <td>61.526038</td>\n",
       "      <td>21.880716</td>\n",
       "      <td>92.355580</td>\n",
       "      <td>8.409545</td>\n",
       "      <td>44.537781</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346997</th>\n",
       "      <td>chr2;109987191;109993691;+</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.450760</td>\n",
       "      <td>1.684586</td>\n",
       "      <td>1.678609</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.140192</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346998</th>\n",
       "      <td>chr2;109987191;109994400;+</td>\n",
       "      <td>16.069005</td>\n",
       "      <td>5.077661</td>\n",
       "      <td>5.053759</td>\n",
       "      <td>10.071656</td>\n",
       "      <td>8.652099</td>\n",
       "      <td>9.572813</td>\n",
       "      <td>4.560769</td>\n",
       "      <td>10.511931</td>\n",
       "      <td>8.660124</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346999</th>\n",
       "      <td>chr2;109987191;110017561;+</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>11.792105</td>\n",
       "      <td>1.678609</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.140192</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>18.557409</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6092818 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 jx  TCGA-24-1431-01A-01R-1566-13  \\\n",
       "0               chr10;48726;48803;-                      0.000000   \n",
       "0               chr10;48726;48803;-                      0.000000   \n",
       "0               chr10;48726;48803;-                      0.000000   \n",
       "0               chr10;48726;48803;-                      0.000000   \n",
       "0               chr10;48726;48803;-                      0.000000   \n",
       "...                             ...                           ...   \n",
       "1346995  chr2;109934895;109947363;+                     14.921219   \n",
       "1346996  chr2;109987191;109987560;+                     34.433582   \n",
       "1346997  chr2;109987191;109993691;+                      0.000000   \n",
       "1346998  chr2;109987191;109994400;+                     16.069005   \n",
       "1346999  chr2;109987191;110017561;+                      0.000000   \n",
       "\n",
       "         TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \\\n",
       "0                            2.176140                      0.842293   \n",
       "0                            2.176140                      0.842293   \n",
       "0                            2.176140                      0.842293   \n",
       "0                            2.176140                      0.842293   \n",
       "0                            2.176140                      0.842293   \n",
       "...                               ...                           ...   \n",
       "1346995                      3.626901                    175.196991   \n",
       "1346996                     53.678131                    138.136089   \n",
       "1346997                      1.450760                      1.684586   \n",
       "1346998                      5.077661                      5.053759   \n",
       "1346999                      0.000000                     11.792105   \n",
       "\n",
       "         TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \\\n",
       "0                            2.517914                      0.961344   \n",
       "0                            2.517914                      0.961344   \n",
       "0                            2.517914                      0.961344   \n",
       "0                            2.517914                      0.961344   \n",
       "0                            2.517914                      0.961344   \n",
       "...                               ...                           ...   \n",
       "1346995                     87.287682                      1.922689   \n",
       "1346996                     42.804536                     61.526038   \n",
       "1346997                      1.678609                      0.000000   \n",
       "1346998                     10.071656                      8.652099   \n",
       "1346999                      1.678609                      0.000000   \n",
       "\n",
       "         TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \\\n",
       "0                            0.000000                      5.700962   \n",
       "0                            0.000000                      5.700962   \n",
       "0                            0.000000                      5.700962   \n",
       "0                            0.000000                      5.700962   \n",
       "0                            0.000000                      5.700962   \n",
       "...                               ...                           ...   \n",
       "1346995                      0.000000                      1.140192   \n",
       "1346996                     21.880716                     92.355580   \n",
       "1346997                      0.000000                      1.140192   \n",
       "1346998                      9.572813                      4.560769   \n",
       "1346999                      0.000000                      1.140192   \n",
       "\n",
       "         TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \\\n",
       "0                            2.102386                      1.237161  ...   \n",
       "0                            2.102386                      1.237161  ...   \n",
       "0                            2.102386                      1.237161  ...   \n",
       "0                            2.102386                      1.237161  ...   \n",
       "0                            2.102386                      1.237161  ...   \n",
       "...                               ...                           ...  ...   \n",
       "1346995                      2.102386                      0.000000  ...   \n",
       "1346996                      8.409545                     44.537781  ...   \n",
       "1346997                      2.102386                      0.000000  ...   \n",
       "1346998                     10.511931                      8.660124  ...   \n",
       "1346999                      2.102386                     18.557409  ...   \n",
       "\n",
       "                                     modified_upstream_txs  \\\n",
       "0        ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "0        ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "0        ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "0        ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "0        ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "...                                                    ...   \n",
       "1346995                                                NaN   \n",
       "1346996                                                NaN   \n",
       "1346997                                                NaN   \n",
       "1346998                                                NaN   \n",
       "1346999                                                NaN   \n",
       "\n",
       "                           in-frame_all-transcript_biexons  \\\n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "...                                                    ...   \n",
       "1346995                                                NaN   \n",
       "1346996                                                NaN   \n",
       "1346997                                                NaN   \n",
       "1346998                                                NaN   \n",
       "1346999                                                NaN   \n",
       "\n",
       "                            in-frame_nonhanging-tx_biexons  \\\n",
       "0        MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "0        MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "0        MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "0        MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "0        MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "...                                                    ...   \n",
       "1346995                                                NaN   \n",
       "1346996                                                NaN   \n",
       "1346997                                                NaN   \n",
       "1346998                                                NaN   \n",
       "1346999                                                NaN   \n",
       "\n",
       "                  in-frame_peptide_sequence  \\\n",
       "0        NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "0        NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "0        NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "0        NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "0        NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "...                                     ...   \n",
       "1346995                                 NaN   \n",
       "1346996                                 NaN   \n",
       "1346997                                 NaN   \n",
       "1346998                                 NaN   \n",
       "1346999                                 NaN   \n",
       "\n",
       "         hanging_txs_included_inframe_pepseqs  prefiltered_in-frame_epitopes  \\\n",
       "0          NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG                      NVYYNEASG   \n",
       "0          NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG                      VYYNEASGG   \n",
       "0          NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG                      YYNEASGGR   \n",
       "0          NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG                      YNEASGGRY   \n",
       "0          NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG                      NEASGGRYV   \n",
       "...                                       ...                            ...   \n",
       "1346995                                   NaN                            NaN   \n",
       "1346996                                   NaN                            NaN   \n",
       "1346997                                   NaN                            NaN   \n",
       "1346998                                   NaN                            NaN   \n",
       "1346999                                   NaN                            NaN   \n",
       "\n",
       "         prefiltered_inframe_epitope_count  \\\n",
       "0                                     17.0   \n",
       "0                                     17.0   \n",
       "0                                     17.0   \n",
       "0                                     17.0   \n",
       "0                                     17.0   \n",
       "...                                    ...   \n",
       "1346995                                NaN   \n",
       "1346996                                NaN   \n",
       "1346997                                NaN   \n",
       "1346998                                NaN   \n",
       "1346999                                NaN   \n",
       "\n",
       "                                      in-frame_neoepitopes  \\\n",
       "0        TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "0        TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "0        TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "0        TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "0        TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "...                                                    ...   \n",
       "1346995                                                NaN   \n",
       "1346996                                                NaN   \n",
       "1346997                                                NaN   \n",
       "1346998                                                NaN   \n",
       "1346999                                                NaN   \n",
       "\n",
       "         in-frame_neoepitope_count  \\\n",
       "0                              5.0   \n",
       "0                              5.0   \n",
       "0                              5.0   \n",
       "0                              5.0   \n",
       "0                              5.0   \n",
       "...                            ...   \n",
       "1346995                        NaN   \n",
       "1346996                        NaN   \n",
       "1346997                        NaN   \n",
       "1346998                        NaN   \n",
       "1346999                        NaN   \n",
       "\n",
       "                     frame-agnostic_all-transcript_biexons  \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "0        LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "...                                                    ...  \n",
       "1346995                                                NaN  \n",
       "1346996                                                NaN  \n",
       "1346997                                                NaN  \n",
       "1346998                                                NaN  \n",
       "1346999                                                NaN  \n",
       "\n",
       "[6092818 rows x 73 columns]"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_old"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "aef7f723",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [jx, TCGA-24-1431-01A-01R-1566-13, TCGA-24-2298-01A-01R-1569-13, TCGA-25-1313-01A-01R-1565-13, TCGA-25-1319-01A-01R-1565-13, TCGA-61-2008-01A-02R-1568-13, TCGA-A2-A0D2-01A-21R-A034-07, TCGA-A2-A0SX-01A-12R-A084-07, TCGA-AO-A0JM-01A-21R-A056-07, TCGA-BH-A18V-01A-11R-A12D-07, TCGA-C8-A12P-01A-11R-A115-07, GTEx_brain_testis_over0, GTEx_brain_testis_over1, GTEx_brain_testis_over10, GTEx_brain_testis_over2, GTEx_brain_testis_over3, GTEx_brain_testis_over5, GTEx_breast_over0, GTEx_breast_over1, GTEx_breast_over10, GTEx_breast_over2, GTEx_breast_over3, GTEx_breast_over5, GTEx_core_over0, GTEx_core_over1, GTEx_core_over10, GTEx_core_over2, GTEx_core_over3, GTEx_core_over5, GTEx_ovary_over0, GTEx_ovary_over1, GTEx_ovary_over10, GTEx_ovary_over2, GTEx_ovary_over3, GTEx_ovary_over5, all_TCGA_over0, all_TCGA_over1, all_TCGA_over10, all_TCGA_over2, all_TCGA_over3, all_TCGA_over5, BRCA_over_5, OV_over_2, OV_over_0, BRCA_over_3, BRCA_over_0, OV_over_10, BRCA_over_2, OV_over_3, BRCA_over_10, OV_over_1, OV_over_5, BRCA_over_1, motif, annotation, 5';3'_coding_regions, has_cds, gene, gene_id, upstream_exon, upstream_txs, downstream_exon, downstream_txs, modified_upstream_txs, in-frame_all-transcript_biexons, in-frame_nonhanging-tx_biexons, in-frame_peptide_sequence, hanging_txs_included_inframe_pepseqs, prefiltered_in-frame_epitopes, prefiltered_inframe_epitope_count, in-frame_neoepitopes, in-frame_neoepitope_count, frame-agnostic_all-transcript_biexons]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 73 columns]"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmer = 'APGVSRVKP'\n",
    "ohsu_old.loc[ohsu_old[ep_col] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "fb19a99e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>135280</th>\n",
       "      <td>chr1;1336516;1337017;-</td>\n",
       "      <td>3.443358</td>\n",
       "      <td>6.528421</td>\n",
       "      <td>3.369173</td>\n",
       "      <td>3.357219</td>\n",
       "      <td>7.690755</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.140192</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000632445.1.MOD.CHR1.1336516.1337017.M...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135281</th>\n",
       "      <td>chr1;1338159;1338268;-</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>4.352281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>1.367545</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>YVFGDLCSM</td>\n",
       "      <td>9.0</td>\n",
       "      <td>YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135281</th>\n",
       "      <td>chr1;1338159;1338268;-</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>4.352281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>1.367545</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>VFGDLCSMA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135281</th>\n",
       "      <td>chr1;1338159;1338268;-</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>4.352281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>1.367545</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>FGDLCSMAP</td>\n",
       "      <td>9.0</td>\n",
       "      <td>YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135281</th>\n",
       "      <td>chr1;1338159;1338268;-</td>\n",
       "      <td>4.591144</td>\n",
       "      <td>4.352281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.196523</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>1.367545</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>YVFGDLCSMAPVGLRIR</td>\n",
       "      <td>GDLCSMAPV</td>\n",
       "      <td>9.0</td>\n",
       "      <td>YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216474</th>\n",
       "      <td>chr1;1342139;1342362;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.725380</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000378888.10.MOD.CHR1.1342139.1342362....</td>\n",
       "      <td>PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...</td>\n",
       "      <td>PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...</td>\n",
       "      <td>GDSRPPSFHRDGMDNET</td>\n",
       "      <td>GDSRPPSFHRDGMDNET</td>\n",
       "      <td>FHRDGMDNE</td>\n",
       "      <td>9.0</td>\n",
       "      <td>DSRPPSFHR;SRPPSFHRD;RPPSFHRDG;PPSFHRDGM;PSFHRD...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>FPASTAAWSPGWSWLRVLTRMRGPRARTATQTCPRLLSGQAASGTP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216474</th>\n",
       "      <td>chr1;1342139;1342362;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.725380</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000378888.10.MOD.CHR1.1342139.1342362....</td>\n",
       "      <td>PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...</td>\n",
       "      <td>PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...</td>\n",
       "      <td>GDSRPPSFHRDGMDNET</td>\n",
       "      <td>GDSRPPSFHRDGMDNET</td>\n",
       "      <td>HRDGMDNET</td>\n",
       "      <td>9.0</td>\n",
       "      <td>DSRPPSFHR;SRPPSFHRD;RPPSFHRDG;PPSFHRDGM;PSFHRD...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>FPASTAAWSPGWSWLRVLTRMRGPRARTATQTCPRLLSGQAASGTP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220095</th>\n",
       "      <td>chr1;1338654;1340397;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.725380</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220096</th>\n",
       "      <td>chr1;1339396;1339484;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.725380</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220097</th>\n",
       "      <td>chr1;1341806;1341915;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.725380</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1523 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             jx  TCGA-24-1431-01A-01R-1566-13  \\\n",
       "135280   chr1;1336516;1337017;-                      3.443358   \n",
       "135281   chr1;1338159;1338268;-                      4.591144   \n",
       "135281   chr1;1338159;1338268;-                      4.591144   \n",
       "135281   chr1;1338159;1338268;-                      4.591144   \n",
       "135281   chr1;1338159;1338268;-                      4.591144   \n",
       "...                         ...                           ...   \n",
       "1216474  chr1;1342139;1342362;-                      0.000000   \n",
       "1216474  chr1;1342139;1342362;-                      0.000000   \n",
       "1220095  chr1;1338654;1340397;-                      0.000000   \n",
       "1220096  chr1;1339396;1339484;-                      0.000000   \n",
       "1220097  chr1;1341806;1341915;-                      0.000000   \n",
       "\n",
       "         TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \\\n",
       "135280                       6.528421                      3.369173   \n",
       "135281                       4.352281                      0.000000   \n",
       "135281                       4.352281                      0.000000   \n",
       "135281                       4.352281                      0.000000   \n",
       "135281                       4.352281                      0.000000   \n",
       "...                               ...                           ...   \n",
       "1216474                      0.725380                      0.000000   \n",
       "1216474                      0.725380                      0.000000   \n",
       "1220095                      0.725380                      0.000000   \n",
       "1220096                      0.725380                      0.000000   \n",
       "1220097                      0.725380                      0.000000   \n",
       "\n",
       "         TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \\\n",
       "135280                       3.357219                      7.690755   \n",
       "135281                       4.196523                      0.961344   \n",
       "135281                       4.196523                      0.961344   \n",
       "135281                       4.196523                      0.961344   \n",
       "135281                       4.196523                      0.961344   \n",
       "...                               ...                           ...   \n",
       "1216474                      0.000000                      0.000000   \n",
       "1216474                      0.000000                      0.000000   \n",
       "1220095                      0.000000                      0.000000   \n",
       "1220096                      0.000000                      0.000000   \n",
       "1220097                      0.000000                      0.000000   \n",
       "\n",
       "         TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \\\n",
       "135280                       0.000000                      1.140192   \n",
       "135281                       1.367545                      5.700962   \n",
       "135281                       1.367545                      5.700962   \n",
       "135281                       1.367545                      5.700962   \n",
       "135281                       1.367545                      5.700962   \n",
       "...                               ...                           ...   \n",
       "1216474                      0.000000                      0.000000   \n",
       "1216474                      0.000000                      0.000000   \n",
       "1220095                      0.000000                      0.000000   \n",
       "1220096                      0.000000                      0.000000   \n",
       "1220097                      0.000000                      0.000000   \n",
       "\n",
       "         TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \\\n",
       "135280                            0.0                      1.237161  ...   \n",
       "135281                            0.0                      2.474321  ...   \n",
       "135281                            0.0                      2.474321  ...   \n",
       "135281                            0.0                      2.474321  ...   \n",
       "135281                            0.0                      2.474321  ...   \n",
       "...                               ...                           ...  ...   \n",
       "1216474                           0.0                      0.000000  ...   \n",
       "1216474                           0.0                      0.000000  ...   \n",
       "1220095                           0.0                      0.000000  ...   \n",
       "1220096                           0.0                      0.000000  ...   \n",
       "1220097                           0.0                      0.000000  ...   \n",
       "\n",
       "                                     modified_upstream_txs  \\\n",
       "135280   ['ENST00000632445.1.MOD.CHR1.1336516.1337017.M...   \n",
       "135281   ['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...   \n",
       "135281   ['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...   \n",
       "135281   ['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...   \n",
       "135281   ['ENST00000631679.1.MOD.CHR1.1338159.1338268.M...   \n",
       "...                                                    ...   \n",
       "1216474  ['ENST00000378888.10.MOD.CHR1.1342139.1342362....   \n",
       "1216474  ['ENST00000378888.10.MOD.CHR1.1342139.1342362....   \n",
       "1220095                                                NaN   \n",
       "1220096                                                NaN   \n",
       "1220097                                                NaN   \n",
       "\n",
       "                           in-frame_all-transcript_biexons  \\\n",
       "135280                                                 NaN   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "...                                                    ...   \n",
       "1216474  PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...   \n",
       "1216474  PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...   \n",
       "1220095                                                NaN   \n",
       "1220096                                                NaN   \n",
       "1220097                                                NaN   \n",
       "\n",
       "                            in-frame_nonhanging-tx_biexons  \\\n",
       "135280                                                 NaN   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...   \n",
       "...                                                    ...   \n",
       "1216474  PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...   \n",
       "1216474  PCFNGRVVSWLVLAEGAHSDAGSQGTDSHTDLPPPLERTGGIGDSR...   \n",
       "1220095                                                NaN   \n",
       "1220096                                                NaN   \n",
       "1220097                                                NaN   \n",
       "\n",
       "         in-frame_peptide_sequence  hanging_txs_included_inframe_pepseqs  \\\n",
       "135280                         NaN                                   NaN   \n",
       "135281           YVFGDLCSMAPVGLRIR                     YVFGDLCSMAPVGLRIR   \n",
       "135281           YVFGDLCSMAPVGLRIR                     YVFGDLCSMAPVGLRIR   \n",
       "135281           YVFGDLCSMAPVGLRIR                     YVFGDLCSMAPVGLRIR   \n",
       "135281           YVFGDLCSMAPVGLRIR                     YVFGDLCSMAPVGLRIR   \n",
       "...                            ...                                   ...   \n",
       "1216474          GDSRPPSFHRDGMDNET                     GDSRPPSFHRDGMDNET   \n",
       "1216474          GDSRPPSFHRDGMDNET                     GDSRPPSFHRDGMDNET   \n",
       "1220095                        NaN                                   NaN   \n",
       "1220096                        NaN                                   NaN   \n",
       "1220097                        NaN                                   NaN   \n",
       "\n",
       "         prefiltered_in-frame_epitopes  prefiltered_inframe_epitope_count  \\\n",
       "135280                             NaN                                0.0   \n",
       "135281                       YVFGDLCSM                                9.0   \n",
       "135281                       VFGDLCSMA                                9.0   \n",
       "135281                       FGDLCSMAP                                9.0   \n",
       "135281                       GDLCSMAPV                                9.0   \n",
       "...                                ...                                ...   \n",
       "1216474                      FHRDGMDNE                                9.0   \n",
       "1216474                      HRDGMDNET                                9.0   \n",
       "1220095                            NaN                                NaN   \n",
       "1220096                            NaN                                NaN   \n",
       "1220097                            NaN                                NaN   \n",
       "\n",
       "                                      in-frame_neoepitopes  \\\n",
       "135280                                                 NaN   \n",
       "135281   YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...   \n",
       "135281   YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...   \n",
       "135281   YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...   \n",
       "135281   YVFGDLCSM;VFGDLCSMA;FGDLCSMAP;GDLCSMAPV;DLCSMA...   \n",
       "...                                                    ...   \n",
       "1216474  DSRPPSFHR;SRPPSFHRD;RPPSFHRDG;PPSFHRDGM;PSFHRD...   \n",
       "1216474  DSRPPSFHR;SRPPSFHRD;RPPSFHRDG;PPSFHRDGM;PSFHRD...   \n",
       "1220095                                                NaN   \n",
       "1220096                                                NaN   \n",
       "1220097                                                NaN   \n",
       "\n",
       "         in-frame_neoepitope_count  \\\n",
       "135280                         0.0   \n",
       "135281                         9.0   \n",
       "135281                         9.0   \n",
       "135281                         9.0   \n",
       "135281                         9.0   \n",
       "...                            ...   \n",
       "1216474                        8.0   \n",
       "1216474                        8.0   \n",
       "1220095                        NaN   \n",
       "1220096                        NaN   \n",
       "1220097                        NaN   \n",
       "\n",
       "                     frame-agnostic_all-transcript_biexons  \n",
       "135280                                                 NaN  \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...  \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...  \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...  \n",
       "135281   ADVVDWLYTHVEGFKERREARKYASSLLKHGFLRHTVNKITFSEQC...  \n",
       "...                                                    ...  \n",
       "1216474  FPASTAAWSPGWSWLRVLTRMRGPRARTATQTCPRLLSGQAASGTP...  \n",
       "1216474  FPASTAAWSPGWSWLRVLTRMRGPRARTATQTCPRLLSGQAASGTP...  \n",
       "1220095                                                NaN  \n",
       "1220096                                                NaN  \n",
       "1220097                                                NaN  \n",
       "\n",
       "[1523 rows x 73 columns]"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_id = 'ENSG00000107404.20'\n",
    "ohsu_old.loc[ohsu_old['gene_id'] == gene_id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "7c5cb807",
   "metadata": {},
   "outputs": [],
   "source": [
    "old_ohsu_set = set(ohsu_old.loc[ohsu_old['gene_id'] == gene_id, ep_col])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "a84a3192",
   "metadata": {},
   "outputs": [],
   "source": [
    "old_eth_set = set(eth_old.loc[eth_old['junctionAnnotated'] == 1, 'kmer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "76b33ccf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ALPRYELEE',\n",
       " 'DSSGSTPAA',\n",
       " 'ELEEAPLTV',\n",
       " 'GALPRYELE',\n",
       " 'GSQQSEALD',\n",
       " 'GTPGPPPST',\n",
       " 'GVSRVKGAK',\n",
       " 'IRKYASSLL',\n",
       " 'LIRKYASSL',\n",
       " 'LPRYELEEA',\n",
       " 'LTGALPRYE',\n",
       " 'PGVSRVKGA',\n",
       " 'PRYELEEAP',\n",
       " 'RLIRKYASS',\n",
       " 'RYELEEAPL',\n",
       " 'SRLIRKYAS',\n",
       " 'SSRLIRKYA',\n",
       " 'STGSQQSEA',\n",
       " 'TGALPRYEL',\n",
       " 'TGSQQSEAL',\n",
       " 'TSSRLIRKY',\n",
       " 'YELEEAPLT'}"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_eth_set.intersection(old_ohsu_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "a1fdd6cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "kmer = 'ALPRYELEE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "f8a1a662",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>895337</th>\n",
       "      <td>chr1;1338654;1339361;-</td>\n",
       "      <td>270.877511</td>\n",
       "      <td>144.350649</td>\n",
       "      <td>67.383458</td>\n",
       "      <td>80.573245</td>\n",
       "      <td>241.29743</td>\n",
       "      <td>30.085984</td>\n",
       "      <td>103.757504</td>\n",
       "      <td>39.945338</td>\n",
       "      <td>45.774942</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000631679.1.MOD.CHR1.1338654.1339361.M...</td>\n",
       "      <td>IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSG...</td>\n",
       "      <td>TGPISLTVAKCWDPTPRSYFTVPRADPVRPIDPAAWLSHTAALTGA...</td>\n",
       "      <td>LTGALPRYELEEAPLTV</td>\n",
       "      <td>LTGALPRYELEEAPLTV</td>\n",
       "      <td>ALPRYELEE</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            jx  TCGA-24-1431-01A-01R-1566-13  \\\n",
       "895337  chr1;1338654;1339361;-                    270.877511   \n",
       "\n",
       "        TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \\\n",
       "895337                    144.350649                     67.383458   \n",
       "\n",
       "        TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \\\n",
       "895337                     80.573245                     241.29743   \n",
       "\n",
       "        TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \\\n",
       "895337                     30.085984                    103.757504   \n",
       "\n",
       "        TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \\\n",
       "895337                     39.945338                     45.774942  ...   \n",
       "\n",
       "                                    modified_upstream_txs  \\\n",
       "895337  ['ENST00000631679.1.MOD.CHR1.1338654.1339361.M...   \n",
       "\n",
       "                          in-frame_all-transcript_biexons  \\\n",
       "895337  IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSG...   \n",
       "\n",
       "                           in-frame_nonhanging-tx_biexons  \\\n",
       "895337  TGPISLTVAKCWDPTPRSYFTVPRADPVRPIDPAAWLSHTAALTGA...   \n",
       "\n",
       "        in-frame_peptide_sequence  hanging_txs_included_inframe_pepseqs  \\\n",
       "895337          LTGALPRYELEEAPLTV                     LTGALPRYELEEAPLTV   \n",
       "\n",
       "        prefiltered_in-frame_epitopes  prefiltered_inframe_epitope_count  \\\n",
       "895337                      ALPRYELEE                                9.0   \n",
       "\n",
       "        in-frame_neoepitopes  in-frame_neoepitope_count  \\\n",
       "895337                   NaN                        0.0   \n",
       "\n",
       "                    frame-agnostic_all-transcript_biexons  \n",
       "895337  IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSG...  \n",
       "\n",
       "[1 rows x 73 columns]"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "ohsu_old.loc[ohsu_old[ep_col] == kmer]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "60237e78",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'coord'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexes/base.py:3080\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3079\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3080\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3081\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep_other/lib/python3.9/site-packages/pandas-1.5.0-py3.9-linux-x86_64.egg/pandas/_libs/index.pyx:70\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep_other/lib/python3.9/site-packages/pandas-1.5.0-py3.9-linux-x86_64.egg/pandas/_libs/index.pyx:101\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:4554\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:4562\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'coord'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [204], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43meth_old\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mloc\u001b[49m\u001b[43m[\u001b[49m\u001b[43meth_old\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mkmer\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m==\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mkmer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mcoord\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:889\u001b[0m, in \u001b[0;36m_LocationIndexer.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    886\u001b[0m         \u001b[38;5;28;01mwith\u001b[39;00m suppress(\u001b[38;5;167;01mKeyError\u001b[39;00m, \u001b[38;5;167;01mIndexError\u001b[39;00m, \u001b[38;5;167;01mAttributeError\u001b[39;00m):\n\u001b[1;32m    887\u001b[0m             \u001b[38;5;66;03m# AttributeError for IntervalTree get_value\u001b[39;00m\n\u001b[1;32m    888\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39m_get_value(\u001b[38;5;241m*\u001b[39mkey, takeable\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_takeable)\n\u001b[0;32m--> 889\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_getitem_tuple\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    890\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    891\u001b[0m     \u001b[38;5;66;03m# we by definition only have the 0th axis\u001b[39;00m\n\u001b[1;32m    892\u001b[0m     axis \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maxis \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m0\u001b[39m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1060\u001b[0m, in \u001b[0;36m_LocIndexer._getitem_tuple\u001b[0;34m(self, tup)\u001b[0m\n\u001b[1;32m   1058\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_getitem_tuple\u001b[39m(\u001b[38;5;28mself\u001b[39m, tup: Tuple):\n\u001b[1;32m   1059\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m suppress(IndexingError):\n\u001b[0;32m-> 1060\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_getitem_lowerdim\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtup\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1062\u001b[0m     \u001b[38;5;66;03m# no multi-index, so validate all of the indexers\u001b[39;00m\n\u001b[1;32m   1063\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_has_valid_tuple(tup)\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:807\u001b[0m, in \u001b[0;36m_LocationIndexer._getitem_lowerdim\u001b[0;34m(self, tup)\u001b[0m\n\u001b[1;32m    803\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i, key \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(tup):\n\u001b[1;32m    804\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_label_like(key):\n\u001b[1;32m    805\u001b[0m         \u001b[38;5;66;03m# We don't need to check for tuples here because those are\u001b[39;00m\n\u001b[1;32m    806\u001b[0m         \u001b[38;5;66;03m#  caught by the _is_nested_tuple_indexer check above.\u001b[39;00m\n\u001b[0;32m--> 807\u001b[0m         section \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_getitem_axis\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mi\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    809\u001b[0m         \u001b[38;5;66;03m# We should never have a scalar section here, because\u001b[39;00m\n\u001b[1;32m    810\u001b[0m         \u001b[38;5;66;03m#  _getitem_lowerdim is only called after a check for\u001b[39;00m\n\u001b[1;32m    811\u001b[0m         \u001b[38;5;66;03m#  is_scalar_access, which that would be.\u001b[39;00m\n\u001b[1;32m    812\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m section\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mndim:\n\u001b[1;32m    813\u001b[0m             \u001b[38;5;66;03m# we're in the middle of slicing through a MultiIndex\u001b[39;00m\n\u001b[1;32m    814\u001b[0m             \u001b[38;5;66;03m# revise the key wrt to `section` by inserting an _NS\u001b[39;00m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1124\u001b[0m, in \u001b[0;36m_LocIndexer._getitem_axis\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1122\u001b[0m \u001b[38;5;66;03m# fall thru to straight lookup\u001b[39;00m\n\u001b[1;32m   1123\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_key(key, axis)\n\u001b[0;32m-> 1124\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_label\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43maxis\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexing.py:1073\u001b[0m, in \u001b[0;36m_LocIndexer._get_label\u001b[0;34m(self, label, axis)\u001b[0m\n\u001b[1;32m   1071\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_get_label\u001b[39m(\u001b[38;5;28mself\u001b[39m, label, axis: \u001b[38;5;28mint\u001b[39m):\n\u001b[1;32m   1072\u001b[0m     \u001b[38;5;66;03m# GH#5667 this will fail if the label is not present in the axis.\u001b[39;00m\n\u001b[0;32m-> 1073\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mxs\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlabel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43maxis\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/generic.py:3724\u001b[0m, in \u001b[0;36mNDFrame.xs\u001b[0;34m(self, key, axis, level, drop_level)\u001b[0m\n\u001b[1;32m   3722\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m axis \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   3723\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m drop_level:\n\u001b[0;32m-> 3724\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m   3725\u001b[0m     index \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\n\u001b[1;32m   3726\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/frame.py:3024\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3022\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   3023\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 3024\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3025\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   3026\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m/cluster/customapps/biomed/grlab/users/prelotla/software/miniconda/envs/pep3/lib/python3.9/site-packages/pandas/core/indexes/base.py:3082\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3080\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3081\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m-> 3082\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3084\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m tolerance \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   3085\u001b[0m     tolerance \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_convert_tolerance(tolerance, np\u001b[38;5;241m.\u001b[39masarray(key))\n",
      "\u001b[0;31mKeyError\u001b[0m: 'coord'"
     ]
    }
   ],
   "source": [
    "eth_old.loc[eth_old['kmer'] == kmer, 'coord']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a9e530b",
   "metadata": {},
   "outputs": [],
   "source": [
    "(pep3) prelotla@login-biomed-01:.../tmp_out_ref_batch_101$ parquet-tools csv *meta* | grep ALPRYELEE\n",
    "IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSGLEIRDRMWLKITIANAVI,ENSG00000107404.20:33_29:0:1339422:2-exons,1,1,ENSG00000107404.20,chr1,-,ref,1,0,nan,0,nan,nan,1339361;1339422;1338522;1338653,1339361;1339439;1338521;1338653,33;29,[[39. 20. 15. ... 35. 20. 48.]],nan,2-exons\n",
    "DPVRPIDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSGLEIRDRMWLKITIANAVI,ENSG00000107404.20:33_29:0:1339437:2-exons,1,0,ENSG00000107404.20,chr1,-,ref,1,0,nan,0,nan,nan,1339361;1339437;1338522;1338653,1339361;1339439;1338521;1338653,33;29,[[39. 20. 15. ... 35. 20. 48.]],nan,2-exons\n",
    "IDPAAWLSHTAALTGALPRYELEEAPLTVKSDMSAVVRVMQLPDSGLEIRDRMWLKITIANAVI,ENSG00000107404.20:34_29:0:1339422:2-exons,1,1,ENSG00000107404.20,chr1,-,ref,1,0,nan,0,nan,nan,1339361;1339422;1338522;1338653,1339361;1339649;1338521;1338653,34;29,[[39. 20. 15. ... 35. 20. 48.]],nan,2-exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b04caf36",
   "metadata": {},
   "outputs": [],
   "source": [
    "Coordinates \n",
    "1339361;1339422;1338522;1338653 \n",
    "=> 1338653, 1339361"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b08aa06c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ANNOTATION FINDINGS \n",
    "1338653, 1339362\n",
    "NOT FOUND \n",
    "1338654 1339361"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "9d283068",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1338654 < 1339361"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "287242c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1339439 > 1339362"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "18bd6feb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1338522 < 1338653"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "4853a548",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1338653 < 1339362"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "e6b6b101",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1338522 < 1338653 < 1339362 < 1339439"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "4bcc40d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "8a7b610f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>48726</td>\n",
       "      <td>48803</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>48726</td>\n",
       "      <td>48803</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>48726</td>\n",
       "      <td>48803</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>48726</td>\n",
       "      <td>48803</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>48726</td>\n",
       "      <td>48803</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346995</th>\n",
       "      <td>chr2</td>\n",
       "      <td>109934895</td>\n",
       "      <td>109947363</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346996</th>\n",
       "      <td>chr2</td>\n",
       "      <td>109987191</td>\n",
       "      <td>109987560</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346997</th>\n",
       "      <td>chr2</td>\n",
       "      <td>109987191</td>\n",
       "      <td>109993691</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346998</th>\n",
       "      <td>chr2</td>\n",
       "      <td>109987191</td>\n",
       "      <td>109994400</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346999</th>\n",
       "      <td>chr2</td>\n",
       "      <td>109987191</td>\n",
       "      <td>110017561</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4157968 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             0          1          2  3\n",
       "0        chr10      48726      48803  -\n",
       "0        chr10      48726      48803  -\n",
       "0        chr10      48726      48803  -\n",
       "0        chr10      48726      48803  -\n",
       "0        chr10      48726      48803  -\n",
       "...        ...        ...        ... ..\n",
       "1346995   chr2  109934895  109947363  +\n",
       "1346996   chr2  109987191  109987560  +\n",
       "1346997   chr2  109987191  109993691  +\n",
       "1346998   chr2  109987191  109994400  +\n",
       "1346999   chr2  109987191  110017561  +\n",
       "\n",
       "[4157968 rows x 4 columns]"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_junction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "39163b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0              48725\n",
       "0              48725\n",
       "0              48725\n",
       "0              48725\n",
       "0              48725\n",
       "             ...    \n",
       "1346995    109934894\n",
       "1346996    109987190\n",
       "1346997    109987190\n",
       "1346998    109987190\n",
       "1346999    109987190\n",
       "Name: 1, Length: 4157968, dtype: object"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "id": "5cc014f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ohsu_to_eth_coord(df, col = 'jx', sep = ';'):\n",
    "    tmp_jx = df[col].str.split(sep,  expand = True)\n",
    "    df[col] = tmp_jx[0] + sep + (tmp_jx[1].astype(int) - 1).astype(str) + sep + tmp_jx[2] + sep + tmp_jx[3]\n",
    "    return df\n",
    "\n",
    "ohsu_df = ohsu_to_eth_coord(ohsu_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "f09d66c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>jx</th>\n",
       "      <th>TCGA-24-1431-01A-01R-1566-13</th>\n",
       "      <th>TCGA-24-2298-01A-01R-1569-13</th>\n",
       "      <th>TCGA-25-1313-01A-01R-1565-13</th>\n",
       "      <th>TCGA-25-1319-01A-01R-1565-13</th>\n",
       "      <th>TCGA-61-2008-01A-02R-1568-13</th>\n",
       "      <th>TCGA-A2-A0D2-01A-21R-A034-07</th>\n",
       "      <th>TCGA-A2-A0SX-01A-12R-A084-07</th>\n",
       "      <th>TCGA-AO-A0JM-01A-21R-A056-07</th>\n",
       "      <th>TCGA-BH-A18V-01A-11R-A12D-07</th>\n",
       "      <th>...</th>\n",
       "      <th>modified_upstream_txs</th>\n",
       "      <th>in-frame_all-transcript_biexons</th>\n",
       "      <th>in-frame_nonhanging-tx_biexons</th>\n",
       "      <th>in-frame_peptide_sequence</th>\n",
       "      <th>hanging_txs_included_inframe_pepseqs</th>\n",
       "      <th>prefiltered_in-frame_epitopes</th>\n",
       "      <th>prefiltered_inframe_epitope_count</th>\n",
       "      <th>in-frame_neoepitopes</th>\n",
       "      <th>in-frame_neoepitope_count</th>\n",
       "      <th>frame-agnostic_all-transcript_biexons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10;48725;48803;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.176140</td>\n",
       "      <td>0.842293</td>\n",
       "      <td>2.517914</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "      <td>MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG</td>\n",
       "      <td>NVYYNEASG;VYYNEASGG;YYNEASGGR;YNEASGGRY;NEASGG...</td>\n",
       "      <td>17.0</td>\n",
       "      <td>TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10;277577;281199;-</td>\n",
       "      <td>2.295572</td>\n",
       "      <td>30.465966</td>\n",
       "      <td>16.003571</td>\n",
       "      <td>10.071656</td>\n",
       "      <td>26.917642</td>\n",
       "      <td>51.966700</td>\n",
       "      <td>53.589040</td>\n",
       "      <td>31.535793</td>\n",
       "      <td>50.723584</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.277578.281199.M...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RAHKSVTECAVFTWTNL</td>\n",
       "      <td>RAHKSVTECAVFTWTNL</td>\n",
       "      <td>RAHKSVTEC;AHKSVTECA;HKSVTECAV;KSVTECAVF;SVTECA...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10;280260;281199;-</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.961344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.307159</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.280261.281199.M...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>RAHKSVTEC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10;281323;283271;-</td>\n",
       "      <td>3.443358</td>\n",
       "      <td>23.212165</td>\n",
       "      <td>18.530451</td>\n",
       "      <td>15.107483</td>\n",
       "      <td>20.188231</td>\n",
       "      <td>32.821074</td>\n",
       "      <td>33.065578</td>\n",
       "      <td>44.150110</td>\n",
       "      <td>40.826300</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.281324.283271.M...</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "      <td>TELTDANGERHDALYVV</td>\n",
       "      <td>TELTDANGERHDALYVV</td>\n",
       "      <td>TELTDANGE;ELTDANGER;LTDANGERH;TDANGERHD;DANGER...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10;283446;286272;-</td>\n",
       "      <td>1.147786</td>\n",
       "      <td>41.346668</td>\n",
       "      <td>19.372744</td>\n",
       "      <td>6.714437</td>\n",
       "      <td>28.840330</td>\n",
       "      <td>61.539513</td>\n",
       "      <td>62.710579</td>\n",
       "      <td>37.842952</td>\n",
       "      <td>92.787044</td>\n",
       "      <td>...</td>\n",
       "      <td>['ENST00000280886.12.MOD.CHR10.283447.286272.M...</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "      <td>LGDSHLGEIWVHSAHN</td>\n",
       "      <td>LGDSHLGEIWVHSAHN</td>\n",
       "      <td>LGDSHLGEI;GDSHLGEIW;DSHLGEIWV;SHLGEIWVH;HLGEIW...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      jx  TCGA-24-1431-01A-01R-1566-13  \\\n",
       "0    chr10;48725;48803;-                      0.000000   \n",
       "1  chr10;277577;281199;-                      2.295572   \n",
       "2  chr10;280260;281199;-                      0.000000   \n",
       "3  chr10;281323;283271;-                      3.443358   \n",
       "4  chr10;283446;286272;-                      1.147786   \n",
       "\n",
       "   TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \\\n",
       "0                      2.176140                      0.842293   \n",
       "1                     30.465966                     16.003571   \n",
       "2                      0.000000                      0.000000   \n",
       "3                     23.212165                     18.530451   \n",
       "4                     41.346668                     19.372744   \n",
       "\n",
       "   TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \\\n",
       "0                      2.517914                      0.961344   \n",
       "1                     10.071656                     26.917642   \n",
       "2                      0.000000                      0.961344   \n",
       "3                     15.107483                     20.188231   \n",
       "4                      6.714437                     28.840330   \n",
       "\n",
       "   TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \\\n",
       "0                      0.000000                      5.700962   \n",
       "1                     51.966700                     53.589040   \n",
       "2                      0.000000                      0.000000   \n",
       "3                     32.821074                     33.065578   \n",
       "4                     61.539513                     62.710579   \n",
       "\n",
       "   TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \\\n",
       "0                      2.102386                      1.237161  ...   \n",
       "1                     31.535793                     50.723584  ...   \n",
       "2                      6.307159                      2.474321  ...   \n",
       "3                     44.150110                     40.826300  ...   \n",
       "4                     37.842952                     92.787044  ...   \n",
       "\n",
       "                               modified_upstream_txs  \\\n",
       "0  ['ENST00000561967.1.MOD.CHR10.48726.48803.MINU...   \n",
       "1  ['ENST00000280886.12.MOD.CHR10.277578.281199.M...   \n",
       "2  ['ENST00000280886.12.MOD.CHR10.280261.281199.M...   \n",
       "3  ['ENST00000280886.12.MOD.CHR10.281324.283271.M...   \n",
       "4  ['ENST00000280886.12.MOD.CHR10.283447.286272.M...   \n",
       "\n",
       "                     in-frame_all-transcript_biexons  \\\n",
       "0  LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...   \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...   \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...   \n",
       "\n",
       "                      in-frame_nonhanging-tx_biexons  \\\n",
       "0  MNMPSTPLAPTTGTATCSWSASTCTTTRPAVAGQCGAGNNWAKGHY...   \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...   \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...   \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...   \n",
       "\n",
       "            in-frame_peptide_sequence  hanging_txs_included_inframe_pepseqs  \\\n",
       "0  NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG    NVYYNEASGGRYVPRAV;TCTTTRPAVAGQCGAG   \n",
       "1                   RAHKSVTECAVFTWTNL                     RAHKSVTECAVFTWTNL   \n",
       "2                           RAHKSVTEC                             RAHKSVTEC   \n",
       "3                   TELTDANGERHDALYVV                     TELTDANGERHDALYVV   \n",
       "4                    LGDSHLGEIWVHSAHN                      LGDSHLGEIWVHSAHN   \n",
       "\n",
       "                       prefiltered_in-frame_epitopes  \\\n",
       "0  NVYYNEASG;VYYNEASGG;YYNEASGGR;YNEASGGRY;NEASGG...   \n",
       "1  RAHKSVTEC;AHKSVTECA;HKSVTECAV;KSVTECAVF;SVTECA...   \n",
       "2                                          RAHKSVTEC   \n",
       "3  TELTDANGE;ELTDANGER;LTDANGERH;TDANGERHD;DANGER...   \n",
       "4  LGDSHLGEI;GDSHLGEIW;DSHLGEIWV;SHLGEIWVH;HLGEIW...   \n",
       "\n",
       "   prefiltered_inframe_epitope_count  \\\n",
       "0                               17.0   \n",
       "1                                9.0   \n",
       "2                                1.0   \n",
       "3                                9.0   \n",
       "4                                8.0   \n",
       "\n",
       "                                in-frame_neoepitopes  \\\n",
       "0  TTRPAVAGQ;TRPAVAGQC;RPAVAGQCG;PAVAGQCGA;AVAGQCGAG   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "   in-frame_neoepitope_count  \\\n",
       "0                        5.0   \n",
       "1                        0.0   \n",
       "2                        0.0   \n",
       "3                        0.0   \n",
       "4                        0.0   \n",
       "\n",
       "               frame-agnostic_all-transcript_biexons  \n",
       "0  LTQIGQCGNQIGAKFWEVISDEHAIDSAGTYHGDSHLQLERINVYY...  \n",
       "1  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...  \n",
       "2  RTELTDANGERHDALYVVGALDEAMELRGMRYHPIDIETSVIRAHK...  \n",
       "3  IWVHSAHNASGYFTIYGDESLQSDHFNSRLSFGDTQTIWARTGYLG...  \n",
       "4  ALRHDRVRLVERGSPHSLPLMESGKILPGVRIIIANPETKGPLGDS...  \n",
       "\n",
       "[5 rows x 73 columns]"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "76c7c8a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    chr10;48726;48803;-\n",
       "0    chr10;48726;48803;-\n",
       "Name: jx, dtype: object"
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ohsu_df['jx'].head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65ade1f6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
