{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2eb08b22-c822-4b33-a5bc-8ed9425ad20b",
   "metadata": {},
   "source": [
    "## Table of Contents\n",
    "- [Import libraries](#1)\n",
    "- [processing tables](#2)\n",
    "- [Download tables](#3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8b15b3e-ac0d-40f4-8164-89d0419e18c1",
   "metadata": {},
   "source": [
    "<a name='1'></a>\n",
    "## Import libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "003ab2a3-f018-4e0e-af3c-1cf45c766ad5",
   "metadata": {},
   "source": [
    "The script is focused on setting up an environment for data analysis and visualization. It imports a suite of libraries and modules that are essential for statistical computing, data manipulation, progress tracking, file system operations, and generating visualizations such as plots and Venn diagrams. The specific libraries imported include pandas for data structures, numpy for numerical operations, tqdm for progress bars, glob for file path retrieval, os for operating system interaction, matplotlib and seaborn for plotting and graphical representations, and matplotlib_venn for creating Venn diagrams.\n",
    "\n",
    "Additionally, the script modifies the system path to include a custom directory, which suggests that the script will use additional custom modules and configuration settings located in this directory. These custom modules, imported with wildcard imports (from config import * and from functions import *)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "79a1ec5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %load /cluster/home/myurchikova/github/projects2020_ohsu/eth/learning_Master_thesis/TASKS/func/base_imports.py\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import tqdm \n",
    "import glob\n",
    "import os\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import tarfile\n",
    "import re\n",
    "from matplotlib_venn import venn2, venn2_circles, venn2_unweighted\n",
    "from matplotlib_venn import venn3, venn3_circles\n",
    "import sys\n",
    "sys.path.append(r\"/cluster/home/myurchikova/github/projects2020_ohsu/eth/learning_Master_thesis/TASKS/func\")\n",
    "from config import *\n",
    "from functions import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd01341f-299a-4695-b1c9-883f19c08828",
   "metadata": {},
   "source": [
    "<a name='2'></a>\n",
    "## Processing table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1aa816f-99ee-4440-a100-6e14ea797611",
   "metadata": {},
   "source": [
    "Two empty lists, ETH and OSHU, are initialized. These may be intended to store data or results related to the subsequent data processing steps.\n",
    "The pd.read_csv function from the pandas library is called three times to read different CSV files, each potentially associated with different datasets or aspects of the data:\n",
    "eth_df is created by reading a CSV file with a path defined by ETH_PATH_BRCA, using specific columns defined by COLS_ETH and a tab ('\\t') as the delimiter. The low_memory=False parameter is set, which is typically used to prevent low memory issues when processing large files.\n",
    "batch_to_gene is created by reading another CSV file with a path defined by BATCH_PATH_BRCA. It names the columns explicitly as 'gene_id' and 'batch', which suggests that this DataFrame will associate gene identifiers with specific batch numbers or identifiers.\n",
    "ohsu_df is created by reading a CSV file with a path defined by OHSU_PATH_BRCA, also using a tab as the delimiter and the low_memory=False parameter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d7da2e75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>batch</th>\n",
       "      <th>TCGAC8A12P01A11RA11507all</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607all</th>\n",
       "      <th>TCGABHA18V01A11RA12D07all</th>\n",
       "      <th>TCGAA2A0D201A21RA03407all</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WYITRSGIA</td>\n",
       "      <td>92347505:92347506:92349915:92349941:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.102634</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISSQSRVEK</td>\n",
       "      <td>92379851:92379859:92493866:92493885:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSGDEEKYP</td>\n",
       "      <td>92600493:92600508:92611313:92611325:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>2.102386</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLKMKMFQI</td>\n",
       "      <td>92379850:92379859:92496416:92496434:None:None</td>\n",
       "      <td>56543</td>\n",
       "      <td>2.922641</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348810</th>\n",
       "      <td>DSAVDFTGR</td>\n",
       "      <td>121183457:121183462:121181326:121181348:None:None</td>\n",
       "      <td>2858</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.711482</td>\n",
       "      <td>2.735089</td>\n",
       "      <td>1.140192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348811</th>\n",
       "      <td>LKWLLLLSL</td>\n",
       "      <td>121181270:121181278:121180861:121180880:None:None</td>\n",
       "      <td>2858</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348812</th>\n",
       "      <td>CLKWCKHPT</td>\n",
       "      <td>121181270:121181281:121168819:121168835:None:None</td>\n",
       "      <td>2858</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.560769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348813</th>\n",
       "      <td>KWLLLLSLF</td>\n",
       "      <td>121181270:121181275:121180858:121180880:None:None</td>\n",
       "      <td>2858</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348814</th>\n",
       "      <td>KWCKHPTLG</td>\n",
       "      <td>121181270:121181275:121168813:121168835:None:None</td>\n",
       "      <td>2858</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.560769</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22348815 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               kmer                                              coord  batch  \\\n",
       "0         WYITRSGIA      92347505:92347506:92349915:92349941:None:None  56543   \n",
       "1         WYITRSGIA      92347505:92347506:92349915:92349941:None:None  56543   \n",
       "2         ISSQSRVEK      92379851:92379859:92493866:92493885:None:None  56543   \n",
       "3         RSGDEEKYP      92600493:92600508:92611313:92611325:None:None  56543   \n",
       "4         HLKMKMFQI      92379850:92379859:92496416:92496434:None:None  56543   \n",
       "...             ...                                                ...    ...   \n",
       "22348810  DSAVDFTGR  121183457:121183462:121181326:121181348:None:None   2858   \n",
       "22348811  LKWLLLLSL  121181270:121181278:121180861:121180880:None:None   2858   \n",
       "22348812  CLKWCKHPT  121181270:121181281:121168819:121168835:None:None   2858   \n",
       "22348813  KWLLLLSLF  121181270:121181275:121180858:121180880:None:None   2858   \n",
       "22348814  KWCKHPTLG  121181270:121181275:121168813:121168835:None:None   2858   \n",
       "\n",
       "          TCGAC8A12P01A11RA11507all  TCGAAOA0JM01A21RA05607all  \\\n",
       "0                          0.000000                   0.000000   \n",
       "1                          0.000000                   0.000000   \n",
       "2                          0.000000                   0.000000   \n",
       "3                          2.922641                   2.102386   \n",
       "4                          2.922641                   0.000000   \n",
       "...                             ...                        ...   \n",
       "22348810                   0.000000                   0.000000   \n",
       "22348811                   0.000000                   0.000000   \n",
       "22348812                   0.000000                   0.000000   \n",
       "22348813                   0.000000                   0.000000   \n",
       "22348814                   0.000000                   0.000000   \n",
       "\n",
       "          TCGABHA18V01A11RA12D07all  TCGAA2A0D201A21RA03407all  \\\n",
       "0                          0.000000                   4.102634   \n",
       "1                          0.000000                   4.102634   \n",
       "2                          2.474321                   0.000000   \n",
       "3                          1.237161                   0.000000   \n",
       "4                          0.000000                   0.000000   \n",
       "...                             ...                        ...   \n",
       "22348810                   3.711482                   2.735089   \n",
       "22348811                   1.237161                   0.000000   \n",
       "22348812                   1.237161                   0.000000   \n",
       "22348813                   1.237161                   0.000000   \n",
       "22348814                   1.237161                   0.000000   \n",
       "\n",
       "          TCGAA2A0SX01A12RA08407all  \n",
       "0                          0.000000  \n",
       "1                          0.000000  \n",
       "2                          0.000000  \n",
       "3                          0.000000  \n",
       "4                          0.000000  \n",
       "...                             ...  \n",
       "22348810                   1.140192  \n",
       "22348811                   0.000000  \n",
       "22348812                   4.560769  \n",
       "22348813                   0.000000  \n",
       "22348814                   4.560769  \n",
       "\n",
       "[22348815 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ETH = []\n",
    "OSHU = []\n",
    "\n",
    "eth_df=pd.read_csv(ETH_PATH_BRCA, usecols = COLS_ETH, sep = '\\t', low_memory=False)\n",
    "batch_to_gene = pd.read_csv(BATCH_PATH_BRCA, names = ['gene_id', 'batch'], low_memory=False)\n",
    "ohsu_df=pd.read_csv(OHSU_PATH_BRCA, sep = '\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68b78946-99fe-4bcf-847d-5101043bff21",
   "metadata": {},
   "source": [
    "eth_df is modified by calling a function to get junction coordinates, which suggests that the DataFrame contains genomic or biological data where junctions are relevant.\n",
    "The pd.merge function is used to merge batch_to_gene DataFrame with eth_df on the 'batch' column, which implies combining two datasets based on a common identifier related to batches of samples or experiments.\n",
    "The drop method is used on ETH_df to remove the 'batch' column after the merge operation, indicating that this identifier is no longer needed in the merged DataFrame.\n",
    "The rename method is called twice to rename columns in ohsu_df and ETH_df using a dictionary of new column names provided by ETH_COLUMNS. This step standardizes the column names across DataFrames for consistency or clarity in further analysis.\n",
    "Finally, ETH_df is printed or returned, which would display the current state of this DataFrame after the operations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "812dea63",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "22348815it [1:18:07, 4767.27it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>kmer</th>\n",
       "      <th>coord</th>\n",
       "      <th>TCGAC8A12P01A11RA11507</th>\n",
       "      <th>TCGAAOA0JM01A21RA05607</th>\n",
       "      <th>TCGABHA18V01A11RA12D07</th>\n",
       "      <th>TCGAA2A0D201A21RA03407</th>\n",
       "      <th>TCGAA2A0SX01A12RA08407</th>\n",
       "      <th>strand</th>\n",
       "      <th>junction_coordinate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>TGKTQTTSP</td>\n",
       "      <td>1333364:1333379:1333505:1333517:None:None</td>\n",
       "      <td>11.690565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.121539</td>\n",
       "      <td>+</td>\n",
       "      <td>1333379:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>TGKTQTTSP</td>\n",
       "      <td>1333364:1333379:1333505:1333517:None:None</td>\n",
       "      <td>11.690565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.121539</td>\n",
       "      <td>+</td>\n",
       "      <td>1333379:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>TGKTQTTSP</td>\n",
       "      <td>1333364:1333379:1333505:1333517:None:None</td>\n",
       "      <td>11.690565</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.474321</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.121539</td>\n",
       "      <td>+</td>\n",
       "      <td>1333379:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>DNQVRSPCP</td>\n",
       "      <td>1333115:1333129:1333258:1333271:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>1333129:1333258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000169962.5</td>\n",
       "      <td>NQTTSPAPF</td>\n",
       "      <td>1333118:1333124:1333505:1333526:None:None</td>\n",
       "      <td>5.845283</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>1333124:1333505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348810</th>\n",
       "      <td>ENSG00000166391.15</td>\n",
       "      <td>VELLPEERT</td>\n",
       "      <td>75728985:75728989:75730193:75730216:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.700962</td>\n",
       "      <td>+</td>\n",
       "      <td>75728989:75730193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348811</th>\n",
       "      <td>ENSG00000166391.15</td>\n",
       "      <td>NADSILGTP</td>\n",
       "      <td>75728226:75728244:75728789:75728798:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>75728244:75728789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348812</th>\n",
       "      <td>ENSG00000166391.15</td>\n",
       "      <td>LGTPGANLL</td>\n",
       "      <td>75728241:75728244:75728789:75728813:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>75728244:75728789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348813</th>\n",
       "      <td>ENSG00000166391.15</td>\n",
       "      <td>LGTPGANLL</td>\n",
       "      <td>75728241:75728244:75728789:75728813:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>75728244:75728789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22348814</th>\n",
       "      <td>ENSG00000166391.15</td>\n",
       "      <td>ADSILGTPG</td>\n",
       "      <td>75728229:75728244:75728789:75728801:None:None</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.237161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>75728244:75728789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22348815 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     gene_id       kmer  \\\n",
       "0          ENSG00000169962.5  TGKTQTTSP   \n",
       "1          ENSG00000169962.5  TGKTQTTSP   \n",
       "2          ENSG00000169962.5  TGKTQTTSP   \n",
       "3          ENSG00000169962.5  DNQVRSPCP   \n",
       "4          ENSG00000169962.5  NQTTSPAPF   \n",
       "...                      ...        ...   \n",
       "22348810  ENSG00000166391.15  VELLPEERT   \n",
       "22348811  ENSG00000166391.15  NADSILGTP   \n",
       "22348812  ENSG00000166391.15  LGTPGANLL   \n",
       "22348813  ENSG00000166391.15  LGTPGANLL   \n",
       "22348814  ENSG00000166391.15  ADSILGTPG   \n",
       "\n",
       "                                                  coord  \\\n",
       "0             1333364:1333379:1333505:1333517:None:None   \n",
       "1             1333364:1333379:1333505:1333517:None:None   \n",
       "2             1333364:1333379:1333505:1333517:None:None   \n",
       "3             1333115:1333129:1333258:1333271:None:None   \n",
       "4             1333118:1333124:1333505:1333526:None:None   \n",
       "...                                                 ...   \n",
       "22348810  75728985:75728989:75730193:75730216:None:None   \n",
       "22348811  75728226:75728244:75728789:75728798:None:None   \n",
       "22348812  75728241:75728244:75728789:75728813:None:None   \n",
       "22348813  75728241:75728244:75728789:75728813:None:None   \n",
       "22348814  75728229:75728244:75728789:75728801:None:None   \n",
       "\n",
       "          TCGAC8A12P01A11RA11507  TCGAAOA0JM01A21RA05607  \\\n",
       "0                      11.690565                     0.0   \n",
       "1                      11.690565                     0.0   \n",
       "2                      11.690565                     0.0   \n",
       "3                       0.000000                     0.0   \n",
       "4                       5.845283                     0.0   \n",
       "...                          ...                     ...   \n",
       "22348810                0.000000                     0.0   \n",
       "22348811                0.000000                     0.0   \n",
       "22348812                0.000000                     0.0   \n",
       "22348813                0.000000                     0.0   \n",
       "22348814                0.000000                     0.0   \n",
       "\n",
       "          TCGABHA18V01A11RA12D07  TCGAA2A0D201A21RA03407  \\\n",
       "0                       2.474321                     0.0   \n",
       "1                       2.474321                     0.0   \n",
       "2                       2.474321                     0.0   \n",
       "3                       1.237161                     0.0   \n",
       "4                       0.000000                     0.0   \n",
       "...                          ...                     ...   \n",
       "22348810                0.000000                     0.0   \n",
       "22348811                1.237161                     0.0   \n",
       "22348812                1.237161                     0.0   \n",
       "22348813                1.237161                     0.0   \n",
       "22348814                1.237161                     0.0   \n",
       "\n",
       "          TCGAA2A0SX01A12RA08407 strand junction_coordinate  \n",
       "0                       9.121539      +     1333379:1333505  \n",
       "1                       9.121539      +     1333379:1333505  \n",
       "2                       9.121539      +     1333379:1333505  \n",
       "3                       0.000000      +     1333129:1333258  \n",
       "4                       0.000000      +     1333124:1333505  \n",
       "...                          ...    ...                 ...  \n",
       "22348810                5.700962      +   75728989:75730193  \n",
       "22348811                0.000000      +   75728244:75728789  \n",
       "22348812                0.000000      +   75728244:75728789  \n",
       "22348813                0.000000      +   75728244:75728789  \n",
       "22348814                0.000000      +   75728244:75728789  \n",
       "\n",
       "[22348815 rows x 10 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eth_df=table_processing.get_junction_coordinates(eth_df,'coord')\n",
    "ETH_df = pd.merge(batch_to_gene,eth_df, on=['batch'])\n",
    "ETH_df.drop(columns=['batch'],inplace=True)\n",
    "ohsu_df.rename(columns = ETH_COLMNS, inplace = True)\n",
    "ETH_df.rename(columns = ETH_COLMNS, inplace = True)\n",
    "ETH_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f0f58b7-1599-4cf8-bf93-a0ddbbe3d8b1",
   "metadata": {},
   "source": [
    "The rename method is used to change the column names in ohsu_df, replacing the name 'in-frame_neopeptides' with 'kmer', which suggests a focus on specific peptide sequences or genetic markers.\n",
    "Several functions from a module called table_processing are applied to ohsu_df, indicating that this DataFrame is undergoing a series of predefined processing steps:\n",
    "ohsu_to_eth_coord function might be transforming or aligning the coordinates from OHSU (Oregon Health & Science University) data to ETH (presumably a different institution or dataset format).\n",
    "change_column_names function is likely standardizing the column names for consistency across different datasets or analyses.\n",
    "preprocess_ohsu function suggests a general preprocessing of the data, which could involve cleaning, normalization, or preparation for further analysis.\n",
    "A new column, 'junction_coordinate', is created in ohsu_df by applying a lambda function to the 'x_shifted' column. This lambda function splits the 'x_shifted' values by a colon, takes the second to fourth elements, and joins them with a colon. This operation is typical for parsing and reformatting complex or compound data entries.\n",
    "The content of the 'junction_coordinate' column is then printed, which may be for the purpose of verification or inspection of the processed data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c144d8f0",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "invalid literal for int() with base 10: 'TRPAVAGQC'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m ohsu_df\u001b[38;5;241m.\u001b[39mrename(columns \u001b[38;5;241m=\u001b[39m {\u001b[38;5;124m'\u001b[39m\u001b[38;5;124min-frame_neoepitopes\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mkmer\u001b[39m\u001b[38;5;124m'\u001b[39m}, inplace \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m----> 2\u001b[0m ohsu_df \u001b[38;5;241m=\u001b[39m \u001b[43mtable_processing\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mohsu_to_eth_coord\u001b[49m\u001b[43m(\u001b[49m\u001b[43mohsu_df\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m ohsu_df\n\u001b[1;32m      4\u001b[0m ohsu_df \u001b[38;5;241m=\u001b[39m table_processing\u001b[38;5;241m.\u001b[39mchange_column_names(ohsu_df)\n",
      "File \u001b[0;32m~/github/projects2020_ohsu/eth/learning_Master_thesis/TASKS/func/functions/table_processing.py:26\u001b[0m, in \u001b[0;36mohsu_to_eth_coord\u001b[0;34m(df, col, new_col, sep)\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mohsu_to_eth_coord\u001b[39m(df, col \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mkmer\u001b[39m\u001b[38;5;124m'\u001b[39m, new_col \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mjx_shifted\u001b[39m\u001b[38;5;124m'\u001b[39m, sep \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m;\u001b[39m\u001b[38;5;124m'\u001b[39m):\n\u001b[1;32m     25\u001b[0m     tmp_jx \u001b[38;5;241m=\u001b[39m df[col]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39msplit(sep,  expand \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m---> 26\u001b[0m     df[new_col] \u001b[38;5;241m=\u001b[39m tmp_jx[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m+\u001b[39m sep \u001b[38;5;241m+\u001b[39m (\u001b[43mtmp_jx\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mastype\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mint\u001b[39;49m\u001b[43m)\u001b[49m \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m1\u001b[39m)\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mstr\u001b[39m) \u001b[38;5;241m+\u001b[39m sep \u001b[38;5;241m+\u001b[39m tmp_jx[\u001b[38;5;241m2\u001b[39m] \u001b[38;5;241m+\u001b[39m sep \u001b[38;5;241m+\u001b[39m tmp_jx[\u001b[38;5;241m3\u001b[39m]\n\u001b[1;32m     27\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m df\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/generic.py:6532\u001b[0m, in \u001b[0;36mNDFrame.astype\u001b[0;34m(self, dtype, copy, errors)\u001b[0m\n\u001b[1;32m   6528\u001b[0m     results \u001b[38;5;241m=\u001b[39m [ser\u001b[38;5;241m.\u001b[39mastype(dtype, copy\u001b[38;5;241m=\u001b[39mcopy) \u001b[38;5;28;01mfor\u001b[39;00m _, ser \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mitems()]\n\u001b[1;32m   6530\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   6531\u001b[0m     \u001b[38;5;66;03m# else, only a single dtype is given\u001b[39;00m\n\u001b[0;32m-> 6532\u001b[0m     new_data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_mgr\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mastype\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   6533\u001b[0m     res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_constructor_from_mgr(new_data, axes\u001b[38;5;241m=\u001b[39mnew_data\u001b[38;5;241m.\u001b[39maxes)\n\u001b[1;32m   6534\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m res\u001b[38;5;241m.\u001b[39m__finalize__(\u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mastype\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/internals/managers.py:414\u001b[0m, in \u001b[0;36mBaseBlockManager.astype\u001b[0;34m(self, dtype, copy, errors)\u001b[0m\n\u001b[1;32m    411\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m using_copy_on_write():\n\u001b[1;32m    412\u001b[0m     copy \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[0;32m--> 414\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    415\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mastype\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    416\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    417\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    418\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    419\u001b[0m \u001b[43m    \u001b[49m\u001b[43musing_cow\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43musing_copy_on_write\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    420\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/internals/managers.py:354\u001b[0m, in \u001b[0;36mBaseBlockManager.apply\u001b[0;34m(self, f, align_keys, **kwargs)\u001b[0m\n\u001b[1;32m    352\u001b[0m         applied \u001b[38;5;241m=\u001b[39m b\u001b[38;5;241m.\u001b[39mapply(f, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[1;32m    353\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 354\u001b[0m         applied \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mgetattr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mf\u001b[49m\u001b[43m)\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    355\u001b[0m     result_blocks \u001b[38;5;241m=\u001b[39m extend_blocks(applied, result_blocks)\n\u001b[1;32m    357\u001b[0m out \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mtype\u001b[39m(\u001b[38;5;28mself\u001b[39m)\u001b[38;5;241m.\u001b[39mfrom_blocks(result_blocks, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maxes)\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/internals/blocks.py:616\u001b[0m, in \u001b[0;36mBlock.astype\u001b[0;34m(self, dtype, copy, errors, using_cow)\u001b[0m\n\u001b[1;32m    596\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    597\u001b[0m \u001b[38;5;124;03mCoerce to the new dtype.\u001b[39;00m\n\u001b[1;32m    598\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    612\u001b[0m \u001b[38;5;124;03mBlock\u001b[39;00m\n\u001b[1;32m    613\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    614\u001b[0m values \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvalues\n\u001b[0;32m--> 616\u001b[0m new_values \u001b[38;5;241m=\u001b[39m \u001b[43mastype_array_safe\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    618\u001b[0m new_values \u001b[38;5;241m=\u001b[39m maybe_coerce_values(new_values)\n\u001b[1;32m    620\u001b[0m refs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/dtypes/astype.py:238\u001b[0m, in \u001b[0;36mastype_array_safe\u001b[0;34m(values, dtype, copy, errors)\u001b[0m\n\u001b[1;32m    235\u001b[0m     dtype \u001b[38;5;241m=\u001b[39m dtype\u001b[38;5;241m.\u001b[39mnumpy_dtype\n\u001b[1;32m    237\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 238\u001b[0m     new_values \u001b[38;5;241m=\u001b[39m \u001b[43mastype_array\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    239\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mValueError\u001b[39;00m, \u001b[38;5;167;01mTypeError\u001b[39;00m):\n\u001b[1;32m    240\u001b[0m     \u001b[38;5;66;03m# e.g. _astype_nansafe can fail on object-dtype of strings\u001b[39;00m\n\u001b[1;32m    241\u001b[0m     \u001b[38;5;66;03m#  trying to convert to float\u001b[39;00m\n\u001b[1;32m    242\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m errors \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/dtypes/astype.py:183\u001b[0m, in \u001b[0;36mastype_array\u001b[0;34m(values, dtype, copy)\u001b[0m\n\u001b[1;32m    180\u001b[0m     values \u001b[38;5;241m=\u001b[39m values\u001b[38;5;241m.\u001b[39mastype(dtype, copy\u001b[38;5;241m=\u001b[39mcopy)\n\u001b[1;32m    182\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 183\u001b[0m     values \u001b[38;5;241m=\u001b[39m \u001b[43m_astype_nansafe\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    185\u001b[0m \u001b[38;5;66;03m# in pandas we don't store numpy str dtypes, so convert to object\u001b[39;00m\n\u001b[1;32m    186\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(dtype, np\u001b[38;5;241m.\u001b[39mdtype) \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28missubclass\u001b[39m(values\u001b[38;5;241m.\u001b[39mdtype\u001b[38;5;241m.\u001b[39mtype, \u001b[38;5;28mstr\u001b[39m):\n",
      "File \u001b[0;32m/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/scipy-stack/2023b/lib/python3.10/site-packages/pandas/core/dtypes/astype.py:134\u001b[0m, in \u001b[0;36m_astype_nansafe\u001b[0;34m(arr, dtype, copy, skipna)\u001b[0m\n\u001b[1;32m    130\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(msg)\n\u001b[1;32m    132\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m copy \u001b[38;5;129;01mor\u001b[39;00m arr\u001b[38;5;241m.\u001b[39mdtype \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mobject\u001b[39m \u001b[38;5;129;01mor\u001b[39;00m dtype \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mobject\u001b[39m:\n\u001b[1;32m    133\u001b[0m     \u001b[38;5;66;03m# Explicit copy, or required since NumPy can't view from / to object.\u001b[39;00m\n\u001b[0;32m--> 134\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43marr\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mastype\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m    136\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m arr\u001b[38;5;241m.\u001b[39mastype(dtype, copy\u001b[38;5;241m=\u001b[39mcopy)\n",
      "\u001b[0;31mValueError\u001b[0m: invalid literal for int() with base 10: 'TRPAVAGQC'"
     ]
    }
   ],
   "source": [
    "ohsu_df.rename(columns = {'in-frame_neoepitopes': 'kmer'}, inplace = True)\n",
    "ohsu_df = table_processing.ohsu_to_eth_coord(ohsu_df)\n",
    "ohsu_df\n",
    "ohsu_df = table_processing.change_column_names(ohsu_df)\n",
    "ohsu_df = table_processing.preprocess_ohsu(ohsu_df)\n",
    "ohsu_df['junction_coordinate'] = ohsu_df['jx_shifted'].apply(lambda x: ':'.join(x.split(';')[1:3]))\n",
    "print(ohsu_df['junction_coordinate'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1304a307-6316-48c3-924a-d3e80d3f4f7a",
   "metadata": {},
   "source": [
    "ohsu_df is being reassigned to a filtered version of itself, selecting only a subset of columns. This typically happens when only specific data columns are needed for further analysis.\n",
    "ETH_df undergoes two drop operations to remove columns named 'strand' and 'coord'. The inplace=True parameter indicates that the DataFrame should be modified in place, without creating a new DataFrame."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68ce378a",
   "metadata": {},
   "outputs": [],
   "source": [
    "oshu_df=ohsu_df[['gene_id','kmer','junction_coordinate','TCGAC8A12P01A11RA11507', 'TCGAAOA0JM01A21RA05607', 'TCGABHA18V01A11RA12D07', 'TCGAA2A0D201A21RA03407','TCGAA2A0SX01A12RA08407']]\n",
    "ETH_df.drop(columns=['strand'],inplace=True)\n",
    "ETH_df.drop(columns=['coord'],inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32214613-6e15-48e2-81f4-292f89d5135a",
   "metadata": {},
   "source": [
    "path_eth_df and path_ohsu_df are created using a create_path function, indicating the generation of file paths for saving the data.\n",
    "ETH_df is saved to a CSV file at the location specified by path_eth_df. The to_csv function is used with header=True, which includes the column names in the output file, and sep='\\t', which means the data is tab-separated.\n",
    "Similarly, ohsu_df is saved to a CSV file at the location specified by path_ohsu_df with the same parameters for headers and separator."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f9d570f",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_eth_df=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_BRCA,NAME_TABLES,NAME_ETH_BRCA])\n",
    "path_ohsu_df=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_BRCA,NAME_TABLES,NAME_OHSU_BRCA])\n",
    "ETH_df.to_csv(path_eth_df, header = True, sep='\\t')\n",
    "ohsu_df.to_csv(path_ohsu_df, header = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6f18f47-1ae3-472e-b1e4-0fceda3a217a",
   "metadata": {},
   "source": [
    "A dictionary named out_table is initialized with various keys, each associated with an empty list. These keys are suggestive of data attributes such as sample identifiers, k-mer counts, sizes, and coordinates which are often used in genomic data analysis.\n",
    "A for loop begins, iterating over columns of ETH_df DataFrame using the tqdm library, which provides a progress bar for loops.\n",
    "Within the loop, multiple conditional and data manipulation steps are performed:\n",
    "Checking if the current column's k-mers are zero and performing set operations.\n",
    "Appending data to the lists in out_table, such as 'junction_coordinate' and various size and coordinate related data.\n",
    "Calculating differences between sets of k-mers and appending the results to the appropriate lists in out_table.\n",
    "The snippet contains several operations that transform and aggregate data, possibly to prepare for statistical analysis or visualization. For example, it deals with intersection and union of genomic coordinates, suggesting a comparison between different datasets or conditions within the study.\n",
    "The code ends with appending array data converted to lists, likely for uniformity in data structure within the out_table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3f7af7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_table = {\n",
    "    'sample':[],\n",
    "    'eth_without_ohsu':[],\n",
    "    'kmer_eth':[],\n",
    "    'ohsu_without_eth':[],\n",
    "    'kmer_ohsu':[],\n",
    "    'inter':[],\n",
    "    'kmer_inter':[],\n",
    "    'full_size_ohsu': [],\n",
    "    'full_size_eth': [],\n",
    "\n",
    "    'coord_OHSU':[],\n",
    "    'coord_ETH':[],\n",
    "    'size_ohsu_coor' : [], \n",
    "    'size_eth_coor' : [], \n",
    "    'size_intersection_coor' : [], \n",
    "    'size_ohsu\\eth_coor' : [], \n",
    "    'size_eth\\ohsu_coor' : [],\n",
    "    'eth_coor\\inter_coor':[],\n",
    "    'ohsu_coor\\inter_coor':[],\n",
    "    'inter_coor':[],\n",
    "    'ohsu_coor\\eth_coor':[],\n",
    "    'eth_coor\\ohsu_coor':[]\n",
    "\n",
    "}\n",
    "\n",
    "for exp_col in tqdm.tqdm(EXPRESSION):\n",
    "    df1=ETH_df[ETH_df[exp_col] > 0]\n",
    "    df2=oshu_df[oshu_df[exp_col] > 0]\n",
    "    eth_kmers=set(df1['kmer'])\n",
    "    ohsu_kmers=set(df2['kmer'])\n",
    "    out_table['coord_ETH'].append(df1['junction_coordinate'] if not df1.empty else 'None' )\n",
    "    out_table['coord_OHSU'].append(df2['junction_coordinate'] if not df2.empty else 'None')\n",
    "    ser1 = set(df1['junction_coordinate'])\n",
    "    ser2 = set(df2['junction_coordinate'])\n",
    "    out_table['size_ohsu_coor'].append(len(ser2))\n",
    "    out_table['size_eth_coor'].append(len(ser1))\n",
    "    out_table['size_ohsu\\eth_coor'].append(len(df_ohsu_filter_coor:=ser2.difference(ser1)))\n",
    "    out_table['size_eth\\ohsu_coor'].append(len(df_eth_filter_coor:=ser1.difference(ser2)))\n",
    "    \n",
    "    out_table['ohsu_coor\\eth_coor'].append(df_ohsu_filter_coor)\n",
    "    out_table['eth_coor\\ohsu_coor'].append(df_eth_filter_coor)\n",
    "    \n",
    "    out_table['size_intersection_coor'].append(len(df_inter_filter_coor:=ser2 & ser1))\n",
    "    out_table['eth_coor\\inter_coor'].append(df_eth_witout_inter_coor:=list(df_eth_filter_coor.difference(df_inter_filter_coor)))\n",
    "    out_table['ohsu_coor\\inter_coor'].append(df_ohsu_witout_inter_coor:=list(df_ohsu_filter_coor.difference(df_inter_filter_coor)))\n",
    "    out_table['inter_coor'].append(list(df_inter_filter_coor))\n",
    "    \n",
    "    out_table['kmer_eth'].append(np.array(eth_kmers))\n",
    "    out_table['kmer_ohsu'].append(np.array(ohsu_kmers))\n",
    "    #append() takes a single item as an input and adds it to the and of the list\n",
    "    out_table['sample'].append(exp_col)\n",
    "    out_table['full_size_ohsu'].append(n_ohsu := len(ohsu_kmers))\n",
    "    out_table['full_size_eth'].append(n_eth:=len(eth_kmers))\n",
    "    \n",
    "    out_table['eth_without_ohsu'].append(len(eth_without_ohsu_kmers:=eth_kmers.difference(ohsu_kmers)))\n",
    "    out_table['ohsu_without_eth'].append(len(oshu_without_ohsu_kmers:=ohsu_kmers.difference(eth_kmers)))\n",
    "    out_table['inter'].append(intersection:=len(intersection_kmers:= (eth_kmers & ohsu_kmers)))\n",
    "    out_table['kmer_inter'].append(list(intersection_kmers))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b8c38ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_df_original=pd.DataFrame(out_table)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fbffed8c-bdaf-48fe-bdab-22ccb56d1935",
   "metadata": {},
   "source": [
    "<a name='3'></a>\n",
    "## Download tables"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9945efc5-e0c0-4b6c-9022-16562049c5e1",
   "metadata": {},
   "source": [
    "path_non_filtering is generated using a create_path function, which likely constructs a file path by concatenating directory and file names specified in the parameters. The directory and file names suggest that the data may be related to a specific type of breast cancer research data (as indicated by BRCA).\n",
    "out_df_original is a DataFrame that is written to a CSV file at the location specified by path_non_filtering. The to_csv method uses header=True, indicating that the CSV file should include column headers, and sep='\\t', specifying that the data fields in the CSV are separated by tabs rather than commas, which is typical for TSV (tab-separated values) files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5335b67b",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_non_filtering=create_path.create_path(SAVE_DIR,[DIR_CSV,DIR_BRCA,NAME_TABLES,NAME_NON_FILTERING_BRCA])\n",
    "out_df_original.to_csv(path_non_filtering, header = True, sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
